<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Comparative Protein Structure Analysis with Bio3D • Bio3D</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../../../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../../../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../../../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../../../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../../../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../../../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="../../../tidyverse.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../../bootstrap-toc.css">
<script src="../../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../../pkgdown.css" rel="stylesheet">
<script src="../../../pkgdown.js"></script><link href="../../../tidyverse-2.css" rel="stylesheet">
<!--optional theme
<link href="../../../.css" rel="stylesheet">
--><!-- docsearch --><script src="../../../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../../../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Comparative Protein Structure Analysis with Bio3D">
<meta property="og:description" content="bio3d">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../../../index.html">
        bio3d</a>
        <div class="info hidden-xs hidden-sm">
<!--          <span class="partof">part of the <a href="https://tidyverse.org">tidyverse</a></span>
-->
          <span class="version version-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4-1.9000</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">INTRODUCTORY TUTORIALS</li>
    <li>
      <a href="../../../articles/online/install_vignette/Bio3D_install.html">Installing Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/intro_vignette/Bio3D_introduction.html">Getting started with Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/pdb_vignette/Bio3D_pdb.html">PDB structure manipulation and analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/pca_vignette/Bio3D_pca.html">Comparative sequence and structure analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/nma_vignette/Bio3D_nma.html">Enhanced Normal Modes Analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/traj_vignette/Bio3D_md.html">Beginning trajectory analysis with Bio3D </a>
    </li>
    <li class="divider">
    <li class="dropdown-header">ADVANCED TUTORIALS</li>
    <li>
      <a href="../../../articles/online/enma_vignettes/Bio3D_nma-dhfr-partI.html">Ensemble NMA of E.coli DHFR structures</a>
    </li>
    <li>
      <a href="../../../articles/online/enma_vignettes/Bio3D_nma-dhfr-partII.html">Ensemble NMA across multiple species of DHFR</a>
    </li>
    <li>
      <a href="../../../articles/online/cna_vignette/Bio3D_cna-transducin.html">Correlation network analysis with Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/cna_vignette/cna_vignette.spin.html">Protein structure network analysis with Bio3D ***</a>
    </li>
    <li>
      <a href="../../../articles/online/eddm_vignette/Bio3D_eddm.spin.html">Ensemble Difference Distance Matrix (eDDM) analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">WEBAPP TUTORIALS</li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/download/webapp-tutorials?download=22:bio3d-webapp-pca">Online protein structure analysis with Bio3D-Web (PDF)</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/webapps">Online Normal Mode Analysis with Bio3D WebApps (To Update)</a>
    </li>
    <li class="divider">
    <li>
      <a href="../../../articles/index.html">All vignettes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Documentation
  </a>
</li>
        <li>
  <a href="../../../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../../../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    News
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket"></span>
     
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Bio3D_pca_files/header-attrs-2.3.2/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Comparative Protein Structure Analysis with Bio3D</h1>
                        <h4 class="author">Xin-Qiu Yao, Lars Skjaerven &amp; Barry J. Grant</h4>
            
      
      
      <div class="hidden name"><code>Bio3D_pca.Rmd</code></div>

    </div>

    
    
<div style="page-break-after: always;"></div>
<div id="background" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#background" class="anchor"></a>Background</h1>
<p>Bio3D<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> is a group of R packages that provide interactive tools for the analysis of bimolecular structure, sequence and simulation data. The aim of this document, termed a vignette<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a> in R parlance, is to provide a brief task-oriented introduction to facilities for analyzing protein structure data with the Bio3D core package, <strong>bio3d</strong> <span class="citation">(Grant et al. 2006)</span>.</p>
<div id="requirements" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#requirements" class="anchor"></a>Requirements</h2>
<p>Detailed instructions for obtaining and installing the Bio3D packages on various platforms can be found in the <a href="http://thegrantlab.org/bio3d">Installing Bio3D</a> vignette available online. Note that to follow along with this vignette the <a href="http://www.drive5.com/muscle/">MUSCLE multiple sequence alignment program</a> must be installed on your system and in the search path for executables. Please see the installation vignette for full details. This particular vignette was generated using <strong>bio3d version 2.4.1.9000</strong>.</p>
</div>
</div>
<div id="getting-started" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting Started</h1>
<p>Start R, load bio3d and use the command <code><a href="https://rdrr.io/r/utils/demo.html">demo("pdb")</a></code> and then <code><a href="https://rdrr.io/r/utils/demo.html">demo("pca")</a></code> to get a quick feel for some of the tasks that we will be introducing in the following sections.</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="http://thegrantlab.org/bio3d">bio3d</a></span>)
<span class="fu"><a href="https://rdrr.io/r/utils/demo.html">demo</a></span>(<span class="st">"pdb"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/demo.html">demo</a></span>(<span class="st">"pca"</span>)
</pre></div>
<div id="side-note" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#side-note" class="anchor"></a>Side-note:</h4>
<p>You will be prompted to hit the <code>RETURN</code> key at each step of the demos as this will allow you to see the particular functions being called. Also note that detailed documentation and example code for each function can be accessed via the <code>help()</code> and <code><a href="https://rdrr.io/r/utils/example.html">example()</a></code> commands (e.g. <code><a href="../../../reference/read.pdb.html">help(read.pdb)</a></code>). You can also copy and paste any of the example code from the documentation of a particular function, or indeed this vignette, directly into your R session to see how things work. You can also find this <a href="http://thegrantlab.org/bio3d">documentation online</a>.</p>
</div>
<div id="working-with-single-pdb-structures" class="section level2">
<h2 class="hasAnchor">
<a href="#working-with-single-pdb-structures" class="anchor"></a>Working with single PDB structures</h2>
<p>A comprehensive introduction to working with PDB format structures with Bio3D can be found in <code>PDB structure manipulation and analysis</code> vignette. Here we confine ourselves to a very brief overview. The code snippet below calls the <code><a href="../../../reference/read.pdb.html">read.pdb()</a></code> function with a single input argument, the four letter Protein Data Bank (PDB) identifier code <code>"1tag"</code>. This will cause the <code><a href="../../../reference/read.pdb.html">read.pdb()</a></code> function to read directly from the online RCSB PDB database and return a new object <code>pdb</code> for further manipulation.</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="kw">pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"1tag"</span>)
</pre></div>
<pre><code>##   Note: Accessing on-line PDB file</code></pre>
<p>Alternatively, you can read a PDB file directly from your local file system using the file name (or the full path to the file) as an argument to <code><a href="../../../reference/read.pdb.html">read.pdb()</a></code>:</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="kw">pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"myfile.pdb"</span>)
<span class="kw">pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"/path/to/my/data/myfile.pdb"</span>)
</pre></div>
<p>A short summary of the <code>pdb</code> object can be obtained by calling the function <code><a href="../../../reference/read.pdb.html">print.pdb()</a></code> (or simply typing <code>pdb</code>, which is equivalent in this case):</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="kw">pdb</span>
</pre></div>
<pre><code>## 
##  Call:  read.pdb(file = "1tag")
## 
##    Total Models#: 1
##      Total Atoms#: 2890,  XYZs#: 8670  Chains#: 1  (values: A)
## 
##      Protein Atoms#: 2521  (residues/Calpha atoms#: 314)
##      Nucleic acid Atoms#: 0  (residues/phosphate atoms#: 0)
## 
##      Non-protein/nucleic Atoms#: 369  (residues: 342)
##      Non-protein/nucleic resid values: [ GDP (1), HOH (340), MG (1) ]
## 
##    Protein sequence:
##       ARTVKLLLLGAGESGKSTIVKQMKIIHQDGYSLEECLEFIAIIYGNTLQSILAIVRAMTT
##       LNIQYGDSARQDDARKLMHMADTIEEGTMPKEMSDIIQRLWKDSGIQACFDRASEYQLND
##       SAGYYLSDLERLVTPGYVPTEQDVLRSRVKTTGIIETQFSFKDLNFRMFDVGGQRSERKK
##       WIHCFEGVTCIIFIAALSAYDMVLVEDDEVNRMHESLHLFNSICN...&lt;cut&gt;...DIII
## 
## + attr: atom, xyz, seqres, helix, sheet,
##         calpha, remark, call</code></pre>
<p>To examine the contents of the <code>pdb</code> object in more detail we can use the <code>attributes</code> function:</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/graph_attr_names.html">attributes</a></span>(<span class="kw">pdb</span>)
</pre></div>
<pre><code>## $names
## [1] "atom"   "xyz"    "seqres" "helix"  "sheet"  "calpha" "remark" "call"  
## 
## $class
## [1] "pdb" "sse"</code></pre>
<p>These attributes describe the list components that comprise the <code>pdb</code> object, and each individual component can be accessed using the <code><a href="https://rdrr.io/r/base/Extract.html">$</a></code> symbol (e.g. <code>pdb$atom</code>). Their complete description can be found on the <code><a href="../../../reference/read.pdb.html">read.pdb()</a></code> functions help page accessible with the command: <code><a href="../../../reference/read.pdb.html">help(read.pdb)</a></code>. Note that the <code>atom</code> component is a data frame (matrix like object) consisting of all atomic coordinate ATOM/HETATM data, with a row per ATOM and a column per record type. The column names can be used as a convenient means of data access, for example to access coordinate data for the first three atoms in our newly created <code>pdb</code> object:</p>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="kw">pdb</span><span class="op">$</span><span class="kw">atom</span>[<span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"resno"</span>,<span class="st">"resid"</span>,<span class="st">"elety"</span>,<span class="st">"x"</span>,<span class="st">"y"</span>,<span class="st">"z"</span>)]
</pre></div>
<pre><code>##   resno resid elety      x      y      z
## 1    27   ALA     N 38.238 18.018 61.225
## 2    27   ALA    CA 38.552 16.715 60.576
## 3    27   ALA     C 40.042 16.687 60.253</code></pre>
<p>In the example above we used numeric indices to access atoms 1 to 3, and a character vector of column names to access the specific record types. In a similar fashion the <code><a href="../../../reference/atom.select.html">atom.select()</a></code> function returns numeric indices that can be used for accessing desired subsets of the <code>pdb</code> data. For example:</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="kw">ca.inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>, <span class="st">"calpha"</span>)
</pre></div>
<p>The returned <code>ca.inds</code> object is a list containing atom and xyz numeric indices corresponding to the selection (all C-alpha atoms in this particular case). The indices can be used to access e.g. the Cartesian coordinates of the selected atoms (<code>pdb$xyz[, ca.inds$xyz]</code>), or residue numbers and B-factor data for the selected atoms. For example:</p>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="co"># See Figure 1</span>
<span class="kw">bf</span> <span class="op">&lt;-</span> <span class="kw">pdb</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">b</span>[<span class="kw">ca.inds</span><span class="op">$</span><span class="kw">atom</span>]
<span class="fu"><a href="../../../reference/plot.bio3d.html">plot.bio3d</a></span>(<span class="kw">bf</span>, resno=<span class="kw">pdb</span>, sse=<span class="kw">pdb</span>, ylab=<span class="st">"B-factor"</span>, xlab=<span class="st">"Residue"</span>, typ=<span class="st">"l"</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_pca_files/figure-html/unnamed-chunk-9-1.png" alt="Residue B-factor data for PDB id 1TAG. Grey boxes depict secondary structure elements in the structure (dark grey: alpha helices; light grey: beta sheets)." width="700"><p class="caption">
Residue B-factor data for PDB id 1TAG. Grey boxes depict secondary structure elements in the structure (dark grey: alpha helices; light grey: beta sheets).
</p>
</div>
<p>In the above example we use these indices to plot residue number (taken from the <code>pdb</code> object, note these residue numbers do not start at 1) vs B-factor along with a basic secondary structure schematic (again taken from the <code>pdb</code> object and provided with the argument <code>sse=pdb</code>; Figure 1). Note also that we could access this same data with <code>pdb$atom[ca.inds$atom, "b"]</code>. As a further example of data access lets extract the sequence for the loop region (P-loop) between beta 1 (the third strand in the beta sheet) and alpha 1 in our <code>pdb</code> object.</p>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="kw">loop</span> <span class="op">&lt;-</span> <span class="kw">pdb</span><span class="op">$</span><span class="kw">sheet</span><span class="op">$</span><span class="kw">end</span>[<span class="fl">3</span>]<span class="op">:</span><span class="kw">pdb</span><span class="op">$</span><span class="kw">helix</span><span class="op">$</span><span class="kw">start</span>[<span class="fl">1</span>]
<span class="kw">loop.inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>, resno=<span class="kw">loop</span>, elety=<span class="st">"CA"</span>)
<span class="kw">pdb</span><span class="op">$</span><span class="kw">atom</span>[<span class="kw">loop.inds</span><span class="op">$</span><span class="kw">atom</span>, <span class="st">"resid"</span>]
</pre></div>
<pre><code>## [1] "LEU" "GLY" "ALA" "GLY" "GLU" "SER" "GLY" "LYS"</code></pre>
<p>In the above example the residue numbers in the <code>sheet</code> and <code>helix</code> components of <code>pdb</code> are accessed and used in a subsequent atom selection, the output of which is used as indices to extract residue names.</p>
<p>Since Bio3D version 2.1 the  component in the PDB object is in a matrix format (as compared to a vector format in previous versions). Thus, notice the extra comma in the square bracket operator when accessing Cartesian coordinates from the  object ().</p>
<div id="question" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question" class="anchor"></a>Question:</h4>
<p>How would you extract the one-letter amino acid sequence for the loop region mentioned above? HINT: The <code><a href="../../../reference/aa123.html">aa321()</a></code> function converts between three-letter and one-letter IUPAC amino acid codes.</p>
</div>
<div id="question-1" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-1" class="anchor"></a>Question:</h4>
<p>How would you select all backbone or sidechain atoms? HINT: see the example section of <code><a href="../../../reference/atom.select.html">help(atom.select)</a></code> and the <code>string</code> option.</p>
</div>
<div id="side-note-1" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#side-note-1" class="anchor"></a>Side-note:</h4>
<p>Consider the <code>combine.select</code> function for combining multiple atom selections. See <code><a href="../../../reference/combine.select.html">help(combine.select)</a></code> and additional vignettes for more details.</p>
<p>Note that if you know a particular sequence pattern or motif characteristic of a region of interest you could access it via a sequence search as follows:</p>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="kw">aa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbseq.html">pdbseq</a></span>(<span class="kw">pdb</span>)
<span class="kw">aa</span>[ <span class="fu"><a href="../../../reference/motif.find.html">motif.find</a></span>(<span class="st">".G....GK[ST]"</span>, <span class="kw">aa</span> )]
</pre></div>
<pre><code>##  35  36  37  38  39  40  41  42  43 
## "L" "G" "A" "G" "E" "S" "G" "K" "S"</code></pre>
</div>
</div>
<div id="working-with-multiple-pdb-structures" class="section level2">
<h2 class="hasAnchor">
<a href="#working-with-multiple-pdb-structures" class="anchor"></a>Working with multiple PDB structures</h2>
<p>The Bio3D package was designed to specifically facilitate the analysis of multiple structures from both experiment and simulation. The challenge of working with these structures is that they are usually different in their composition (i.e. contain differing number of atoms, sequences, chains, ligands, structures, conformations etc. even for the same protein as we will see below) and it is these differences that are frequently of most interest.</p>
<p>For this reason Bio3D contains extensive utilities to enable the reading and writing of sequence and structure data, sequence and structure alignment, performing homologous protein searches, structure annotation, atom selection, re-orientation, superposition, rigid core identification, clustering, torsion analysis, distance matrix analysis, structure and sequence conservation analysis, normal mode analysis across related structures, and principal component analysis of structural ensembles. We will demonstrate some of these utilities in the following sections. More comprehensive demonstrations for special tasks, e.g. various operations of PDB structures, analysis of MD trajectories and normal mode analysis can be found in corresponding vignettes.</p>
<p>Before delving into more advanced analysis lets examine how we can read multiple PDB structures from the RCSB PDB for a particular protein and perform some basic analysis:</p>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="co"># Download some example PDB files</span>
<span class="kw">ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"1TND_B"</span>,<span class="st">"1AGR_A"</span>,<span class="st">"1FQJ_A"</span>,<span class="st">"1TAG_A"</span>,<span class="st">"1GG2_A"</span>,<span class="st">"1KJY_A"</span>)
<span class="kw">raw.files</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/get.pdb.html">get.pdb</a></span>(<span class="kw">ids</span>)
</pre></div>
<p>The <code><a href="../../../reference/get.pdb.html">get.pdb()</a></code> function will download the requested files, below we extract the particular chains we are most interested in with the function <code><a href="../../../reference/pdbsplit.html">pdbsplit()</a></code> (note these <code>ids</code> could come from the results of a <code><a href="../../../reference/blast.pdb.html">blast.pdb()</a></code> search as described in subsequent sections). The requested chains are then aligned and their structural data stored in a new object <code>pdbs</code> that can be used for further analysis and manipulation.</p>
<div class="sourceCode" id="cb18"><pre class="downlit">
<span class="co"># Extract and align the chains we are interested in</span>
<span class="kw">files</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbsplit.html">pdbsplit</a></span>(<span class="kw">raw.files</span>, <span class="kw">ids</span>)
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbaln.html">pdbaln</a></span>(<span class="kw">files</span>)
</pre></div>
<p>Below we examine their sequence and structural similarity.</p>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="co"># Calculate sequence identity</span>
<span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span>),<span class="fl">1</span>,<span class="fl">6</span>)
<span class="fu"><a href="../../../reference/seqidentity.html">seqidentity</a></span>(<span class="kw">pdbs</span>)
</pre></div>
<pre><code>##        1TND_B 1AGR_A 1FQJ_A 1TAG_A 1GG2_A 1KJY_A
## 1TND_B  1.000  0.693  0.914  1.000  0.690  0.696
## 1AGR_A  0.693  1.000  0.779  0.694  0.997  0.994
## 1FQJ_A  0.914  0.779  1.000  0.914  0.776  0.782
## 1TAG_A  1.000  0.694  0.914  1.000  0.691  0.697
## 1GG2_A  0.690  0.997  0.776  0.691  1.000  0.991
## 1KJY_A  0.696  0.994  0.782  0.697  0.991  1.000</code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit">
<span class="co">## Calculate RMSD</span>
<span class="fu"><a href="../../../reference/rmsd.html">rmsd</a></span>(<span class="kw">pdbs</span>, fit=<span class="fl">TRUE</span>)
</pre></div>
<pre><code>##        1TND_B 1AGR_A 1FQJ_A 1TAG_A 1GG2_A 1KJY_A
## 1TND_B  0.000  0.965  0.609  1.283  1.612  2.100
## 1AGR_A  0.965  0.000  0.873  1.575  1.777  1.914
## 1FQJ_A  0.609  0.873  0.000  1.265  1.737  2.042
## 1TAG_A  1.283  1.575  1.265  0.000  1.687  1.841
## 1GG2_A  1.612  1.777  1.737  1.687  0.000  1.879
## 1KJY_A  2.100  1.914  2.042  1.841  1.879  0.000</code></pre>
<div id="question-2" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-2" class="anchor"></a>Question:</h4>
<p>What effect does setting the <code>fit=TRUE</code> option have in the RMSD calculation? What would the results indicate if you set <code>fit=FALSE</code> or disparaged this option? HINT: Bio3D functions have various default options that will be used if the option is not explicitly specified by the user, see <code><a href="../../../reference/rmsd.html">help(rmsd)</a></code> for an example and note that the input options with an equals sign (e.g. <code>fit=FALSE</code>) have default values.</p>
</div>
</div>
<div id="exploring-example-data-for-the-transducin-heterotrimeric-g-protein" class="section level2">
<h2 class="hasAnchor">
<a href="#exploring-example-data-for-the-transducin-heterotrimeric-g-protein" class="anchor"></a>Exploring example data for the transducin heterotrimeric G Protein</h2>
<p>A number of example datasets are included with the Bio3D core package. The main purpose of including this data (which may be generated by the user by following the extended examples documented within the various Bio3D functions) is to allow users to more quickly appreciate the capabilities of functions that would otherwise require extensive data downloads before execution.</p>
<p>For a number of the examples in the current vignette we will utilize the included transducin dataset that contains over 50 publicly available structures. This dataset formed the basis of the work described in <span class="citation">(Yao and Grant 2013)</span> and we refer the motivated reader to this publication and references therein for extensive background information. Briefly, heterotrimeric G proteins are molecular switches that turn on and off intracellular signaling cascades in response to the activation of G protein coupled receptors (GPCRs). Receptor activation by extracellular stimuli promotes a cycle of GTP binding and hydrolysis on the G protein alpha subunit that leads to conformational rearrangements (i.e. internal structural changes) that activate multiple downstream effectors. The current dataset consists of transducin (including Gt and Gi/o) alpha subunit sequence and structural data and can be accessed with the command <code><a href="https://rdrr.io/r/base/attach.html">attach(transducin)</a></code>:</p>
<div class="sourceCode" id="cb23"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/attach.html">attach</a></span>(<span class="kw">transducin</span>)
</pre></div>
<div id="side-note-2" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#side-note-2" class="anchor"></a>Side-note:</h4>
<p>This dataset can be assembled from scratch with commands similar to those detailed in the next section and those listed in section 2.2. Also see <code><a href="../../../reference/example.data.html">help(example.data)</a></code> for a full description of this dataset’s contents.</p>
</div>
</div>
</div>
<div id="constructing-experimental-structure-ensembles-for-a-protein-family" class="section level1">
<h1 class="hasAnchor">
<a href="#constructing-experimental-structure-ensembles-for-a-protein-family" class="anchor"></a>Constructing Experimental Structure Ensembles for a Protein Family</h1>
<p>Comparing multiple structures of homologous proteins and carefully analyzing large multiple sequence alignments can help identify patterns of sequence and structural conservation and highlight conserved interactions that are crucial for protein stability and function <span class="citation">(Grant et al. 2007)</span>. Bio3D provides a useful framework for such studies and can facilitate the integration of sequence, structure and dynamics data in the analysis of protein evolution.</p>
<div id="finding-available-sets-of-similar-structures" class="section level2">
<h2 class="hasAnchor">
<a href="#finding-available-sets-of-similar-structures" class="anchor"></a>Finding Available Sets of Similar Structures</h2>
<p>In this tutorial, to collect available transducin crystal structures, we first use BLAST to query the PDB database to find similar sequences (and hence structures) to our chosen representative (PDB Id: 1TAG):</p>
<div class="sourceCode" id="cb24"><pre class="downlit">
<span class="kw">pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"1tag"</span>)
<span class="kw">seq</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbseq.html">pdbseq</a></span>(<span class="kw">pdb</span>)
<span class="kw">blast</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/blast.pdb.html">blast.pdb</a></span>(<span class="kw">seq</span>)
</pre></div>
<p>Examining the alignment scores and their associated E-values (with the function <code><a href="../../../reference/blast.pdb.html">plot.blast()</a></code>) indicates a sensible normalized score (-log(E-Value)) cutoff of 240 bits (Figure 2).</p>
<div class="sourceCode" id="cb25"><pre class="downlit">
<span class="co"># See Figure 2.</span>
<span class="kw">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/blast.pdb.html">plot.blast</a></span>(<span class="kw">blast</span>, cutoff=<span class="fl">240</span>)
</pre></div>
<pre><code>##   * Possible cutoff values:    242 6 
##             Yielding Nhits:    66 106 
## 
##   * Chosen cutoff value of:    240 
##             Yielding Nhits:    66</code></pre>
<div class="figure">
<img src="Bio3D_pca_files/figure-html/blasthits-1.png" alt="Summary of BLAST results for query 1tag against the PDB chain database" width="700"><p class="caption">
Summary of BLAST results for query 1tag against the PDB chain database
</p>
</div>
<p>We can then list a subset of our top hits, for example:</p>
<div class="sourceCode" id="cb27"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">hits</span><span class="op">$</span><span class="kw">hits</span>)
</pre></div>
<pre><code>##   pdb.id   acc      group
## 1 "1TAD_A" "1TAD_A" "1"  
## 2 "6OY9_A" "6OY9_A" "1"  
## 3 "3V00_A" "3V00_A" "1"  
## 4 "1FQJ_A" "1FQJ_A" "1"  
## 5 "6OYA_A" "6OYA_A" "1"  
## 6 "1GOT_A" "1GOT_A" "1"</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">hits</span><span class="op">$</span><span class="kw">pdb.id</span>)
</pre></div>
<pre><code>## [1] "1TAD_A" "6OY9_A" "3V00_A" "1FQJ_A" "6OYA_A" "1GOT_A"</code></pre>
<div id="sidenote" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#sidenote" class="anchor"></a>Sidenote:</h4>
<p>The function <code><a href="../../../reference/pdb.annotate.html">pdb.annotate()</a></code> can fetch detailed information about the corresponding structures (e.g. title, experimental method, resolution, ligand name(s), citation, etc.). For example:</p>
<div class="sourceCode" id="cb31"><pre class="downlit">
<span class="kw">anno</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdb.annotate.html">pdb.annotate</a></span>(<span class="kw">hits</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">anno</span>[, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"resolution"</span>, <span class="st">"ligandId"</span>, <span class="st">"citation"</span>)])
</pre></div>
<pre><code>##        resolution       ligandId                         citation
## 1TAD_A        1.7 ALF,CA,CAC,GDP      Sondek et al. Nature (1994)
## 6OY9_A        3.9           &lt;NA&gt;       Gao et al. Mol.Cell (2019)
## 3V00_A        2.9            GDP Singh et al. Biochemistry (2012)
## 1FQJ_A       2.02     ALF,GDP,MG        Slep et al. Nature (2001)
## 6OYA_A        3.3           &lt;NA&gt;       Gao et al. Mol.Cell (2019)
## 1GOT_A        2.0            MSE   Lambright et al. Nature (1996)</code></pre>
</div>
</div>
<div id="multiple-sequence-alignment" class="section level2">
<h2 class="hasAnchor">
<a href="#multiple-sequence-alignment" class="anchor"></a>Multiple Sequence Alignment</h2>
<p>Next we download the complete list of structures from the PDB (with function <code><a href="../../../reference/get.pdb.html">get.pdb()</a></code>), and use function <code><a href="../../../reference/pdbsplit.html">pdbsplit()</a></code> to split the structures into separate chains and store them for subsequent access. Finally, function <code><a href="../../../reference/pdbaln.html">pdbaln()</a></code> will extract the sequence of each structure and perform a multiple sequence alignment to determine residue-residue correspondences (<strong>NOTE:</strong> requires external program <em>MUSCLE</em> be in search path for executables):</p>
<div class="sourceCode" id="cb33"><pre class="downlit">
<span class="co"># Download PDBs and split by chain ID</span>
<span class="kw">files</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/get.pdb.html">get.pdb</a></span>(<span class="kw">hits</span>, path=<span class="st">"raw_pdbs"</span>, split = <span class="fl">TRUE</span>)

<span class="co"># Extract and align sequences</span>
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbaln.html">pdbaln</a></span>(<span class="kw">files</span>)
</pre></div>
<p>You can now inspect the alignment (the automatically generated “aln.fa” file) with your favorite alignment viewer (we recommend SEAVIEW, available from: <a href="http://pbil.univ-lyon1.fr/software/seaview.html">http://pbil.univ-lyon1.fr/software/seaview.html</a>).</p>
<div id="side-note-3" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#side-note-3" class="anchor"></a>Side-note:</h4>
<p>You may find a number of structures with missing residues (i.e. gaps in the alignment) at sites of particular interest to you. If this is the case you may consider removing these structures from your hit list and generating a smaller, but potentially higher quality, dataset for further exploration.</p>
</div>
<div id="question-3" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-3" class="anchor"></a>Question:</h4>
<p>How could you automatically identify gap positions in your alignment? HINT: try the command <code><a href="https://rdrr.io/r/utils/help.search.html">help.search("gap", package="bio3d")</a></code>.</p>
</div>
</div>
</div>
<div id="comparative-structure-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#comparative-structure-analysis" class="anchor"></a>Comparative Structure Analysis</h1>
<p>The detailed comparison of homologous protein structures can be used to infer pathways for evolutionary adaptation and, at closer evolutionary distances, mechanisms for conformational change. Bio3D employs both conventional methods for structural analysis (alignment, RMSD, difference distance matrix analysis, etc.) as well as refined structural superposition and principal component analysis (PCA) to facilitate comparative structure analysis.</p>
<div id="structure-superposition" class="section level2">
<h2 class="hasAnchor">
<a href="#structure-superposition" class="anchor"></a>Structure Superposition</h2>
<p>Conventional structural superposition of proteins minimizes the root mean square difference between their full set of equivalent residues. This can be performed with Bio3D functions <code><a href="../../../reference/pdbfit.html">pdbfit()</a></code> and <code><a href="../../../reference/fit.xyz.html">fit.xyz()</a></code> as outlined previously. However, for certain applications such a superposition procedure can be inappropriate. For example, in the comparison of a multi-domain protein that has undergone a hinge-like rearrangement of its domains, standard all atom superposition would result in an underestimate of the true atomic displacement by attempting superposition over all domains (whole structure superposition). A more appropriate and insightful superposition would be anchored at the most invariant region and hence more clearly highlight the domain rearrangement (sub-structure superposition).</p>
<p>The Bio3D <code><a href="../../../reference/core.find.html">core.find()</a></code> function implements an iterated superposition procedure, where residues displaying the largest positional differences are identified and excluded at each round. The function returns an ordered list of excluded residues, from which the user can select a subset of ’core’ residues upon which superposition can be based.</p>
<div class="sourceCode" id="cb34"><pre class="downlit">
<span class="kw">core</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/core.find.html">core.find</a></span>(<span class="kw">pdbs</span>)
</pre></div>
<p>The <code><a href="../../../reference/plot.core.html">plot.core()</a></code> and <code><a href="../../../reference/print.core.html">print.core()</a></code> functions allow one to further examine the output of the <code><a href="../../../reference/core.find.html">core.find()</a></code> procedure (see below and Figure 3).</p>
<div class="sourceCode" id="cb35"><pre class="downlit">
<span class="co"># See Figure 3.</span>
<span class="kw">col</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"black"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw">core</span><span class="op">$</span><span class="kw">volume</span>))
<span class="kw">col</span>[<span class="kw">core</span><span class="op">$</span><span class="kw">volume</span><span class="op">&lt;</span><span class="fl">2</span>]<span class="op">=</span><span class="st">"pink"</span>; <span class="kw">col</span>[<span class="kw">core</span><span class="op">$</span><span class="kw">volume</span><span class="op">&lt;</span><span class="fl">1</span>]<span class="op">=</span><span class="st">"red"</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">core</span>, col=<span class="kw">col</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_pca_files/figure-html/unnamed-chunk-16-1.png" alt="Identification of core residues" width="700"><p class="caption">
Identification of core residues
</p>
</div>
<p>The <code><a href="../../../reference/print.core.html">print.core()</a></code> function also returns <code>atom</code> and <code>xyz</code> indices similar to those returned from the <code><a href="../../../reference/atom.select.html">atom.select()</a></code> function. Below we use these indices for core superposition and to write a quick PDB file for viewing in a molecular graphics program such as VMD (Figure 4).</p>
<div class="sourceCode" id="cb36"><pre class="downlit">
<span class="kw">core.inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">core</span>, vol=<span class="fl">1.0</span>)
</pre></div>
<pre><code>## # 88 positions (cumulative volume &lt;= 1 Angstrom^3) 
##    start end length
## 1     32  52     21
## 2    195 195      1
## 3    216 226     11
## 4    239 239      1
## 5    242 247      6
## 6    260 274     15
## 7    279 279      1
## 8    282 283      2
## 9    295 304     10
## 10   317 336     20</code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit">
<span class="fu"><a href="../../../reference/write.pdb.html">write.pdb</a></span>(xyz=<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>[<span class="fl">1</span>,<span class="kw">core.inds</span><span class="op">$</span><span class="kw">xyz</span>], file=<span class="st">"quick_core.pdb"</span>)
</pre></div>
<div class="figure">
<img src="figures/core.png" alt=""><p class="caption">The most structural invariant core positions in the transducin family</p>
</div>
<p>We can now superpose all structures on the selected core indices with the <code><a href="../../../reference/fit.xyz.html">fit.xyz()</a></code> or <code><a href="../../../reference/pdbfit.html">pdbfit()</a></code> function.</p>
<div class="sourceCode" id="cb39"><pre class="downlit">
<span class="kw">xyz</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbfit.html">pdbfit</a></span>( <span class="kw">pdbs</span>, <span class="kw">core.inds</span> )
</pre></div>
<p>The above command performs the actual superposition and stores the new coordinates in the matrix object <code>xyz</code>.</p>
<div id="side-note-4" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#side-note-4" class="anchor"></a>Side-note:</h4>
<p>By providing an extra <code>outpath="somedir"</code> argument to <code>pdbfit</code> the superposed structures can be output for viewing (in this case to the local directory <code>somedir</code> which you can obviously change). These fitted structures can then be viewed in your favorite molecular graphics program (Figure 5).</p>
<div class="figure">
<img src="figures/fit.png" alt=""><p class="caption">Structure ensemble of transducin family superposed based on core positions</p>
</div>
</div>
</div>
<div id="standard-structural-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#standard-structural-analysis" class="anchor"></a>Standard Structural Analysis</h2>
<p>Bio3D contains functions to perform standard structural analysis, such as root mean-square deviation (RMSD), root mean-square fluctuation (RMSF), secondary structure, dihedral angles, difference distance matrices etc. The current section provides a brief exposure to using Bio3D in this vein. However, do feel free to skip ahead to the arguably more interesting section on PCA analysis.</p>
<div id="root-mean-square-deviation-rmsd" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#root-mean-square-deviation-rmsd" class="anchor"></a>Root mean square deviation (RMSD):</h4>
<p>RMSD is a standard measure of structural distance between coordinate sets. Here we examine the pairwise RMSD values and cluster our structures based on these values:</p>
<div class="sourceCode" id="cb40"><pre class="downlit">
<span class="kw">rd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/rmsd.html">rmsd</a></span>(<span class="kw">xyz</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span>(<span class="kw">rd</span>, breaks=<span class="fl">40</span>, xlab=<span class="st">"RMSD (Å)"</span>, main=<span class="st">"Histogram of RMSD"</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_pca_files/figure-html/unnamed-chunk-19-1.png" alt="Histogram of RMSD among transducin structures" width="700"><p class="caption">
Histogram of RMSD among transducin structures
</p>
</div>
<div class="sourceCode" id="cb41"><pre class="downlit">
<span class="co"># RMSD clustering</span>
<span class="kw">hc.rd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html">hclust</a></span>(<span class="fu"><a href="https://rdrr.io/r/stats/dist.html">as.dist</a></span>(<span class="kw">rd</span>))
</pre></div>
<p>The result can be illustrated as a simple dendrogram with the command:</p>
<div class="sourceCode" id="cb42"><pre class="downlit">
<span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span>), <span class="fl">1</span>, <span class="fl">6</span>)
<span class="fu"><a href="../../../reference/hclustplot.html">hclustplot</a></span>(<span class="kw">hc.rd</span>, colors=<span class="kw">annotation</span>[, <span class="st">"color"</span>], labels=<span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span>, cex=<span class="fl">0.5</span>,
          ylab=<span class="st">"RMSD (Å)"</span>, main=<span class="st">"RMSD Cluster Dendrogram"</span>, fillbox=<span class="fl">FALSE</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_pca_files/figure-html/unnamed-chunk-21-1.png" alt="RMSD clustering of transducin structures" width="672"><p class="caption">
RMSD clustering of transducin structures
</p>
</div>
</div>
<div id="question-4" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-4" class="anchor"></a>Question:</h4>
<p>How many structure groups/clusters do we have according to this clustering? How would determine which structures are assigned to which cluster? HINT: See <code><a href="https://rdrr.io/r/stats/cutree.html">help(cutree)</a></code>.</p>
</div>
<div id="question-5" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-5" class="anchor"></a>Question:</h4>
<p>What kind of plot would the command <code><a href="https://rdrr.io/r/stats/heatmap.html">heatmap(rd)</a></code> produce? How would you label this plot with PDB codes? HINT: labCol and labRow.</p>
</div>
<div id="root-mean-squared-fluctuations-rmsf" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#root-mean-squared-fluctuations-rmsf" class="anchor"></a>Root mean squared fluctuations (RMSF):</h4>
<p>RMSF is another often used measure of conformational variance. The Bio3D <code><a href="../../../reference/rmsf.html">rmsf()</a></code> function returns a vector of atom-wise (or residue-wise) variance instead of a single numeric value. The below sequence of commands returns the indices for gap containing positions, which we then exclude from subsequent RMSF calculation:</p>
<div class="sourceCode" id="cb43"><pre class="downlit">
<span class="co"># Ignore gap containing positions</span>
<span class="kw">gaps.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/gap.inspect.html">gap.inspect</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">ali</span>)
<span class="kw">gaps.pos</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/gap.inspect.html">gap.inspect</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>)

<span class="co"># Tailor the PDB structure to exclude gap positions for SSE annotation</span>
<span class="kw">id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"1TAG"</span>, <span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span>)
<span class="kw">inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>, resno = <span class="kw">pdbs</span><span class="op">$</span><span class="kw">resno</span>[<span class="kw">id</span>, <span class="kw">gaps.res</span><span class="op">$</span><span class="kw">f.inds</span>])
<span class="kw">ref.pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/trim.html">trim.pdb</a></span>(<span class="kw">pdb</span>, inds = <span class="kw">inds</span>)

<span class="co"># Plot RMSF with SSE annotation and labeled with residue numbers (Figure 8.)</span>
<span class="kw">rf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/rmsf.html">rmsf</a></span>(<span class="kw">xyz</span>[, <span class="kw">gaps.pos</span><span class="op">$</span><span class="kw">f.inds</span>])
<span class="fu"><a href="../../../reference/plot.bio3d.html">plot.bio3d</a></span>(<span class="kw">rf</span>, resno=<span class="kw">ref.pdb</span>, sse=<span class="kw">ref.pdb</span>, ylab=<span class="st">"RMSF (Å)"</span>, 
           xlab=<span class="st">"Residue No."</span>, typ=<span class="st">"l"</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_pca_files/figure-html/unnamed-chunk-22-1.png" alt="RMSF plot" width="700"><p class="caption">
RMSF plot
</p>
</div>
</div>
<div id="torsiondihedral-analysis" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#torsiondihedral-analysis" class="anchor"></a>Torsion/Dihedral analysis:</h4>
<p>The conformation of a polypeptide or nucleotide chain can be usefully described in terms of angles of internal rotation around its constituent bonds.</p>
<div class="sourceCode" id="cb44"><pre class="downlit">
<span class="kw">tor</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/torsion.pdb.html">torsion.pdb</a></span>(<span class="kw">pdb</span>)

<span class="co"># Basic Ramachandran plot (Figure 9)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">tor</span><span class="op">$</span><span class="kw">phi</span>, <span class="kw">tor</span><span class="op">$</span><span class="kw">psi</span>, xlab=<span class="st">"phi"</span>, ylab=<span class="st">"psi"</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_pca_files/figure-html/unnamed-chunk-23-1.png" alt="Basic Ramachandran plot" width="700"><p class="caption">
Basic Ramachandran plot
</p>
</div>
<p>Lets compare the Calpha atom based pseudo-torsion angles between two structures, the inactive (PDB Id: 1TAG) and active (PDB Id: 1TND) transducin alpha subunit:</p>
<div class="sourceCode" id="cb45"><pre class="downlit">
<span class="co"># Locate the two structures in pdbs</span>
<span class="kw">ind.a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"1TAG_A"</span>, <span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span>)
<span class="kw">ind.b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"1TND_B"</span>, <span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span>)

<span class="co"># Exclude gaps in the two structures to make them comparable</span>
<span class="kw">gaps.xyz2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/gap.inspect.html">gap.inspect</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>[<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">ind.a</span>, <span class="kw">ind.b</span>), ])
<span class="kw">a.xyz</span> <span class="op">&lt;-</span> <span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>[<span class="kw">ind.a</span>, <span class="kw">gaps.xyz2</span><span class="op">$</span><span class="kw">f.inds</span>]
<span class="kw">b.xyz</span> <span class="op">&lt;-</span> <span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>[<span class="kw">ind.b</span>, <span class="kw">gaps.xyz2</span><span class="op">$</span><span class="kw">f.inds</span>]

<span class="co"># Compare CA based pseudo-torsion angles between the two structures</span>
<span class="kw">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/torsion.xyz.html">torsion.xyz</a></span>(<span class="kw">a.xyz</span>, atm.inc=<span class="fl">1</span>)
<span class="kw">b</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/torsion.xyz.html">torsion.xyz</a></span>(<span class="kw">b.xyz</span>, atm.inc=<span class="fl">1</span>)
<span class="kw">d.ab</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/wrap.tor.html">wrap.tor</a></span>(<span class="kw">a</span><span class="op">-</span><span class="kw">b</span>)
<span class="kw">d.ab</span>[<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="kw">d.ab</span>)] <span class="op">&lt;-</span> <span class="fl">0</span>

<span class="co"># Plot results with SSE annotation</span>
<span class="fu"><a href="../../../reference/plot.bio3d.html">plot.bio3d</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span>(<span class="kw">d.ab</span>), resno=<span class="kw">pdb</span>, sse=<span class="kw">pdb</span>, typ=<span class="st">"h"</span>, xlab=<span class="st">"Residue No."</span>, 
           ylab = <span class="st">"Difference Angle"</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_pca_files/figure-html/unnamed-chunk-24-1.png" alt="Torsion angle difference between structures in inactive GDP bound (1tag) and active GTP analogue bound (1tnd) states" width="700"><p class="caption">
Torsion angle difference between structures in inactive GDP bound (1tag) and active GTP analogue bound (1tnd) states
</p>
</div>
</div>
<div id="difference-distance-matrix-analysis-ddm" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#difference-distance-matrix-analysis-ddm" class="anchor"></a>Difference distance matrix analysis (DDM)</h4>
<p>Distance matrices can be calculated with the function <code><a href="../../../reference/dm.html">dm()</a></code> and contact maps with the function <code><a href="../../../reference/cmap.html">cmap()</a></code>. In the example below we calculate the difference distance matrix by simply subtracting one distance matrix from another. Note the vectorized nature of the this calculation (i.e. we do not have to explicitly iterate through each element of the matrix):</p>
<div class="sourceCode" id="cb46"><pre class="downlit">
<span class="kw">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/dm.html">dm.xyz</a></span>(<span class="kw">a.xyz</span>)
<span class="kw">b</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/dm.html">dm.xyz</a></span>(<span class="kw">b.xyz</span>)

<span class="fu"><a href="../../../reference/plot.dmat.html">plot.dmat</a></span>( (<span class="kw">a</span> <span class="op">-</span> <span class="kw">b</span>), nlevels=<span class="fl">10</span>, grid.col=<span class="st">"gray"</span>, xlab=<span class="st">"1tag"</span>, ylab=<span class="st">"1tnd"</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_pca_files/figure-html/unnamed-chunk-25-1.png" alt="Difference of distance matrices between structures in GDP(1tag) and GTP(1tnd) nucleotide states" width="528"><p class="caption">
Difference of distance matrices between structures in GDP(1tag) and GTP(1tnd) nucleotide states
</p>
</div>
</div>
<div id="question-6" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-6" class="anchor"></a>Question:</h4>
<p>Can you think of the pros and cons of these different analysis methods?</p>
</div>
</div>
</div>
<div id="principal-component-analysis-pca" class="section level1">
<h1 class="hasAnchor">
<a href="#principal-component-analysis-pca" class="anchor"></a>Principal Component Analysis (PCA)</h1>
<p>Following core identification and subsequent superposition, PCA can be employed to examine the relationship between different structures based on their equivalent residues. The application of PCA to both distributions of experimental structures and molecular dynamics trajectories, along with its ability to provide considerable insight into the nature of conformational differences is also discussed in the molecular dynamics trajectory analysis vignette.</p>
<p>Briefly, the resulting principal components (orthogonal eigenvectors) describe the axes of maximal variance of the distribution of structures. Projection of the distribution onto the space defined by the largest principal components results in a lower dimensional representation of the structural dataset. The percentage of the total mean square displacement (or variance) of atom positional fluctuations captured in each dimension is characterized by their corresponding eigenvalue. Experience suggests that 3–5 dimensions are often sufficient to capture over 70 percent of the total variance in a given family of structures. Thus, a handful of principal components are sufficient to provide a useful description while still retaining most of the variance in the original distribution <span class="citation">(Grant et al. 2006)</span>.</p>
<p>The below command excludes the gap positions identified above from the PCA (note you can also simply run <code><a href="../../../reference/pca.html">pca(xyz, rm.gaps=TRUE)</a></code>).</p>
<div class="sourceCode" id="cb47"><pre class="downlit">
<span class="co"># Do PCA</span>
<span class="kw">pc.xray</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pca.xyz.html">pca.xyz</a></span>(<span class="kw">xyz</span>[, <span class="kw">gaps.pos</span><span class="op">$</span><span class="kw">f.inds</span>])
<span class="kw">pc.xray</span>
</pre></div>
<pre><code>## 
## Call:
##   pca.xyz(xyz = xyz[, gaps.pos$f.inds])
## 
## Class:
##   pca
## 
## Number of eigenvalues:
##   915
## 
##         Eigenvalue Variance Cumulative
##    PC 1    259.521   49.794     49.794
##    PC 2     81.302   15.599     65.393
##    PC 3     40.095    7.693     73.086
##    PC 4     29.701    5.699     78.785
##    PC 5     24.048    4.614     83.399
##    PC 6     19.611    3.763     87.162
## 
##    (Obtained from 53 conformers with 915 xyz input values).
## 
## + attr: L, U, z, au, sdev, mean, call</code></pre>
<div id="question-7" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-7" class="anchor"></a>Question:</h4>
<p>Why is the input to function <code><a href="../../../reference/pca.xyz.html">pca.xyz()</a></code> given as <code>xyz</code> rather than <code>pdbs$xyz</code>?</p>
</div>
<div id="question-8" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-8" class="anchor"></a>Question:</h4>
<p>Why would you need superposition before using <code>pca.xyz</code> but not need it for <code>pca.tor</code>?</p>
<p>A quick overview of the results of <code><a href="../../../reference/pca.xyz.html">pca.xyz()</a></code> can be obtained by calling <code><a href="../../../reference/pca.xyz.html">print.pca()</a></code> and <code><a href="../../../reference/plot.pca.html">plot.pca()</a></code> (Figure 12).</p>
<div class="sourceCode" id="cb49"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">pc.xray</span>, col=<span class="kw">annotation</span>[, <span class="st">"color"</span>])
</pre></div>
<div class="figure">
<img src="Bio3D_pca_files/figure-html/unnamed-chunk-27-1.png" alt="Overview of PCA results for transducin crystallographic structures" width="700"><p class="caption">
Overview of PCA results for transducin crystallographic structures
</p>
</div>
<p>Setting the option <code>pc.axes</code> will allow for single plots to also be produced in this way, e.g.:</p>
<div class="sourceCode" id="cb50"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">pc.xray</span>, pc.axes=<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, col=<span class="kw">annotation</span>[, <span class="st">"color"</span>])
</pre></div>
<div class="figure">
<img src="Bio3D_pca_files/figure-html/unnamed-chunk-28-1.png" alt="Results of setting pc.axes=1:2 in plot.pca() function" width="700"><p class="caption">
Results of setting pc.axes=1:2 in plot.pca() function
</p>
</div>
<p>One can then use the <code><a href="https://rdrr.io/r/graphics/identify.html">identify()</a></code> function to label and individual points.</p>
<div class="sourceCode" id="cb51"><pre class="downlit">
<span class="co"># Left-click on a point to label and right-click to end</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/identify.html">identify</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>], labels=<span class="fu"><a href="../../../reference/basename.pdb.html">basename.pdb</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span>))
</pre></div>
<p>We can also call <code><a href="../../../reference/plot.bio3d.html">plot.bio3d()</a></code> to examine the contribution of each residue to the first three principal components with the following commands (Figure 13).</p>
<div class="sourceCode" id="cb52"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(mfrow = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">3</span>, <span class="fl">1</span>), cex = <span class="fl">0.75</span>, mar = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">3</span>, <span class="fl">4</span>, <span class="fl">1</span>, <span class="fl">1</span>))
<span class="fu"><a href="../../../reference/plot.bio3d.html">plot.bio3d</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">au</span>[,<span class="fl">1</span>], resno=<span class="kw">ref.pdb</span>, sse=<span class="kw">ref.pdb</span>, ylab=<span class="st">"PC1"</span>)
<span class="fu"><a href="../../../reference/plot.bio3d.html">plot.bio3d</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">au</span>[,<span class="fl">2</span>], resno=<span class="kw">ref.pdb</span>, sse=<span class="kw">ref.pdb</span>, ylab=<span class="st">"PC2"</span>)
<span class="fu"><a href="../../../reference/plot.bio3d.html">plot.bio3d</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">au</span>[,<span class="fl">3</span>], resno=<span class="kw">ref.pdb</span>, sse=<span class="kw">ref.pdb</span>, ylab=<span class="st">"PC3"</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_pca_files/figure-html/unnamed-chunk-31-1.png" alt="Contribution of each residue to the first three principal components" width="576"><p class="caption">
Contribution of each residue to the first three principal components
</p>
</div>
<p>The plots in Figures 12-14 display the relationships between different conformers, highlight positions responsible for the major differences between structures and enable the interpretation and characterization of multiple interconformer relationships.</p>
<p>To further aid interpretation, a PDB format trajectory can be produced that interpolates between the most dissimilar structures in the distribution along a given principal component. This involves dividing the difference between the conformers into a number of evenly spaced steps along the principal components, forming the frames of the trajectory. Such trajectories can be directly visualized in a molecular graphics program, such as VMD <span class="citation">(Humphrey 1996)</span> (Figure 15). Furthermore, the PCA results can be compared to those from simulations (see the molecular dynamics and normal mode analysis vignettes), as well as guiding dynamic network analysis, being analyzed for possible dynamic domains (with the Bio3D <code><a href="../../../reference/geostas.html">geostas()</a></code> function from the <strong>bio3d.geostas</strong> package), domain and shear movements with the DynDom package <span class="citation">(Hayward and Berendsen 1998)</span>, or used as initial seed structures for reaction path refinement methods such as Conjugate Peak Refinement <span class="citation">(Fischer and Karplus 1992)</span>.</p>
<div class="sourceCode" id="cb53"><pre class="downlit">
<span class="co"># See Figure 15</span>
<span class="fu"><a href="../../../reference/mktrj.html">mktrj.pca</a></span>(<span class="kw">pc.xray</span>, pc=<span class="fl">1</span>, file=<span class="st">"pc1.pdb"</span>)
</pre></div>
<div class="figure">
<img src="figures/pca.png" alt=""><p class="caption">Interpolated structures along PC1 produced by the mktrj.pca() function</p>
</div>
</div>
<div id="conformer-clustering-in-pc-space" class="section level2">
<h2 class="hasAnchor">
<a href="#conformer-clustering-in-pc-space" class="anchor"></a>Conformer Clustering in PC Space</h2>
<p>Clustering structures in PC space can often enable one to focus on the relationships between individual structures in terms of their major structural displacements, with a controllable level of dynamic details (via specifying the number of PCs used in the clustering). For example, with clustering along PCs 1 and 2, we can investigate how the X-ray structures of transducin relate to each other with respect to the major conformation change that covers over 65% of their structural variance (See Figures 12). This can reveal functional relationships that are often hard to find by conventional pairwise methods such as the RMSD clustering detailed previously (Figure 7). For example in the PC1-PC2 plane, the inactive “GDP” structures (green points in Figure 12) are further split into two sub-groups (Figures 16 and 17). The bottom-right sub-group (blue) exclusively correspond to the structures complexed with GDP dissociation inhibitor (GDI). This functionaly distinct state is clearly evident in the PC plot and clustering dendrogram that can be generated with the following commands:</p>
<div class="sourceCode" id="cb54"><pre class="downlit">
<span class="kw">hc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html">hclust</a></span>(<span class="fu"><a href="https://rdrr.io/r/stats/dist.html">dist</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>]))
<span class="kw">grps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cutree.html">cutree</a></span>(<span class="kw">hc</span>, h=<span class="fl">30</span>)
<span class="kw">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"red"</span>, <span class="st">"green"</span>, <span class="st">"blue"</span>)[<span class="kw">grps</span>]
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">pc.xray</span>, pc.axes=<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, col=<span class="kw">cols</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_pca_files/figure-html/unnamed-chunk-33-1.png" alt="Clustering based on PC1-PC2. Note the new blue cluster are all GDI bound forms (see text for details)" width="480"><p class="caption">
Clustering based on PC1-PC2. Note the new blue cluster are all GDI bound forms (see text for details)
</p>
</div>
<div class="sourceCode" id="cb55"><pre class="downlit">
<span class="co"># Dendrogram plot</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">cols</span>) <span class="op">&lt;-</span> <span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span>
<span class="fu"><a href="../../../reference/hclustplot.html">hclustplot</a></span>(<span class="kw">hc</span>, colors=<span class="kw">cols</span>, ylab=<span class="st">"Distance in PC Space"</span>, main=<span class="st">"PC1-2"</span>, cex=<span class="fl">0.5</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_pca_files/figure-html/unnamed-chunk-34-1.png" alt="Clustering based on PC1-PC2" width="672"><p class="caption">
Clustering based on PC1-PC2
</p>
</div>
<div id="sidenote-1" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#sidenote-1" class="anchor"></a>Sidenote:</h4>
<p>On the PC1 vs PC2 conformer plot in Figure 16 you can interactively identify and label individual structures by using the <code><a href="https://rdrr.io/r/graphics/identify.html">identify()</a></code> function clicking with your mouse (left to select, right to end). In this particular case the command would be:</p>
<div class="sourceCode" id="cb56"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/identify.html">identify</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">1</span>], <span class="kw">pc.xray</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">2</span>], labels=<span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span>)
</pre></div>
</div>
<div id="question-9" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-9" class="anchor"></a>Question:</h4>
<p>Which clustering appears to be most informative, that based on RMSD or that based on PCA? Why might this be the case? HINT: It can be useful to think of PCA as a filter for large scale conformational changes.</p>
</div>
</div>
</div>
<div id="where-to-next" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#where-to-next" class="anchor"></a>Where to Next</h1>
<p>If you have read this far, congratulations! We are ready to have some fun and move to <a href="http://thegrantlab.org/bio3d">other package vignettes</a> that describe more interesting analysis including Correlation Network Analysis (where we will build and dissect dynamic networks form different correlated motion data), enhanced methods for Normal Mode Analysis (where we will explore the dynamics of large protein families and superfamilies using predictive calculations), and advanced Comparative Structure Analysis (where we will mine available experimental data and supplement it with simulation results to map the conformational dynamics and coupled motions of proteins).</p>
</div>
<div id="document-details" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#document-details" class="anchor"></a>Document Details</h1>
<p>This document is available from the <a href="http://thegrantlab.org/bio3d">Bio3D website</a> in R markdown, PDF, and HTML formats. All code can be extracted and automatically executed to generate Figures and/or the PDF with the following commands:</p>
<div class="sourceCode" id="cb57"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://github.com/rstudio/rmarkdown">rmarkdown</a></span>)
<span class="fu"><a href="https://rdrr.io/pkg/rmarkdown/man/render.html">render</a></span>(<span class="st">"Bio3D_pca.Rmd"</span>, <span class="st">"all"</span>)
</pre></div>
</div>
<div id="information-about-the-current-bio3d-session" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#information-about-the-current-bio3d-session" class="anchor"></a>Information About the Current Bio3D Session</h1>
<div class="sourceCode" id="cb58"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()
</pre></div>
<pre><code>## R version 3.6.0 (2019-04-26)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS  10.15.5
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] bio3d_2.4-1.9000
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.5         XML_3.99-0.3       digest_0.6.25      crayon_1.3.4      
##  [5] rprojroot_1.3-2    assertthat_0.2.1   bitops_1.0-6       grid_3.6.0        
##  [9] R6_2.4.1           backports_1.1.8    magrittr_1.5       evaluate_0.14     
## [13] highr_0.8          stringi_1.4.6      rlang_0.4.6.9000   fs_1.4.2          
## [17] rmarkdown_2.3.2    pkgdown_1.5.1.9000 desc_1.2.0         tools_3.6.0       
## [21] stringr_1.4.0      RCurl_1.98-1.2     parallel_3.6.0     yaml_2.2.1        
## [25] xfun_0.15          compiler_3.6.0     memoise_1.1.0      htmltools_0.5.0   
## [29] knitr_1.29</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references hanging-indent">
<div id="ref-cpr">
<p>Fischer, S., and M. Karplus. 1992. “Conjugate Peak Refinement: An Algorithm for Finding Reaction Paths and Accurate Transition States in Systems with Many Degrees of Freedom.” <em>Chem. Phys. Lett</em> 194: 252–61.</p>
</div>
<div id="ref-grant07">
<p>Grant, B. J., A. J. Mccammon, L. S. D. Caves, and R. A. Cross. 2007. “Multivariate Analysis of Conserved Sequence-Structure Relationships in Kinesins: Coupling of the Active Site and a Tubulin-Binding Sub-Domain.” <em>J. Mol. Biol</em> 5: 1231–48.</p>
</div>
<div id="ref-grant06">
<p>Grant, B. J., A. P. D. C Rodrigues, K. M. Elsawy, A. J. Mccammon, and L. S. D. Caves. 2006. “Bio3d: An R Package for the Comparative Analysis of Protein Structures.” <em>Bioinformatics</em> 22: 2695–6.</p>
</div>
<div id="ref-dyndom">
<p>Hayward, S., and H. Berendsen. 1998. “Systematic Analysis of Domain Motions in Proteins from Conformational Change: New Results on Citrate Synthase and T4 Lysozyme.” <em>Proteins</em> 30: 144–54.</p>
</div>
<div id="ref-vmd">
<p>Humphrey, et al., W. 1996. “VMD: Visual Molecular Dynamics.” <em>J. Mol. Graph</em> 14: 33–38.</p>
</div>
<div id="ref-yao13">
<p>Yao, X. Q., and B. J. Grant. 2013. “Domain-Opening and Dynamic Coupling in the Alpha-Subunit of Heterotrimeric G Proteins.” <em>Biophys. J</em> 105: L08–10.</p>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>The latest version of the package, full documentation and further vignettes (including detailed installation instructions) can be obtained from the main Bio3D website: <a href="http://thegrantlab.org/bio3d/">http://thegrantlab.org/bio3d/</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>This vignette contains executable examples, see <code><a href="https://rdrr.io/r/utils/vignette.html">help(vignette)</a></code> for further details.<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="tidyverse">
  <p>Authored by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="author">
  <p>
    Built with ❤️ and <a href="https://pkgdown.r-lib.org">pkgdown</a> © 2020.
  </p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '1270c4079b26a138795263974bbf302d',
    indexName: 'tidyverse',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
