<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Atom Selection from PDB and PRMTOP Structure Objects — atom.select • Bio3D</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Atom Selection from PDB and PRMTOP Structure Objects — atom.select" />
<meta property="og:description" content="Return the &amp;#8216;atom&amp;#8217; and &amp;#8216;xyz&amp;#8217; coordinate indices of
  &amp;#8216;pdb&amp;#8217; or &amp;#8216;prmtop&amp;#8217; structure objects corresponding to the
  intersection of a hierarchical selection." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Bio3D</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4-1.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-info"></span>
     
    Getting Started
  </a>
</li>
<li>
  <a href="../index.html">
    <span class="fa fa-book"></span>
     
    Tutorials
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Functions
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../CODE_OF_CONDUCT.html">
    <span class="fa fa-newspaper-o"></span>
     
    News &amp; Updates
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Atom Selection from PDB and PRMTOP Structure Objects</h1>
    
    <div class="hidden name"><code>atom.select.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Return the &#8216;atom&#8217; and &#8216;xyz&#8217; coordinate indices of
  &#8216;pdb&#8217; or &#8216;prmtop&#8217; structure objects corresponding to the
  intersection of a hierarchical selection.</p>
    </div>

    <pre class="usage"><span class='fu'>atom.select</span>(<span class='kw'>...</span>)

<span class='co'># S3 method for pdb</span>
<span class='fu'>atom.select</span>(<span class='kw'>pdb</span>, string = <span class='kw'>NULL</span>,
                          type  = <span class='kw'>NULL</span>, eleno = <span class='kw'>NULL</span>, elety = <span class='kw'>NULL</span>,
                          resid = <span class='kw'>NULL</span>, chain = <span class='kw'>NULL</span>, resno = <span class='kw'>NULL</span>,
                          insert = <span class='kw'>NULL</span>, segid = <span class='kw'>NULL</span>, 
                          operator = <span class='st'>"AND"</span>, inverse = <span class='fl'>FALSE</span>,
                          value = <span class='fl'>FALSE</span>, verbose=<span class='fl'>FALSE</span>, <span class='kw'>...</span>)
<span class='co'># S3 method for pdbs</span>
<span class='fu'>atom.select</span>(<span class='kw'>pdbs</span>, string = <span class='kw'>NULL</span>, 
                           resno = <span class='kw'>NULL</span>, chain = <span class='kw'>NULL</span>, resid = <span class='kw'>NULL</span>,
                           operator=<span class='st'>"AND"</span>, inverse = <span class='fl'>FALSE</span>,
                           value = <span class='fl'>FALSE</span>, verbose=<span class='fl'>FALSE</span>, <span class='kw'>...</span>)
<span class='co'># S3 method for mol2</span>
<span class='fu'>atom.select</span>(<span class='kw'>mol</span>, string=<span class='kw'>NULL</span>,
                           eleno = <span class='kw'>NULL</span>, elena = <span class='kw'>NULL</span>, elety = <span class='kw'>NULL</span>,
                           resid = <span class='kw'>NULL</span>, chain = <span class='kw'>NULL</span>, resno = <span class='kw'>NULL</span>,
                           statbit = <span class='kw'>NULL</span>,
			   operator = "AND", inverse = FALS<span class='st'>"AND"</span><span class='fl'>FALSE</span>
                           value = <span class='fl'>FALSE</span>, verbose=<span class='fl'>FALSE</span>,  <span class='kw'>...</span>)

<span class='co'># S3 method for prmtop</span>
<span class='fu'>atom.select</span>(<span class='kw'>prmtop</span>, <span class='kw'>...</span>)

<span class='co'># S3 method for select</span>
<span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span>(<span class='kw'>x</span>, <span class='kw'>...</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>...</th>
      <td><p>arguments passed to <code>atom.select.pdb</code>,
    <code>atom.select.prmtop</code>, or <code>print</code>.</p></td>
    </tr>
    <tr>
      <th>pdb</th>
      <td><p>a structure object of class <code>"pdb"</code>, obtained from
    <code>read.pdb</code>.</p></td>
    </tr>
    <tr>
      <th>pdbs</th>
      <td><p>a numeric matrix of aligned C-alpha xyz Cartesian
    coordinates as obtained with <code>read.fasta.pdb</code> or <code>pdbaln</code>.</p></td>
    </tr>
    <tr>
      <th>string</th>
      <td><p>a single selection keyword from <code>calpha</code>
    <code>cbeta</code> <code>backbone</code> <code>sidechain</code> <code>protein</code>
    <code>nucleic</code> <code>ligand</code> <code>water</code> <code>h</code> or <code>noh</code>.</p></td>
    </tr>
    <tr>
      <th>type</th>
      <td><p>a single element character vector for selecting &#8216;ATOM&#8217;
    or &#8216;HETATM&#8217; record types.</p></td>
    </tr>
    <tr>
      <th>eleno</th>
      <td><p>a numeric vector of element numbers.</p></td>
    </tr>
    <tr>
      <th>elena</th>
      <td><p>a character vector of atom names.</p></td>
    </tr>
    <tr>
      <th>elety</th>
      <td><p>a character vector of atom names.</p></td>
    </tr>
    <tr>
      <th>resid</th>
      <td><p>a character vector of residue name identifiers.</p></td>
    </tr>
    <tr>
      <th>chain</th>
      <td><p>a character vector of chain identifiers.</p></td>
    </tr>
    <tr>
      <th>resno</th>
      <td><p>a numeric vector of residue numbers.</p></td>
    </tr>
    <tr>
      <th>insert</th>
      <td><p>a character vector of insert identifiers. Non-insert 
    residues can be selected with <code>NA</code> or &#8216;&#8217; values. The default value 
    of <code>NULL</code> will select both insert and non-insert residues.</p></td>
    </tr>
    <tr>
      <th>segid</th>
      <td><p>a character vector of segment identifiers. Empty segid
    values can be selected with <code>NA</code> or &#8216;&#8217; values. The default value
    of <code>NULL</code> will select both empty and non-empty segment identifiers.</p></td>
    </tr>
    <tr>
      <th>operator</th>
      <td><p>a single element character specifying either the AND
    or OR operator by which individual selection components should be
    combined. Allowed values are &#8216;"AND"&#8217; and &#8216;"OR"&#8217;.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>logical, if TRUE details of the selection are printed.</p></td>
    </tr>
    <tr>
      <th>inverse</th>
      <td><p>logical, if TRUE the inversed selection is retured
    (i.e. all atoms NOT in the selection).</p></td>
    </tr>
    <tr>
      <th>value</th>
      <td><p>logical, if FALSE, vectors containing the (integer) indices of
    the matches determined by <code>atom.select</code> are returned, and if TRUE,
    a <code>pdb</code> object containing the matching atoms themselves is
    returned.</p></td>
    </tr>
    <tr>
      <th>mol</th>
      <td><p>a structure object of class <code>"mol2"</code>, obtained from
    <code>read.mol2</code>.</p></td>
    </tr>
    <tr>
      <th>statbit</th>
      <td><p>a character vector of statbit identifiers.</p></td>
    </tr>
    <tr>
      <th>prmtop</th>
      <td><p>a structure object of class <code>"prmtop"</code>, obtained from
    <code>read.prmtop</code>.</p></td>
    </tr>
    <tr>
      <th>x</th>
      <td><p>a atom.select object as obtained from
    <code>atom.select</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function allows for the selection of atom and coordinate data
  corresponding to the intersection of various input criteria.</p>
<p>Input selection criteria include selection <code>string</code> keywords (such as
  <code>"calpha"</code>, <code>"backbone"</code>, <code>"sidechain"</code>,  <code>"protein"</code>,
  <code>"nucleic"</code>, <code>"ligand"</code>, etc.) and individual named
  selection components (including &#8216;chain&#8217;,  &#8216;resno&#8217;,
  &#8216;resid&#8217;, &#8216;elety&#8217; etc.).</p>
<p>For example, <code>atom.select(pdb, "calpha")</code> will return indices for
  all C-alpha (CA) atoms found in protein residues in the <code>pdb</code>
  object, <code>atom.select(pdb, "backbone")</code> will return indices
  for all protein N,CA,C,O atoms, and <code>atom.select(pdb, "cbeta")</code>
  for all protein N,CA,C,O,CB atoms.</p>
<p>Note that keyword <code>string</code> shortcuts can be combined with individual
  selection components, e.g. <code>atom.select(pdb, "protein", chain="A")</code>
  will select all protein atoms found in chain A.</p>
<p>Selection criteria are combined according to the provided
  <code>operator</code> argument. The default operator <code>AND</code> (or <code>&amp;</code>) will
  combine by intersection while <code>OR</code> (or <code><a href='https://rdrr.io/r/base/Logic.html'>|</a></code>) will take the union.</p>
<p>For example, <code>atom.select(pdb, "protein", elety=c("N", "CA", "C"),
  resno=65:103)</code> will select the N, CA, C atoms in the protein residues
  65 through 103, while <code>atom.select(pdb, "protein", resid="ATP",
  operator="OR")</code> will select all protein atoms as well as any ATP
  residue(s).</p>
<p>Other <code>string</code> shortcuts include:
  <code>"calpha"</code>, <code>"back"</code>, <code>"backbone"</code>, <code>"cbeta"</code>,
  <code>"protein"</code>, <code>"notprotein"</code>, <code>"ligand"</code>,
  <code>"water"</code>, <code>"notwater"</code>, <code>"h"</code>, <code>"noh"</code>,
  <code>"nucleic"</code>, and <code>"notnucleic"</code>.</p>
<p>In addition, the <code>combine.select</code> function can further combine atom
  selections using &#8216;AND&#8217;, &#8216;OR&#8217;, or &#8216;NOT&#8217; logical
  operations.</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>Protein atoms are defined as any atom in a residue matching the
  residue name in the attached <code>aa.table</code> data frame. See
  <code>aa.table$aa3</code> for a complete list of residue names.</p>
<p>Nucleic atoms are defined as all atoms found in residues with names A,
  U, G, C, T, I, DA, DU, DG, DC, DT, or DI.</p>
<p>Water atoms/residues are defined as those with residue names H2O, OH2,
  HOH, HHO, OHH, SOL, WAT, TIP, TIP, TIP3, or TIP4.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns a list of class <code>"select"</code> with the following components:</p>
<dt>atom</dt><dd><p>a numeric matrix of atomic indices.</p></dd>
  <dt>xyz </dt><dd><p>a numeric matrix of xyz indices.</p></dd>
  <dt>call </dt><dd><p>the matched call.</p></dd>

    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695--2696.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Barry Grant, Lars Skjaerven</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code>read.pdb</code>, <code>as.select</code>,
  <code>combine.select</code>, <code>trim.pdb</code>,
  <code>write.pdb</code>, <code>read.prmtop</code>,
  <code>read.crd</code>, <code>read.dcd</code>,
  <code>read.ncdf</code>.</p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples">
<span class='co'>##- PDB example</span>
<span class='co'># Read a PDB file</span>
<span class='kw'>pdb</span> <span class='op'>&lt;-</span> <span class='fu'>read.pdb</span>( <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"examples/1hel.pdb"</span>, package=<span class='st'>"bio3d"</span>) )

<span class='co'># Select protein atoms of chain A</span>
<span class='fu'>atom.select</span>(<span class='kw'>pdb</span>, <span class='st'>"protein"</span>, chain=<span class='st'>"A"</span>)

<span class='co'># Select all atoms except from the protein</span>
<span class='fu'>atom.select</span>(<span class='kw'>pdb</span>, <span class='st'>"protein"</span>, inverse=<span class='fl'>TRUE</span>, verbose=<span class='fl'>TRUE</span>)

<span class='co'># Select all C-alpha atoms with residues numbers between 43 and 54</span>
<span class='kw'>sele</span> <span class='op'>&lt;-</span> <span class='fu'>atom.select</span>(<span class='kw'>pdb</span>, <span class='st'>"calpha"</span>, resno=<span class='fl'>43</span><span class='op'>:</span><span class='fl'>54</span>, verbose=<span class='fl'>TRUE</span>)

<span class='co'># Access the PDB data with the selection indices</span>
<span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span>( <span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>atom</span>[ <span class='kw'>sele</span><span class='op'>$</span><span class='kw'>atom</span>, <span class='st'>"resid"</span> ] )
<span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span>( <span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>xyz</span>[ <span class='kw'>sele</span><span class='op'>$</span><span class='kw'>xyz</span> ] )

<span class='co'># Trim PDB to selection</span>
<span class='kw'>ca.pdb</span> <span class='op'>&lt;-</span> <span class='fu'>trim.pdb</span>(<span class='kw'>pdb</span>, <span class='kw'>sele</span>)

<span class='co'>if</span> (<span class='fl'>FALSE</span>) {

<span class='co'>##- PRMTOP example</span>
<span class='kw'>prmtop</span> <span class='op'>&lt;-</span> <span class='fu'>read.prmtop</span>(<span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"examples/crambin.prmtop"</span>, package=<span class='st'>"bio3d"</span>))

<span class='co'>## Atom selection</span>
<span class='kw'>ca.inds</span> <span class='op'>&lt;-</span> <span class='fu'>atom.select</span>(<span class='kw'>prmtop</span>, <span class='st'>"calpha"</span>)

}</pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


