<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Distance Matrix Analysis — dm • Bio3D</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Distance Matrix Analysis — dm" />
<meta property="og:description" content="Construct a distance matrix for a given protein structure." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Bio3D</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4-1.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-info"></span>
     
    Getting Started
  </a>
</li>
<li>
  <a href="../index.html">
    <span class="fa fa-book"></span>
     
    Tutorials
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Functions
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../CODE_OF_CONDUCT.html">
    <span class="fa fa-newspaper-o"></span>
     
    News &amp; Updates
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Distance Matrix Analysis</h1>
    
    <div class="hidden name"><code>dm.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Construct a distance matrix for a given protein structure.</p>
    </div>

    <pre class="usage"><span class='fu'>dm</span>(<span class='kw'>...</span>)

<span class='co'># S3 method for pdb</span>
<span class='fu'>dm</span>(<span class='kw'>pdb</span>, inds = <span class='kw'>NULL</span>, grp = <span class='fl'>TRUE</span>, verbose=<span class='fl'>TRUE</span>, <span class='kw'>...</span>)
<span class='co'># S3 method for pdbs</span>
<span class='fu'>dm</span>(<span class='kw'>pdbs</span>, rm.gaps=<span class='fl'>FALSE</span>, all.atom=<span class='fl'>FALSE</span>, 
                        aligned.atoms.only=<span class='kw'>NULL</span>, <span class='kw'>...</span>)

<span class='co'># S3 method for xyz</span>
<span class='fu'>dm</span>(<span class='kw'>xyz</span>, grpby = <span class='kw'>NULL</span>, scut = <span class='kw'>NULL</span>, mask.lower = <span class='fl'>TRUE</span>, 
                  gc.first=<span class='fl'>FALSE</span>, ncore=<span class='fl'>1</span>, <span class='kw'>...</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>pdb</th>
      <td><p>a <code>pdb</code> structure object as returned by
    <code>read.pdb</code> or a numeric vector of &#8216;xyz&#8217; coordinates.</p></td>
    </tr>
    <tr>
      <th>inds</th>
      <td><p>atom and xyz coordinate indices obtained from <code>atom.select</code>
    that selects the elements of <code>pdb</code> upon which the calculation
    should be based.</p></td>
    </tr>
    <tr>
      <th>grp</th>
      <td><p>logical, if TRUE atomic distances will be grouped according
    to their residue membership. See &#8216;grpby&#8217;.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>logical, if TRUE possible warnings are printed.</p></td>
    </tr>
    <tr>
      <th>pdbs</th>
      <td><p>a &#8216;pdbs&#8217; object as returned by <code>read.fasta.pdb</code>,
    <code>read.all</code>, or <code>pdbaln</code>.</p></td>
    </tr>
    <tr>
      <th>rm.gaps</th>
      <td><p>logical, if TRUE gapped positions are removed in the returned value.</p></td>
    </tr>
    <tr>
      <th>all.atom</th>
      <td><p>logical, if TRUE all-atom coordinates from <code>read.all</code>
    are used.</p></td>
    </tr>
    <tr>
      <th>aligned.atoms.only</th>
      <td><p>logical, if TRUE only equivalent (aligned) atoms are considered. 
    Only meaningful when <code>all.atom=TRUE</code>. Default: FALSE.</p></td>
    </tr>
    <tr>
      <th>xyz</th>
      <td><p>a numeric vector or matrix of Cartesian coordinates.</p></td>
    </tr>
    <tr>
      <th>grpby</th>
      <td><p>a vector counting connective duplicated elements that
    indicate the elements of <code>xyz</code> that should be considered as a group
    (e.g. atoms from a particular residue).</p></td>
    </tr>
    <tr>
      <th>scut</th>
      <td><p>a cutoff neighbour value which has the effect of
    excluding atoms, or groups, that are sequentially within this
    value.</p></td>
    </tr>
    <tr>
      <th>mask.lower</th>
      <td><p>logical, if TRUE the lower matrix elements
    (i.e. those below the diagonal) are returned as NA.</p></td>
    </tr>
    <tr>
      <th>gc.first</th>
      <td><p>logical, if TRUE will call gc() first before calculation of 
    distance matrix. This is to solve the memory overload problem when <code>ncore &gt; 1</code>
    and <code>xyz</code> has many rows/columns, with a bit sacrifice on speed.</p></td>
    </tr>
    <tr>
      <th>ncore </th>
      <td><p>number of CPU cores used to do the calculation.
    <code>ncore&gt;1</code> requires package &#8216;parallel&#8217; installed.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>arguments passed to and from functions.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Distance matrices, also called distance plots or distance maps, are an
  established means of describing and comparing protein conformations
  (e.g. Phillips, 1970; Holm, 1993).</p>
<p>A distance matrix is a 2D representation of 3D structure that is
  independent of the coordinate reference frame and, ignoring
  chirality, contains enough information to reconstruct the 3D Cartesian
  coordinates (e.g. Havel, 1983).</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns a numeric matrix of class <code>"dmat"</code>, with all N by N
  distances, where N is the number of selected atoms. With multiple
  frames the output is provided in a three dimensional array.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695--2696.</p>
<p>Phillips (1970) <em>Biochem. Soc. Symp.</em> <b>31</b>, 11--28.</p>  
<p>Holm (1993) <em>J. Mol. Biol.</em> <b>233</b>, 123--138.</p>  
<p>Havel (1983) <em>Bull. Math. Biol.</em> <b>45</b>, 665--720.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Barry Grant</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>The input <code>selection</code> can be any character string or pattern
  interpretable by the function <code>atom.select</code>.  For example,
  shortcuts <code>"calpha"</code>, <code>"back"</code>, <code>"all"</code> and selection
  strings of the form <code>/segment/chain/residue number/residue
    name/element number/element name/</code>; see <code>atom.select</code>
  for details.</p>
<p>If a coordinate vector is provided as input (rather than a <code>pdb</code>
  object) the <code>selection</code> option is redundant and the input vector
  should be pruned instead to include only desired positions.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code>plot.dmat</code>, <code>read.pdb</code>, <code>atom.select</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='co'># \donttest{</span>
<span class='co'># PDB server connection required - testing excluded</span>

<span class='co'>##--- Distance Matrix Plot</span>
<span class='kw'>pdb</span> <span class='op'>&lt;-</span> <span class='fu'>read.pdb</span>( <span class='st'>"4q21"</span> )
<span class='kw'>k</span> <span class='op'>&lt;-</span> <span class='fu'>dm</span>(<span class='kw'>pdb</span>,inds=<span class='st'>"calpha"</span>)
<span class='fu'><a href='https://rdrr.io/r/graphics/filled.contour.html'>filled.contour</a></span>(<span class='kw'>k</span>, nlevels = <span class='fl'>10</span>)

<span class='co'>## NOTE: FOLLOWING EXAMPLE NEEDS MUSCLE INSTALLED</span>
<span class='co'>if</span>(<span class='fu'>check.utility</span>(<span class='st'>"muscle"</span>)) {

<span class='co'>##--- DDM: Difference Distance Matrix</span>
<span class='co'># Downlaod and align two PDB files</span>
<span class='kw'>pdbs</span> <span class='op'>&lt;-</span> <span class='fu'>pdbaln</span>( <span class='fu'>get.pdb</span>( <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>( <span class='st'>"4q21"</span>, <span class='st'>"521p"</span>), path = <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span>() ), outfile = <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempfile</a></span>() )

<span class='co'># Get distance matrix</span>
<span class='kw'>a</span> <span class='op'>&lt;-</span> <span class='fu'>dm.xyz</span>(<span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>xyz</span>[<span class='fl'>1</span>,])
<span class='kw'>b</span> <span class='op'>&lt;-</span> <span class='fu'>dm.xyz</span>(<span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>xyz</span>[<span class='fl'>2</span>,])

<span class='co'># Calculate DDM</span>
<span class='kw'>c</span> <span class='op'>&lt;-</span> <span class='kw'>a</span> <span class='op'>-</span> <span class='kw'>b</span>

<span class='co'># Plot DDM</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/plot.html'>plot</a></span>(<span class='kw'>c</span>,key=<span class='fl'>FALSE</span>, grid=<span class='fl'>FALSE</span>)

<span class='fu'><a href='https://rdrr.io/r/graphics/plot.html'>plot</a></span>(<span class='kw'>c</span>, axis.tick.space=<span class='fl'>10</span>,
     resnum.1=<span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>resno</span>[<span class='fl'>1</span>,],
     resnum.2=<span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>resno</span>[<span class='fl'>2</span>,],
     grid.col=<span class='st'>"black"</span>,
     xlab=<span class='st'>"Residue No. (4q21)"</span>, ylab=<span class='st'>"Residue No. (521p)"</span>)
}
<span class='co'># }</span>

<span class='co'>if</span> (<span class='fl'>FALSE</span>) {
<span class='co'>##-- Residue-wise distance matrix based on the</span>
<span class='co'>##   minimal distance between all available atoms</span>
<span class='kw'>l</span> <span class='op'>&lt;-</span> <span class='fu'>dm.xyz</span>(<span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>xyz</span>, grpby=<span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>atom</span>[,<span class='st'>"resno"</span>], scut=<span class='fl'>3</span>)
}</pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


