<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Secondary Structure Analysis with DSSP or STRIDE — dssp • Bio3D</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Secondary Structure Analysis with DSSP or STRIDE — dssp" />
<meta property="og:description" content="Secondary structure assignment according to the method of Kabsch and Sander (DSSP)
  or the method of Frishman and Argos (STRIDE)." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Bio3D</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4-1.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-info"></span>
     
    Getting Started
  </a>
</li>
<li>
  <a href="../index.html">
    <span class="fa fa-book"></span>
     
    Tutorials
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Functions
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../CODE_OF_CONDUCT.html">
    <span class="fa fa-newspaper-o"></span>
     
    News &amp; Updates
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Secondary Structure Analysis with DSSP or STRIDE</h1>
    
    <div class="hidden name"><code>dssp.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Secondary structure assignment according to the method of Kabsch and Sander (DSSP)
  or the method of Frishman and Argos (STRIDE).</p>
    </div>

    <pre class="usage"><span class='fu'>dssp</span>(<span class='kw'>...</span>)

<span class='co'># S3 method for pdb</span>
<span class='fu'>dssp</span>(<span class='kw'>pdb</span>, exefile = <span class='st'>"dssp"</span>, resno=<span class='fl'>TRUE</span>, full=<span class='fl'>FALSE</span>, verbose=<span class='fl'>FALSE</span>, <span class='kw'>...</span>)

<span class='co'># S3 method for pdbs</span>
<span class='fu'>dssp</span>(<span class='kw'>pdbs</span>, <span class='kw'>...</span>)

<span class='co'># S3 method for xyz</span>
<span class='fu'>dssp</span>(<span class='kw'>xyz</span>, <span class='kw'>pdb</span>, <span class='kw'>...</span>)

<span class='fu'>stride</span>(<span class='kw'>pdb</span>, exefile = <span class='st'>"stride"</span>, resno=<span class='fl'>TRUE</span>)

<span class='co'># S3 method for sse</span>
<span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span>(<span class='kw'>x</span>, <span class='kw'>...</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>pdb</th>
      <td><p>a structure object of class <code>"pdb"</code>, obtained from
    <code>read.pdb</code>.</p></td>
    </tr>
    <tr>
      <th>exefile</th>
      <td><p>file path to the &#8216;DSSP&#8217; or &#8216;STRIDE&#8217; program 
    on your system (i.e. how is &#8216;DSSP&#8217; or &#8216;STRIDE&#8217; invoked).</p></td>
    </tr>
    <tr>
      <th>resno</th>
      <td><p>logical, if TRUE output is in terms of residue numbers
    rather than residue index (position in sequence).</p></td>
    </tr>
    <tr>
      <th>full</th>
      <td><p>logical, if TRUE bridge pairs and hbonds columns are
    parsed.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>logical, if TRUE &#8216;DSSP&#8217; warning and error
    messages are printed.</p></td>
    </tr>
    <tr>
      <th>pdbs</th>
      <td><p>a list object of class <code>"pdbs"</code> (obtained with
    <code>pdbaln</code> or <code>read.fasta.pdb</code>).</p></td>
    </tr>
    <tr>
      <th>xyz</th>
      <td><p>a trajectory object of class <code>"xyz"</code>, obtained from
    <code>read.ncdf</code>, <code>read.dcd</code>,
    <code>read.crd</code>.</p></td>
    </tr>
    <tr>
      <th>x</th>
      <td><p>an <code>sse</code> object obtained from <code>dssp.pdb</code> or 
    <code>stride</code>.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>additional arguments to and from functions.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function calls the &#8216;DSSP&#8217; or &#8216;STRIDE&#8217; program to define 
  secondary structure and psi and phi torsion angles.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns a list with the following components:</p>
<dt>helix</dt><dd><p>&#8216;start&#8217;, &#8216;end&#8217;, &#8216;length&#8217;,
    &#8216;chain&#8217; and &#8216;type&#8217; of helix, where start and end are
    residue numbers or residue index positions depending on the value of
    &#8220;resno&#8221; input argument.</p></dd>
  <dt>sheet</dt><dd><p>&#8216;start&#8217;, &#8216;end&#8217; and &#8216;length&#8217; of E
    type sse, where start and end are residue numbers &#8220;resno&#8221;.</p></dd>
  <dt>turn</dt><dd><p>&#8216;start&#8217;, &#8216;end&#8217; and &#8216;length&#8217; of T
    type sse, where start and end are residue numbers &#8220;resno&#8221;.</p></dd>
  <dt>phi</dt><dd><p>a numeric vector of phi angles.</p></dd>
  <dt>psi</dt><dd><p>a numeric vector of psi angles.</p></dd>
  <dt>acc</dt><dd><p>a numeric vector of solvent accessibility.</p></dd>
  <dt>sse</dt><dd><p>a character vector of secondary structure type per residue.</p></dd>
  <dt>hbonds</dt><dd><p>a 10 or 16 column matrix containing the bridge pair
    records as well as backbone NH--&gt;O and O--&gt;NH H-bond records. 
    (Only available for <code>dssp</code></p></dd>

    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695--2696.</p>
<p>&#8216;DSSP&#8217; is the work of Kabsch and Sander:
  Kabsch and Sander (1983) <em>Biopolymers.</em>  <b>12</b>, 2577--2637.</p>
<p>For information on obtaining &#8216;DSSP&#8217;, see:<br />
  <a href='http://swift.cmbi.ru.nl/gv/dssp/'>http://swift.cmbi.ru.nl/gv/dssp/</a>.</p>
<p>&#8216;STRIDE&#8217; is the work of Frishman and Argos:
  Frishman and Argos (1995) <em>Proteins.</em>  <b>3</b>, 566--579.</p>
<p>For information on obtaining the &#8216;STRIDE&#8217; program, see:<br />
  <a href='http://webclu.bio.wzw.tum.de/stride/'>http://webclu.bio.wzw.tum.de/stride/</a>,
  or copy it from an installation of VMD.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Barry Grant, Lars Skjaerven (dssp.pdbs)</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>A system call is made to the &#8216;DSSP&#8217; or &#8216;STRIDE&#8217; program, 
  which must be installed on your system and in the search path for executables.
  See <a href='http://thegrantlab.org/bio3d/tutorials/installing-bio3d'>http://thegrantlab.org/bio3d/tutorials/installing-bio3d</a> for 
  instructions of how to install these programs.</p>  
<p>For the <code>hbonds</code> list component the column names can be
  used as a convenient means of data access, namely:<br />
  Bridge pair 1            &#8220;BP1&#8221;,<br />
  Bridge pair 2            &#8220;BP2&#8221;,<br />
  Backbone H-bond (NH--&gt;O) &#8220;NH-O.1&#8221;,<br />
  H-bond energy of NH--&gt;O  &#8220;E1&#8221;,<br />
  Backbone H-bond (O--&gt;NH) &#8220;O-HN.1&#8221;,<br />
  H-bond energy of O--&gt;NH  &#8220;E2&#8221;,<br />
  Backbone H-bond (NH--&gt;O) &#8220;NH-O.2&#8221;,<br />
  H-bond energy of NH--&gt;O  &#8220;E3&#8221;,<br />
  Backbone H-bond (O--&gt;NH) &#8220;O-HN.2&#8221;,<br />
  H-bond energy of O--&gt;NH  &#8220;E4&#8221;.<br /></p>
<p>If &#8216;resno=TRUE&#8217; the following additional columns are included:<br />
  Chain ID of resno &#8220;BP1&#8221;:    &#8220;ChainBP1&#8221;,<br />
  Chain ID of resno &#8220;BP2&#8221;:    &#8220;ChainBP2&#8221;,<br />
  Chain ID of resno &#8220;O-HN.1&#8221;: &#8220;Chain1&#8221;,<br />
  Chain ID of resno &#8220;NH-O.2&#8221;: &#8220;Chain2&#8221;,<br />
  Chain ID of resno &#8220;O-HN.1&#8221;: &#8220;Chain3&#8221;,<br />
  Chain ID of resno &#8220;NH-O.2&#8221;: &#8220;Chain4&#8221;.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code>read.pdb</code>,
  <code>torsion.pdb</code>, <code>torsion.xyz</code>,
  <code>plot.bio3d</code>,</p>  
<p><code>read.ncdf</code>, <code>read.dcd</code>,
  <code>read.prmtop</code>, <code>read.crd</code>,</p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='co'>if</span> (<span class='fl'>FALSE</span>) {
<span class='co'>##- PDB example</span>
<span class='co'># Read a PDB file</span>
<span class='kw'>pdb</span> <span class='op'>&lt;-</span> <span class='fu'>read.pdb</span>(<span class='st'>"1bg2"</span>)
<span class='kw'>sse</span> <span class='op'>&lt;-</span> <span class='fu'>dssp</span>(<span class='kw'>pdb</span>)
<span class='kw'>sse2</span> <span class='op'>&lt;-</span> <span class='fu'>stride</span>(<span class='kw'>pdb</span>)

<span class='co'>## Short summary</span>
<span class='kw'>sse</span>
<span class='kw'>sse2</span>

<span class='co'># Helix data</span>
<span class='kw'>sse</span><span class='op'>$</span><span class='kw'>helix</span>

<span class='co'># Precent SSE content</span>
<span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span>(<span class='kw'>sse</span><span class='op'>$</span><span class='kw'>helix</span><span class='op'>$</span><span class='kw'>length</span>)<span class='op'>/</span><span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span>(<span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>calpha</span>) <span class='op'>*</span> <span class='fl'>100</span>
<span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span>(<span class='kw'>sse</span><span class='op'>$</span><span class='kw'>sheet</span><span class='op'>$</span><span class='kw'>length</span>)<span class='op'>/</span><span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span>(<span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>calpha</span>) <span class='op'>*</span> <span class='fl'>100</span>



<span class='co'>##- PDBs example</span>
<span class='kw'>aln</span>  <span class='op'>&lt;-</span> <span class='fu'>read.fasta</span>( <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"examples/kif1a.fa"</span>,package=<span class='st'>"bio3d"</span>) )
<span class='kw'>pdbs</span> <span class='op'>&lt;-</span> <span class='fu'>read.fasta.pdb</span>( <span class='kw'>aln</span> )

<span class='co'>## Aligned PDB defined secondary structure</span>
<span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>sse</span>

<span class='co'>## Aligned DSSP defined secondary structure </span>
<span class='kw'>sse</span> <span class='op'>&lt;-</span> <span class='fu'>dssp</span>(<span class='kw'>pdbs</span>)


<span class='co'>##- XYZ Trajectory</span>
<span class='kw'>pdb</span> <span class='op'>&lt;-</span> <span class='fu'>read.pdb</span>(<span class='st'>"2mda"</span>, multi=<span class='fl'>TRUE</span>)
<span class='fu'>dssp.xyz</span>(<span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>xyz</span>, <span class='kw'>pdb</span>)

<span class='co'>## Note. for large MD trajectories you may want to skip some frames, e.g.</span>
<span class='kw'>xyz</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/cbind.html'>rbind</a></span>(<span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>xyz</span>, <span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>xyz</span>)       <span class='co'>## dummy trajectory</span>
<span class='kw'>frames</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>1</span>, to=<span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span>(<span class='kw'>xyz</span>), by=<span class='fl'>4</span>) <span class='co'>## frame numbers to examine</span>
<span class='kw'>ss</span> <span class='op'>&lt;-</span> <span class='fu'>dssp.xyz</span>(<span class='kw'>xyz</span>[<span class='kw'>frames</span>, ], <span class='kw'>pdb</span>)      <span class='co'>## matrix of sse frame x residue</span>

}</pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


