<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Bio3d Example Data — example.data • Bio3D</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Bio3d Example Data — example.data" />
<meta property="og:description" content="These data sets contain the results of running various Bio3D
  functions on example kinesin and transducin structural data, and on a short 
  coarse-grained MD simulation data for HIV protease. The main 
  purpose of including this data (which may be generated by the user by following the extended examples documented within the various Bio3D functions) is to
  speed up example execution. It should allow users to more quickly
  appreciate the capabilities of functions that would otherwise require
  raw data download, input and processing before execution.
Note that related datasets formed the basis of
  the work described in (Grant, 2007) and (Yao &amp;amp; Grant, 2013) for kinesin
  and transducin examples, respectively." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Bio3D</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4-1.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-info"></span>
     
    Getting Started
  </a>
</li>
<li>
  <a href="../index.html">
    <span class="fa fa-book"></span>
     
    Tutorials
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Functions
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../CODE_OF_CONDUCT.html">
    <span class="fa fa-newspaper-o"></span>
     
    News &amp; Updates
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Bio3d Example Data</h1>
    
    <div class="hidden name"><code>example.data.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>These data sets contain the results of running various Bio3D
  functions on example kinesin and transducin structural data, and on a short 
  coarse-grained MD simulation data for HIV protease. The main 
  purpose of including this data (which may be generated by the user by following the extended examples documented within the various Bio3D functions) is to
  speed up example execution. It should allow users to more quickly
  appreciate the capabilities of functions that would otherwise require
  raw data download, input and processing before execution.</p>
<p>Note that related datasets formed the basis of
  the work described in (Grant, 2007) and (Yao &amp; Grant, 2013) for <code>kinesin</code>
  and <code>transducin</code> examples, respectively.</p>
    </div>

    <pre class="usage"><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='kw'>kinesin</span>)
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='kw'>transducin</span>)
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='kw'>hivp</span>)</pre>


    <h2 class="hasAnchor" id="format"><a class="anchor" href="#format"></a>Format</h2>

    <p>Three objects from analysis of the <code>kinesin</code> and <code>transducin</code> sequence and structure
  data:</p><ol>
<li><p>pdbs is a list of class <code>pdbs</code> containing aligned PDB
      structure data.  In the case of transducin this is the output of running 
      <code>pdbaln</code> on a set of 53 G[alpha]i structures from the PDB database (see <code>pdbs$id</code> 
      or <code>annotation</code> described below for details). The coordinates
      are fitted onto the first structure based on <code>"core"</code> positions obtained from <code>core.find</code> and superposed using the function <code>pdbfit</code>.</p></li>
<li><p>core is a list of class <code>"core"</code> obtained by running the
      function <code>core.find</code> on the <code>pdbs</code> object as described above.</p></li>
<li><p>annotation is a character matrix describing the nucleotide state and bound 
      ligand species for each structure in <code>pdbs</code> as obtained from the function <code>pdb.annotate</code>.</p></li>
</ol>

<p>One object named <code>net</code> in the hivp example data stores the correlation 
  network obtained from the analysis of the MD simulation trajectory of HIV 
  protease using the <code>cna</code> function. The original trajectory file can be 
  accessed by the command &#8216;system.file("examples/hivp.dcd", package="bio3d")&#8217;.</p>
    <h2 class="hasAnchor" id="source"><a class="anchor" href="#source"></a>Source</h2>

    <p>A related but more extensive dataset formed the basis of
  the work described in (Grant, 2007) and (Yao &amp; Grant, 2013) for <code>kinesin</code>
  and <code>transducin</code> examples, respectively.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695--2696.</p>
<p>Grant, B.J. et al. (2007) <em>J. Mol. Biol.</em> <b>368</b>, 1231--1248.</p>  
<p>Yao, X.Q. et al. (2013) <em>Biophys. J.</em> <b>105</b>, L08--L10.</p>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


