<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Filter for Cross-correlation Matrices (Cij) — filter.dccm • Bio3D</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>

<!-- Bootstrap -->
<link href="../tidyverse.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>
<link href="../tidyverse-2.css" rel="stylesheet">

<!--optional theme
<link href="../.css" rel="stylesheet">
-->


<!-- docsearch -->
<script src="../docsearch.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous" />
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>


<meta property="og:title" content="Filter for Cross-correlation Matrices (Cij) — filter.dccm" />
<meta property="og:description" content="This function builds various cij matrix for correlation network analysis" />
<meta property="og:image" content="/logo.png" />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../index.html">
        bio3d</a>
        <div class="info hidden-xs hidden-sm">
<!--          <span class="partof">part of the <a href="https://tidyverse.org">tidyverse</a></span>
-->
          <span class="version version-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4-1.9000</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">INTRODUCTORY TUTORIALS</li>
    <li>
      <a href="../articles/online/install_vignette/Bio3D_install.html">Installing Bio3D</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/user-guide">New user guide</a>
    </li>
    <li>
      <a href="../articles/online/intro_vignette/Bio3D_introduction.html">Getting started with Bio3D</a>
    </li>
    <li>
      <a href="../articles/online/pdb_vignette/Bio3D_pdb.html">PDB structure manipulation and analysis</a>
    </li>
    <li>
      <a href="../articles/online/pca_vignette/Bio3D_pca.html">Comparative sequence and structure analysis</a>
    </li>
    <li>
      <a href="../articles/online/nma_vignette/Bio3D_nma.html">Enhanced Normal Modes Analysis</a>
    </li>
    <li>
      <a href="../articles/online/traj_vignette/Bio3D_md.html">Beginning trajectory analysis with Bio3D </a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">ADVANCED TUTORIALS</li>
    <li>
      <a href="../articles/online/enma_vignettes/Bio3D_nma-dhfr-partI.html">Ensemble NMA of E.coli DHFR structures</a>
    </li>
    <li>
      <a href="../articles/online/enma_vignettes/Bio3D_nma-dhfr-partII.html">Ensemble NMA across multiple species of DHFR</a>
    </li>
    <li>
      <a href="../articles/online/cna_vignette/cna_vignette.spin.html">Protein structure network analysis with Bio3D</a>
    </li>
    <li>
      <a href="../articles/online/eddm_vignette/Bio3D_eddm.spin.html">Ensemble Difference Distance Matrix (eDDM) analysis</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">WEBAPP TUTORIALS</li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/download/webapp-tutorials?download=22:bio3d-webapp-pca">Online protein structure analysis with Bio3D-Web (PDF)</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/webapps">Online Normal Mode Analysis with Bio3D WebApps (To Update)</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/index.html">All vignettes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Documentation
  </a>
</li>
        <li>
  <a href="../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    News
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket"></span>
     
  </a>
</li>
      </ul>
      
      <form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
        </div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Filter for Cross-correlation Matrices (Cij)</h1>
    
    <div class="hidden name"><code>filter.dccm.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function builds various cij matrix for correlation network analysis</p>
    </div>

    <pre class="usage"><span class='fu'>filter.dccm</span>(<span class='kw'>x</span>, cutoff.cij = <span class='kw'>NULL</span>, cmap = <span class='kw'>NULL</span>, xyz = <span class='kw'>NULL</span>, fac = <span class='kw'>NULL</span>, 
      cutoff.sims = <span class='kw'>NULL</span>, collapse = <span class='fl'>TRUE</span>, extra.filter = <span class='kw'>NULL</span>, <span class='kw'>...</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>A matrix (nXn), a numeric array with 3 dimensions (nXnXm), 
    a list with m cells each containing nXn matrix, or a list with
    &#8216;all.dccm&#8217; component, containing atomic correlation values, 
    where "n" is the number of residues and "m" the number of calculations. 
    The matrix elements should be in between -1 and 1. See &#8216;dccm&#8217; 
    function in bio3d package for further details.</p></td>
    </tr>
    <tr>
      <th>cutoff.cij</th>
      <td><p>Threshold for each individual correlation value. 
    If NULL, a guessed value will be used. See below for details.</p></td>
    </tr>
    <tr>
      <th>cmap</th>
      <td><p>logical or numerical matrix indicating the contact map.
    If logical and TRUE, contact map will be calculated with input
    <code>xyz</code>.</p></td>
    </tr>
    <tr>
      <th>xyz</th>
      <td><p>XYZ coordinates, or a &#8216;pdbs&#8217; object obtained from 
    <code>pdbaln</code> or <code>read.fasta.pdb</code>, for contact map calculations.</p></td>
    </tr>
    <tr>
      <th>fac</th>
      <td><p>factor indicating distinct categories of input correlation matrices.</p></td>
    </tr>
    <tr>
      <th>cutoff.sims</th>
      <td><p>Threshold for the number of simulations with observed correlation
    value above <code>cutoff.cij</code> for the same residue/atomic pairs. See below for details.</p></td>
    </tr>
    <tr>
      <th>collapse</th>
      <td><p>logical, if TRUE the mean matrix will be returned.</p></td>
    </tr>
    <tr>
      <th>extra.filter</th>
      <td><p>Filter to apply in addition to the model chosen.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>extra arguments passed to function <code>cmap</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns a matrix of class "dccm" or a 3D array of filtered cross-correlations.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695--2696.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Xin-Qiu Yao, Guido Scarabelli &amp; Barry Grant</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>If cmap is TRUE or provided a numerical matrix, the function inspects
     a set of cross-correlation matrices, or DCCM, and decides edges for 
     correlation network analysis based on:</p>
<p>1. min(abs(cij)) &gt;= cutoff.cij, or
        2. max(abs(cij)) &gt;= cutoff.cij &amp;&amp; residues contact each other 
           based on results from <code>cmap</code>.</p>
<p>Otherwise, the function filters DCCMs with <code>cutoff.cij</code> and 
     return the mean of correlations present in at least 
     <code>cutoff.sims</code> calculated matrices.</p>
<p>An internally guessed <code>cuoff.cij</code> is used if <code>cutoff.cij=NULL</code> is provided. 
     By default, the cutoff is determined by keeping 5% of all residue pairs connected.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code>cna</code>, <code>dccm</code>, <code>dccm.nma</code>, <code>dccm.xyz</code>,
   <code>cmap</code>, <code>plot.dccm</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'>if</span> (<span class='fl'>FALSE</span>) {

<span class='co'># Example of transducin</span>
<span class='fu'><a href='https://rdrr.io/r/base/attach.html'>attach</a></span>(<span class='kw'>transducin</span>)

<span class='kw'>gaps.pos</span> <span class='op'>&lt;-</span> <span class='fu'>gap.inspect</span>(<span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>xyz</span>)
<span class='kw'>modes</span> <span class='op'>&lt;-</span> <span class='fu'>nma.pdbs</span>(<span class='kw'>pdbs</span>, ncore=<span class='kw'>NULL</span>)
<span class='kw'>dccms</span> <span class='op'>&lt;-</span> <span class='fu'>dccm.enma</span>(<span class='kw'>modes</span>, ncore=<span class='kw'>NULL</span>)

<span class='kw'>cij</span> <span class='op'>&lt;-</span> <span class='fu'>filter.dccm</span>(<span class='kw'>dccms</span>, xyz=<span class='kw'>pdbs</span>)

<span class='co'># Example protein kinase</span>
<span class='co'># Select Protein Kinase PDB IDs</span>
<span class='kw'>ids</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"4b7t_A"</span>, <span class='st'>"2exm_A"</span>, <span class='st'>"1opj_A"</span>, <span class='st'>"4jaj_A"</span>, <span class='st'>"1a9u_A"</span>,
                 <span class='st'>"1tki_A"</span>, <span class='st'>"1csn_A"</span>, <span class='st'>"1lp4_A"</span>)

<span class='co'># Download and split by chain ID</span>
<span class='kw'>files</span> <span class='op'>&lt;-</span> <span class='fu'>get.pdb</span>(<span class='kw'>ids</span>, path = <span class='st'>"raw_pdbs"</span>, split=<span class='fl'>TRUE</span>)

<span class='co'># Alignment of structures</span>
<span class='kw'>pdbs</span> <span class='op'>&lt;-</span> <span class='fu'>pdbaln</span>(<span class='kw'>files</span>) <span class='co'># Sequence identity</span>
<span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fu'>seqidentity</span>(<span class='kw'>pdbs</span>)))

<span class='co'># NMA on all structures</span>
<span class='kw'>modes</span> <span class='op'>&lt;-</span> <span class='fu'>nma.pdbs</span>(<span class='kw'>pdbs</span>, ncore=<span class='kw'>NULL</span>)

<span class='co'># Calculate correlation matrices for each structure</span>
<span class='kw'>cij</span> <span class='op'>&lt;-</span> <span class='fu'>dccm</span>(<span class='kw'>modes</span>)

<span class='co'># Set DCCM plot panel names for combined figure</span>
<span class='fu'><a href='https://rdrr.io/r/base/dimnames.html'>dimnames</a></span>(<span class='kw'>cij</span><span class='op'>$</span><span class='kw'>all.dccm</span>) <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span>(<span class='kw'>NULL</span>, <span class='kw'>NULL</span>, <span class='kw'>ids</span>)
<span class='fu'>plot.dccm</span>(<span class='kw'>cij</span><span class='op'>$</span><span class='kw'>all.dccm</span>)

<span class='co'># Filter to display only correlations present in all structures</span>
<span class='kw'>cij.all</span> <span class='op'>&lt;-</span> <span class='fu'>filter.dccm</span>(<span class='kw'>cij</span>, cutoff.sims = <span class='fl'>8</span>, cutoff.cij = <span class='fl'>0</span>)
<span class='fu'>plot.dccm</span>(<span class='kw'>cij.all</span>, main = <span class='st'>"Consensus Residue Cross Correlation"</span>)

<span class='fu'><a href='https://rdrr.io/r/base/detach.html'>detach</a></span>(<span class='kw'>transducin</span>)
}
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="tidyverse">
  <p>Authored by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="author">
  <p>
    Built with &#10084;&#65039; and <a href="https://pkgdown.r-lib.org">pkgdown</a> &copy; 2020.
  </p>
</div>

      </footer>
   </div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script>
<script>
  docsearch({
    
    
    apiKey: '1270c4079b26a138795263974bbf302d',
    indexName: 'tidyverse',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>



  </body>
</html>


