<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Coordinate Superposition — fit.xyz • Bio3D</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Coordinate Superposition — fit.xyz" />
<meta property="og:description" content="Coordinate superposition with the Kabsch algorithm." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Bio3D</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4-1.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-info"></span>
     
    Getting Started
  </a>
</li>
<li>
  <a href="../index.html">
    <span class="fa fa-book"></span>
     
    Tutorials
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Functions
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../CODE_OF_CONDUCT.html">
    <span class="fa fa-newspaper-o"></span>
     
    News &amp; Updates
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Coordinate Superposition</h1>
    
    <div class="hidden name"><code>fit.xyz.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Coordinate superposition with the Kabsch algorithm.</p>
    </div>

    <pre class="usage"><span class='fu'>fit.xyz</span>(<span class='kw'>fixed</span>, <span class='kw'>mobile</span>,
        fixed.inds  = <span class='kw'>NULL</span>,
        mobile.inds = <span class='kw'>NULL</span>,
        verbose=<span class='fl'>FALSE</span>,
        prefix= <span class='st'>""</span>, pdbext = <span class='st'>""</span>,
        outpath = <span class='st'>"fitlsq"</span>, full.pdbs=<span class='fl'>FALSE</span>, 
        ncore = <span class='fl'>1</span>, nseg.scale = <span class='fl'>1</span>, <span class='kw'>...</span>)

<span class='fu'>rot.lsq</span>(<span class='kw'>xx</span>, <span class='kw'>yy</span>,
        xfit = <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span>(<span class='fl'>TRUE</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='kw'>xx</span>)), yfit = <span class='kw'>xfit</span>,
        verbose = <span class='fl'>FALSE</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>fixed </th>
      <td><p>numeric vector of xyz coordinates.</p></td>
    </tr>
    <tr>
      <th>mobile</th>
      <td><p>numeric vector, numeric matrix, or an object with an
    <code>xyz</code> component containing one or more coordinate sets.</p></td>
    </tr>
    <tr>
      <th>fixed.inds</th>
      <td><p>a vector of indices that selects the elements of
    <code>fixed</code> upon which fitting should be based.</p></td>
    </tr>
    <tr>
      <th>mobile.inds</th>
      <td><p>a vector of indices that selects the elements
    of <code>mobile</code> upon which fitting should be based.</p></td>
    </tr>
    <tr>
      <th>full.pdbs</th>
      <td><p>logical, if TRUE &#8220;full&#8221; coordinate files
    (i.e. all atoms) are written to the location specified by
    <code>outpath</code>.</p></td>
    </tr>
    <tr>
      <th>prefix</th>
      <td><p>prefix to mobile$id to locate &#8220;full&#8221; input PDB files.  Only
        required if <code>full.pdbs</code> is TRUE.</p></td>
    </tr>
    <tr>
      <th>pdbext</th>
      <td><p>the file name extension of the input PDB files.</p></td>
    </tr>
    <tr>
      <th>outpath</th>
      <td><p>character string specifing the output directory when
    <code>full.pdbs</code> is TRUE.</p></td>
    </tr>
    <tr>
      <th>xx</th>
      <td><p>numeric vector corresponding to the moving &#8216;subject&#8217;
    coordinate set.</p></td>
    </tr>
    <tr>
      <th>yy</th>
      <td><p>numeric vector corresponding to the fixed &#8216;target&#8217;
    coordinate set.</p></td>
    </tr>
    <tr>
      <th>xfit</th>
      <td><p>logical vector with the same length as <code>xx</code>,  with
    TRUE elements corresponding to the subset of positions upon which
    fitting is to be performed.</p></td>
    </tr>
    <tr>
      <th>yfit</th>
      <td><p>logical vector with the same length as <code>yy</code>,  with
    TRUE elements corresponding to the subset of positions upon which
    fitting is to be performed.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>logical, if TRUE more details are printed.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>other parameters for <code>read.pdb</code>.</p></td>
    </tr>
    <tr>
      <th>ncore </th>
      <td><p>number of CPU cores used to do the calculation.
    <code>ncore&gt;1</code> requires package &#8216;parallel&#8217; installed.</p></td>
    </tr>
    <tr>
      <th>nseg.scale </th>
      <td><p>split input data into specified number of segments
    prior to running multiple core calculation.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The function <code>fit.xyz</code> is a wrapper for the function
  <code>rot.lsq</code>, which performs the actual coordinate superposition.
  The function <code>rot.lsq</code> is an implementation of the Kabsch
  algorithm (Kabsch, 1978) and evaluates the optimal rotation matrix
  to minimize the RMSD between two structures.</p>
<p>Since the Kabsch algorithm assumes that the number of points are the
  same in the two input structures, care should be taken to ensure that
  consistent atom sets are selected with <code>fixed.inds</code> and
  <code>mobile.inds</code>.</p>
<p>Optionally, &#8220;full&#8221; PDB file superposition and output can be
  accomplished by setting <br /> <code>full.pdbs=TRUE</code>.  In that case, the
  input (<code>mobile</code>) passed to <code>fit.xyz</code> should be a list object
  obtained with the function <code>read.fasta.pdb</code>, since the
  components <code>id</code>, <code>resno</code> and <code>xyz</code> are required to
  establish correspondences.  See the examples below.</p>
<p>In dealing with large vector and matrix, running on multiple
  cores, especially when <code>ncore&gt;&gt;1</code>, may ask for a large portion
  of system memory. To avoid the overuse of memory, input data is first
  split into segments (for xyz matrix, the splitting is along the row). 
  The number of data segments is equal to <code>nseg.scale*nseg.base</code>, where
  <code>nseg.base </code> is an integer determined by the dimension of the data.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns moved coordinates.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695--2696.</p>
<p>Kabsch <em>Acta Cryst</em> (1978) <b>A34</b>, 827--828.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Barry Grant with <code>rot.lsq</code> contributions from Leo Caves</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code>rmsd</code>, <code>read.pdb</code>,
  <code>read.fasta.pdb</code>, <code>read.dcd</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='co'># \donttest{</span>
<span class='co'># PDB server connection required - testing excluded</span>

<span class='co'>##--- Read an alignment &amp; Fit aligned structures</span>
<span class='kw'>aln</span>  <span class='op'>&lt;-</span> <span class='fu'>read.fasta</span>(<span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"examples/kif1a.fa"</span>,package=<span class='st'>"bio3d"</span>))
<span class='kw'>pdbs</span> <span class='op'>&lt;-</span> <span class='fu'>read.fasta.pdb</span>(<span class='kw'>aln</span>)

<span class='kw'>gaps</span> <span class='op'>&lt;-</span> <span class='fu'>gap.inspect</span>(<span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>xyz</span>)

<span class='kw'>xyz</span> <span class='op'>&lt;-</span> <span class='fu'>fit.xyz</span>( fixed  = <span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>xyz</span>[<span class='fl'>1</span>,],
               mobile = <span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>xyz</span>,
               fixed.inds  = <span class='kw'>gaps</span><span class='op'>$</span><span class='kw'>f.inds</span>,
               mobile.inds = <span class='kw'>gaps</span><span class='op'>$</span><span class='kw'>f.inds</span> )

<span class='co'>#rmsd( xyz[, gaps$f.inds] )</span>
<span class='co'>#rmsd( pdbs$xyz[, gaps$f.inds] )</span>

<span class='co'>##-- Superpose again this time outputing PDBs</span>
<span class='kw'>xyz</span> <span class='op'>&lt;-</span> <span class='fu'>fit.xyz</span>( fixed = <span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>xyz</span>[<span class='fl'>1</span>,],
               mobile = <span class='kw'>pdbs</span>,
               fixed.inds  = <span class='kw'>gaps</span><span class='op'>$</span><span class='kw'>f.inds</span>,
               mobile.inds = <span class='kw'>gaps</span><span class='op'>$</span><span class='kw'>f.inds</span>,
               outpath = <span class='st'>"rough_fit"</span>,
               full.pdbs = <span class='fl'>TRUE</span>)

<span class='co'>##--- Fit two PDBs</span>
<span class='kw'>A</span> <span class='op'>&lt;-</span> <span class='fu'>read.pdb</span>(<span class='st'>"1bg2"</span>)
<span class='kw'>A.ind</span> <span class='op'>&lt;-</span> <span class='fu'>atom.select</span>(<span class='kw'>A</span>, resno=<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>256</span><span class='op'>:</span><span class='fl'>269</span>), elety=<span class='st'>'CA'</span>)

<span class='kw'>B</span> <span class='op'>&lt;-</span> <span class='fu'>read.pdb</span>(<span class='st'>"2kin"</span>)
<span class='kw'>B.ind</span> <span class='op'>&lt;-</span> <span class='fu'>atom.select</span>(<span class='kw'>B</span>, resno=<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>257</span><span class='op'>:</span><span class='fl'>270</span>), elety=<span class='st'>'CA'</span>)

<span class='kw'>xyz</span> <span class='op'>&lt;-</span> <span class='fu'>fit.xyz</span>(fixed=<span class='kw'>A</span><span class='op'>$</span><span class='kw'>xyz</span>, mobile=<span class='kw'>B</span><span class='op'>$</span><span class='kw'>xyz</span>,
               fixed.inds=<span class='kw'>A.ind</span><span class='op'>$</span><span class='kw'>xyz</span>,
               mobile.inds=<span class='kw'>B.ind</span><span class='op'>$</span><span class='kw'>xyz</span>)

<span class='co'># Write out moved PDB</span>
<span class='kw'>C</span> <span class='op'>&lt;-</span> <span class='kw'>B</span>; <span class='kw'>C</span><span class='op'>$</span><span class='kw'>xyz</span> <span class='op'>=</span> <span class='kw'>xyz</span>
<span class='fu'>write.pdb</span>(pdb=<span class='kw'>C</span>, file = <span class='st'>"moved.pdb"</span>)
<span class='co'># }</span>
</pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


