<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Biomolecular Visualization with PyMOL — pymol • Bio3D</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>

<!-- Bootstrap -->
<link href="../tidyverse.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>
<link href="../tidyverse-2.css" rel="stylesheet">

<!--optional theme
<link href="../.css" rel="stylesheet">
-->


<!-- docsearch -->
<script src="../docsearch.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous" />
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>


<meta property="og:title" content="Biomolecular Visualization with PyMOL — pymol" />
<meta property="og:description" content="Visualize Bio3D structure objects in PyMOL" />
<meta property="og:image" content="/logo.png" />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../index.html">
        bio3d</a>
        <div class="info hidden-xs hidden-sm">
<!--          <span class="partof">part of the <a href="https://tidyverse.org">tidyverse</a></span>
-->
          <span class="version version-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4-1.9000</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">INTRODUCTORY TUTORIALS</li>
    <li>
      <a href="../articles/online/install_vignette/Bio3D_install.html">Installing Bio3D</a>
    </li>
    <li>
      <a href="../articles/online/intro_vignette/Bio3D_introduction.html">Getting started with Bio3D</a>
    </li>
    <li>
      <a href="../articles/online/pdb_vignette/Bio3D_pdb.html">PDB structure manipulation and analysis</a>
    </li>
    <li>
      <a href="../articles/online/pca_vignette/Bio3D_pca.html">Comparative sequence and structure analysis</a>
    </li>
    <li>
      <a href="../articles/online/nma_vignette/Bio3D_nma.html">Enhanced Normal Modes Analysis</a>
    </li>
    <li>
      <a href="../articles/online/traj_vignette/Bio3D_md.html">Beginning trajectory analysis with Bio3D </a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">ADVANCED TUTORIALS</li>
    <li>
      <a href="../articles/online/enma_vignettes/Bio3D_nma-dhfr-partI.html">Ensemble NMA of E.coli DHFR structures</a>
    </li>
    <li>
      <a href="../articles/online/enma_vignettes/Bio3D_nma-dhfr-partII.html">Ensemble NMA across multiple species of DHFR</a>
    </li>
    <li>
      <a href="../articles/online/cna_vignette/Bio3D_cna-transducin.html">Correlation network analysis with Bio3D</a>
    </li>
    <li>
      <a href="../articles/online/cna_vignette/cna_vignette.spin.html">Protein structure network analysis with Bio3D ***</a>
    </li>
    <li>
      <a href="../articles/online/eddm_vignette/Bio3D_eddm.spin.html">Ensemble Difference Distance Matrix (eDDM) analysis</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">WEBAPP TUTORIALS</li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/download/webapp-tutorials?download=22:bio3d-webapp-pca">Online protein structure analysis with Bio3D-Web (PDF)</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/webapps">Online Normal Mode Analysis with Bio3D WebApps (To Update)</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/index.html">All vignettes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Documentation
  </a>
</li>
        <li>
  <a href="../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    News
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket"></span>
     
  </a>
</li>
      </ul>
      
      <form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
        </div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Biomolecular Visualization with PyMOL</h1>
    
    <div class="hidden name"><code>pymol.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Visualize Bio3D structure objects in PyMOL</p>
    </div>

    <pre class="usage"><span class='fu'>pymol</span>(<span class='kw'>...</span>)

<span class='co'># S3 method for pdbs</span>
<span class='fu'>pymol</span>(<span class='kw'>pdbs</span>, col=<span class='kw'>NULL</span>, as=<span class='st'>"ribbon"</span>, file=<span class='kw'>NULL</span>, type=<span class='st'>"script"</span>,
                     exefile=<span class='st'>"pymol"</span>, user.vec=<span class='kw'>NULL</span>, <span class='kw'>...</span>)

<span class='co'># S3 method for nma</span>
<span class='fu'>pymol</span>(<span class='kw'>...</span>)

<span class='co'># S3 method for pca</span>
<span class='fu'>pymol</span>(<span class='kw'>...</span>)

<span class='co'># S3 method for modes</span>
<span class='fu'>pymol</span>(<span class='kw'>modes</span>, mode=<span class='kw'>NULL</span>, file=<span class='kw'>NULL</span>, scale=<span class='fl'>5</span>, dual=<span class='fl'>FALSE</span>,
                     type=<span class='st'>"script"</span>, exefile=<span class='st'>"pymol"</span>, <span class='kw'>...</span>)

<span class='co'># S3 method for dccm</span>
<span class='fu'>pymol</span>(<span class='kw'>dccm</span>, <span class='kw'>pdb</span>, file=<span class='kw'>NULL</span>,
                     step=<span class='fl'>0.2</span>, omit=<span class='fl'>0.2</span>, radius = <span class='fl'>0.15</span>,
                     type=<span class='st'>"script"</span>, exefile=<span class='st'>"pymol"</span>, <span class='kw'>...</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>pdbs</th>
      <td><p>aligned C-alpha Cartesian coordinates as obtained with 
    <code>read.fasta.pdb</code> or <code>pdbaln</code>.</p></td>
    </tr>
    <tr>
      <th>col</th>
      <td><p>a single element character vector specifying the coloring
    of the structures. Options are: &#8216;index&#8217;, &#8216;index2&#8217;,
    &#8216;gaps&#8217;, &#8216;rmsf&#8217;, &#8216;user&#8217;.</p>
<p>Special cases: Provide a &#8216;core&#8217; object as obtained by
    <code>core.find</code> to color on the invariant core. Alternatively,
    provide a vector containing the color code for each structure in the
    &#8216;pdbs&#8217; object.</p></td>
    </tr>
    <tr>
      <th>user.vec</th>
      <td><p>User defined vector for coloring. Only used if <code>col="user"</code>.</p></td>
    </tr>
    <tr>
      <th>as</th>
      <td><p>show as &#8216;ribbon&#8217;, &#8216;cartoon&#8217;, &#8216;lines&#8217;, &#8216;putty&#8217;.</p></td>
    </tr>
    <tr>
      <th>file</th>
      <td><p>a single element character vector specifying the file name
    of the PyMOL session/script file.</p></td>
    </tr>
    <tr>
      <th>type</th>
      <td><p>a single element character vector specifying the output
    type: &#8216;script&#8217; generates a .pml script; &#8216;session&#8217;
    generates a .pse session file; &#8216;launch&#8217; launches pymol.</p></td>
    </tr>
    <tr>
      <th>exefile</th>
      <td><p>file path to the &#8216;PYMOL&#8217; program on your system (i.e.
    how is &#8216;PYMOL&#8217; invoked). If <code>NULL</code>, use OS-dependent 
    default path to the program.</p></td>
    </tr>
    <tr>
      <th>modes</th>
      <td><p>an object of class <code>nma</code> or <code>pca</code> as obtained
    from functions <code>nma</code> or <code>pca.xyz</code>.</p></td>
    </tr>
    <tr>
      <th>mode</th>
      <td><p>the mode number for which the vector field should be
    made.</p></td>
    </tr>
    <tr>
      <th>scale</th>
      <td><p>global scaling factor.</p></td>
    </tr>
    <tr>
      <th>dual</th>
      <td><p>logical, if TRUE mode vectors are also drawn in both
    direction.</p></td>
    </tr>
    <tr>
      <th>dccm</th>
      <td><p>an object of class <code>dccm</code> as obtained from
    function <code>dccm</code>.</p></td>
    </tr>
    <tr>
      <th>pdb</th>
      <td><p>an object of class <code>pdb</code> as obtained from
    function <code>read.pdb</code> or a numerical vector of Cartesian
    coordinates.</p></td>
    </tr>
    <tr>
      <th>step</th>
      <td><p>binning interval of cross-correlation coefficents.</p></td>
    </tr>
    <tr>
      <th>omit</th>
      <td><p>correlation coefficents with values (0-omit, 0+omit) will
    be omitted from visualization.</p></td>
    </tr>
    <tr>
      <th>radius</th>
      <td><p>numeric, radius of visualized correlation cylinders in
    PyMol. Alternatively, a matrix with the same dimesions as
    <code>dccm</code> can be provided, e.g. to draw cylinders with radii
    associated to the pairwise correlation value.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>arguments passed to function <code>pymol.modes</code> for
    &#8216;nma&#8217; and &#8216;pca&#8217; objects.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>These functions provides a convenient approach for the visualization of
  Bio3D objects in PyMOL. See examples for more details.</p>
<p>DCCM PyMOL visualization:
  This function generates a PyMOL (python) script that will draw colored
  lines between (anti)correlated residues. The PyMOL script file is
  stored in the working directory with filename &#8220;R.py&#8221;.
  PyMOL will only be launched (and opened)  when using argument
  &#8216;type='launch'&#8217;. Alternatively a PDB file with CONECT records
  will be generated (when argument <code>type='pdb'</code>).</p>   
<p>For the PyMOL version, PyMOL CGO objects are generated - each object
  representing a range of correlation values (corresponding to the
  actual correlation values as found in the correlation
  matrix). E.g. the PyMOL object with name &#8220;cor_-1_-08&#8221; would
  display all pairs of correlations with values between -1 and -0.8.</p>
<p>NMA / PCA PyMOL vector field visualization:
  This function generates a PyMOL (python) script for drawing
  mode vectors on a PDB structure. The PyMOL script file is stored in
  the working directory with filename &#8220;R.py&#8221;.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Called for its action</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Skjaerven, L. et al. (2014) <em>BMC Bioinformatics</em> <b>15</b>, 399.
  Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695--2696.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Lars Skjaerven</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code>view</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>if</span> (<span class='fl'>FALSE</span>) {

<span class='co'>##- pymol with a 'pdbs' object</span>
<span class='fu'><a href='https://rdrr.io/r/base/attach.html'>attach</a></span>(<span class='kw'>transducin</span>)

<span class='co'># build a pymol session containing all structures in the PDBs object</span>
<span class='fu'>pymol</span>(<span class='kw'>pdbs</span>)

<span class='co'># color by invariant core (</span>
<span class='co'># core &lt;- core.find(pdbs)</span>
<span class='fu'>pymol</span>(<span class='kw'>pdbs</span>, col=<span class='kw'>core</span>)

<span class='co'># color by RMSF</span>
<span class='fu'>pymol</span>(<span class='kw'>pdbs</span>, col=<span class='st'>"rmsf"</span>)

<span class='co'># color by a user defined vector</span>
<span class='co'># For example, colored by averaged contact density around each residue</span>
<span class='kw'>cm</span> <span class='op'>&lt;-</span> <span class='fu'>cmap</span>(<span class='kw'>pdbs</span>, binary=<span class='fl'>FALSE</span>)
<span class='kw'>vec</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colSums.html'>rowSums</a></span>(<span class='kw'>cm</span>, na.rm=<span class='fl'>TRUE</span>)
<span class='fu'>pymol</span>(<span class='kw'>pdbs</span>, col=<span class='st'>"user"</span>, user.vec=<span class='kw'>vec</span>)

<span class='co'># color by clustering</span>
<span class='kw'>rd</span> <span class='op'>&lt;-</span> <span class='fu'>rmsd</span>(<span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>xyz</span>)
<span class='kw'>hc</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/hclust.html'>hclust</a></span>(<span class='fu'><a href='https://rdrr.io/r/stats/dist.html'>as.dist</a></span>(<span class='kw'>rd</span>))
<span class='kw'>grps</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/cutree.html'>cutree</a></span>(<span class='kw'>hc</span>, k=<span class='fl'>3</span>)
<span class='fu'>pymol</span>(<span class='kw'>pdbs</span>, col=<span class='kw'>grps</span>)



<span class='co'>##- pymol with a 'dccm' object</span>
<span class='co'>## Fetch stucture</span>
<span class='kw'>pdb</span> <span class='op'>&lt;-</span> <span class='fu'>read.pdb</span>( <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"examples/1hel.pdb"</span>, package=<span class='st'>"bio3d"</span>) )

<span class='co'>## Calculate normal modes</span>
<span class='kw'>modes</span> <span class='op'>&lt;-</span> <span class='fu'>nma</span>(<span class='kw'>pdb</span>)

<span class='co'>## Calculate correlation matrix</span>
<span class='kw'>cm</span> <span class='op'>&lt;-</span> <span class='fu'>dccm.nma</span>(<span class='kw'>modes</span>)

<span class='fu'>pymol</span>(<span class='kw'>cm</span>, <span class='kw'>modes</span><span class='op'>$</span><span class='kw'>xyz</span>)



<span class='co'>##- pymol with a 'nma' or 'pca' object</span>
<span class='fu'>pymol</span>(<span class='kw'>modes</span>, mode=<span class='fl'>7</span>)

<span class='fu'><a href='https://rdrr.io/r/base/detach.html'>detach</a></span>(<span class='kw'>transducin</span>)
}</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="tidyverse">
  <p>Authored by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="author">
  <p>
    Built with &#10084;&#65039; and <a href="https://pkgdown.r-lib.org">pkgdown</a> &copy; 2020.
  </p>
</div>

      </footer>
   </div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script>
<script>
  docsearch({
    
    
    apiKey: '1270c4079b26a138795263974bbf302d',
    indexName: 'tidyverse',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>



  </body>
</html>


