<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Read PDB File — read.pdb • Bio3D</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>

<!-- Bootstrap -->
<link href="../tidyverse.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>
<link href="../tidyverse-2.css" rel="stylesheet">

<!--optional theme
<link href="../.css" rel="stylesheet">
-->


<!-- docsearch -->
<script src="../docsearch.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous" />
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>


<meta property="og:title" content="Read PDB File — read.pdb" />
<meta property="og:description" content="Read a Protein Data Bank (PDB) coordinate file." />
<meta property="og:image" content="/logo.png" />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../index.html">
        bio3d</a>
        <div class="info hidden-xs hidden-sm">
<!--          <span class="partof">part of the <a href="https://tidyverse.org">tidyverse</a></span>
-->
          <span class="version version-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4-1.9000</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">INTRODUCTORY TUTORIALS</li>
    <li>
      <a href="../articles/online/install_vignette/Bio3D_install.html">Installing Bio3D</a>
    </li>
    <li>
      <a href="../articles/online/intro_vignette/Bio3D_introduction.html">Getting started with Bio3D</a>
    </li>
    <li>
      <a href="../articles/online/pdb_vignette/Bio3D_pdb.html">PDB structure manipulation and analysis</a>
    </li>
    <li>
      <a href="../articles/online/pca_vignette/Bio3D_pca.html">Comparative sequence and structure analysis</a>
    </li>
    <li>
      <a href="../articles/online/nma_vignette/Bio3D_nma.html">Enhanced Normal Modes Analysis</a>
    </li>
    <li>
      <a href="../articles/online/traj_vignette/Bio3D_md.html">Beginning trajectory analysis with Bio3D </a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">ADVANCED TUTORIALS</li>
    <li>
      <a href="../articles/online/enma_vignettes/Bio3D_nma-dhfr-partI.html">Ensemble NMA of E.coli DHFR structures</a>
    </li>
    <li>
      <a href="../articles/online/enma_vignettes/Bio3D_nma-dhfr-partII.html">Ensemble NMA across multiple species of DHFR</a>
    </li>
    <li>
      <a href="../articles/online/cna_vignette/Bio3D_cna-transducin.html">Correlation network analysis with Bio3D</a>
    </li>
    <li>
      <a href="../articles/online/cna_vignette/cna_vignette.spin.html">Protein structure network analysis with Bio3D ***</a>
    </li>
    <li>
      <a href="../articles/online/eddm_vignette/Bio3D_eddm.spin.html">Ensemble Difference Distance Matrix (eDDM) analysis</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">WEBAPP TUTORIALS</li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/download/webapp-tutorials?download=22:bio3d-webapp-pca">Online protein structure analysis with Bio3D-Web (PDF)</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/webapps">Online Normal Mode Analysis with Bio3D WebApps (To Update)</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/index.html">All vignettes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Documentation
  </a>
</li>
        <li>
  <a href="../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    News
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket"></span>
     
  </a>
</li>
      </ul>
      
      <form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
        </div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Read PDB File</h1>
    
    <div class="hidden name"><code>read.pdb.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Read a Protein Data Bank (PDB) coordinate file.</p>
    </div>

    <pre class="usage"><span class='fu'>read.pdb</span>(<span class='kw'>file</span>, maxlines = <span class='op'>-</span><span class='fl'>1</span>, multi = <span class='fl'>FALSE</span>, rm.insert = <span class='fl'>FALSE</span>,
         rm.alt = <span class='fl'>TRUE</span>, ATOM.only = <span class='fl'>FALSE</span>, hex = <span class='fl'>FALSE</span>, verbose = <span class='fl'>TRUE</span>)

<span class='fu'>read.pdb2</span>(<span class='kw'>file</span>, maxlines = <span class='op'>-</span><span class='fl'>1</span>, multi = <span class='fl'>FALSE</span>, rm.insert = <span class='fl'>FALSE</span>,
         rm.alt = <span class='fl'>TRUE</span>, ATOM.only = <span class='fl'>FALSE</span>, verbose = <span class='fl'>TRUE</span>)

<span class='co'># S3 method for pdb</span>
<span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span>(<span class='kw'>x</span>, printseq=<span class='fl'>TRUE</span>, <span class='kw'>...</span>)

<span class='co'># S3 method for pdb</span>
<span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span>(<span class='kw'>object</span>, printseq=<span class='fl'>FALSE</span>, <span class='kw'>...</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>file</th>
      <td><p>a single element character vector containing the name of
    the PDB file to be read, or the four letter PDB identifier for
    online file access.</p></td>
    </tr>
    <tr>
      <th>maxlines</th>
      <td><p>the maximum number of lines to read before giving up
    with large files. By default if will read up to the end of input on
    the connection.</p></td>
    </tr>
    <tr>
      <th>multi</th>
      <td><p>logical, if TRUE multiple ATOM records are read for all
    models in multi-model files and their coordinates returned.</p></td>
    </tr>
    <tr>
      <th>rm.insert</th>
      <td><p>logical, if TRUE PDB insert records are ignored.</p></td>
    </tr>
    <tr>
      <th>rm.alt</th>
      <td><p>logical, if TRUE PDB alternate records are ignored.</p></td>
    </tr>
    <tr>
      <th>ATOM.only</th>
      <td><p>logical, if TRUE only ATOM/HETATM records are stored.
    Useful for speed enhancements with large files where secondary structure, 
    biological unit and other remark records are not required.</p></td>
    </tr>
    <tr>
      <th>hex</th>
      <td><p>logical, if TRUE enable parsing of hexadecimal atom
    numbers (&gt; 99.999) and residue numbers (&gt; 9.999) (e.g. from VMD). Note
    that numbering is assumed to be consecutive (with no missing numbers)
    and the hexadecimals should start at atom number 100.000 and residue
    number 10.000 and proceed to the end of file.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>print details of the reading process.</p></td>
    </tr>
    <tr>
      <th>x</th>
      <td><p>a PDB structure object obtained from
    <code>read.pdb</code>.</p></td>
    </tr>
    <tr>
      <th>object</th>
      <td><p>a PDB structure object obtained from
    <code>read.pdb</code>.</p></td>
    </tr>
    <tr>
      <th>printseq</th>
      <td><p>logical, if TRUE the PDB ATOM sequence will be printed
    to the screen. See also <code>pdbseq</code>.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>additional arguments to &#8216;print&#8217;.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p><code>read.pdb</code> is a re-implementation (using Rcpp) of the slower but
  more tested R implementation of the same function (called
  <code>read.pdb2</code> since bio3d-v2.3).</p>  
<p><code>maxlines</code> may be set so as to restrict the reading to a portion
  of input files. Note that the preferred means of reading large 
  multi-model files is via binary DCD or NetCDF format trajectory files 
  (see the <code>read.dcd</code> and <code>read.ncdf</code> functions).</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns a list of class <code>"pdb"</code> with the following components:</p>
<dt>atom</dt><dd><p>a data.frame containing all atomic coordinate ATOM and 
    HETATM data, with a row per ATOM/HETATM and a column per record type.  
    See below for details of the record type naming convention (useful for 
    accessing columns).</p></dd>
  <dt>helix </dt><dd><p>&#8216;start&#8217;, &#8216;end&#8217; and &#8216;length&#8217; of H
    type sse, where start and end are residue numbers &#8220;resno&#8221;.</p></dd>
  <dt>sheet </dt><dd><p>&#8216;start&#8217;, &#8216;end&#8217; and &#8216;length&#8217; of E
    type sse, where start and end are residue numbers &#8220;resno&#8221;.</p></dd>
  <dt>seqres </dt><dd><p>sequence from SEQRES field.</p></dd>
  <dt>xyz </dt><dd><p>a numeric matrix of class <code>"xyz"</code> containing the
    ATOM and HETATM coordinate data.</p></dd>
  <dt>calpha </dt><dd><p>logical vector with length equal to <code><a href='https://rdrr.io/r/base/nrow.html'>nrow(atom)</a></code>
    with TRUE values indicating a C-alpha &#8220;elety&#8221;.</p></dd>
  <dt>remark </dt><dd><p>a list object containing information taken from 'REMARK'
    records of a <code>"pdb"</code>. It can be used for building biological units
    (See <code>biounit</code>).</p></dd>
  <dt>call </dt><dd><p>the matched call.</p></dd>

    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695--2696.</p>  
<p>For a description of PDB format (version3.3) see:<br />
  <a href='http://www.wwpdb.org/documentation/format33/v3.3.html'>http://www.wwpdb.org/documentation/format33/v3.3.html</a>.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Barry Grant</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>For both <code>atom</code> and <code>het</code> list components the column names can be
  used as a convenient means of data access, namely:
  Atom serial number &#8220;eleno&#8221; ,
  Atom type &#8220;elety&#8221;,
  Alternate location indicator &#8220;alt&#8221;,
  Residue name &#8220;resid&#8221;,
  Chain identifier &#8220;chain&#8221;,
  Residue sequence number &#8220;resno&#8221;,
  Code for insertion of residues &#8220;insert&#8221;,
  Orthogonal coordinates &#8220;x&#8221;,
  Orthogonal coordinates &#8220;y&#8221;,
  Orthogonal coordinates &#8220;z&#8221;,
  Occupancy &#8220;o&#8221;, and
  Temperature factor &#8220;b&#8221;.
  See examples for further details.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code>atom.select</code>, <code>write.pdb</code>,
  <code>trim.pdb</code>, <code>cat.pdb</code>, 
  <code>read.prmtop</code>, <code>as.pdb</code>,
  <code>read.dcd</code>, <code>read.ncdf</code>, 
  <code>read.fasta.pdb</code>, <code>read.fasta</code>, 
  <code>biounit</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## Read a PDB file from the RCSB online database</span>
<span class='co'>#pdb &lt;- read.pdb("4q21")</span>

<span class='co'>## Read a PDB file from those included with the package</span>
<span class='kw'>pdb</span> <span class='op'>&lt;-</span> <span class='fu'>read.pdb</span>( <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"examples/1hel.pdb"</span>, package=<span class='st'>"bio3d"</span>) )

<span class='co'>## Print a brief composition summary</span>
<span class='kw'>pdb</span></div><div class='output co'>#&gt; 
#&gt;  Call:  read.pdb(file = system.file("examples/1hel.pdb", package = "bio3d"))
#&gt; 
#&gt;    Total Models#: 1
#&gt;      Total Atoms#: 1001,  XYZs#: 3003  Chains#: 1  (values: A)
#&gt; 
#&gt;      Protein Atoms#: 1001  (residues/Calpha atoms#: 129)
#&gt;      Nucleic acid Atoms#: 0  (residues/phosphate atoms#: 0)
#&gt; 
#&gt;      Non-protein/nucleic Atoms#: 0  (residues: 0)
#&gt;      Non-protein/nucleic resid values: [ none ]
#&gt; 
#&gt;    Protein sequence:
#&gt;       KVFGRCELAAAMKRHGLDNYRGYSLGNWVCAAKFESNFNTQATNRNTDGSTDYGILQINS
#&gt;       RWWCNDGRTPGSRNLCNIPCSALLSSDITASVNCAKKIVSDGNGMNAWVAWRNRCKGTDV
#&gt;       QAWIRGCRL
#&gt; 
#&gt; + attr: atom, xyz, seqres, helix, sheet,
#&gt;         calpha, remark, call</div><div class='input'>
<span class='co'>## Examine the storage format (or internal *str*ucture)</span>
<span class='fu'><a href='https://rdrr.io/r/utils/str.html'>str</a></span>(<span class='kw'>pdb</span>)</div><div class='output co'>#&gt; List of 8
#&gt;  $ atom  :'data.frame':	1001 obs. of  16 variables:
#&gt;   ..$ type  : chr [1:1001] "ATOM" "ATOM" "ATOM" "ATOM" ...
#&gt;   ..$ eleno : int [1:1001] 1 2 3 4 5 6 7 8 9 10 ...
#&gt;   ..$ elety : chr [1:1001] "N" "CA" "C" "O" ...
#&gt;   ..$ alt   : chr [1:1001] NA NA NA NA ...
#&gt;   ..$ resid : chr [1:1001] "LYS" "LYS" "LYS" "LYS" ...
#&gt;   ..$ chain : chr [1:1001] "A" "A" "A" "A" ...
#&gt;   ..$ resno : int [1:1001] 1 1 1 1 1 1 1 1 1 2 ...
#&gt;   ..$ insert: chr [1:1001] NA NA NA NA ...
#&gt;   ..$ x     : num [1:1001] 3.294 2.388 2.438 2.406 0.949 ...
#&gt;   ..$ y     : num [1:1001] 10.2 10.5 12 12.9 10.1 ...
#&gt;   ..$ z     : num [1:1001] 10.27 9.17 8.89 9.81 9.56 ...
#&gt;   ..$ o     : num [1:1001] 1 1 1 1 1 1 1 1 1 1 ...
#&gt;   ..$ b     : num [1:1001] 11.18 9.68 14 14 13.29 ...
#&gt;   ..$ segid : chr [1:1001] NA NA NA NA ...
#&gt;   ..$ elesy : chr [1:1001] "N" "C" "C" "O" ...
#&gt;   ..$ charge: chr [1:1001] NA NA NA NA ...
#&gt;  $ xyz   : 'xyz' num [1, 1:3003] 3.29 10.16 10.27 2.39 10.53 ...
#&gt;  $ seqres: Named chr [1:129] "LYS" "VAL" "PHE" "GLY" ...
#&gt;   ..- attr(*, "names")= chr [1:129] "A" "A" "A" "A" ...
#&gt;  $ helix :List of 4
#&gt;   ..$ start: Named num [1:7] 4 25 79 88 104 109 120
#&gt;   .. ..- attr(*, "names")= chr [1:7] "" "" "" "" ...
#&gt;   ..$ end  : Named num [1:7] 16 34 84 100 108 114 124
#&gt;   .. ..- attr(*, "names")= chr [1:7] "" "" "" "" ...
#&gt;   ..$ chain: chr [1:7] "A" "A" "A" "A" ...
#&gt;   ..$ type : chr [1:7] "1" "1" "5" "1" ...
#&gt;  $ sheet :List of 4
#&gt;   ..$ start: Named num [1:3] 42 49 56
#&gt;   .. ..- attr(*, "names")= chr [1:3] "" "" ""
#&gt;   ..$ end  : Named num [1:3] 46 54 60
#&gt;   .. ..- attr(*, "names")= chr [1:3] "" "" ""
#&gt;   ..$ chain: chr [1:3] "A" "A" "A"
#&gt;   ..$ sense: chr [1:3] "0" "-1" "-1"
#&gt;  $ calpha: logi [1:1001] FALSE TRUE FALSE FALSE FALSE FALSE ...
#&gt;  $ remark:List of 1
#&gt;   ..$ biomat:List of 4
#&gt;   .. ..$ num   : int 1
#&gt;   .. ..$ chain :List of 1
#&gt;   .. .. ..$ : chr "A"
#&gt;   .. ..$ mat   :List of 1
#&gt;   .. .. ..$ :List of 1
#&gt;   .. .. .. ..$ A: num [1:3, 1:4] 1 0 0 0 1 0 0 0 1 0 ...
#&gt;   .. ..$ method: chr "AUTHOR"
#&gt;  $ call  : language read.pdb(file = system.file("examples/1hel.pdb", package = "bio3d"))
#&gt;  - attr(*, "class")= chr [1:2] "pdb" "sse"</div><div class='input'>
<span class='co'>## Print data for the first four atom</span>
<span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>atom</span>[<span class='fl'>1</span><span class='op'>:</span><span class='fl'>4</span>,]</div><div class='output co'>#&gt;   type eleno elety  alt resid chain resno insert     x      y      z o     b
#&gt; 1 ATOM     1     N &lt;NA&gt;   LYS     A     1   &lt;NA&gt; 3.294 10.164 10.266 1 11.18
#&gt; 2 ATOM     2    CA &lt;NA&gt;   LYS     A     1   &lt;NA&gt; 2.388 10.533  9.168 1  9.68
#&gt; 3 ATOM     3     C &lt;NA&gt;   LYS     A     1   &lt;NA&gt; 2.438 12.049  8.889 1 14.00
#&gt; 4 ATOM     4     O &lt;NA&gt;   LYS     A     1   &lt;NA&gt; 2.406 12.898  9.815 1 14.00
#&gt;   segid elesy charge
#&gt; 1  &lt;NA&gt;     N   &lt;NA&gt;
#&gt; 2  &lt;NA&gt;     C   &lt;NA&gt;
#&gt; 3  &lt;NA&gt;     C   &lt;NA&gt;
#&gt; 4  &lt;NA&gt;     O   &lt;NA&gt;</div><div class='input'>
<span class='co'>## Print some coordinate data</span>
<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span>(<span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>atom</span>[, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"x"</span>,<span class='st'>"y"</span>,<span class='st'>"z"</span>)])</div><div class='output co'>#&gt;        x      y      z
#&gt; 1  3.294 10.164 10.266
#&gt; 2  2.388 10.533  9.168
#&gt; 3  2.438 12.049  8.889
#&gt; 4  2.406 12.898  9.815
#&gt; 5  0.949 10.101  9.559
#&gt; 6 -0.050 10.621  8.573</div><div class='input'>
<span class='co'>## Or coordinates as a numeric vector</span>
<span class='co'>#head(pdb$xyz)</span>

<span class='co'>## Print C-alpha coordinates (can also use 'atom.select' function)</span>
<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span>(<span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>atom</span>[<span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>calpha</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"resid"</span>,<span class='st'>"elety"</span>,<span class='st'>"x"</span>,<span class='st'>"y"</span>,<span class='st'>"z"</span>)])</div><div class='output co'>#&gt;    resid elety      x      y     z
#&gt; 2    LYS    CA  2.388 10.533 9.168
#&gt; 11   VAL    CA  2.524 13.840 7.282
#&gt; 18   PHE    CA -1.021 15.173 7.169
#&gt; 29   GLY    CA -2.645 17.273 4.511
#&gt; 33   ARG    CA -4.106 20.560 5.674
#&gt; 44   CYS    CA -7.767 19.572 5.189</div><div class='input'><span class='kw'>inds</span> <span class='op'>&lt;-</span> <span class='fu'>atom.select</span>(<span class='kw'>pdb</span>, elety=<span class='st'>"CA"</span>)
<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span>( <span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>atom</span>[<span class='kw'>inds</span><span class='op'>$</span><span class='kw'>atom</span>, ] )</div><div class='output co'>#&gt;    type eleno elety  alt resid chain resno insert      x      y     z o     b
#&gt; 2  ATOM     2    CA &lt;NA&gt;   LYS     A     1   &lt;NA&gt;  2.388 10.533 9.168 1  9.68
#&gt; 11 ATOM    11    CA &lt;NA&gt;   VAL     A     2   &lt;NA&gt;  2.524 13.840 7.282 1 10.02
#&gt; 18 ATOM    18    CA &lt;NA&gt;   PHE     A     3   &lt;NA&gt; -1.021 15.173 7.169 1 15.34
#&gt; 29 ATOM    29    CA &lt;NA&gt;   GLY     A     4   &lt;NA&gt; -2.645 17.273 4.511 1 13.97
#&gt; 33 ATOM    33    CA &lt;NA&gt;   ARG     A     5   &lt;NA&gt; -4.106 20.560 5.674 1 11.63
#&gt; 44 ATOM    44    CA &lt;NA&gt;   CYS     A     6   &lt;NA&gt; -7.767 19.572 5.189 1 12.93
#&gt;    segid elesy charge
#&gt; 2   &lt;NA&gt;     C   &lt;NA&gt;
#&gt; 11  &lt;NA&gt;     C   &lt;NA&gt;
#&gt; 18  &lt;NA&gt;     C   &lt;NA&gt;
#&gt; 29  &lt;NA&gt;     C   &lt;NA&gt;
#&gt; 33  &lt;NA&gt;     C   &lt;NA&gt;
#&gt; 44  &lt;NA&gt;     C   &lt;NA&gt;</div><div class='input'>
<span class='co'>## The atom.select() function returns 'indices' (row numbers)</span>
<span class='co'>## that can be used for accessing subsets of PDB objects, e.g.</span>
<span class='kw'>inds</span> <span class='op'>&lt;-</span> <span class='fu'>atom.select</span>(<span class='kw'>pdb</span>,<span class='st'>"ligand"</span>)
<span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>atom</span>[<span class='kw'>inds</span><span class='op'>$</span><span class='kw'>atom</span>,]</div><div class='output co'>#&gt;  [1] type   eleno  elety  alt    resid  chain  resno  insert x      y     
#&gt; [11] z      o      b      segid  elesy  charge
#&gt; &lt;0 rows&gt; (or 0-length row.names)</div><div class='input'><span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>xyz</span>[<span class='kw'>inds</span><span class='op'>$</span><span class='kw'>xyz</span>]</div><div class='output co'>#&gt; numeric(0)</div><div class='input'>
<span class='co'>## See the help page for atom.select() function for more details.</span>


<span class='co'>if</span> (<span class='fl'>FALSE</span>) {
<span class='co'>## Print SSE data for helix and sheet,</span>
<span class='co'>##  see also dssp() and stride() functions</span>
<span class='fu'>print.sse</span>(<span class='kw'>pdb</span>)
<span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>helix</span>
<span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>sheet</span><span class='op'>$</span><span class='kw'>start</span>
  
<span class='co'>## Print SEQRES data</span>
<span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>seqres</span>

<span class='co'>## SEQRES as one letter code</span>
<span class='fu'>aa321</span>(<span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>seqres</span>)

<span class='co'>## Where is the P-loop motif in the ATOM sequence</span>
<span class='kw'>inds.seq</span> <span class='op'>&lt;-</span> <span class='fu'>motif.find</span>(<span class='st'>"G....GKT"</span>, <span class='fu'>pdbseq</span>(<span class='kw'>pdb</span>))
<span class='fu'>pdbseq</span>(<span class='kw'>pdb</span>)[<span class='kw'>inds.seq</span>]

<span class='co'>## Where is it in the structure</span>
<span class='kw'>inds.pdb</span> <span class='op'>&lt;-</span> <span class='fu'>atom.select</span>(<span class='kw'>pdb</span>,resno=<span class='kw'>inds.seq</span>, elety=<span class='st'>"CA"</span>)
<span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>atom</span>[<span class='kw'>inds.pdb</span><span class='op'>$</span><span class='kw'>atom</span>,]
<span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>xyz</span>[<span class='kw'>inds.pdb</span><span class='op'>$</span><span class='kw'>xyz</span>]

<span class='co'>## View in interactive 3D mode</span>
<span class='co'>#view(pdb)</span>
}
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="tidyverse">
  <p>Authored by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="author">
  <p>
    Built with &#10084;&#65039; and <a href="https://pkgdown.r-lib.org">pkgdown</a> &copy; 2020.
  </p>
</div>

      </footer>
   </div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script>
<script>
  docsearch({
    
    
    apiKey: '1270c4079b26a138795263974bbf302d',
    indexName: 'tidyverse',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>



  </body>
</html>


