<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Read Aligned Structure Data — read.fasta.pdb • Bio3D</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Read Aligned Structure Data — read.fasta.pdb" />
<meta property="og:description" content="Read aligned PDB structures and store their C-alpha atom data, including xyz
  coordinates, residue numbers, residue type and B-factors." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Bio3D</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4-1.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-info"></span>
     
    Getting Started
  </a>
</li>
<li>
  <a href="../index.html">
    <span class="fa fa-book"></span>
     
    Tutorials
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Functions
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../CODE_OF_CONDUCT.html">
    <span class="fa fa-newspaper-o"></span>
     
    News &amp; Updates
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Read Aligned Structure Data</h1>
    
    <div class="hidden name"><code>read.fasta.pdb.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Read aligned PDB structures and store their C-alpha atom data, including xyz
  coordinates, residue numbers, residue type and B-factors.</p>
    </div>

    <pre class="usage"><span class='fu'>read.fasta.pdb</span>(<span class='kw'>aln</span>, prefix = <span class='st'>""</span>, pdbext = <span class='st'>""</span>, fix.ali = <span class='fl'>FALSE</span>,
             pdblist=<span class='kw'>NULL</span>, ncore = <span class='fl'>1</span>, nseg.scale = <span class='fl'>1</span>, progress = <span class='kw'>NULL</span>, <span class='kw'>...</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>aln</th>
      <td><p>an alignment data structure obtained with
    <code>read.fasta</code>.</p></td>
    </tr>
    <tr>
      <th>prefix</th>
      <td><p>prefix to aln$id to locate PDB files.</p></td>
    </tr>
    <tr>
      <th>pdbext</th>
      <td><p>the file name extention of the PDB files.</p></td>
    </tr>
    <tr>
      <th>fix.ali</th>
      <td><p>logical, if TRUE check consistence between <code>$ali</code> 
    and <code>$resno</code>, and correct <code>$ali</code> if they don't match.</p></td>
    </tr>
    <tr>
      <th>pdblist</th>
      <td><p>an optional list of <code>pdb</code> objects with sequence
    corresponding to the alignments in <code>aln</code>. Primarily used
    through function <code>pdbaln</code> when the PDB objects already exists
    (avoids reading PDBs from file).</p></td>
    </tr>
    <tr>
      <th>ncore </th>
      <td><p>number of CPU cores used to do the calculation.
    <code>ncore&gt;1</code> requires package &#8216;parallel&#8217; installed.</p></td>
    </tr>
    <tr>
      <th>nseg.scale </th>
      <td><p>split input data into specified number of segments
    prior to running multiple core calculation. See <code>fit.xyz</code>.</p></td>
    </tr>
    <tr>
      <th>progress</th>
      <td><p>progress bar for use with shiny web app.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>other parameters for <code>read.pdb</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The input <code>aln</code>, produced with <code>read.fasta</code>, must
  have identifers (i.e. sequence names) that match the PDB file
  names. For example the sequence corresponding to the structure
  &#8220;1bg2.pdb&#8221; should have the identifer &#8216;1bg2&#8217;. See
  examples below.</p>  
<p>Sequence miss-matches will generate errors.  Thus, care should be taken
  to ensure that the sequences in the alignment match the sequences in
  their associated PDB files.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns a list of class <code>"pdbs"</code> with the following five
  components:</p>
<dt>xyz</dt><dd><p>numeric matrix of aligned C-alpha coordinates.</p></dd>
  <dt>resno</dt><dd><p>character matrix of aligned residue numbers.</p></dd>
  <dt>b</dt><dd><p>numeric matrix of aligned B-factor values.</p></dd>
  <dt>chain</dt><dd><p>character matrix of aligned chain identifiers.</p></dd>
  <dt>id</dt><dd><p>character vector of PDB sequence/structure names.</p></dd>
  <dt>ali</dt><dd><p>character matrix of aligned sequences.</p></dd>
  <dt>resid</dt><dd><p>character matrix of aligned 3-letter residue names.</p></dd>
  <dt>sse</dt><dd><p>character matrix of aligned helix and strand secondary
    structure elements as defined in each PDB file.</p></dd>  
  <dt>call</dt><dd><p>the matched call.</p></dd>

    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695--2696.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Barry Grant</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>The sequence character &#8216;X&#8217; is useful for masking unusual
   or unknown residues, as it can match any other residue type.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code>read.fasta</code>, <code>read.pdb</code>,
  <code>core.find</code>, <code>fit.xyz</code>,
  <code>read.all</code>, <code>pymol.pdbs</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='co'># \donttest{</span>
<span class='co'># Redundant testing excluded</span>

<span class='co'># Read sequence alignment</span>
<span class='kw'>file</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"examples/kif1a.fa"</span>,package=<span class='st'>"bio3d"</span>)
<span class='kw'>aln</span>  <span class='op'>&lt;-</span> <span class='fu'>read.fasta</span>(<span class='kw'>file</span>)

<span class='co'># Read aligned PDBs</span>
<span class='kw'>pdbs</span> <span class='op'>&lt;-</span> <span class='fu'>read.fasta.pdb</span>(<span class='kw'>aln</span>)

<span class='co'># Structure/sequence names/ids</span>
<span class='fu'><a href='https://rdrr.io/r/base/basename.html'>basename</a></span>( <span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>id</span> )

<span class='co'># Alignment positions 335 to 339</span>
<span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>ali</span>[,<span class='fl'>335</span><span class='op'>:</span><span class='fl'>339</span>]
<span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>resid</span>[,<span class='fl'>335</span><span class='op'>:</span><span class='fl'>339</span>]
<span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>resno</span>[,<span class='fl'>335</span><span class='op'>:</span><span class='fl'>339</span>]
<span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>b</span>[,<span class='fl'>335</span><span class='op'>:</span><span class='fl'>339</span>]

<span class='co'># Alignment C-alpha coordinates for these positions</span>
<span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>xyz</span>[, <span class='fu'>atom2xyz</span>(<span class='fl'>335</span><span class='op'>:</span><span class='fl'>339</span>)]

<span class='co'># See 'fit.xyz()' function for actual coordinate superposition</span>
<span class='co'>#  e.g. fit to first structure</span>
<span class='co'># xyz &lt;- fit.xyz(pdbs$xyz[1,], pdbs)</span>
<span class='co'># xyz[, atom2xyz(335:339)]</span>

<span class='co'># }</span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


