<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Read Aligned Structure Data — read.all • Bio3D</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>

<!-- Bootstrap -->
<link href="../tidyverse.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>
<link href="../tidyverse-2.css" rel="stylesheet">

<!--optional theme
<link href="../.css" rel="stylesheet">
-->


<!-- docsearch -->
<script src="../docsearch.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous" />
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>


<meta property="og:title" content="Read Aligned Structure Data — read.all" />
<meta property="og:description" content="Read aligned PDB structures and store their equalvalent atom data, including xyz
  coordinates, residue numbers, residue type and B-factors." />
<meta property="og:image" content="/logo.png" />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../index.html">
        bio3d</a>
        <div class="info hidden-xs hidden-sm">
<!--          <span class="partof">part of the <a href="https://tidyverse.org">tidyverse</a></span>
-->
          <span class="version version-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4-1.9000</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">INTRODUCTORY TUTORIALS</li>
    <li>
      <a href="../articles/online/install_vignette/Bio3D_install.html">Installing Bio3D</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/user-guide">New user guide</a>
    </li>
    <li>
      <a href="../articles/online/intro_vignette/Bio3D_introduction.html">Getting started with Bio3D</a>
    </li>
    <li>
      <a href="../articles/online/pdb_vignette/Bio3D_pdb.html">PDB structure manipulation and analysis</a>
    </li>
    <li>
      <a href="../articles/online/pca_vignette/Bio3D_pca.html">Comparative sequence and structure analysis</a>
    </li>
    <li>
      <a href="../articles/online/nma_vignette/Bio3D_nma.html">Enhanced Normal Modes Analysis</a>
    </li>
    <li>
      <a href="../articles/online/traj_vignette/Bio3D_md.html">Beginning trajectory analysis with Bio3D </a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">ADVANCED TUTORIALS</li>
    <li>
      <a href="../articles/online/enma_vignettes/Bio3D_nma-dhfr-partI.html">Ensemble NMA of E.coli DHFR structures</a>
    </li>
    <li>
      <a href="../articles/online/enma_vignettes/Bio3D_nma-dhfr-partII.html">Ensemble NMA across multiple species of DHFR</a>
    </li>
    <li>
      <a href="../articles/online/cna_vignette/cna_vignette.spin.html">Protein structure network analysis with Bio3D</a>
    </li>
    <li>
      <a href="../articles/online/eddm_vignette/Bio3D_eddm.spin.html">Ensemble Difference Distance Matrix (eDDM) analysis</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">WEBAPP TUTORIALS</li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/download/webapp-tutorials?download=22:bio3d-webapp-pca">Online protein structure analysis with Bio3D-Web (PDF)</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/webapps">Online Normal Mode Analysis with Bio3D WebApps (To Update)</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/index.html">All vignettes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Documentation
  </a>
</li>
        <li>
  <a href="../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    News
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket"></span>
     
  </a>
</li>
      </ul>
      
      <form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
        </div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Read Aligned Structure Data</h1>
    
    <div class="hidden name"><code>read.all.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Read aligned PDB structures and store their equalvalent atom data, including xyz
  coordinates, residue numbers, residue type and B-factors.</p>
    </div>

    <pre class="usage"><span class='fu'>read.all</span>(<span class='kw'>aln</span>, prefix = <span class='st'>""</span>, pdbext = <span class='st'>""</span>, sel = <span class='kw'>NULL</span>, rm.wat=<span class='fl'>TRUE</span>, rm.ligand=<span class='fl'>FALSE</span>,
         compact = <span class='fl'>TRUE</span>, ncore = <span class='kw'>NULL</span>, <span class='kw'>...</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>aln</th>
      <td><p>an alignment data structure obtained with
    <code>read.fasta</code>.</p></td>
    </tr>
    <tr>
      <th>prefix</th>
      <td><p>prefix to aln$id to locate PDB files.</p></td>
    </tr>
    <tr>
      <th>pdbext</th>
      <td><p>the file name extention of the PDB files.</p></td>
    </tr>
    <tr>
      <th>sel</th>
      <td><p>a selection string detailing the atom type data to store
    (see function store.atom)</p></td>
    </tr>
    <tr>
      <th>rm.wat</th>
      <td><p>logical, if TRUE water atoms are removed.</p></td>
    </tr>
    <tr>
      <th>rm.ligand</th>
      <td><p>logical, if TRUE ligand atoms are removed.</p></td>
    </tr>
    <tr>
      <th>compact</th>
      <td><p>logical, if TRUE the number of atoms stored for each aligned
    residue varies according to the amino acid type. If FALSE, the constant 
    maximum possible number of atoms are stored for all aligned residues.</p></td>
    </tr>
    <tr>
      <th>ncore </th>
      <td><p>number of CPU cores used to do the calculation.
    By default (<code>ncore=NULL</code>) use all detected CPU cores.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>other parameters for <code>read.pdb</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The input <code>aln</code>, produced with <code>read.fasta</code>, must
  have identifers (i.e. sequence names) that match the PDB file
  names. For example the sequence corresponding to the structure
  file &#8220;mypdbdir/1bg2.pdb&#8221; should have the identifer
  &#8216;mypdbdir/1bg2.pdb&#8217; or &#8216;1bg2&#8217; if input &#8216;prefix&#8217;
  and &#8216;pdbext&#8217; equal &#8216;mypdbdir/&#8217; and &#8216;pdb&#8217;. See the
  examples below.</p>
<p>Sequence miss-matches will generate errors.  Thus, care should be taken
  to ensure that the sequences in the alignment match the sequences in
  their associated PDB files.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns a list of class <code>"pdbs"</code> with the following five
  components:</p>
<dt>xyz</dt><dd><p>numeric matrix of aligned C-alpha coordinates.</p></dd>
  <dt>resno</dt><dd><p>character matrix of aligned residue numbers.</p></dd>
  <dt>b</dt><dd><p>numeric matrix of aligned B-factor values.</p></dd>
  <dt>chain</dt><dd><p>character matrix of aligned chain identifiers.</p></dd>
  <dt>id</dt><dd><p>character vector of PDB sequence/structure names.</p></dd>
  <dt>ali</dt><dd><p>character matrix of aligned sequences.</p></dd>
  <dt>resid</dt><dd><p>character matrix of aligned 3-letter residue names.</p></dd>
  <dt>all </dt><dd><p>numeric matrix of aligned equalvelent atom coordinates.</p></dd>
  <dt>all.elety</dt><dd><p>numeric matrix of aligned atom element types.</p></dd>
  <dt>all.resid</dt><dd><p>numeric matrix of aligned three-letter residue codes.</p></dd>
  <dt>all.resno</dt><dd><p>numeric matrix of aligned residue numbers.</p></dd>
  <dt>all.grpby</dt><dd><p>numeric vector indicating the group of atoms belonging to
    the same aligned residue.</p></dd>
  <dt>all.hetatm</dt><dd><p>a list of &#8216;pdb&#8217; objects for non-protein atoms.</p></dd>

    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695--2696.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Barry Grant</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>This function is still in development and is NOT part of the
  offical bio3d package.</p>  
<p>The sequence character &#8216;X&#8217; is useful for masking unusual
  or unknown residues, as it can match any other residue type.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code>read.fasta</code>, <code>read.pdb</code>,
  <code>core.find</code>, <code>fit.xyz</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># still working on speeding this guy up</span>
<span class='co'>if</span> (<span class='fl'>FALSE</span>) {
<span class='co'>## Read sequence alignment</span>
<span class='kw'>file</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"examples/kif1a.fa"</span>,package=<span class='st'>"bio3d"</span>)
<span class='kw'>aln</span>  <span class='op'>&lt;-</span> <span class='fu'>read.fasta</span>(<span class='kw'>file</span>)

<span class='co'>## Read aligned PDBs storing all data for 'sel'</span>
<span class='kw'>sel</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"N"</span>, <span class='st'>"CA"</span>, <span class='st'>"C"</span>, <span class='st'>"O"</span>, <span class='st'>"CB"</span>, <span class='st'>"*G"</span>, <span class='st'>"*D"</span>,  <span class='st'>"*E"</span>, <span class='st'>"*Z"</span>)
<span class='kw'>pdbs</span> <span class='op'>&lt;-</span> <span class='fu'>read.all</span>(<span class='kw'>aln</span>, sel=<span class='kw'>sel</span>)

<span class='kw'>atm</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span>(<span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>all</span>)
<span class='kw'>ca.ind</span>  <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/which.html'>which</a></span>(<span class='kw'>atm</span> <span class='op'>==</span> <span class='st'>"CA"</span>)
<span class='kw'>core</span> <span class='op'>&lt;-</span> <span class='fu'>core.find</span>(<span class='kw'>pdbs</span>)
<span class='kw'>core.ind</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>( <span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>matrix</a></span>(<span class='kw'>ca.ind</span>, nrow=<span class='fl'>3</span>)[,<span class='kw'>core</span><span class='op'>$</span><span class='kw'>c0.5A.atom</span>] )

<span class='co'>## Fit structures</span>
<span class='kw'>nxyz</span> <span class='op'>&lt;-</span> <span class='fu'>fit.xyz</span>(<span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>all</span>[<span class='fl'>1</span>,], <span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>all</span>,
               fixed.inds  = <span class='kw'>core.ind</span>,
               mobile.inds = <span class='kw'>core.ind</span>)

<span class='kw'>ngap.col</span> <span class='op'>&lt;-</span> <span class='fu'>gap.inspect</span>(<span class='kw'>nxyz</span>)

<span class='co'>#npc.xray &lt;- pca.xyz(nxyz[ ,ngap.col$f.inds])</span>

<span class='co'>#a &lt;- mktrj.pca(npc.xray, pc=1, file="pc1-all.pdb",</span>
<span class='co'>#               elety=pdbs$all.elety[1,unique( ceiling(ngap.col$f.inds/3) )],</span>
<span class='co'>#               resid=pdbs$all.resid[1,unique( ceiling(ngap.col$f.inds/3) )],</span>
<span class='co'>#               resno=pdbs$all.resno[1,unique( ceiling(ngap.col$f.inds/3) )] )</span>

}
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="tidyverse">
  <p>Authored by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="author">
  <p>
    Built with &#10084;&#65039; and <a href="https://pkgdown.r-lib.org">pkgdown</a> &copy; 2020.
  </p>
</div>

      </footer>
   </div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script>
<script>
  docsearch({
    
    
    apiKey: '1270c4079b26a138795263974bbf302d',
    indexName: 'tidyverse',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>



  </body>
</html>


