<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>PDB structure manipulation and analysis • Bio3D</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../../../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../../../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../../../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../../../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../../../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../../../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="../../../tidyverse.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../../bootstrap-toc.css">
<script src="../../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../../pkgdown.css" rel="stylesheet">
<script src="../../../pkgdown.js"></script><link href="../../../tidyverse-2.css" rel="stylesheet">
<!--optional theme
<link href="../../../.css" rel="stylesheet">
--><!-- docsearch --><script src="../../../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../../../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="PDB structure manipulation and analysis">
<meta property="og:description" content="bio3d">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../../../index.html">
        bio3d</a>
        <div class="info hidden-xs hidden-sm">
<!--          <span class="partof">part of the <a href="https://tidyverse.org">tidyverse</a></span>
-->
          <span class="version version-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4-1.9000</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">INTRODUCTORY TUTORIALS</li>
    <li>
      <a href="../../../articles/online/install_vignette/Bio3D_install.html">Installing Bio3D</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/user-guide">New user guide</a>
    </li>
    <li>
      <a href="../../../articles/online/intro_vignette/Bio3D_introduction.html">Getting started with Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/pdb_vignette/Bio3D_pdb.html">PDB structure manipulation and analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/pca_vignette/Bio3D_pca.html">Comparative sequence and structure analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/nma_vignette/Bio3D_nma.html">Enhanced Normal Modes Analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/traj_vignette/Bio3D_md.html">Beginning trajectory analysis with Bio3D </a>
    </li>
    <li class="divider">
    <li class="dropdown-header">ADVANCED TUTORIALS</li>
    <li>
      <a href="../../../articles/online/enma_vignettes/Bio3D_nma-dhfr-partI.html">Ensemble NMA of E.coli DHFR structures</a>
    </li>
    <li>
      <a href="../../../articles/online/enma_vignettes/Bio3D_nma-dhfr-partII.html">Ensemble NMA across multiple species of DHFR</a>
    </li>
    <li>
      <a href="../../../articles/online/cna_vignette/cna_vignette.spin.html">Protein structure network analysis with Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/eddm_vignette/Bio3D_eddm.spin.html">Ensemble Difference Distance Matrix (eDDM) analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">WEBAPP TUTORIALS</li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/download/webapp-tutorials?download=22:bio3d-webapp-pca">Online protein structure analysis with Bio3D-Web (PDF)</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/webapps">Online Normal Mode Analysis with Bio3D WebApps (To Update)</a>
    </li>
    <li class="divider">
    <li>
      <a href="../../../articles/index.html">All vignettes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Documentation
  </a>
</li>
        <li>
  <a href="../../../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../../../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    News
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket"></span>
     
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Bio3D_pdb_files/header-attrs-2.3.2/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>PDB structure manipulation and analysis</h1>
                        <h4 class="author">Lars Skjaerven, Xin-Qiu Yao &amp; Barry J. Grant</h4>
            
      
      
      <div class="hidden name"><code>Bio3D_pdb.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Bio3D<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> is a family of related R packages containing utilities for the analysis of biomolecular structure, sequence and trajectory data <span class="citation">(Grant et al. 2006; Skjaerven et al. 2014)</span>. Features include the ability to read and write biomolecular structure, sequence and dynamic trajectory data, perform atom selection, re-orientation, superposition, rigid core identification, clustering, distance matrix analysis, conservation analysis, normal mode analysis, principal component analysis, and protein structure and correlation network analysis. Bio3D takes advantage of the extensive graphical and statistical capabilities of the <a href="http://www.r-project.org">R environment</a> and thus represents a useful framework for exploratory analysis of structural data.</p>
<div id="new-to-r" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#new-to-r" class="anchor"></a>New to R?</h2>
<p>There are numerous on–line resources that can help you get started using R effectively. A number of these can be found from the main R website at <a href="http://www.r-project.org" class="uri">http://www.r-project.org</a>. We particularly like the following:</p>
<ul>
<li>
<a href="https://rstudio.cloud/learn/primers">R primers from RStudio</a>: an interactive R tutorial in your web browser,<br>
</li>
<li>
<a href="https://www.datacamp.com/courses/free-introduction-to-r">DataCamp’s Introduction to R</a>: Learn by doing in your web browser (requires free registration),<br>
</li>
<li>
<a href="https://youtu.be/Asm2PHOZAcE">Introduction to R and RStudio video series</a>: Some of Barry’s course material from UC San Diego,<br>
</li>
<li>
<a href="http://cran.r-project.org/doc/manuals/R-intro.pdf">Offical introduction to R</a>: The offical R manual.</li>
</ul>
</div>
<div id="using-this-vignette" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#using-this-vignette" class="anchor"></a>Using this vignette</h2>
<p>The aim of this document, termed a vignette<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a> in R parlance, is to provide a brief introduction to PDB structure manipulation and analysis with the Bio3D core package (<strong>bio3d</strong>). A number of other Bio3D package vignettes and tutorials are available online at <a href="http://thegrantlab.org/bio3d" class="uri">http://thegrantlab.org/bio3d</a>. In particular, detailed instructions for obtaining and installing the Bio3D packages on various platforms can be found in the <a href="http://thegrantlab.org/bio3d">Installing Bio3D Vignette</a>. Note that this vignette was generated using <strong>bio3d version 2.4.1.9000</strong>.</p>
</div>
</div>
<div id="getting-started" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h1>
<p>Start R (type R at the command prompt or, on Windows, double click on the R icon) and load the bio3d package by typing <code><a href="http://thegrantlab.org/bio3d">library(bio3d)</a></code> at the R console prompt.</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="http://thegrantlab.org/bio3d">bio3d</a></span>)
</pre></div>
<p>Then use the command <code><a href="../../../reference/lbio3d.html">lbio3d()</a></code> or <code><a href="https://rdrr.io/pkg/bio3d/man">help(package=bio3d)</a></code> to list the functions within the package and <code>help(FunctionName)</code> to obtain more information about an individual function.</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="co"># List of bio3d functions with brief description</span>
<span class="fu"><a href="https://rdrr.io/r/utils/help.html">help</a></span>(package=<span class="kw">bio3d</span>)

<span class="co"># Detailed help on a particular function, e.g. 'pca.xyz'</span>
<span class="fu"><a href="https://rdrr.io/r/utils/help.html">help</a></span>(<span class="kw">pca.xyz</span>)
</pre></div>
<p>To search the help system for documentation matching a particular word or topic use the command <code><a href="https://rdrr.io/r/utils/help.search.html">help.search("topic")</a></code>. For example, <code><a href="https://rdrr.io/r/utils/help.search.html">help.search("pdb")</a></code></p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/help.search.html">help.search</a></span>(<span class="st">"pdb"</span>)
</pre></div>
<p>Typing <code><a href="https://rdrr.io/r/utils/help.start.html">help.start()</a></code> will start a local HTML interface to the R documentation and help system. After initiating <code><a href="https://rdrr.io/r/utils/help.start.html">help.start()</a></code> in a session the <code>help()</code> commands will open as HTML pages in your web browser.</p>
<div id="bio3d-functions-and-their-typical-usage" class="section level2">
<h2 class="hasAnchor">
<a href="#bio3d-functions-and-their-typical-usage" class="anchor"></a>bio3d functions and their typical usage</h2>
<p>The bio3d package consists of input/output functions, conversion and manipulation functions, analysis functions, and graphics functions all of which are fully documented both <a href="http://thegrantlab.org/bio3d">online</a> and within the R help system introduced above.</p>
<p>To better understand how a particular function operates it is often helpful to view and execute an example. Every function within the Bio3D packages is documented with example code that you can view by issuing the <code>help()</code> command.</p>
<p>Running the command <code>example(function)</code> will directly execute the example for a given function. In addition, a number of longer worked examples are available as <a href="http://thegrantlab.org/bio3d">Tutorials</a> on the Bio3D website.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/example.html">example</a></span>(<span class="kw">plot.bio3d</span>)
</pre></div>
</div>
</div>
<div id="working-with-individual-pdb-files" class="section level1">
<h1 class="hasAnchor">
<a href="#working-with-individual-pdb-files" class="anchor"></a>Working with individual PDB files</h1>
<p>Protein Data Bank files (or PDB files) are the most common format for the distribution and storage of high-resolution biomolecular coordinate data. The bio3d package contains functions for the reading (<strong>read.pdb()</strong>, <strong>read.fasta.pdb()</strong>, <strong>get.pdb()</strong>, <strong>convert.pdb()</strong>, <strong>basename.pdb()</strong>), writing (<strong>write.pdb()</strong>) and manipulation (<strong>trim.pdb()</strong>, <strong>cat.pdb()</strong>, <strong>pdbsplit()</strong>, <strong>atom.select()</strong>, <strong>pdbseq()</strong>) of PDB files. Indeed numerous Bio3D analysis functions are intended to operate on PDB file derived data (e.g. <strong>blast.pdb()</strong>, <strong>chain.pdb()</strong>, <strong>nma.pdb()</strong>, <strong>pdb.annotate()</strong>, <strong>pdbaln()</strong>, <strong>pdbfit()</strong>, <strong>struct.aln()</strong>, <strong>dssp()</strong>, <strong>pca.pdbs()</strong> etc.)</p>
<p>At their most basic, PDB coordinate files contain a list of all the atoms of one or more molecular structures. Each atom position is defined by its x, y, z coordinates in a conventional orthogonal coordinate system. Additional data, including listings of observed secondary structure elements, are also commonly (but not always) detailed in PDB files.</p>
<div id="read-a-pdb-file" class="section level2">
<h2 class="hasAnchor">
<a href="#read-a-pdb-file" class="anchor"></a>Read a PDB file</h2>
<p>To read a single PDB file with Bio3D we can use the <strong>read.pdb()</strong> function. The minimal input required for this function is a specification of the file to be read. This can be either the file name of a local file on disc or the RCSB PDB identifier of a file to read directly from the on-line PDB repository. For example to read and inspect the on-line file with PDB ID <code>4q21</code>:</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="kw">pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"4q21"</span>)
</pre></div>
<pre><code>##   Note: Accessing on-line PDB file</code></pre>
<p>To get a quick summary of the contents of the <code>pdb</code> object you just created you can issue the command <code><a href="https://rdrr.io/r/base/print.html">print(pdb)</a></code> or simply type <code>pdb</code> (which is equivalent in this case):</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="kw">pdb</span>
</pre></div>
<pre><code>## 
##  Call:  read.pdb(file = "4q21")
## 
##    Total Models#: 1
##      Total Atoms#: 1447,  XYZs#: 4341  Chains#: 1  (values: A)
## 
##      Protein Atoms#: 1340  (residues/Calpha atoms#: 168)
##      Nucleic acid Atoms#: 0  (residues/phosphate atoms#: 0)
## 
##      Non-protein/nucleic Atoms#: 107  (residues: 80)
##      Non-protein/nucleic resid values: [ GDP (1), HOH (78), MG (1) ]
## 
##    Protein sequence:
##       MTEYKLVVVGAGGVGKSALTIQLIQNHFVDEYDPTIEDSYRKQVVIDGETCLLDILDTAG
##       QEEYSAMRDQYMRTGEGFLCVFAINNTKSFEDIHQYREQIKRVKDSDDVPMVLVGNKCDL
##       AARTVESRQAQDLARSYGIPYIETSAKTRQGVEDAFYTLVREIRQHKL
## 
## + attr: atom, xyz, seqres, helix, sheet,
##         calpha, remark, call</code></pre>
<p>Note that the attributes (<code>+ attr:</code>) of this object are listed on the last couple of lines. To find the attributes of any such object you can use:</p>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/graph_attr_names.html">attributes</a></span>(<span class="kw">pdb</span>)
</pre></div>
<pre><code>## $names
## [1] "atom"   "xyz"    "seqres" "helix"  "sheet"  "calpha" "remark" "call"  
## 
## $class
## [1] "pdb" "sse"</code></pre>
<p>To access these individual attributes we use the <code>dollar-attribute</code> name convention that is common with R list objects. For example, to access the <code>atom</code> attribute or component use <code>pdb$atom</code>:</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">pdb</span><span class="op">$</span><span class="kw">atom</span>)
</pre></div>
<pre><code>##   type eleno elety  alt resid chain resno insert      x      y      z o     b
## 1 ATOM     1     N &lt;NA&gt;   MET     A     1   &lt;NA&gt; 64.080 50.529 32.509 1 28.66
## 2 ATOM     2    CA &lt;NA&gt;   MET     A     1   &lt;NA&gt; 64.044 51.615 33.423 1 29.19
## 3 ATOM     3     C &lt;NA&gt;   MET     A     1   &lt;NA&gt; 63.722 52.849 32.671 1 30.27
## 4 ATOM     4     O &lt;NA&gt;   MET     A     1   &lt;NA&gt; 64.359 53.119 31.662 1 34.93
## 5 ATOM     5    CB &lt;NA&gt;   MET     A     1   &lt;NA&gt; 65.373 51.805 34.158 1 28.49
## 6 ATOM     6    CG &lt;NA&gt;   MET     A     1   &lt;NA&gt; 65.122 52.780 35.269 1 32.18
##   segid elesy charge
## 1  &lt;NA&gt;     N   &lt;NA&gt;
## 2  &lt;NA&gt;     C   &lt;NA&gt;
## 3  &lt;NA&gt;     C   &lt;NA&gt;
## 4  &lt;NA&gt;     O   &lt;NA&gt;
## 5  &lt;NA&gt;     C   &lt;NA&gt;
## 6  &lt;NA&gt;     C   &lt;NA&gt;</code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="co"># Print $atom data for the first two atoms</span>
<span class="kw">pdb</span><span class="op">$</span><span class="kw">atom</span>[<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, ]
</pre></div>
<pre><code>##   type eleno elety  alt resid chain resno insert      x      y      z o     b
## 1 ATOM     1     N &lt;NA&gt;   MET     A     1   &lt;NA&gt; 64.080 50.529 32.509 1 28.66
## 2 ATOM     2    CA &lt;NA&gt;   MET     A     1   &lt;NA&gt; 64.044 51.615 33.423 1 29.19
##   segid elesy charge
## 1  &lt;NA&gt;     N   &lt;NA&gt;
## 2  &lt;NA&gt;     C   &lt;NA&gt;</code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="co"># Print a subset of $atom data for the first two atoms</span>
<span class="kw">pdb</span><span class="op">$</span><span class="kw">atom</span>[<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"eleno"</span>, <span class="st">"elety"</span>, <span class="st">"x"</span>,<span class="st">"y"</span>,<span class="st">"z"</span>)]
</pre></div>
<pre><code>##   eleno elety      x      y      z
## 1     1     N 64.080 50.529 32.509
## 2     2    CA 64.044 51.615 33.423</code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="co"># Note that individual $atom records can also be accessed like this</span>
<span class="kw">pdb</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">elety</span>[<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>]
</pre></div>
<pre><code>## [1] "N"  "CA"</code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="co"># Which allows us to do the following (see Figure 1.)</span>
<span class="fu"><a href="../../../reference/plot.bio3d.html">plot.bio3d</a></span>(<span class="kw">pdb</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">b</span>[<span class="kw">pdb</span><span class="op">$</span><span class="kw">calpha</span>], sse=<span class="kw">pdb</span>, typ=<span class="st">"l"</span>, ylab=<span class="st">"B-factor"</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_pdb_files/figure-html/unnamed-chunk-8-1.png" alt="Residue temperature factors for PDB ID `4q21` with secondary structure element (SSE) annotation in marginal regions plotted with function **plot.bio3d()**" width="700"><p class="caption">
Residue temperature factors for PDB ID <code>4q21</code> with secondary structure element (SSE) annotation in marginal regions plotted with function <strong>plot.bio3d()</strong>
</p>
</div>
<p>Note that the main <code>xyz</code> coordinate attribute is a numeric matrix with 3N columns (each atom has three values x, y and z). The number of rows here correspond to the number of models in the PDB file (typically one for X-ray structures and multiple for NMR structures).</p>
<div class="sourceCode" id="cb20"><pre class="downlit">
<span class="co"># Print a summary of the coordinate data in $xyz</span>
<span class="kw">pdb</span><span class="op">$</span><span class="kw">xyz</span>
</pre></div>
<pre><code>## 
##    Total Frames#: 1
##    Total XYZs#:   4341,  (Atoms#:  1447)
## 
##     [1]  64.08  50.529  32.509  &lt;...&gt;  74.159  76.923  41.999  [4341] 
## 
## + attr: Matrix DIM = 1 x 4341</code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit">
<span class="co"># Examine the row and column dimensions</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="kw">pdb</span><span class="op">$</span><span class="kw">xyz</span>)
</pre></div>
<pre><code>## [1]    1 4341</code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit">
<span class="kw">pdb</span><span class="op">$</span><span class="kw">xyz</span>[ <span class="fl">1</span>, <span class="fu"><a href="../../../reference/atom2xyz.html">atom2xyz</a></span>(<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>) ]
</pre></div>
<pre><code>## [1] 64.080 50.529 32.509 64.044 51.615 33.423</code></pre>
<div id="side-note-the-pdb-class" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#side-note-the-pdb-class" class="anchor"></a>Side-note: The ‘pdb’ class</h4>
<p>Objects created by the <strong>read.pdb()</strong> function are of class “pdb”. This is recognized by other so called generic Bio3D functions (for example <strong>atom.select()</strong>, <strong>nma()</strong>, <strong>print()</strong>, <strong>summary()</strong> etc.). A generic function is a function that examines the class of its first argument, and then decides what type of operation to perform (more specifically it decides which specific method to dispatch to). So for example, the generic <strong>atom.select()</strong> function knows that the input is of class “pdb”, rather than for example an AMBER parameter and topology file, and will act accordingly.</p>
<p>A careful reader will also of noted that our “pdb” object created above also has a second class, namely “sse” (see the output of <code><a href="https://rdrr.io/pkg/igraph/man/graph_attr_names.html">attributes(pdb)</a></code> or <code><a href="https://rdrr.io/r/base/class.html">class(pdb)</a></code>). This stands for <em>secondary structure elements</em> and is recognized by the <strong>plot.bio3d()</strong> function to annotate the positions of major secondary structure elements in the marginal regions of these plots (see Figure 1). This is all part of the R S3 object orientation system. This S3 system os used throughout Bio3D to simplify and facilitate our work with these types of objects.</p>
</div>
</div>
<div id="atom-selection" class="section level2">
<h2 class="hasAnchor">
<a href="#atom-selection" class="anchor"></a>Atom selection</h2>
<p>The bio3d <strong>atom.select()</strong> function is arguably one of the most challenging for newcomers to master. It is however central to PDB structure manipulation and analysis. At its most basic, this function operates on PDB structure objects (as created by <strong>read.pdb()</strong>) and returns the numeric indices of a selected atom subset. These indices can then be used to access the <code>$atom</code> and <code>$xyz</code> attributes of PDB structure related objects.</p>
<p>For example to select the indices for all C-alpha atoms we can use the following command:</p>
<div class="sourceCode" id="cb26"><pre class="downlit">
<span class="co"># Select all C-alpha atoms (return their indices)</span>
<span class="kw">ca.inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>, <span class="st">"calpha"</span>)
<span class="kw">ca.inds</span>
</pre></div>
<pre><code>## 
##  Call:  atom.select.pdb(pdb = pdb, string = "calpha")
## 
##    Atom Indices#: 168  ($atom)
##    XYZ  Indices#: 504  ($xyz)
## 
## + attr: atom, xyz, call</code></pre>
<p>Note that the attributes of the returned <code>ca.inds</code> from <strong>atom.select()</strong> include both <code>atom</code> and <code>xyz</code> components. These are numeric vectors that can be used as indices to access the corresponding atom and xyz components of the input PDB structure object. For example:</p>
<div class="sourceCode" id="cb28"><pre class="downlit">
<span class="co"># Print details of the first few selected atoms</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>( <span class="kw">pdb</span><span class="op">$</span><span class="kw">atom</span>[<span class="kw">ca.inds</span><span class="op">$</span><span class="kw">atom</span>, ] )
</pre></div>
<pre><code>##    type eleno elety  alt resid chain resno insert      x      y      z o     b
## 2  ATOM     2    CA &lt;NA&gt;   MET     A     1   &lt;NA&gt; 64.044 51.615 33.423 1 29.19
## 10 ATOM    10    CA &lt;NA&gt;   THR     A     2   &lt;NA&gt; 62.439 54.794 32.359 1 28.10
## 17 ATOM    17    CA &lt;NA&gt;   GLU     A     3   &lt;NA&gt; 63.968 58.232 32.801 1 30.95
## 26 ATOM    26    CA &lt;NA&gt;   TYR     A     4   &lt;NA&gt; 61.817 61.333 33.161 1 23.42
## 38 ATOM    38    CA &lt;NA&gt;   LYS     A     5   &lt;NA&gt; 63.343 64.814 33.163 1 21.34
## 47 ATOM    47    CA &lt;NA&gt;   LEU     A     6   &lt;NA&gt; 61.321 67.068 35.557 1 18.99
##    segid elesy charge
## 2   &lt;NA&gt;     C   &lt;NA&gt;
## 10  &lt;NA&gt;     C   &lt;NA&gt;
## 17  &lt;NA&gt;     C   &lt;NA&gt;
## 26  &lt;NA&gt;     C   &lt;NA&gt;
## 38  &lt;NA&gt;     C   &lt;NA&gt;
## 47  &lt;NA&gt;     C   &lt;NA&gt;</code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit">
<span class="co"># And selected xyz coordinates</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>( <span class="kw">pdb</span><span class="op">$</span><span class="kw">xyz</span>[, <span class="kw">ca.inds</span><span class="op">$</span><span class="kw">xyz</span>] )
</pre></div>
<pre><code>## [1] 64.044 51.615 33.423 62.439 54.794 32.359</code></pre>
<p>In addition to the common selection strings (such as ‘calpha’ ‘cbeta’ ‘backbone’ ‘protein’ ‘notprotein’ ‘ligand’ ‘water’ ‘notwater’ ‘h’ and ‘noh’) various individual atom properties can be used for selection.</p>
<div class="sourceCode" id="cb32"><pre class="downlit">
<span class="co"># Select chain A</span>
<span class="kw">a.inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>, chain=<span class="st">"A"</span>)

<span class="co"># Select C-alphas of chain A</span>
<span class="kw">ca.inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>, <span class="st">"calpha"</span>, chain=<span class="st">"A"</span>)

<span class="co"># We can combine multiple selection criteria to return their intersection</span>
<span class="kw">cab.inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>, elety=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"CA"</span>,<span class="st">"CB"</span>), chain=<span class="st">"A"</span>, resno=<span class="fl">10</span><span class="op">:</span><span class="fl">20</span>)
</pre></div>
<div id="inverse-selection" class="section level3">
<h3 class="hasAnchor">
<a href="#inverse-selection" class="anchor"></a>Inverse selection</h3>
<p>The <strong>atom.select()</strong> function also contain options to inverse the final selection. For example, including the argument <code>inverse=TRUE</code> in the statement below will select all atoms except the water atoms:</p>
<div class="sourceCode" id="cb33"><pre class="downlit">
<span class="co"># Select all atoms except waters</span>
<span class="kw">nowat.inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>, <span class="st">"water"</span>, inverse=<span class="fl">TRUE</span>)
</pre></div>
</div>
<div id="combining-selections" class="section level3">
<h3 class="hasAnchor">
<a href="#combining-selections" class="anchor"></a>Combining selections</h3>
<p>The <code>operator</code> argument of <strong>atom.select()</strong> determines how to combine the individual selection statements. The operator is by default <code>"AND"</code>, which specifies that the final selection is combined by intersection of the individual selection statements. In contrast, by setting <code>operator="OR"</code> the final selection is determined by the union of the individual statements. For example, the following selection will select all protein atoms as well as those of any residue named GDP:</p>
<div class="sourceCode" id="cb34"><pre class="downlit">
<span class="co"># Select protein + GDP</span>
<span class="kw">sele</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>, <span class="st">"protein"</span>, resid=<span class="st">"GDP"</span>, operator=<span class="st">"OR"</span>)
</pre></div>
<p>The argument <code>verbose=TRUE</code> can be used to print more information on how the selection components are combined:</p>
<div class="sourceCode" id="cb35"><pre class="downlit">
<span class="kw">sele</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>, <span class="st">"protein"</span>, elety=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"N"</span>, <span class="st">"CA"</span>, <span class="st">"C"</span>), resno=<span class="fl">50</span><span class="op">:</span><span class="fl">60</span>, verbose=<span class="kw">T</span>)
</pre></div>
<pre><code>## 
##  .. 00001340 atom(s) from 'string' selection 
##  .. 00000505 atom(s) from 'elety' selection 
##  .. 00000077 atom(s) from 'resno' selection 
##  .. 00000033 atom(s) in final combined selection</code></pre>
<p>Function <strong>combine.select()</strong> provides further functionality to combine selections. For example the above selection of protein and GDP can be completed with:</p>
<div class="sourceCode" id="cb37"><pre class="downlit">
<span class="kw">a.inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>, <span class="st">"protein"</span>)
<span class="kw">b.inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>, resid=<span class="st">"GDP"</span>)
<span class="kw">sele</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/combine.select.html">combine.select</a></span>(<span class="kw">a.inds</span>, <span class="kw">b.inds</span>, operator=<span class="st">"OR"</span>)
</pre></div>
<p>See <code><a href="../../../reference/combine.select.html">help(combine.select)</a></code> for more information.</p>
<div id="question" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question" class="anchor"></a>Question:</h4>
<p>Using <code><a href="../../../reference/atom.select.html">atom.select()</a></code> how would you extract the amino acid sequence of your structure in 3-letter and 1-letter forms?</p>
</div>
<div id="answer" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#answer" class="anchor"></a>Answer:</h4>
<p>First select the C-alpha atoms, and use the returned atom indices to access the <code>resid</code> values of <code>pdb$atom</code>.</p>
<div class="sourceCode" id="cb38"><pre class="downlit">
<span class="kw">ca.inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>, <span class="st">"calpha"</span>)
<span class="kw">aa3</span> <span class="op">&lt;-</span> <span class="kw">pdb</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">resid</span>[ <span class="kw">ca.inds</span><span class="op">$</span><span class="kw">atom</span> ]
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">aa3</span>)
</pre></div>
<pre><code>## [1] "MET" "THR" "GLU" "TYR" "LYS" "LEU"</code></pre>
<p>From there you can use the utility function <strong>aa321()</strong> to convert to 1-letter from</p>
<div class="sourceCode" id="cb40"><pre class="downlit">
<span class="fu"><a href="../../../reference/aa123.html">aa321</a></span>(<span class="kw">aa3</span>)
</pre></div>
<pre><code>##   [1] "M" "T" "E" "Y" "K" "L" "V" "V" "V" "G" "A" "G" "G" "V" "G" "K" "S" "A"
##  [19] "L" "T" "I" "Q" "L" "I" "Q" "N" "H" "F" "V" "D" "E" "Y" "D" "P" "T" "I"
##  [37] "E" "D" "S" "Y" "R" "K" "Q" "V" "V" "I" "D" "G" "E" "T" "C" "L" "L" "D"
##  [55] "I" "L" "D" "T" "A" "G" "Q" "E" "E" "Y" "S" "A" "M" "R" "D" "Q" "Y" "M"
##  [73] "R" "T" "G" "E" "G" "F" "L" "C" "V" "F" "A" "I" "N" "N" "T" "K" "S" "F"
##  [91] "E" "D" "I" "H" "Q" "Y" "R" "E" "Q" "I" "K" "R" "V" "K" "D" "S" "D" "D"
## [109] "V" "P" "M" "V" "L" "V" "G" "N" "K" "C" "D" "L" "A" "A" "R" "T" "V" "E"
## [127] "S" "R" "Q" "A" "Q" "D" "L" "A" "R" "S" "Y" "G" "I" "P" "Y" "I" "E" "T"
## [145] "S" "A" "K" "T" "R" "Q" "G" "V" "E" "D" "A" "F" "Y" "T" "L" "V" "R" "E"
## [163] "I" "R" "Q" "H" "K" "L"</code></pre>
<p>Note that if you try searching with <code><a href="https://rdrr.io/r/utils/help.search.html">help.search("PDB sequence", package="bio3d")</a></code>, you will likely find a bio3d function that essentially does this all for you; Namely <strong>pdbseq()</strong>:</p>
<div class="sourceCode" id="cb42"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>( <span class="fu"><a href="../../../reference/pdbseq.html">pdbseq</a></span>(<span class="kw">pdb</span>) )
</pre></div>
<pre><code>##   1   2   3   4   5   6 
## "M" "T" "E" "Y" "K" "L"</code></pre>
<p>Given the large number of functions in the bio3d package, using <strong>help.search()</strong> can be an effective way to find functionality related to your specific task.</p>
</div>
</div>
</div>
<div id="write-a-pdb-object" class="section level2">
<h2 class="hasAnchor">
<a href="#write-a-pdb-object" class="anchor"></a>Write a PDB object</h2>
<p>Use the command <strong>write.pdb()</strong> to output a structure object to a PDB formatted file on your local hard drive. Below we use function <strong>atom.select()</strong> to select only the backbone atoms, and <strong>trim.pdb()</strong> to create a new PDB object based on our selection of backbone atoms. Finally we use the function <strong>write.pdb()</strong> to write the newly generated PDB file containing only the backbone atoms:</p>
<div class="sourceCode" id="cb44"><pre class="downlit">
<span class="co"># Output a backbone only PDB file to disc</span>
<span class="kw">b.inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>, <span class="st">"back"</span>)
<span class="kw">backpdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/trim.html">trim.pdb</a></span>(<span class="kw">pdb</span>, <span class="kw">b.inds</span>)
<span class="fu"><a href="../../../reference/write.pdb.html">write.pdb</a></span>(<span class="kw">backpdb</span>, file=<span class="st">"4q21_back.pdb"</span>)
</pre></div>
<div id="side-note" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#side-note" class="anchor"></a>Side-note:</h4>
<p>The selection statement can directly be provided into function <strong>trim.pdb()</strong>. Alternatively, function <strong>atom.select()</strong> can also return the resulting trimmed <code>pdb</code> object using the optional argument <code>value=TRUE</code>. See examples below:</p>
<div class="sourceCode" id="cb45"><pre class="downlit">
<span class="co"># Selection statements can be passed directly to trim.pdb()</span>
<span class="kw">backpdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/trim.html">trim.pdb</a></span>(<span class="kw">pdb</span>, <span class="st">"backbone"</span>)

<span class="co"># The 'value=TRUE' option of atom.select() will result in a PDB object being returned</span>
<span class="kw">backpdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>, <span class="st">"backbone"</span>, value=<span class="fl">TRUE</span>)
</pre></div>
<p>Function <strong>write.pdb()</strong> contains further arguments to change the specific data in the PDB structure object. For example, using argument <code>resno</code> the residue numbers in the PDB object will be changed according to the input values, e.g. for the purpose of renumbering a PDB object (see also the <strong>force.renumber</strong> argument in <strong>clean.pdb()</strong> or the <strong>renumber</strong> argument in the <strong>convert.pdb()</strong> function):</p>
<div class="sourceCode" id="cb46"><pre class="downlit">
<span class="co"># Renumber all residues</span>
<span class="fu"><a href="../../../reference/write.pdb.html">write.pdb</a></span>(<span class="kw">backpdb</span>, resno=<span class="kw">backpdb</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">resno</span><span class="op">+</span><span class="fl">10</span>)

<span class="co"># Assign chain B to all residues</span>
<span class="fu"><a href="../../../reference/write.pdb.html">write.pdb</a></span>(<span class="kw">backpdb</span>, chain=<span class="st">"B"</span>)
</pre></div>
</div>
</div>
<div id="manipulate-a-pdb-object" class="section level2">
<h2 class="hasAnchor">
<a href="#manipulate-a-pdb-object" class="anchor"></a>Manipulate a PDB object</h2>
<p>Basic functions for concatenating, trimming, splitting, converting, rotating, translating and superposing PDB files are available but often you will want to manipulate PDB objects in a custom way.</p>
<p>Below we provide a basic example of such a manipulation process where we read in a multi-chained PDB structure, reassign chain identifiers, and renumber selected residues.</p>
<div class="sourceCode" id="cb47"><pre class="downlit">
<span class="kw">pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"4lhy"</span>)

<span class="co"># select chains A, E and F</span>
<span class="kw">inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>, chain=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"A"</span>, <span class="st">"E"</span>, <span class="st">"F"</span>))

<span class="co"># trim PDB to selection</span>
<span class="kw">pdb2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/trim.html">trim.pdb</a></span>(<span class="kw">pdb</span>, <span class="kw">inds</span>)

<span class="co"># assign new chain identifiers</span>
<span class="kw">pdb2</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">chain</span>[ <span class="kw">pdb2</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">chain</span><span class="op">==</span><span class="st">"E"</span> ] <span class="op">&lt;-</span> <span class="st">"B"</span>
<span class="kw">pdb2</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">chain</span>[ <span class="kw">pdb2</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">chain</span><span class="op">==</span><span class="st">"F"</span> ] <span class="op">&lt;-</span> <span class="st">"C"</span>

<span class="co"># re-number chain B and C</span>
<span class="kw">pdb2</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">resno</span>[ <span class="kw">pdb2</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">chain</span><span class="op">==</span><span class="st">"B"</span> ] <span class="op">&lt;-</span> <span class="kw">pdb2</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">resno</span>[ <span class="kw">pdb2</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">chain</span><span class="op">==</span><span class="st">"B"</span> ] <span class="op">-</span> <span class="fl">156</span>
<span class="kw">pdb2</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">resno</span>[ <span class="kw">pdb2</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">chain</span><span class="op">==</span><span class="st">"C"</span> ] <span class="op">&lt;-</span> <span class="kw">pdb2</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">resno</span>[ <span class="kw">pdb2</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">chain</span><span class="op">==</span><span class="st">"C"</span> ] <span class="op">-</span> <span class="fl">156</span>

<span class="co"># assign the GDP residue a residue number of 500</span>
<span class="kw">pdb2</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">resno</span>[ <span class="kw">pdb2</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">resid</span><span class="op">==</span><span class="st">"GDP"</span> ] <span class="op">&lt;-</span> <span class="fl">500</span>

<span class="co"># use chain D for the GDP residue</span>
<span class="kw">pdb2</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">chain</span>[ <span class="kw">pdb2</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">resid</span><span class="op">==</span><span class="st">"GDP"</span> ] <span class="op">&lt;-</span> <span class="st">"D"</span>

<span class="co"># Center, to the coordinate origin, and orient, by principal axes,</span>
<span class="co"># the coordinates of a given PDB structure or xyz vector.</span>
<span class="kw">xyz</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/orient.pdb.html">orient.pdb</a></span>(<span class="kw">pdb2</span>)

<span class="co"># write the new pdb object to file</span>
<span class="fu"><a href="../../../reference/write.pdb.html">write.pdb</a></span>(<span class="kw">pdb2</span>, xyz=<span class="kw">xyz</span>, file=<span class="st">"4LHY_AEF-oriented.pdb"</span>)
</pre></div>
</div>
<div id="concatenate-multiple-pdbs" class="section level2">
<h2 class="hasAnchor">
<a href="#concatenate-multiple-pdbs" class="anchor"></a>Concatenate multiple PDBs</h2>
<p>Function <strong>cat.pdb()</strong> can be used to concatenate two or more PDB files. This function contains further arguments to re-assign residue numbers and chain identifiers. In the example below we illustrate how to concatenate <code>4q21</code> with specific components of <code>4lhy</code> into a new PDB object:</p>
<div class="sourceCode" id="cb48"><pre class="downlit">
<span class="co"># read two G-protein structures</span>
<span class="kw">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"4q21"</span>)
</pre></div>
<pre><code>## Warning in get.pdb(file, path = tempdir(), verbose = FALSE): /var/folders/xf/
## qznxnpf91vb1wm4xwgnbt0xr0000gn/T//Rtmp6Xsoc0/4q21.pdb exists. Skipping download</code></pre>
<div class="sourceCode" id="cb50"><pre class="downlit">
<span class="kw">b</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"4lhy"</span>)
</pre></div>
<pre><code>## Warning in get.pdb(file, path = tempdir(), verbose = FALSE): /var/folders/xf/
## qznxnpf91vb1wm4xwgnbt0xr0000gn/T//Rtmp6Xsoc0/4lhy.pdb exists. Skipping download</code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit">
<span class="kw">a1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/trim.html">trim.pdb</a></span>(<span class="kw">a</span>, chain=<span class="st">"A"</span>)

<span class="kw">b1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/trim.html">trim.pdb</a></span>(<span class="kw">b</span>, chain=<span class="st">"A"</span>)
<span class="kw">b2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/trim.html">trim.pdb</a></span>(<span class="kw">b</span>, chain=<span class="st">"E"</span>)
<span class="kw">b3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/trim.html">trim.pdb</a></span>(<span class="kw">b</span>, chain=<span class="st">"F"</span>)

<span class="co"># concatenate PDBs</span>
<span class="kw">new</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/cat.pdb.html">cat.pdb</a></span>(<span class="kw">a1</span>, <span class="kw">b1</span>, <span class="kw">b2</span>, <span class="kw">b3</span>, rechain=<span class="fl">TRUE</span>)
</pre></div>
<pre><code>## Warning in clean.pdb(new, consecutive = !rechain, force.renumber = renumber, :
## PDB is still not clean. Try fix.chain=TRUE and/or fix.aa=TRUE</code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="kw">new</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">chain</span>)

<span class="co"># write new PDB object to file</span>
<span class="fu"><a href="../../../reference/write.pdb.html">write.pdb</a></span>(<span class="kw">new</span>, file=<span class="st">"4Q21-4LHY.pdb"</span>)
</pre></div>
</div>
<div id="coordinate-superposition-and-structural-alignment" class="section level2">
<h2 class="hasAnchor">
<a href="#coordinate-superposition-and-structural-alignment" class="anchor"></a>Coordinate superposition and structural alignment</h2>
<p>Structure superposition is often essential for the direct comparison of multiple structures. Bio3D offers versatile functionality for coordinate superposition at various levels. The simplest level is sequence only based superposition:</p>
<div class="sourceCode" id="cb55"><pre class="downlit">
<span class="co"># Align and superpose two or more structures</span>
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbaln.html">pdbaln</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"4q21"</span>, <span class="st">"521p"</span>), fit=<span class="fl">TRUE</span>)
</pre></div>
<p>Here the returned object is of class <code>pdbs</code>, which we will discuss in detail further below. For now note that it contains a <code>xyz</code> numeric matrix of aligned C-alpha coordinates.</p>
<div class="sourceCode" id="cb56"><pre class="downlit">
<span class="kw">pdbs</span>
</pre></div>
<pre><code>##                              1        .         .         .         .         50 
## [Truncated_Name:1]4q21.pdb   MTEYKLVVVGAGGVGKSALTIQLIQNHFVDEYDPTIEDSYRKQVVIDGET
## [Truncated_Name:2]521p.pdb   MTEYKLVVVGAVGVGKSALTIQLIQNHFVDEYDPTIEDSYRKQVVIDGET
##                              *********** ************************************** 
##                              1        .         .         .         .         50 
## 
##                             51        .         .         .         .         100 
## [Truncated_Name:1]4q21.pdb   CLLDILDTAGQEEYSAMRDQYMRTGEGFLCVFAINNTKSFEDIHQYREQI
## [Truncated_Name:2]521p.pdb   CLLDILDTTGQEEYSAMRDQYMRTGEGFLCVFAINNTKSFEDIHQYREQI
##                              ******** ***************************************** 
##                             51        .         .         .         .         100 
## 
##                            101        .         .         .         .         150 
## [Truncated_Name:1]4q21.pdb   KRVKDSDDVPMVLVGNKCDLAARTVESRQAQDLARSYGIPYIETSAKTRQ
## [Truncated_Name:2]521p.pdb   KRVKDSDDVPMVLVGNKCDLAARTVESRQAQDLARSYGIPYIETSAKTRQ
##                              ************************************************** 
##                            101        .         .         .         .         150 
## 
##                            151        .       168 
## [Truncated_Name:1]4q21.pdb   GVEDAFYTLVREIRQHKL
## [Truncated_Name:2]521p.pdb   GVEDAFYTLVREIRQH--
##                              ****************   
##                            151        .       168 
## 
## Call:
##   pdbaln(files = c("4q21", "521p"), fit = TRUE)
## 
## Class:
##   pdbs, fasta
## 
## Alignment dimensions:
##   2 sequence rows; 168 position columns (166 non-gap, 2 gap) 
## 
## + attr: xyz, resno, b, chain, id, ali, resid, sse, call</code></pre>
<p>Underlying the <strong>pdbfit()</strong> function are calls to the <strong>seqaln()</strong> and <strong>fit.xyz()</strong> functions. The later of theses does the actual superposition based on <em>all</em> the aligned positions returned from <strong>seqaln()</strong>. Hence the superposition is said to be <em>sequence based</em>.</p>
<p>An alternative approach is to use a structure only alignment method (e.g. the <strong>mustang()</strong> function) as a basis for superposition. This is particularly useful when the structures (and hence usually the sequences) to be compared are dissimilar to the point where sequence comparison may give erroneous results.</p>
<p>Variations of both these approaches are also implemented in bio3d. For example, the function <strong>struct.aln()</strong> performs cycles of refinement steps of the alignment to improve the fit by removing atoms with a high structural deviation.</p>
<div id="note-the-seqaln-function-requires-muscle" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#note-the-seqaln-function-requires-muscle" class="anchor"></a>Note: The seqaln() function requires MUSCLE</h4>
<p>The <em>MUSCLE</em> multiple sequence alignment program (available from the <a href="http://www.drive5.com/muscle/">muscle home page</a>) must be installed on your system and in the search path for executables in order to run functions <strong>pdbfit()</strong> and <strong>struct.aln()</strong> as these call the <strong>seqaln()</strong> function, which is based on MUSCLE. Please see the installation vignette for further details.</p>
<p>Function <strong>struct.aln()</strong> performs a sequence alignment followed by a structural alignment of two PDB objects. This facilitates rapid superposition of two PDB structures with unequal, but related PDB sequences. Below we use <strong>struct.aln()</strong> to superimpose the multi-chained PDB ID <code>4lhy</code> to PDB ID <code>4q21</code>:</p>
<div class="sourceCode" id="cb58"><pre class="downlit">
<span class="co"># read two G-protein structures</span>
<span class="kw">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"4q21"</span>)
</pre></div>
<pre><code>## Warning in get.pdb(file, path = tempdir(), verbose = FALSE): /var/folders/xf/
## qznxnpf91vb1wm4xwgnbt0xr0000gn/T//Rtmp6Xsoc0/4q21.pdb exists. Skipping download</code></pre>
<div class="sourceCode" id="cb60"><pre class="downlit">
<span class="kw">b</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"4lhy"</span>)
</pre></div>
<pre><code>## Warning in get.pdb(file, path = tempdir(), verbose = FALSE): /var/folders/xf/
## qznxnpf91vb1wm4xwgnbt0xr0000gn/T//Rtmp6Xsoc0/4lhy.pdb exists. Skipping download</code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit">
<span class="co"># perform iterative alignment</span>
<span class="kw">aln</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/struct.aln.html">struct.aln</a></span>(<span class="kw">a</span>, <span class="kw">b</span>)

<span class="co"># store new coordinates of protein B</span>
<span class="kw">b</span><span class="op">$</span><span class="kw">xyz</span> <span class="op">&lt;-</span> <span class="kw">aln</span><span class="op">$</span><span class="kw">xyz</span>
</pre></div>
<p>Note that <strong>struct.aln()</strong> performs cycles of refinement steps of the structural alignment to improve the fit by removing atoms with a high structural deviation. At each cycle it prints associated RMSD value of atoms included in the alignment. The resulting superimposed structures are optionally written to your hard drive at default folder with name <code>fitlsq/</code>.</p>
<div class="figure">
<img src="figures/4Q21-4LHY_structaln.png" alt=""><p class="caption">Result of fitting PDB IDs <code>4lhy</code> (grey) and <code>4q21</code> (red) using function <strong>struct.aln()</strong>. Note that <code>4lhy</code> contains 6 chains (A-F), while <code>4q21</code> consist of only 1 chain. Based on the sequence and structure alignment performed internally in <strong>struct.aln()</strong> the two PDBs align at certain helices in the shared domain.</p>
</div>
<p>While <strong>struct.aln()</strong> operates on a pair of PDB objects, function <strong>fit.xyz()</strong> can perform coordinate superposition on multiple structures. However, indices needs to provided ensuring the same number of atoms for which the fitting should be based. Below we superimpose a specific helix of <code>4lhy</code> onto an equivalent helix of <code>4q21</code>:</p>
<div class="sourceCode" id="cb63"><pre class="downlit">
<span class="co"># indices at which the superposition should be based</span>
<span class="kw">a.ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">a</span>, chain=<span class="st">"A"</span>, resno=<span class="fl">87</span><span class="op">:</span><span class="fl">103</span>, elety=<span class="st">"CA"</span>)
<span class="kw">b.ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">b</span>, chain=<span class="st">"A"</span>, resno=<span class="fl">93</span><span class="op">:</span><span class="fl">109</span>, elety=<span class="st">"CA"</span>)

<span class="co"># perform superposition</span>
<span class="kw">xyz</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/fit.xyz.html">fit.xyz</a></span>(fixed=<span class="kw">a</span><span class="op">$</span><span class="kw">xyz</span>, mobile=<span class="kw">b</span><span class="op">$</span><span class="kw">xyz</span>,
               fixed.inds=<span class="kw">a.ind</span><span class="op">$</span><span class="kw">xyz</span>,
               mobile.inds=<span class="kw">b.ind</span><span class="op">$</span><span class="kw">xyz</span>)

<span class="co"># write coordinates to file</span>
<span class="fu"><a href="../../../reference/write.pdb.html">write.pdb</a></span>(<span class="kw">b</span>, xyz=<span class="kw">xyz</span>, file=<span class="st">"4LHY-at-4Q21.pdb"</span>)
</pre></div>
<div class="figure">
<img src="figures/4Q21-4LHY-fit.png" alt=""><p class="caption">Result of fitting a specific helix in PDB ID <code>4lhy</code> onto an equivalent helix of PDB ID <code>4q21</code> using function <strong>fit.xyz()</strong>.</p>
</div>
<p>For more information on Coordinate superposition see <code><a href="../../../reference/fit.xyz.html">help(fit.xyz)</a></code>.</p>
</div>
</div>
<div id="binding-site-identification" class="section level2">
<h2 class="hasAnchor">
<a href="#binding-site-identification" class="anchor"></a>Binding site identification</h2>
<p>Function <strong>binding.site()</strong> provides functionality to determine interacting residues between two PDB enteties or between two atom selections of a PDB object. This function reports the residues of selection 1 closer than a cutoff to selection 2. By default, <strong>binding.site()</strong> attempts to identify “protein” and “ligand” using <strong>atom.select()</strong>. Thus, to determine the binding site of PDB <code>4q21</code> only the <code>pdb</code> object needs to be provided:</p>
<div class="sourceCode" id="cb64"><pre class="downlit">
<span class="co"># read G-protein structure</span>
<span class="kw">pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"4q21"</span>)
</pre></div>
<pre><code>## Warning in get.pdb(file, path = tempdir(), verbose = FALSE): /var/folders/xf/
## qznxnpf91vb1wm4xwgnbt0xr0000gn/T//Rtmp6Xsoc0/4q21.pdb exists. Skipping download</code></pre>
<div class="sourceCode" id="cb66"><pre class="downlit">
<span class="kw">bs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/binding.site.html">binding.site</a></span>(<span class="kw">pdb</span>)
</pre></div>
<p>The output of <strong>binding.site()</strong> is a list containing (1) an <code>inds</code> component with a <code>select</code> object containing the atom and xyz indices of the identified binding site, (2) a <code>resnames</code> component containing formatted residue names of the binding site residues, (3) <code>resno</code> and <code>chain</code> components with residue numbers and chain identifiers for the binding site residues.</p>
<div class="sourceCode" id="cb67"><pre class="downlit">
<span class="co"># residue names of identified binding site</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">bs</span><span class="op">$</span><span class="kw">resnames</span>)
</pre></div>
<pre><code>##  [1] "ALA 11 (A)"  "GLY 12 (A)"  "GLY 13 (A)"  "VAL 14 (A)"  "GLY 15 (A)" 
##  [6] "LYS 16 (A)"  "SER 17 (A)"  "ALA 18 (A)"  "PHE 28 (A)"  "VAL 29 (A)" 
## [11] "ASP 30 (A)"  "GLU 31 (A)"  "TYR 32 (A)"  "ASP 33 (A)"  "PRO 34 (A)" 
## [16] "ILE 36 (A)"  "ASP 57 (A)"  "THR 58 (A)"  "ALA 59 (A)"  "ASN 116 (A)"
## [21] "LYS 117 (A)" "ASP 119 (A)" "LEU 120 (A)" "THR 144 (A)" "SER 145 (A)"
## [26] "ALA 146 (A)" "LYS 147 (A)"</code></pre>
<p>Alternatively, user defined atom selections can be provided to determine e.g. residues at the binding interface between two proteins:</p>
<div class="sourceCode" id="cb69"><pre class="downlit">
<span class="kw">b</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"4lhy"</span>)
</pre></div>
<pre><code>## Warning in get.pdb(file, path = tempdir(), verbose = FALSE): /var/folders/xf/
## qznxnpf91vb1wm4xwgnbt0xr0000gn/T//Rtmp6Xsoc0/4lhy.pdb exists. Skipping download</code></pre>
<div class="sourceCode" id="cb71"><pre class="downlit">
<span class="co"># atom selection</span>
<span class="kw">a.inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">b</span>, chain=<span class="st">"A"</span>)
<span class="kw">b.inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">b</span>, chain=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"E"</span>, <span class="st">"F"</span>))

<span class="co"># identify interface residues</span>
<span class="kw">bs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/binding.site.html">binding.site</a></span>(<span class="kw">b</span>, a.inds=<span class="kw">a.inds</span>, b.inds=<span class="kw">b.inds</span>)

<span class="co"># use b-factor column to store interface in PDB file</span>
<span class="kw">b</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">b</span>[ <span class="kw">bs</span><span class="op">$</span><span class="kw">inds</span><span class="op">$</span><span class="kw">atom</span> ] <span class="op">&lt;-</span> <span class="fl">1</span>
<span class="kw">b</span><span class="op">$</span><span class="kw">atom</span><span class="op">$</span><span class="kw">b</span>[ <span class="op">-</span><span class="kw">bs</span><span class="op">$</span><span class="kw">inds</span><span class="op">$</span><span class="kw">atom</span> ] <span class="op">&lt;-</span> <span class="fl">0</span>

<span class="co"># write to file</span>
<span class="fu"><a href="../../../reference/write.pdb.html">write.pdb</a></span>(<span class="kw">b</span>, file=<span class="st">"4LHY-interface.pdb"</span>)
</pre></div>
<div class="figure">
<img src="figures/4LHY-interace.png" alt=""><p class="caption">Visualization of interface residues as obtained from function <strong>binding.site()</strong>.</p>
</div>
</div>
<div id="reading-multi-model-pdb-files" class="section level2">
<h2 class="hasAnchor">
<a href="#reading-multi-model-pdb-files" class="anchor"></a>Reading multi-model PDB files</h2>
<p>Bio3D can read and analyze multi-model PDB files, such as those representing NMR ensembles. To enable reading of all models you need to provide argument <code>multi=TRUE</code> to function <strong>read.pdb()</strong>. This will store the Cartesian coordinates of each model in the <code>xyz</code> component so that <code><a href="https://rdrr.io/r/base/nrow.html">nrow(xyz)</a></code> equals the number of models in the PDB structure. Several Bio3D functions works directly on these multi-model PDB objects. Below we provide an example of how to read and access the coordinates of a multi-model PDB file. We then illustrate two functions for the analysis of a multi-model PDB file.</p>
<div class="sourceCode" id="cb72"><pre class="downlit">
<span class="co"># Read multi-model PDB file</span>
<span class="kw">pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"1d1d"</span>, multi=<span class="fl">TRUE</span>)
</pre></div>
<pre><code>##   Note: Accessing on-line PDB file</code></pre>
<div class="sourceCode" id="cb74"><pre class="downlit">
<span class="co"># The xyz component contains 20 frames</span>
<span class="kw">pdb</span><span class="op">$</span><span class="kw">xyz</span>
</pre></div>
<pre><code>## 
##    Total Frames#: 20
##    Total XYZs#:   10185,  (Atoms#:  3395)
## 
##     [1]  1.325  0  0  &lt;...&gt;  -16.888  -81.895  5.777  [203700] 
## 
## + attr: Matrix DIM = 20 x 10185</code></pre>
<div class="sourceCode" id="cb76"><pre class="downlit">
<span class="co"># Select a subset of the protein</span>
<span class="kw">ca.inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>, <span class="st">"calpha"</span>)

<span class="co"># Access C-alpha coordinates of the first 5 models</span>
<span class="co">#pdb$xyz[1:5, ca.inds$xyz]</span>
</pre></div>
</div>
<div id="identification-of-dynamic-domains" class="section level2">
<h2 class="hasAnchor">
<a href="#identification-of-dynamic-domains" class="anchor"></a>Identification of dynamic domains</h2>
<p>Function <strong>geostas()</strong> from the Bio3D package, <strong>bio3d.geostas</strong>, attempts to identify rigid domains in a protein (or nucleic acid) structure based on knowledge of its structural ensemble (as obtained a from multi-model PDB input file, an ensemble of multiple PDB files (see section below), NMA or MD results). Below we demonstrate the identification of such dynamic domains in a multi-model PDB and use function <strong>fit.xyz()</strong> to fit all models to one of the identified domains. We then use <strong>write.pdb()</strong> to store the aligned structures for further visualization, e.g. in VMD:</p>
<div class="sourceCode" id="cb77"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">bio3d.geostas</span>)
</pre></div>
<pre><code>## Loading required package: bio3d.nma</code></pre>
<pre><code>## Registered S3 methods overwritten by 'bio3d.nma':
##   method     from 
##   dccm.egnm  bio3d
##   dccm.enma  bio3d
##   dccm.gnm   bio3d
##   dccm.nma   bio3d
##   mktrj.enma bio3d
##   mktrj.nma  bio3d
##   plot.enma  bio3d
##   plot.nma   bio3d
##   plot.rmsip bio3d
##   print.enma bio3d
##   print.nma  bio3d
##   pymol.nma  bio3d</code></pre>
<pre><code>## 
## Attaching package: 'bio3d.nma'</code></pre>
<pre><code>## The following objects are masked from 'package:bio3d':
## 
##     aanma, bhattacharyya, build.hessian, cov.enma, cov.nma,
##     covsoverlap, difference.vector, ff.aaenm, ff.aaenm2, ff.anm,
##     ff.calpha, ff.pfanm, ff.reach, ff.sdenm, fluct.nma, gnm,
##     load.enmff, nma, overlap, rmsip, rtb, sip, var.pdbs, var.xyz</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::overlap' by 'bio3d.nma::overlap' when
## loading 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::cov.enma' by 'bio3d.nma::cov.enma'
## when loading 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::ff.calpha' by 'bio3d.nma::ff.calpha'
## when loading 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::ff.aaenm2' by 'bio3d.nma::ff.aaenm2'
## when loading 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::load.enmff' by
## 'bio3d.nma::load.enmff' when loading 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::covsoverlap' by
## 'bio3d.nma::covsoverlap' when loading 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::bhattacharyya' by
## 'bio3d.nma::bhattacharyya' when loading 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::nma' by 'bio3d.nma::nma' when loading
## 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::rmsip' by 'bio3d.nma::rmsip' when
## loading 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::ff.aaenm' by 'bio3d.nma::ff.aaenm'
## when loading 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::cov.nma' by 'bio3d.nma::cov.nma' when
## loading 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::aanma' by 'bio3d.nma::aanma' when
## loading 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::var.xyz' by 'bio3d.nma::var.xyz' when
## loading 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::fluct.nma' by 'bio3d.nma::fluct.nma'
## when loading 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::ff.reach' by 'bio3d.nma::ff.reach'
## when loading 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::difference.vector' by
## 'bio3d.nma::difference.vector' when loading 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::ff.anm' by 'bio3d.nma::ff.anm' when
## loading 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::gnm' by 'bio3d.nma::gnm' when loading
## 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::rtb' by 'bio3d.nma::rtb' when loading
## 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::ff.pfanm' by 'bio3d.nma::ff.pfanm'
## when loading 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::ff.sdenm' by 'bio3d.nma::ff.sdenm'
## when loading 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::var.pdbs' by 'bio3d.nma::var.pdbs'
## when loading 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::sip' by 'bio3d.nma::sip' when loading
## 'bio3d.geostas'</code></pre>
<pre><code>## Warning: replacing previous import 'bio3d::build.hessian' by
## 'bio3d.nma::build.hessian' when loading 'bio3d.geostas'</code></pre>
<pre><code>## Registered S3 method overwritten by 'bio3d.geostas':
##   method       from 
##   plot.geostas bio3d</code></pre>
<pre><code>## 
## Attaching package: 'bio3d.geostas'</code></pre>
<pre><code>## The following objects are masked from 'package:bio3d':
## 
##     amsm.xyz, geostas</code></pre>
<div class="sourceCode" id="cb109"><pre class="downlit">
<span class="co"># Domain analysis</span>
<span class="kw">gs</span>  <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/geostas.html">geostas</a></span>(<span class="kw">pdb</span>)
</pre></div>
<pre><code>##   .. 220 'calpha' atoms selected
##   .. 'xyz' coordinate data with 20 frames 
##   .. 'fit=TRUE': running function 'core.find'
##   .. coordinates are superimposed to core region
##   .. calculating atomic movement similarity matrix ('amsm.xyz()') 
##   .. dimensions of AMSM are 220x220
##   .. clustering AMSM using 'kmeans' 
##   .. converting indices to match input 'pdb' object 
##      (additional attribute 'atomgrps' generated)</code></pre>
<div class="sourceCode" id="cb111"><pre class="downlit">
<span class="co"># Fit all frames to the 'first' domain</span>
<span class="kw">domain.inds</span> <span class="op">&lt;-</span> <span class="kw">gs</span><span class="op">$</span><span class="kw">inds</span>[[<span class="fl">1</span>]]

<span class="kw">xyz</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbfit.html">pdbfit</a></span>(<span class="kw">pdb</span>, inds=<span class="kw">domain.inds</span>)

<span class="co"># write fitted coordinates</span>
<span class="fu"><a href="../../../reference/write.pdb.html">write.pdb</a></span>(<span class="kw">pdb</span>, xyz=<span class="kw">xyz</span>, chain=<span class="kw">gs</span><span class="op">$</span><span class="kw">atomgrps</span>, file=<span class="st">"1d1d_fit-domain1.pdb"</span>)

<span class="co"># plot geostas results</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">gs</span>, contour=<span class="fl">FALSE</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_pdb_files/figure-html/unnamed-chunk-33-1.png" alt="Plot of atomic movement similarity matrix with domain annotation for PDB ID `1d1d`" width="480"><p class="caption">
Plot of atomic movement similarity matrix with domain annotation for PDB ID <code>1d1d</code>
</p>
</div>
<div class="figure">
<img src="figures/1d1d_vis-geostas.png" alt=""><p class="caption">Multi-model PDB with 20 frames superimposed on ‘domain 1’ (colored grey) identified using function <strong>geostas()</strong>. See also related function <strong>core.find()</strong>.</p>
</div>
</div>
<div id="invariant-core-identification" class="section level2">
<h2 class="hasAnchor">
<a href="#invariant-core-identification" class="anchor"></a>Invariant core identification</h2>
<p>Function <strong>core.find()</strong> also works on a multi-model PDB file (or <code>pdbs</code> object, see below) to determine the most invariant region a protein ensemble. Below we fit the NMR ensemble to the region identified by <strong>core.find()</strong>:</p>
<div class="sourceCode" id="cb112"><pre class="downlit">
<span class="co"># Invariant core</span>
<span class="kw">core</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/core.find.html">core.find</a></span>(<span class="kw">pdb</span>)

<span class="co"># fit to core region</span>
<span class="kw">xyz</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbfit.html">pdbfit</a></span>(<span class="kw">pdb</span>, inds=<span class="kw">core</span>)

<span class="co"># write fitted coordinates</span>
<span class="fu"><a href="../../../reference/write.pdb.html">write.pdb</a></span>(<span class="kw">pdb</span>, xyz=<span class="kw">xyz</span>, file=<span class="st">"1d1d_fit-core.pdb"</span>)
</pre></div>
</div>
<div id="constructing-biological-units" class="section level2">
<h2 class="hasAnchor">
<a href="#constructing-biological-units" class="anchor"></a>Constructing biological units</h2>
<p>PDB crystal structure files typically detail the atomic coordinates for the contents of just one crystal asymmetric unit. This may or may not be the same as the biologically-relevant assembly or <em>biological unit</em>. For example, the PDB entry ‘2DN1’ of human hemoglobin stores one pair of alpha and beta globin subunits (see Figure 7). However, under physiological conditions the biological unit is known to be a tetramer (composed of two alpha and two beta subunits). In this particular case, the crystal asymmetric unit is part of the biological unit. More generally, the asymmetric unit stored in a PDB file can be: (i) One biological unit; (ii) A portion of a biological unit, or (iii) Multiple biological units.</p>
<div class="figure">
<img src="figures/2dn1.png" alt=""><p class="caption">The alpha/beta dimer of human hemoglobin as stored in the PDB entry ‘2DN1’.</p>
</div>
<p>Reconstruction of the biological unit for a given PDB file can thus be essential for understanding structure-function relationships. PDB files often provide information about how to construct biological units via symmetry operations (i.e. rotation and translation) performed on the stored coordinates. In bio3d, the function <strong>read.pdb()</strong> automatically stores this information if available. The function <strong>biounit()</strong> can then be called to build biological units based on this information stored within an appropriate ‘pdb’ object. The following example illustrates the construction of the hemoglobin tetramer from the dimer in the asymmetric unit of ‘2DN1’.</p>
<div class="sourceCode" id="cb113"><pre class="downlit">
<span class="co"># Read PDB from online database</span>
<span class="kw">pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"2dn1"</span>)
</pre></div>
<pre><code>##   Note: Accessing on-line PDB file</code></pre>
<div class="sourceCode" id="cb115"><pre class="downlit">
<span class="co"># Examine biological unit matrices</span>
<span class="kw">pdb</span><span class="op">$</span><span class="kw">remark</span><span class="op">$</span><span class="kw">biomat</span>
</pre></div>
<pre><code>## $num
## [1] 1
## 
## $chain
## $chain[[1]]
## [1] "A" "B"
## 
## 
## $mat
## $mat[[1]]
## $mat[[1]]$`A B`
##      [,1] [,2] [,3] [,4]
## [1,]    1    0    0    0
## [2,]    0    1    0    0
## [3,]    0    0    1    0
## 
## $mat[[1]]$`A B`
##      [,1] [,2] [,3] [,4]
## [1,]    0    1    0    0
## [2,]    1    0    0    0
## [3,]    0    0   -1    0
## 
## 
## 
## $method
## [1] "AUTHOR"</code></pre>
<p>The <code>$remark$biomat</code> component in a ‘pdb’ object describe the transformation (translation and rotation) matrices that can be applied to construct biological units. It contains:</p>
<ul>
<li>
<strong>num</strong>, the number of biological units described</li>
<li>
<strong>chain</strong>, a ‘list’ object. Each component is a vector of chain labels to which the transformation matrices are applied to get the corresponding biological unit</li>
<li>
<strong>mat</strong>, a ‘list’ of transformation matrices for each biological unit</li>
<li>
<strong>method</strong>, a character vector about the method (‘AUTHOR’ or ‘SOFTWARE’) that determined each biological unit</li>
</ul>
<div class="sourceCode" id="cb117"><pre class="downlit">
<span class="fu"><a href="../../../reference/biounit.html">biounit</a></span>(<span class="kw">pdb</span>)
</pre></div>
<pre><code>## $`AUTHOR.determined.tetramer (4 chains)`
## 
##  Call:  biounit(pdb = pdb)
## 
##    Total Models#: 1
##      Total Atoms#: 5012,  XYZs#: 15036  Chains#: 4  (values: A B C D)
## 
##      Protein Atoms#: 4344  (residues/Calpha atoms#: 570)
##      Nucleic acid Atoms#: 0  (residues/phosphate atoms#: 0)
## 
##      Non-protein/nucleic Atoms#: 668  (residues: 472)
##      Non-protein/nucleic resid values: [ HEM (4), HOH (460), MBN (4), OXY (4) ]
## 
##    Protein sequence:
##       LSPADKTNVKAAWGKVGAHAGEYGAEALERMFLSFPTTKTYFPHFDLSHGSAQVKGHGKK
##       VADALTNAVAHVDDMPNALSALSDLHAHKLRVDPVNFKLLSHCLLVTLAAHLPAEFTPAV
##       HASLDKFLASVSTVLTSKYRHLTPEEKSAVTALWGKVNVDEVGGEALGRLLVVYPWTQRF
##       FESFGDLSTPDAVMGNPKVKAHGKKVLGAFSDGLAHLDNLKGTFA...&lt;cut&gt;...HKYH
## 
## + attr: atom, helix, sheet, seqres, xyz,
##         calpha, call, log</code></pre>
<p>The function <strong>biounit()</strong> returns a ‘list’ object. Each component of the ‘list’ represents a possible biological unit stored as a ‘pdb’ object. In above example, two new chains (chain C and D) are constructed from symmetry operations performed on the original chains (chain A and B) see Figure 8.</p>
<div class="figure">
<img src="figures/2dn1-biounit.png" alt=""><p class="caption">The biological unit of human hemoglobin (tetramer) built from PDB entry ‘2DN1’.</p>
</div>
<div id="side-note-1" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#side-note-1" class="anchor"></a>Side-note:</h4>
<p>The biological unit described in a PDB file is not always unique. Distinct methods, e.g. based on expertise of authors of the crystallographic structure or calucaltion of software (See <a href="https://pdb101.rcsb.org/learn/guide-to-understanding-pdb-data/biological-assemblies">here</a> for more details), may determine different multimeric state. Also, as mentioned above, one crystal asymmetric unit may contain multiple biological units. The function <strong>biounit()</strong> returns all the possible biological units listed in the file (and that is why the return value is a ‘list’ instead of a simple ‘pdb’ object). It is dependent on users which biological unit is finally adopted for further analysis. Check the ‘names’ attributes of the returned ‘list’ for the determination method and multimeric state of each biological unit. For example, the PDB entry ‘1KJY’ describes three possible biological units for the G protein alpha subunit bound with GDI inhibitor. The common way is to select either one of the two dimers (they have slightly distinct conformation) instead of the software predicted tetramer.</p>
<div class="sourceCode" id="cb119"><pre class="downlit">
<span class="kw">pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"1kjy"</span>)
</pre></div>
<pre><code>##   Note: Accessing on-line PDB file</code></pre>
<div class="sourceCode" id="cb121"><pre class="downlit">
<span class="kw">bio</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/biounit.html">biounit</a></span>(<span class="kw">pdb</span>)
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">bio</span>)
</pre></div>
<pre><code>## [1] "AUTHOR.determined.dimer (2 chains)"     
## [2] "AUTHOR.determined.dimer (2 chains)"     
## [3] "SOFTWARE.determined.tetramer (4 chains)"</code></pre>
<p>Optionally, <strong>biounit()</strong> returns the biological unit as a multi-model ‘pdb’ object. In this case, the topology of the ‘pdb’ (e.g. <code>$atom</code>) is the same as input, whereas additional coordinates generated via the symmetry operations are stored in distinc rows of <code>$xyz</code>. This is obtained by setting the argument <code>multi=TRUE</code>. It is particular useful for speed enhancements with biological units containing many symmetric copies, for example a virus capsid:</p>
<div class="sourceCode" id="cb123"><pre class="downlit">
<span class="kw">pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"2bfu"</span>)
</pre></div>
<pre><code>##   Note: Accessing on-line PDB file</code></pre>
<div class="sourceCode" id="cb125"><pre class="downlit">
<span class="kw">bio</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/biounit.html">biounit</a></span>(<span class="kw">pdb</span>, multi = <span class="fl">TRUE</span>)
<span class="kw">bio</span>
</pre></div>
<pre><code>## $`SOFTWARE.determined.multimer (120 chains)`
## 
##  Call:  biounit(pdb = pdb, multi = TRUE)
## 
##    Total Models#: 60
##      Total Atoms#: 5282,  XYZs#: 950760  Chains#: 2  (values: L S)
## 
##      Protein Atoms#: 5282  (residues/Calpha atoms#: 558)
##      Nucleic acid Atoms#: 0  (residues/phosphate atoms#: 0)
## 
##      Non-protein/nucleic Atoms#: 0  (residues: 0)
##      Non-protein/nucleic resid values: [ none ]
## 
##    Protein sequence:
##       MEQNLFALSLDDTSSVRGSLLDTKFAQTRVLLSKAMAGGDVLLDEYLYDVVNGQDFRATV
##       AFLRTHVITGKIKVTATTNISDNSGCCLMLAINSGVRGKYSTDVYTICSQDSMTWNPGCK
##       KNFSFTFNPNPCGDSWSAEMISRSRVRMTVICVSGWTLSPTTDVIAKLDWSIVNEKCEPT
##       IYHLADCQNWLPLNRWMGKLTFPQGVTSEVRRMPLSIGGGAGATQ...&lt;cut&gt;...TPPL
## 
## + attr: atom, helix, sheet, seqres, xyz,
##         calpha, call</code></pre>
<div class="sourceCode" id="cb127"><pre class="downlit">
<span class="kw">bio</span>[[<span class="fl">1</span>]]<span class="op">$</span><span class="kw">xyz</span>
</pre></div>
<pre><code>## 
##    Total Frames#: 60
##    Total XYZs#:   15846,  (Atoms#:  5282)
## 
##     [1]  21.615  -8.176  108.9  &lt;...&gt;  -109.546  -20.972  -91.006  [950760] 
## 
## + attr: Matrix DIM = 60 x 15846</code></pre>
</div>
</div>
</div>
<div id="working-with-multiple-pdb-files" class="section level1">
<h1 class="hasAnchor">
<a href="#working-with-multiple-pdb-files" class="anchor"></a>Working with multiple PDB files</h1>
<p>The Bio3D packages were designed to specifically facilitate the analysis of multiple structures from both experiment and simulation. The challenge of working with these structures is that they are usually different in their composition (i.e. contain differing number of atoms, sequences, chains, ligands, structures, conformations etc. even for the same protein as we will see below) and it is these differences that are frequently of most interest.</p>
<p>For this reason Bio3D contains extensive utilities to enable the reading, writing, manipulation and analysis of such heterogenous structure sets. This topic is detailed extensively in the separate Principal Component Analysis vignette available from <a href="http://thegrantlab.org/bio3d" class="uri">http://thegrantlab.org/bio3d</a>.</p>
<p>Before delving into more advanced analysis (detailed in additional vignettes) lets examine how we can read multiple PDB structures from the RCSB PDB for a particular protein and perform some basic analysis:</p>
<div class="sourceCode" id="cb129"><pre class="downlit">
<span class="co"># Download some example PDB files</span>
<span class="kw">ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"1TND_B"</span>,<span class="st">"1AGR_A"</span>,<span class="st">"1FQJ_A"</span>,<span class="st">"1TAG_A"</span>,<span class="st">"1GG2_A"</span>,<span class="st">"1KJY_A"</span>)
<span class="kw">raw.files</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/get.pdb.html">get.pdb</a></span>(<span class="kw">ids</span>)
</pre></div>
<p>The <code><a href="../../../reference/get.pdb.html">get.pdb()</a></code> function will download the requested files, below we extract the particular chains we are most interested in with the function <code><a href="../../../reference/pdbsplit.html">pdbsplit()</a></code> (note these <code>ids</code> could come from the results of a <code><a href="../../../reference/blast.pdb.html">blast.pdb()</a></code> search as described in other vignettes). The requested chains are then aligned and their structural data stored in a new object <code>pdbs</code> that can be used for further analysis and manipulation.</p>
<div class="sourceCode" id="cb130"><pre class="downlit">
<span class="co"># Extract and align the chains we are interested in</span>
<span class="kw">files</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbsplit.html">pdbsplit</a></span>(<span class="kw">raw.files</span>, <span class="kw">ids</span>)
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbaln.html">pdbaln</a></span>(<span class="kw">files</span>)
</pre></div>
<p>Below we examine the sequence and structural similarity.</p>
<div class="sourceCode" id="cb131"><pre class="downlit">
<span class="co"># Calculate sequence identity</span>
<span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/basename.pdb.html">basename.pdb</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span>)
<span class="fu"><a href="../../../reference/seqidentity.html">seqidentity</a></span>(<span class="kw">pdbs</span>)
</pre></div>
<pre><code>##        1TND_B 1AGR_A 1FQJ_A 1TAG_A 1GG2_A 1KJY_A
## 1TND_B  1.000  0.693  0.914  1.000  0.690  0.696
## 1AGR_A  0.693  1.000  0.779  0.694  0.997  0.994
## 1FQJ_A  0.914  0.779  1.000  0.914  0.776  0.782
## 1TAG_A  1.000  0.694  0.914  1.000  0.691  0.697
## 1GG2_A  0.690  0.997  0.776  0.691  1.000  0.991
## 1KJY_A  0.696  0.994  0.782  0.697  0.991  1.000</code></pre>
<div class="sourceCode" id="cb133"><pre class="downlit">
<span class="co">## Calculate RMSD</span>
<span class="fu"><a href="../../../reference/rmsd.html">rmsd</a></span>(<span class="kw">pdbs</span>, fit=<span class="fl">TRUE</span>)
</pre></div>
<pre><code>## Warning in rmsd(pdbs, fit = TRUE): No indices provided, using the 313 non NA positions</code></pre>
<pre><code>##        1TND_B 1AGR_A 1FQJ_A 1TAG_A 1GG2_A 1KJY_A
## 1TND_B  0.000  0.965  0.609  1.283  1.612  2.100
## 1AGR_A  0.965  0.000  0.873  1.575  1.777  1.914
## 1FQJ_A  0.609  0.873  0.000  1.265  1.737  2.042
## 1TAG_A  1.283  1.575  1.265  0.000  1.687  1.841
## 1GG2_A  1.612  1.777  1.737  1.687  0.000  1.879
## 1KJY_A  2.100  1.914  2.042  1.841  1.879  0.000</code></pre>
<div class="sourceCode" id="cb136"><pre class="downlit">
<span class="co">## Quick PCA (see Figure 9)</span>
<span class="kw">pc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pca.html">pca</a></span>(<span class="fu"><a href="../../../reference/pdbfit.html">pdbfit</a></span>(<span class="kw">pdbs</span>), rm.gaps=<span class="fl">TRUE</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">pc</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_pdb_files/figure-html/pca-1.png" alt="Results of protein structure principal component analysis (PCA) plotted by function **plot.pca()**" width="700"><p class="caption">
Results of protein structure principal component analysis (PCA) plotted by function <strong>plot.pca()</strong>
</p>
</div>
<p>Below we examine protein structural dynamics predicted by normal mode analysis using the <strong>bio3d.nma</strong> package from the Bio3D package family.<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a></p>
<div class="sourceCode" id="cb137"><pre class="downlit">
<span class="co">## Quick NMA of all structures (see Figure 10)</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">bio3d.nma</span>)
<span class="kw">modes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdbs</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">modes</span>, <span class="kw">pdbs</span>, spread=<span class="fl">TRUE</span>)
</pre></div>
<pre><code>## Extracting SSE from pdbs$sse attribute</code></pre>
<div class="figure">
<img src="Bio3D_pdb_files/figure-html/nma-1.png" alt="Results of ensemble normal mode analysis (NMA) plotted by function **plot.enma()**" width="768"><p class="caption">
Results of ensemble normal mode analysis (NMA) plotted by function <strong>plot.enma()</strong>
</p>
</div>
<div id="question-1" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-1" class="anchor"></a>Question:</h4>
<p>What effect does setting the <code>fit=TRUE</code> option have in the RMSD calculation? What would the results indicate if you set <code>fit=FALSE</code> or disparaged this option? HINT: Bio3D functions have various default options that will be used if the option is not explicitly specified by the user, see <code><a href="../../../reference/rmsd.html">help(rmsd)</a></code> for an example and note that the input options with an equals sign (e.g. <code>fit=FALSE</code>) have default values.</p>
</div>
</div>
<div id="where-to-next" class="section level1">
<h1 class="hasAnchor">
<a href="#where-to-next" class="anchor"></a>Where to next</h1>
<p>If you have read this far, congratulations! We are ready to have some fun and move to <a href="http://thegrantlab.org/bio3d">other package vignettes</a> that describe more interesting analysis including advanced <strong>Comparative Structure Analysis</strong> (where we will mine available experimental data and supplement it with simulation results to map the conformational dynamics and coupled motions of proteins), <strong>Trajectory Analysis</strong> (where we analyze molecular dynamics simulation trajectories), enhanced methods for <strong>Normal Mode Analysis</strong> (where we will explore the dynamics of large protein families and superfamilies), and <strong>Correlation Network Analysis</strong> (where we will build and dissect dynamic networks form different correlated motion data).</p>
</div>
<div id="document-and-current-bio3d-session-details" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#document-and-current-bio3d-session-details" class="anchor"></a>Document and current Bio3D session details</h1>
<p>This document is available on the <a href="http://thegrantlab.org/bio3d">Bio3D website</a> in both HTML and PDF formats. All code can be extracted and automatically executed to generate Figures and/or the PDF/HTML with the following commands:</p>
<div class="sourceCode" id="cb139"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://github.com/rstudio/rmarkdown">rmarkdown</a></span>)
<span class="fu"><a href="https://rdrr.io/pkg/rmarkdown/man/render.html">render</a></span>(<span class="st">"Bio3D_pdb.Rmd"</span>, <span class="st">"all"</span>)
</pre></div>
<div class="sourceCode" id="cb140"><pre class="downlit">
<span class="co"># Information about the current Bio3D session</span>
<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()
</pre></div>
<pre><code>## R version 3.6.0 (2019-04-26)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS  10.15.5
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] bio3d.geostas_0.1.0.9000 bio3d.nma_0.1.0.9000     bio3d_2.4-1.9000        
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.5         lattice_0.20-41    digest_0.6.25      crayon_1.3.4      
##  [5] rprojroot_1.3-2    assertthat_0.2.1   grid_3.6.0         R6_2.4.1          
##  [9] backports_1.1.8    magrittr_1.5       evaluate_0.14      highr_0.8         
## [13] stringi_1.4.6      rlang_0.4.6.9000   fs_1.4.2           rmarkdown_2.3.2   
## [17] pkgdown_1.5.1.9000 desc_1.2.0         tools_3.6.0        stringr_1.4.0     
## [21] parallel_3.6.0     yaml_2.2.1         xfun_0.15          compiler_3.6.0    
## [25] memoise_1.1.0      htmltools_0.5.0    knitr_1.29</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references hanging-indent">
<div id="ref-grant06">
<p>Grant, B. J., A. P. D. C Rodrigues, K. M. Elsawy, A. J. Mccammon, and L. S. D. Caves. 2006. “Bio3d: An R Package for the Comparative Analysis of Protein Structures.” <em>Bioinformatics</em> 22: 2695–6. <a href="https://doi.org/10.1093/bioinformatics/btl461">https://doi.org/10.1093/bioinformatics/btl461</a>.</p>
</div>
<div id="ref-skjaerven14">
<p>Skjaerven, L., X. Q. Yao, G. Scarabelli, and B. J. Grant. 2014. “Integrating Protein Structural Dynamics and Evolutionary Analysis with Bio3d.” <em>BMC Bioinformatics</em> 15: 399. <a href="https://doi.org/10.1186/s12859-014-0399-6">https://doi.org/10.1186/s12859-014-0399-6</a>.</p>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>The latest version of the package, full documentation and further vignettes (including detailed installation instructions) can be obtained from the main Bio3D website: <a href="http://thegrantlab.org/bio3d/">thegrantlab.org/bio3d/</a>.<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>This vignette contains executable examples, see <code><a href="https://rdrr.io/r/utils/vignette.html">help(vignette)</a></code> for further details.<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p>See also dedicated vignettes for <em>ensemble NMA</em> provided with the Bio3D package.<a href="#fnref3" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="tidyverse">
  <p>Authored by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="author">
  <p>
    Built with ❤️ and <a href="https://pkgdown.r-lib.org">pkgdown</a> © 2020.
  </p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '1270c4079b26a138795263974bbf302d',
    indexName: 'tidyverse',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
