<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to Ensemble Difference Distance Matrix (eDDM) Analysis • Bio3D</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../../../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../../../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../../../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../../../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../../../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../../../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="../../../tidyverse.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../../bootstrap-toc.css">
<script src="../../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../../pkgdown.css" rel="stylesheet">
<script src="../../../pkgdown.js"></script><link href="../../../tidyverse-2.css" rel="stylesheet">
<!--optional theme
<link href="../../../.css" rel="stylesheet">
--><!-- docsearch --><script src="../../../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../../../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Introduction to Ensemble Difference Distance Matrix (eDDM) Analysis">
<meta property="og:description" content="bio3d">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../../../index.html">
        bio3d</a>
        <div class="info hidden-xs hidden-sm">
<!--          <span class="partof">part of the <a href="https://tidyverse.org">tidyverse</a></span>
-->
          <span class="version version-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4-1.9000</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">INTRODUCTORY TUTORIALS</li>
    <li>
      <a href="../../../articles/online/install_vignette/Bio3D_install.html">Installing Bio3D</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/user-guide">New user guide</a>
    </li>
    <li>
      <a href="../../../articles/online/intro_vignette/Bio3D_introduction.html">Getting started with Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/pdb_vignette/Bio3D_pdb.html">PDB structure manipulation and analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/pca_vignette/Bio3D_pca.html">Comparative sequence and structure analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/nma_vignette/Bio3D_nma.html">Enhanced Normal Modes Analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/traj_vignette/Bio3D_md.html">Beginning trajectory analysis with Bio3D </a>
    </li>
    <li class="divider">
    <li class="dropdown-header">ADVANCED TUTORIALS</li>
    <li>
      <a href="../../../articles/online/enma_vignettes/Bio3D_nma-dhfr-partI.html">Ensemble NMA of E.coli DHFR structures</a>
    </li>
    <li>
      <a href="../../../articles/online/enma_vignettes/Bio3D_nma-dhfr-partII.html">Ensemble NMA across multiple species of DHFR</a>
    </li>
    <li>
      <a href="../../../articles/online/cna_vignette/cna_vignette.spin.html">Protein structure network analysis with Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/eddm_vignette/Bio3D_eddm.spin.html">Ensemble Difference Distance Matrix (eDDM) analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">WEBAPP TUTORIALS</li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/download/webapp-tutorials?download=22:bio3d-webapp-pca">Online protein structure analysis with Bio3D-Web (PDF)</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/webapps">Online Normal Mode Analysis with Bio3D WebApps (To Update)</a>
    </li>
    <li class="divider">
    <li>
      <a href="../../../articles/index.html">All vignettes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Documentation
  </a>
</li>
        <li>
  <a href="../../../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../../../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    News
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket"></span>
     
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Bio3D_eddm.spin_files/header-attrs-2.3.2/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to Ensemble Difference Distance Matrix (eDDM) Analysis</h1>
                        <h4 class="author">Barry J. Grant, Lars Skjærven, and Xin-Qiu Yao</h4>
            
      
      
      <div class="hidden name"><code>Bio3D_eddm.spin.Rmd</code></div>

    </div>

    
    
<div style="page-break-after: always;"></div>
<div id="background" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#background" class="anchor"></a>Background</h1>
<p>The ensemble difference distance matrix (eDDM) analysis is a computational approach to comparing structures of proteins and other biomolecules<span class="citation">(Grant, Skjaerven, and Yao 2020)</span>. It calculates and compares atomic distance matrices across large sets of homologous atomic structures to help identify the residue wise determinants underlying specific functional processes. The approach supports both experimental and theoretical simulation-generated structures. A typical eDDM workflow includes:</p>
<ul>
<li><strong>Collecting and preparing the structure set</strong></li>
<li><strong>Grouping structures</strong></li>
<li><strong>Calculating difference distance matrices and associated statistics</strong></li>
<li><strong>Identifying and visualizing significant distance changes.</strong></li>
</ul>
<p>The approach is implemented in the Bio3D package, <strong>bio3d.eddm</strong>. It can be integrated with other Bio3D methods<span class="citation">(Grant et al. 2006; Skjaerven et al. 2014)</span> for dissecting sequence-structure-function relationships, and can be used in a highly automated and reproducible manner. <strong>bio3d.eddm</strong> is distributed as an open source R package available from: <a href="https://urldefense.com/v3/__http://thegrantlab.org/bio3d__" class="uri">https://urldefense.com/v3/__http://thegrantlab.org/bio3d__</a>;!!Mih3wA!Vq7PIukjkapVBl7l5rylIUTj54SLL0NA4UaAh9K-Nw7WDKBxkEq8FzzryybFTS4$ .</p>
<div id="requirement" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#requirement" class="anchor"></a>Requirement:</h4>
<p>The latest <strong>bio3d</strong> and <strong>bio3d.eddm</strong> are required. See the <a href="https://urldefense.com/v3/__http://thegrantlab.org/bio3d__;!!Mih3wA!Vq7PIukjkapVBl7l5rylIUTj54SLL0NA4UaAh9K-Nw7WDKBxkEq8FzzryybFTS4%24">Installing Bio3D Vignette</a> for more detail. Once installed, type the following command to start using the package (Note that the dependent bio3d package is loaded automatically):</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">bio3d.eddm</span>)
</pre></div>
</div>
</div>
<div id="collecting-and-preparing-the-structure-set" class="section level1">
<h1 class="hasAnchor">
<a href="#collecting-and-preparing-the-structure-set" class="anchor"></a>Collecting and preparing the structure set</h1>
<p>In this vignette, we will use G protein-coupled receptors (GPCRs) as an example to illustrate the eDDM approach. The embedded code below can be used to reproduce the results described in the associated paper<span class="citation">(Grant, Skjaerven, and Yao 2020)</span>. Specifically, we will analyze structures of the beta adrenergic receptor (<span class="math inline">\(\beta\)</span>AR).</p>
<div id="using-the-pre-assembled-structure-set" class="section level2">
<h2 class="hasAnchor">
<a href="#using-the-pre-assembled-structure-set" class="anchor"></a>Using the pre-assembled structure set</h2>
<p>Users are recommended to use the dataset shipped with <strong>bio3d.eddm</strong> for just to reproduce the results described in this document. This dataset contains 20 pre-assembled <span class="math inline">\(\beta\)</span>AR structures. However, to apply the approach to other systems, users need to prepare the structure set from scratch (see next subsection).</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/attach.html">attach</a></span>(<span class="kw">gpcr</span>)
</pre></div>
</div>
<div id="getting-the-structure-set-from-scratch" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-the-structure-set-from-scratch" class="anchor"></a>Getting the structure set from scratch</h2>
<p>In this subsection, the detailed procedure to prepare the pre-assembled “gpcr” dataset will be described.<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> Similar steps are required preceding other Bio3D analysis methods, such as principal component analysis (PCA) and ensemble normal mode analysis (eNMA) (see relevant <a href="https://urldefense.com/v3/__http://thegrantlab.org/bio3d__;!!Mih3wA!Vq7PIukjkapVBl7l5rylIUTj54SLL0NA4UaAh9K-Nw7WDKBxkEq8FzzryybFTS4%24">Bio3D vignettes</a>).</p>
<p><em>For users who simply want to reproduce the analysis reported in this document, this subsection can be skipped.</em></p>
<p>Starting with the sequence of the human <span class="math inline">\(\beta\)</span>-2 adregenic receptor (UNIPROT: P07550), we search the PDB database to find structures with the same or similar sequence to the query, using the BLAST method implemented in Bio3D.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="kw">aa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/get.seq.html">get.seq</a></span>(<span class="st">"P07550"</span>)
<span class="kw">blast</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/blast.pdb.html">blast.pdb</a></span>(<span class="kw">aa</span>)  
</pre></div>
<p>The identified structures are sorted in the descending order of sequence similarity to the query. A similarity threshold (on the E-value) is required to cut the searching result and determine the structure set for subsequent analyses. The Bio3D function <code><a href="../../../reference/blast.pdb.html">plot.blast()</a></code> or simply <code><a href="https://rdrr.io/r/graphics/plot.html">plot()</a></code> can help find a suitable threshold by examining the distribution of the similarity scores.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="kw">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">blast</span>, cutoff=<span class="fl">301</span>)
</pre></div>
<div class="figure">
<img src="figures/blast.png" style="width:50.0%" alt=""><p class="caption"><strong>BLAST report regarding the search for beta adregenic receptor structures.</strong></p>
</div>
<p>The structures can be further filtered by structural resolution (for crystallographic and cryo-EM structures). Such information can be automatically obtained using the Bio3D function, <code><a href="../../../reference/pdb.annotate.html">pdb.annotate()</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="kw">annotation</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdb.annotate.html">pdb.annotate</a></span>(<span class="kw">hits</span>)
<span class="kw">pdb.id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span>(<span class="kw">annotation</span>, <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(<span class="kw">hits</span><span class="op">$</span><span class="kw">pdb.id</span>, <span class="kw">resolution</span><span class="op">&lt;=</span> <span class="fl">3.5</span>))
<span class="kw">annotation</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(<span class="kw">annotation</span>, <span class="kw">resolution</span> <span class="op">&lt;=</span> <span class="fl">3.5</span>)
</pre></div>
<p>Selected structrues are then downloaded through the Bio3D function, <code><a href="../../../reference/get.pdb.html">get.pdb()</a></code>. The optional <strong>split=TRUE</strong> splits the downloaded files into individual chains to facilitate subsequent analyses.</p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="kw">files</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/get.pdb.html">get.pdb</a></span>(<span class="kw">pdb.id</span>, path=<span class="st">"pdbs"</span>, split=<span class="fl">TRUE</span>)
</pre></div>
<p>All selected structures are <strong>aligned</strong>. This is to facilitate structural comparisons between “equivalent” or aligned residues.</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbaln.html">pdbaln</a></span>(<span class="kw">files</span>)
</pre></div>
<div id="side-note" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#side-note" class="anchor"></a>Side-note:</h4>
<p><em>A good practice</em> is to save the generated <code>pdbs</code> and <code>annotation</code> for future uses. This can be done by following commands:</p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="kw">mydata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(pdbs=<span class="kw">pdbs</span>, annotation=<span class="kw">annotation</span>)
<span class="fu"><a href="https://rdrr.io/r/base/save.html">save</a></span>(<span class="kw">mydata</span>, file=<span class="st">"mydata.RData"</span>)
</pre></div>
<p>To load the saved dataset, type following commands:</p>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="st">"mydata.RData"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/attach.html">attach</a></span>(<span class="kw">mydata</span>)
</pre></div>
</div>
</div>
</div>
<div id="grouping-structures" class="section level1">
<h1 class="hasAnchor">
<a href="#grouping-structures" class="anchor"></a>Grouping structures</h1>
<p>The eDDM analysis compares structural ensembles under distinct ligation, activation, etc. conditions. <strong>At least two groups</strong> of structures are required. The grouping of structures can be either from available structural annotations (e.g., the ligand identity bound with each structure; such information is available in the above prepared <code>annotation</code> object) or from a structural clustering analysis. The latter approach is more general and requires minimal prior knowledge about the system. In the following example, we use PCA of the distance matrices (through the bio3d function, <code><a href="../../../reference/pca.array.html">pca.array()</a></code>) followed by a conventional hierarchical clustering in the PC1-PC2 subspace to get the intrinsic grouping of the <span class="math inline">\(\beta\)</span>AR structures.</p>
<p>First, we need to update the aligned structures to include all heavy atoms. This is done by the bio3d function, <code><a href="../../../reference/read.all.html">read.all()</a></code>.</p>
<p>If the pre-assembled “gpcr” dataset shipped with the bio3d.eddm package is used through <code><a href="https://rdrr.io/r/base/attach.html">attach(gpcr)</a></code>, raw PDB data must be downloaded first.</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="fu"><a href="../../../reference/get.pdb.html">get.pdb</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span>, split=<span class="fl">TRUE</span>, path=<span class="st">"pdbs"</span>)
<span class="kw">pdbs.aa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.all.html">read.all</a></span>(<span class="kw">pdbs</span>, prefix=<span class="st">"pdbs/split_chain/"</span>, pdbext=<span class="st">".pdb"</span>)
</pre></div>
<p>Otherwise, if the dataset was rebuilt from scratch, type:</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="kw">pdbs.aa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.all.html">read.all</a></span>(<span class="kw">pdbs</span>)
</pre></div>
<p>In the following, we will calculate distance matrices, perform PCA, and cluster structures.</p>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="co"># Calculate distance matrices.</span>
<span class="co"># The option `all.atom=TRUE` tells that all heavy-atom coordinates will be used.</span>
<span class="kw">dm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/dm.html">dm</a></span>(<span class="kw">pdbs.aa</span>, all.atom=<span class="fl">TRUE</span>)
</pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="co"># Perform PCA of distance matrices.</span>
<span class="kw">pc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pca.array.html">pca.array</a></span>(<span class="kw">dm</span>)
</pre></div>
<pre><code>## NOTE: Removing 178047 upper triangular gap cells with missing data
##       retaining 17578 upper triangular non-gap cells for analysis.</code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="co"># See Figure 2.</span>
<span class="fu"><a href="../../../reference/plot.pca.html">plot.pca.scree</a></span>(<span class="kw">pc</span>)
</pre></div>
<div class="figure">
<img src="figures/screeplot-1.png" alt="**Scree plot of PCA.** It indicates that the first principal component (PC1) is dominant over other PCs." width="1166.66666666667"><p class="caption">
<strong>Scree plot of PCA.</strong> It indicates that the first principal component (PC1) is dominant over other PCs.
</p>
</div>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="co"># Perform structural clustering in the PC1-PC2 subspace.</span>
<span class="kw">hc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html">hclust</a></span>(<span class="fu"><a href="https://rdrr.io/r/stats/dist.html">dist</a></span>(<span class="kw">pc</span><span class="op">$</span><span class="kw">z</span>[, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>]))
<span class="kw">grps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cutree.html">cutree</a></span>(<span class="kw">hc</span>, k=<span class="fl">3</span>)
</pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="co"># See Figure 3.</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">pc</span>, pc.axes=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">2</span>), col=<span class="kw">grps</span>)
</pre></div>
<div class="figure">
<img src="figures/conformerplot-1.png" alt="**Conformer plot.** Each point represents a structure with point color indicating the membership ID of the clustering. Three major clusters or groups are identified: Group 1 (black) representing inactive GPCR structures, Group 2 (red) representing active GPCR structures, and Group 3 (green)." width="1166.66666666667"><p class="caption">
<strong>Conformer plot.</strong> Each point represents a structure with point color indicating the membership ID of the clustering. Three major clusters or groups are identified: Group 1 (black) representing inactive GPCR structures, Group 2 (red) representing active GPCR structures, and Group 3 (green).
</p>
</div>
</div>
<div id="calculating-difference-distance-matrices-and-associated-statistics" class="section level1">
<h1 class="hasAnchor">
<a href="#calculating-difference-distance-matrices-and-associated-statistics" class="anchor"></a>Calculating difference distance matrices and associated statistics</h1>
<p>The <code><a href="https://rdrr.io/pkg/bio3d.eddm/man/eddm.html">eddm()</a></code> function calculates the difference mean distance between groups for each residue pair and statistical significance assessed using a two-sample Wilcoxon test. Long-distance pairs in all structures are omitted. This step is controlled by the <code>mask</code> option of the function. By default, the function uses <code>mask="effdist"</code>, which calculates “effective” distances where changes for long-range residue pairs are scaled down to zero while changes involving short-range interacting residues are kept intact. This “mask” method is designed to assure that at long distances all residue pairs are indistinguishable, resembling an energetic perspective where the residue-residue interaction becomes zero at long distances.</p>
<p>In the following example, we use <code>mask="cmap"</code>. In this “masking” mode, internally calculated contact maps will be used to exclude residue pairs of (constantly) long distance (i.e., pairs that do not show a stable contact in any structural group). An all heavy atom-based method is used by default to calculate contacts. Specifically, a pair of residues are defined to be in contact if their minimal (heavy or non-hydrogen) atomic distance is less than a threshold (by default, 4 angstrom; popular choices are in the range of 4~5 angstrom). Contact maps for each group of structures are summarized – For each residue pair, if the contact probability across all structures within a group is over a probability threshold, 80%, a value of 1 is set for the pair in this group; otherwise, the value is set to be NA (i.e., treated as “missing data”). Other probability thresholds and distance cutoffs can be used by passing arguments to the <code><a href="https://rdrr.io/pkg/bio3d.eddm/man/eddm.html">eddm()</a></code> function (see <code><a href="https://rdrr.io/pkg/bio3d.eddm/man/eddm.html">?eddm</a></code> for more detail).</p>
<p>Using <code>mask="cmap"</code>, we can also identify “switching” residues, i.e., residues showing rearranged contact networks in different groups (see next section).</p>
<div class="sourceCode" id="cb18"><pre class="downlit">
<span class="kw">tbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/bio3d.eddm/man/eddm.html">eddm</a></span>(<span class="kw">pdbs.aa</span>, grps=<span class="kw">grps</span>, dm=<span class="kw">dm</span>, mask=<span class="st">"cmap"</span>)
</pre></div>
</div>
<div id="identifying-and-visualizing-significant-distance-changes" class="section level1">
<h1 class="hasAnchor">
<a href="#identifying-and-visualizing-significant-distance-changes" class="anchor"></a>Identifying and visualizing significant distance changes</h1>
<div id="identifying-significant-distance-changes" class="section level2">
<h2 class="hasAnchor">
<a href="#identifying-significant-distance-changes" class="anchor"></a>Identifying significant distance changes</h2>
<p>In this step, <strong>significant</strong> distance changes are identified by using the <code><a href="https://rdrr.io/pkg/bio3d.eddm/man/subset.eddm.html">subset.eddm()</a></code> or simply <code><a href="https://rdrr.io/r/base/subset.html">subset()</a></code> function. A significant change is defined by a <em>p</em>-value of the statistical test lower than the threshold <strong>alpha</strong> and the absolute mean distance change is above the threshold <strong>beta</strong>. Below, we use <strong>alpha=0.005</strong> and <strong>beta=1.0 (angstrom)</strong>. Also, only <strong>“switching”</strong> residues are returned, i.e., residues showing rearranged contact networks in different groups. This is done by setting <code>switch.only=TRUE</code>. For clarity, only Group 1 (representing “inactive” GPCR structures) and Group 2 (“active” GPCR structures) are compared.</p>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="kw">keys</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(<span class="kw">tbl</span>, alpha=<span class="fl">0.005</span>, beta=<span class="fl">1.0</span>, switch.only=<span class="fl">TRUE</span>)
<span class="kw">keys</span>
</pre></div>
<pre><code>## 
## Class:
##   eddm, data.frame
## 
## # Residue pairs:
##   36 (34 unique residues)
## 
## # Groups:
##   2 (1, 2)
## 
##                 a         b   i   j c.1 c.2  d.1   d.2 sd.1 sd.2 dm.1_2 z.1_2
## 212-274  THR68(A) ASP130(A) 212 274  NA   1 4.23  2.85 0.48 0.34   1.38  2.86
## 274-285 ASP130(A) TYR141(A) 274 285  NA   1 7.10  2.84 0.43 0.27   4.26  9.86
## 274-287 ASP130(A) SER143(A) 274 287   1  NA 2.67 10.34 0.20 0.42   7.66 37.74
## 284-287 LYS140(A) SER143(A) 284 287  NA   1 6.20  3.27 0.60 0.17   2.93  4.92
## 212-285  THR68(A) TYR141(A) 212 285  NA   1 6.15  3.40 0.44 0.41   2.75  6.25
## 275-285 ARG131(A) TYR141(A) 275 285   1  NA 3.42  4.76 0.72 0.69   1.33  1.85
##          pv.1_2
## 212-274 0.00031
## 274-285 0.00016
## 274-287 0.00016
## 284-287 0.00016
## 212-285 0.00016
## 275-285 0.00470
## &lt;...&gt;</code></pre>
<p>In the above table, various results are reported, including residue IDs, contact and distance changes, and associated statistics. For a more detailed explanation of each column, see <code><a href="https://rdrr.io/pkg/bio3d.eddm/man/eddm.html">?eddm</a></code>.</p>
<div id="side-note-1" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#side-note-1" class="anchor"></a>Side-note:</h4>
<p>We recommend that users produce exploratory plots of the <strong>alpha</strong> and <strong>beta</strong> parameters. This is similar to how researchers examine differential expression from RNA-Sequencing experiments – for example, plotting <span class="math inline">\(-log(p-value)\)</span> vs <span class="math inline">\(log_{2}(Fold)\)</span> change. Such plots can help researches examine various cut-off values and ultimately the robustness of their results and conclusions.</p>
</div>
</div>
<div id="visualizing-results" class="section level2">
<h2 class="hasAnchor">
<a href="#visualizing-results" class="anchor"></a>Visualizing results</h2>
<p>Results can be viewed as a 2D plot showing the distribution and magnitude of significant changes along the protein sequence. Optionally, select pairs can be labeled. Here, we label the two regions mentioned in the associated paper<span class="citation">(Grant, Skjaerven, and Yao 2020)</span>, i.e., the region involving the intra-cellular loop 2 (ICL2) (row #1-3 in <code>keys</code>) and the region in the middle of trans-membrane (TM) helices (row #17-19 in <code>keys</code>). Other types of plot are available by setting the <code>type</code> argument of the function (See <code><a href="https://rdrr.io/pkg/bio3d.eddm/man/plot.eddm.html">?plot.eddm</a></code> for more detail).</p>
<div class="sourceCode" id="cb21"><pre class="downlit">
<span class="co"># See Figure 4.</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">keys</span>, pdbs=<span class="kw">pdbs</span>, full=<span class="fl">TRUE</span>, resno=<span class="kw">NULL</span>, sse=<span class="kw">pdbs</span><span class="op">$</span><span class="kw">sse</span>[<span class="fl">1</span>, ], type=<span class="st">"tile"</span>, 
     labels=<span class="fl">TRUE</span>, labels.ind=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">17</span><span class="op">:</span><span class="fl">19</span>))
</pre></div>
<div class="figure">
<img src="figures/eddmplot1-1.png" alt="**The 'tile' plot of identified significant distance changes.** Each tile represents a residue pair with color and size scaled by the associated distance change from Group 1 to Group 2 (unit: angstrom)." width="1166.66666666667"><p class="caption">
<strong>The ‘tile’ plot of identified significant distance changes.</strong> Each tile represents a residue pair with color and size scaled by the associated distance change from Group 1 to Group 2 (unit: angstrom).
</p>
</div>
<p>Box-whisker plots of select pairs can also be generated for more specific comparisons.</p>
<div class="sourceCode" id="cb22"><pre class="downlit">
<span class="co"># See Figure 5.</span>
<span class="kw">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span>(<span class="kw">keys</span>, <span class="kw">dm</span>, <span class="kw">grps</span>, inds = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,<span class="fl">17</span><span class="op">:</span><span class="fl">19</span>))
<span class="kw">p</span> <span class="op">&lt;-</span> <span class="kw">p</span> <span class="op">+</span> <span class="kw">ggplot2</span>::<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span>(values=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"gray30"</span>, <span class="st">"#F8766D"</span>), name=<span class="st">"Group"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">p</span>)
</pre></div>
<div class="figure">
<img src="figures/eddmplot2-1.png" alt="**Box-whisker plots of select residue pairs.**" width="1120"><p class="caption">
<strong>Box-whisker plots of select residue pairs.</strong>
</p>
</div>
<p>Note that in the above, we use the <code>scale_color_manual()</code> function from the <strong>ggplot2</strong> package to overide the default coloring scheme, in order to match the coloring in the PCA conformer plot (<strong>Figure 3</strong>).</p>
<p>Finally, 3D views of identified residue pairs mapped onto GPCR structures are generated. This is done by the <code><a href="https://rdrr.io/pkg/bio3d.eddm/man/pymol.eddm.html">pymol.eddm()</a></code> (invoked by calling <code><a href="../../../reference/pymol.html">pymol()</a></code>) Bio3D function. Before that, structural fitting is performed to facilitate the visual comparision.<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a></p>
<div class="sourceCode" id="cb23"><pre class="downlit">
<span class="kw">core</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/core.find.html">core.find</a></span>(<span class="kw">pdbs</span>)
</pre></div>
<p>Again, the following step is dependent on whether the pre-assembled or reconstructed dataset is used.</p>
<p>If using the pre-assembled “gpcr” dataset:</p>
<div class="sourceCode" id="cb24"><pre class="downlit">
<span class="kw">xyz</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbfit.html">pdbfit</a></span>(<span class="kw">pdbs</span>, inds=<span class="kw">core</span>, outpath=<span class="st">"fitlsq"</span>, prefix=<span class="st">"pdbs/split_chain/"</span>, 
              pdbext=<span class="st">".pdb"</span>)
</pre></div>
<p>Otherwise, if using the dataset constructed from scratch:</p>
<div class="sourceCode" id="cb25"><pre class="downlit">
<span class="kw">xyz</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbfit.html">pdbfit</a></span>(<span class="kw">pdbs</span>, inds=<span class="kw">core</span>, outpath=<span class="st">"fitlsq"</span>)
</pre></div>
<p>For convenience, we update the stored coordinates using the “fitted” structures.</p>
<div class="sourceCode" id="cb26"><pre class="downlit">
<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span> <span class="op">&lt;-</span> <span class="kw">xyz</span>
</pre></div>
<p>We then view all identified key residue pairs showing significant distance changes using the <code><a href="../../../reference/pymol.html">pymol()</a></code> Bio3D function. Note that the function generates a script file that needs to be opened by the <strong>PyMol</strong> software. In the following example, the key residues are displayed as “sticks”, through specifying <code>as="sticks"</code> in the function. See <code><a href="https://rdrr.io/pkg/bio3d.eddm/man/pymol.eddm.html">?pymol.eddm</a></code> for more types of representations.</p>
<div class="sourceCode" id="cb27"><pre class="downlit">
<span class="co"># See Figures 6-8.</span>
<span class="fu"><a href="../../../reference/pymol.html">pymol</a></span>(<span class="kw">keys</span>, pdbs=<span class="kw">pdbs</span>, grps=<span class="kw">grps</span>, as=<span class="st">"sticks"</span>)
</pre></div>
<div class="figure">
<img src="figures/pymol_sw_all.png" style="width:80.0%" alt=""><p class="caption"><strong>Use of <code><a href="../../../reference/pymol.html">pymol()</a></code> to visualize all identified residue pairs (as sticks).</strong> Two GPCR structures are superimposed (green, inactive from PDB: 2R4R; cyan, active from PDB: 3SN6).</p>
</div>
<div class="figure">
<img src="figures/pymol_sw1.png" style="width:50.0%" alt=""><p class="caption"><strong>A close view of the switching region near ICL2.</strong> Two GPCR structures are superimposed (green, inactive from PDB: 2R4R; cyan, active from PDB: 3SN6).</p>
</div>
<div class="figure">
<img src="figures/pymol_sw2.png" style="width:50.0%" alt=""><p class="caption"><strong>A close view of the switching region in the middle of TM helices.</strong> Two GPCR structures are superimposed (green, inactive from PDB: 2R4R; cyan, active from PDB: 3SN6).</p>
</div>
<div class="sourceCode" id="cb28"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/detach.html">detach</a></span>(<span class="kw">gpcr</span>)
</pre></div>
</div>
</div>
<div id="about-this-document" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#about-this-document" class="anchor"></a>About this document</h1>
<p>This document is generated by the <strong>rmarkdown</strong> R package. To reproduce it, simply type following commands:<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a></p>
<div class="sourceCode" id="cb29"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://github.com/rstudio/rmarkdown">rmarkdown</a></span>)
<span class="fu"><a href="https://rdrr.io/pkg/rmarkdown/man/render.html">render</a></span>(<span class="st">"Bio3D_eddm.r"</span>, <span class="st">"all"</span>)
</pre></div>
</div>
<div id="information-about-the-current-bio3d-session" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#information-about-the-current-bio3d-session" class="anchor"></a>Information about the current Bio3D session</h1>
<div class="sourceCode" id="cb30"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>(), <span class="fl">FALSE</span>)
</pre></div>
<pre><code>## R version 3.6.0 (2019-04-26)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS  10.15.5
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] bio3d.eddm_0.1.0.9000 bio3d_2.4-1.9000     
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.5         plyr_1.8.6         compiler_3.6.0     pillar_1.4.4      
##  [5] highr_0.8          tools_3.6.0        digest_0.6.25      evaluate_0.14     
##  [9] memoise_1.1.0      lifecycle_0.2.0    tibble_3.0.1       gtable_0.3.0      
## [13] pkgconfig_2.0.3    rlang_0.4.6.9000   ggrepel_0.8.2      yaml_2.2.1        
## [17] parallel_3.6.0     pkgdown_1.5.1.9000 xfun_0.15          stringr_1.4.0     
## [21] dplyr_0.8.5        knitr_1.29         desc_1.2.0         fs_1.4.2          
## [25] vctrs_0.3.1        rprojroot_1.3-2    grid_3.6.0         tidyselect_1.1.0  
## [29] glue_1.4.1         R6_2.4.1           rmarkdown_2.3.2    reshape2_1.4.4    
## [33] ggplot2_3.3.0      purrr_0.3.4        farver_2.0.3       magrittr_1.5      
## [37] backports_1.1.8    scales_1.1.1       htmltools_0.5.0    ellipsis_0.3.1    
## [41] assertthat_0.2.1   colorspace_1.4-1   labeling_0.3       stringi_1.4.6     
## [45] munsell_0.5.0      crayon_1.3.4</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references hanging-indent">
<div id="ref-grant06">
<p>Grant, B. J., A. P. D. C Rodrigues, K. M. Elsawy, A. J. Mccammon, and L. S. D. Caves. 2006. “Bio3d: An R Package for the Comparative Analysis of Protein Structures.” <em>Bioinformatics</em> 22: 2695–6. <a href="https://doi.org/10.1093/bioinformatics/btl461">https://doi.org/10.1093/bioinformatics/btl461</a>.</p>
</div>
<div id="ref-Grant20">
<p>Grant, B. J., L. Skjaerven, and X. Q. Yao. 2020. “The Bio3D Packages for Structural Bioinformatics.” <em>Protein Sci</em> in press. <a href="https://doi.org/DOI:10.1002/pro.3923">https://doi.org/DOI:10.1002/pro.3923</a>.</p>
</div>
<div id="ref-skjaerven14">
<p>Skjaerven, L., X. Q. Yao, G. Scarabelli, and B. J. Grant. 2014. “Integrating Protein Structural Dynamics and Evolutionary Analysis with Bio3d.” <em>BMC Bioinformatics</em> 15: 399. <a href="https://doi.org/10.1186/s12859-014-0399-6">https://doi.org/10.1186/s12859-014-0399-6</a>.</p>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>Results are subject to the dynamic status of online servers, databases, etc. and may not be exactly same as the reported.<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>Structural fitting is not a required step for the eDDM calculation itself.<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p>The PyMol images need to be prepared separately and to be inserted manually.<a href="#fnref3" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="tidyverse">
  <p>Authored by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="author">
  <p>
    Built with ❤️ and <a href="https://pkgdown.r-lib.org">pkgdown</a> © 2020.
  </p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '1270c4079b26a138795263974bbf302d',
    indexName: 'tidyverse',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
