<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Integrating normal modes and correlation network analysis in Bio3D • Bio3D</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../../../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../../../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../../../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../../../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../../../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../../../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="../../../tidyverse.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../../bootstrap-toc.css">
<script src="../../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../../pkgdown.css" rel="stylesheet">
<script src="../../../pkgdown.js"></script><link href="../../../tidyverse-2.css" rel="stylesheet">
<!--optional theme
<link href="../../../.css" rel="stylesheet">
--><!-- docsearch --><script src="../../../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../../../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Integrating normal modes and correlation network analysis in Bio3D">
<meta property="og:description" content="bio3d">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../../../index.html">
        bio3d</a>
        <div class="info hidden-xs hidden-sm">
<!--          <span class="partof">part of the <a href="https://tidyverse.org">tidyverse</a></span>
-->
          <span class="version version-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4-1.9000</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">INTRODUCTORY TUTORIALS</li>
    <li>
      <a href="../../../articles/online/install_vignette/Bio3D_install.html">Installing Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/intro_vignette/Bio3D_introduction.html">Getting started with Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/pdb_vignette/Bio3D_pdb.html">PDB structure manipulation and analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/pca_vignette/Bio3D_pca.html">Comparative sequence and structure analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/nma_vignette/Bio3D_nma.html">Enhanced Normal Modes Analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/traj_vignette/Bio3D_md.html">Beginning trajectory analysis with Bio3D </a>
    </li>
    <li class="divider">
    <li class="dropdown-header">ADVANCED TUTORIALS</li>
    <li>
      <a href="../../../articles/online/enma_vignettes/Bio3D_nma-dhfr-partI.html">Ensemble NMA of E.coli DHFR structures</a>
    </li>
    <li>
      <a href="../../../articles/online/enma_vignettes/Bio3D_nma-dhfr-partII.html">Ensemble NMA across multiple species of DHFR</a>
    </li>
    <li>
      <a href="../../../articles/online/cna_vignette/Bio3D_cna-transducin.html">Correlation network analysis with Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/cna_vignette/cna_vignette.spin.html">Protein structure network analysis with Bio3D ***</a>
    </li>
    <li>
      <a href="../../../articles/online/eddm_vignette/Bio3D_eddm.spin.html">Ensemble Difference Distance Matrix (eDDM) analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">WEBAPP TUTORIALS</li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/download/webapp-tutorials?download=22:bio3d-webapp-pca">Online protein structure analysis with Bio3D-Web (PDF)</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/webapps">Online Normal Mode Analysis with Bio3D WebApps (To Update)</a>
    </li>
    <li class="divider">
    <li>
      <a href="../../../articles/index.html">All vignettes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Documentation
  </a>
</li>
        <li>
  <a href="../../../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../../../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    News
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket"></span>
     
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Bio3D_cna-transducin_files/header-attrs-2.3.2/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Integrating normal modes and correlation network analysis in Bio3D</h1>
                        <h4 class="author">Xin-Qiu Yao, Guido Scarabelli, Lars Skjaerven &amp; Barry J. Grant</h4>
            
      
      
      <div class="hidden name"><code>Bio3D_cna-transducin.Rmd</code></div>

    </div>

    
    
<div id="background" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#background" class="anchor"></a>Background</h1>
<p>Bio3D<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> is an R package that provides interactive tools for structural bioinformatics. The primary focus of Bio3D is the analysis of bimolecular structure, sequence and simulation data <span class="citation">(Grant et al. 2006)</span>.</p>
<p>Correlation network analysis can be employed to identify protein segments with correlated motions. In this approach, a weighted graph is constructed where each residue represents a node and the weight of the connection between nodes, <em>i</em> and <em>j</em>, represents their respective cross-correlation value, cij, expressed for example by the Pearson-like form or the linear mutual information. In this example, Normal mode analysis (NMA) is employed for the calculation of cross-correlations<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>.</p>
<div id="requirements" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#requirements" class="anchor"></a>Requirements</h2>
<p>Detailed instructions for obtaining and installing the Bio3D package on various platforms can be found in the <a href="http://thegrantlab.org/bio3d/tutorials"><strong>Installing Bio3D Vignette</strong></a> available both on-line and from within the Bio3D package. Note that this vignette also makes use of the <a href="http://igraph.org/">IGRAPH</a> R package, which can be installed with the command</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"igraph"</span>)
</pre></div>
</div>
<div id="about-this-document" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#about-this-document" class="anchor"></a>About this document</h2>
<p>This vignette was generated using <strong>Bio3D version 2.4.1.9000</strong>.</p>
</div>
</div>
<div id="correlation-network-analysis-based-on-single-structure-nma" class="section level1">
<h1 class="hasAnchor">
<a href="#correlation-network-analysis-based-on-single-structure-nma" class="anchor"></a>Correlation network analysis based on single-structure NMA</h1>
<p>In this first example we perform <em>NMA</em> on two crystallographic structures of transducin G protein alpha subunits, an active GTP-analog-bound structure (PDB id <em>1tnd</em>) and an inactive GDP- and GDI (GDP dissociation inhibitor)-bound structure (PDB id <em>1kjy</em>), by calling the function <strong>nma()</strong>. Cross-correlation matrices are calculated with the function <strong>dccm()</strong>. Correlation networks are then constructed with the function <strong>cna()</strong>. Network visualization is finally performed with the function <strong>plot.cna()</strong>. See also <code><a href="../../../reference/cna.html">help(cna)</a></code> for more details and example analysis.</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="http://thegrantlab.org/bio3d">bio3d</a></span>)
</pre></div>
<p>First we read our selected GTP and GDI bound structures, select chain A, and perform NMA on each individually.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="kw">pdb.gtp</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"1TND"</span>)
<span class="kw">pdb.gdi</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"1KJY"</span>)

<span class="kw">pdb.gtp</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/trim.html">trim.pdb</a></span>(<span class="kw">pdb.gtp</span>, inds=<span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb.gtp</span>, chain=<span class="st">"A"</span>))
<span class="kw">pdb.gdi</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/trim.html">trim.pdb</a></span>(<span class="kw">pdb.gdi</span>, inds=<span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb.gdi</span>, chain=<span class="st">"A"</span>))
</pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="kw">modes.gtp</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb.gtp</span>)
</pre></div>
<pre><code>##  Building Hessian...     Done in 0.066 seconds.
##  Diagonalizing Hessian...    Done in 1.107 seconds.</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="kw">modes.gdi</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb.gdi</span>)
</pre></div>
<pre><code>##  Building Hessian...     Done in 0.095 seconds.
##  Diagonalizing Hessian...    Done in 1.08 seconds.</code></pre>
<p>Residue cross-correlations can then be calculated based on these normal mode results.</p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="kw">cij.gtp</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/dccm.html">dccm</a></span>(<span class="kw">modes.gtp</span>)
<span class="kw">cij.gdi</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/dccm.html">dccm</a></span>(<span class="kw">modes.gdi</span>)
</pre></div>
<p>Correlation networks for both conformational states are constructed by applying the function <strong>cna()</strong> to the corresponding correlation matrices. Here, the C-alpha atoms represents nodes which are interconnected by edges with weights corresponding to the pairwise correlation coefficient. Edges are only constructed for pairs of nodes which obtain a coupling strength larger than a specified cutoff value (0.35 in this example). The weight of each edge is calculated as -log(|cij|), where cij is the correlation value between two nodes <em>i</em> and <em>j</em>. For each correlation network, betweenness clustering is performed to generate aggregate nodal clusters, or communities, that are highly intra-connected, but loosely inter-connected. By default, <strong>cna()</strong> returns communities associated with the maximal modularity value.</p>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="kw">net.gtp</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/cna.html">cna</a></span>(<span class="kw">cij.gtp</span>, cutoff.cij=<span class="fl">0.35</span>)
</pre></div>
<pre><code>## Warning in (function (graph, weights = E(graph)$weight, directed = TRUE, :
## At community.c:460 :Membership vector will be selected based on the lowest
## modularity score.</code></pre>
<pre><code>## Warning in (function (graph, weights = E(graph)$weight, directed = TRUE, : At
## community.c:467 :Modularity calculation with weighted edge betweenness community
## detection might not make sense -- modularity treats edge weights as similarities
## while edge betwenness treats them as distances</code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="kw">net.gdi</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/cna.html">cna</a></span>(<span class="kw">cij.gdi</span>, cutoff.cij=<span class="fl">0.35</span>)
</pre></div>
<pre><code>## Warning in (function (graph, weights = E(graph)$weight, directed = TRUE, :
## At community.c:460 :Membership vector will be selected based on the lowest
## modularity score.</code></pre>
<pre><code>## Warning in (function (graph, weights = E(graph)$weight, directed = TRUE, : At
## community.c:467 :Modularity calculation with weighted edge betweenness community
## detection might not make sense -- modularity treats edge weights as similarities
## while edge betwenness treats them as distances</code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="kw">net.gtp</span>
</pre></div>
<pre><code>## 
## Call:
##   cna.dccm(cij = cij.gtp, cutoff.cij = 0.35)
## 
## Structure: 
##  - NETWORK NODES#:   323     EDGES#: 1427 
##  - COMMUNITY NODES#: 12  EDGES#: 22 
## 
##  + attr: network, communities, community.network,
##         community.cij, cij, call</code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="kw">net.gdi</span>
</pre></div>
<pre><code>## 
## Call:
##   cna.dccm(cij = cij.gdi, cutoff.cij = 0.35)
## 
## Structure: 
##  - NETWORK NODES#:   320     EDGES#: 1587 
##  - COMMUNITY NODES#: 8   EDGES#: 11 
## 
##  + attr: network, communities, community.network,
##         community.cij, cij, call</code></pre>
<p>A 3-D visualization of networks can also be performed with the Bio3D function <strong>view.dccm()</strong> (See <code>help(view.dccm)</code> and the vignette “Enhanced Methods for Normal Mode Analysis with Bio3D” available on-line):</p>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="fu">view.dccm</span>(<span class="kw">net.gtp</span><span class="op">$</span><span class="kw">cij</span>, launch=<span class="fl">TRUE</span>)
</pre></div>
<p>Maximization of modularity sometimes creates unexpected community partitions splitting natural structure motifs, such as secondary structure elements, into many small community ‘islands’. To avoid this situation, we look into partitions with modularity close to the maximal value but with an overall smaller number of communities.</p>
<div class="sourceCode" id="cb20"><pre class="downlit">
<span class="kw">mod.select</span> <span class="op">&lt;-</span> <span class="fu">function</span>(<span class="kw">x</span>, <span class="kw">thres</span>=<span class="fl">0.005</span>) {
   <span class="kw">remodel</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/community.tree.html">community.tree</a></span>(<span class="kw">x</span>, rescale = <span class="fl">TRUE</span>)
   <span class="kw">n.max</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="kw">x</span><span class="op">$</span><span class="kw">communities</span><span class="op">$</span><span class="kw">membership</span>))
   <span class="kw">ind.max</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">remodel</span><span class="op">$</span><span class="kw">num.of.comms</span> <span class="op">==</span> <span class="kw">n.max</span>)
   <span class="kw">v</span> <span class="op">=</span> <span class="kw">remodel</span><span class="op">$</span><span class="kw">modularity</span>[<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw">remodel</span><span class="op">$</span><span class="kw">modularity</span>)<span class="op">:</span><span class="kw">ind.max</span>]
   <span class="kw">v</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rev.html">rev</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/diff.html">diff</a></span>(<span class="kw">v</span>))
   <span class="kw">fa</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">v</span><span class="op">&gt;=</span><span class="kw">thres</span>)[<span class="fl">1</span>] <span class="op">-</span> <span class="fl">1</span>
   <span class="kw">ncomm</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="kw">fa</span>), <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span>(<span class="kw">remodel</span><span class="op">$</span><span class="kw">num.of.comms</span>), <span class="kw">n.max</span> <span class="op">-</span> <span class="kw">fa</span>)

   <span class="kw">ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">remodel</span><span class="op">$</span><span class="kw">num.of.comms</span> <span class="op">==</span> <span class="kw">ncomm</span>)
   <span class="fu"><a href="../../../reference/network.amendment.html">network.amendment</a></span>(<span class="kw">x</span>, <span class="kw">remodel</span><span class="op">$</span><span class="kw">tree</span>[<span class="kw">ind</span>, ])
}
</pre></div>
<div class="sourceCode" id="cb21"><pre class="downlit">
<span class="kw">nnet.gtp</span> <span class="op">=</span> <span class="fu">mod.select</span>(<span class="kw">net.gtp</span>)
<span class="kw">nnet.gtp</span>
</pre></div>
<pre><code>## 
## Call:
##   cna.dccm(cij = cij.gtp, cutoff.cij = 0.35)
## 
## Structure: 
##  - NETWORK NODES#:   323     EDGES#: 1427 
##  - COMMUNITY NODES#: 10  EDGES#: 18 
## 
##  + attr: network, communities, community.network,
##         community.cij, cij, call</code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit">
<span class="kw">nnet.gdi</span> <span class="op">=</span> <span class="fu">mod.select</span>(<span class="kw">net.gdi</span>)
<span class="kw">nnet.gdi</span>
</pre></div>
<pre><code>## 
## Call:
##   cna.dccm(cij = cij.gdi, cutoff.cij = 0.35)
## 
## Structure: 
##  - NETWORK NODES#:   320     EDGES#: 1587 
##  - COMMUNITY NODES#: 7   EDGES#: 9 
## 
##  + attr: network, communities, community.network,
##         community.cij, cij, call</code></pre>
<p>The resulting networks can be visualized with the Bio3D function <strong>plot.cna()</strong>, which can generate 2D representations for both full residue-level and coarse-grained community-level networks.</p>
<div class="sourceCode" id="cb25"><pre class="downlit">
<span class="kw">cent.gtp.full</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/layout.cna.html">layout.cna</a></span>(<span class="kw">nnet.gtp</span>, pdb=<span class="kw">pdb.gtp</span>, full=<span class="fl">TRUE</span>, k=<span class="fl">3</span>)[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>]
<span class="kw">cent.gtp</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/layout.cna.html">layout.cna</a></span>(<span class="kw">nnet.gtp</span>, pdb=<span class="kw">pdb.gtp</span>, k=<span class="fl">3</span>)[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>]
<span class="kw">cent.gdi.full</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/layout.cna.html">layout.cna</a></span>(<span class="kw">nnet.gdi</span>, pdb=<span class="kw">pdb.gtp</span>, full=<span class="fl">TRUE</span>, k=<span class="fl">3</span>)[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>]
<span class="kw">cent.gdi</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/layout.cna.html">layout.cna</a></span>(<span class="kw">nnet.gdi</span>, pdb=<span class="kw">pdb.gtp</span>, k=<span class="fl">3</span>)[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>]
</pre></div>
<p>The following code plots the four networks.</p>
<div class="sourceCode" id="cb26"><pre class="downlit">
<span class="fu"><a href="../../../reference/plot.cna.html">plot.cna</a></span>(<span class="kw">nnet.gtp</span>, layout=<span class="kw">cent.gtp.full</span>, full=<span class="fl">TRUE</span>, vertex.label=<span class="fl">NA</span>, vertex.size=<span class="fl">5</span>)
<span class="fu"><a href="../../../reference/plot.cna.html">plot.cna</a></span>(<span class="kw">nnet.gtp</span>, layout=<span class="kw">cent.gtp</span>)
<span class="fu"><a href="../../../reference/plot.cna.html">plot.cna</a></span>(<span class="kw">nnet.gdi</span>, layout=<span class="kw">cent.gdi.full</span>, full=<span class="fl">TRUE</span>, vertex.label=<span class="fl">NA</span>, vertex.size=<span class="fl">5</span>)
<span class="fu"><a href="../../../reference/plot.cna.html">plot.cna</a></span>(<span class="kw">nnet.gdi</span>, layout=<span class="kw">cent.gdi</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_cna-transducin_files/figure-html/figure1-1.png" alt='Correlation networks for GTP "active" and GDI "inhibitory" conformational states of transducin. Networks are derived from NMA applied to single GTP and GDI crystallographic structures.' width="700"><p class="caption">
Correlation networks for GTP “active” and GDI “inhibitory” conformational states of transducin. Networks are derived from NMA applied to single GTP and GDI crystallographic structures.
</p>
</div>
</div>
<div id="correlation-network-analysis-based-on-ensemble-nma" class="section level1">
<h1 class="hasAnchor">
<a href="#correlation-network-analysis-based-on-ensemble-nma" class="anchor"></a>Correlation network analysis based on ensemble NMA</h1>
<p>In this example we perform ensemble NMA on 53 crystallographic transducin G-alpha structures, which can be categorized into active GTP-analog-bound, inactive GDP-bound, and inhibitory GDI-bound states. For this analysis we utilize the example transducin structure dataset (further details of which can be obtained via <code><a href="../../../reference/example.data.html">help(transducin)</a></code> along with the “Comparative Protein Structure Analysis with Bio3D” vignette available on-line). Briefly, this dataset includes aligned PDB coordinates (<code>pdbs</code>), structural invariant core positions (<code>core</code>) and annotations for each PDB structure (<code>annotation</code>). Cross-correlation matrices for all structures in the ensemble are calculated with the function <strong>dccm()</strong>. State-specific ensemble average correlation matrices are then obtained with the function <strong>filter.dccm()</strong>. Correlation networks are finally constructed with the function <strong>cna()</strong>, and visualization of the networks performed with the function <strong>plot.cna()</strong>:</p>
<div class="sourceCode" id="cb27"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/attach.html">attach</a></span>(<span class="kw">transducin</span>)
</pre></div>
<div class="sourceCode" id="cb28"><pre class="downlit">
<span class="kw">modes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdbs</span>, ncore=<span class="fl">8</span>)
</pre></div>
<div class="sourceCode" id="cb29"><pre class="downlit">
<span class="kw">cijs0</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/dccm.html">dccm</a></span>(<span class="kw">modes</span>)
</pre></div>
<p>Below we perform optional filtering and averaging per state of the individual structure cross-correlation matrices. In this example, we also utilize a cutoff for correlation (0.35 in this example) and a cutoff for C-alpha distance (10 angstrom) (See <code><a href="../../../reference/filter.dccm.html">help(filter.dccm)</a></code> and main text for details).</p>
<div class="sourceCode" id="cb30"><pre class="downlit">
<span class="kw">cij</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/filter.dccm.html">filter.dccm</a></span>(<span class="kw">cijs0</span>, xyz=<span class="kw">pdbs</span>, fac=<span class="kw">annotation</span>[, <span class="st">"state3"</span>], cutoff.cij=<span class="fl">0.35</span>, 
                dcut=<span class="fl">10</span>, scut=<span class="fl">0</span>, pcut=<span class="fl">0.75</span>, ncore=<span class="fl">8</span>)
</pre></div>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
## 
  |                                                                            
  |                                                                      |   0%
## 
  |                                                                            
  |                                                                      |   0%</code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit">
<span class="kw">nets</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/cna.html">cna</a></span>(<span class="kw">cij</span>, cutoff.cij = <span class="fl">0</span>)
<span class="kw">net1</span> <span class="op">=</span> <span class="fu">mod.select</span>(<span class="kw">nets</span><span class="op">$</span><span class="kw">GTP</span>)
<span class="kw">net2</span> <span class="op">=</span> <span class="fu">mod.select</span>(<span class="kw">nets</span><span class="op">$</span><span class="kw">GDI</span>)
</pre></div>
<div class="sourceCode" id="cb33"><pre class="downlit">
<span class="kw">ref.pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbs2pdb.html">pdbs2pdb</a></span>(<span class="kw">pdbs</span>, inds=<span class="fl">1</span>, rm.gaps=<span class="fl">TRUE</span>)[[<span class="fl">1</span>]]
<span class="kw">cent.gtp.full</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/layout.cna.html">layout.cna</a></span>(<span class="kw">net1</span>, pdb=<span class="kw">ref.pdb</span>, full=<span class="fl">TRUE</span>, k=<span class="fl">3</span>)[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>]
<span class="kw">cent.gtp</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/layout.cna.html">layout.cna</a></span>(<span class="kw">net1</span>, pdb=<span class="kw">ref.pdb</span>, k=<span class="fl">3</span>)[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>]
<span class="kw">cent.gdi.full</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/layout.cna.html">layout.cna</a></span>(<span class="kw">net2</span>, pdb=<span class="kw">ref.pdb</span>, full=<span class="fl">TRUE</span>, k=<span class="fl">3</span>)[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>]
<span class="kw">cent.gdi</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/layout.cna.html">layout.cna</a></span>(<span class="kw">net2</span>, pdb=<span class="kw">ref.pdb</span>, k=<span class="fl">3</span>)[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>]
</pre></div>
<div class="sourceCode" id="cb34"><pre class="downlit">
<span class="fu"><a href="../../../reference/plot.cna.html">plot.cna</a></span>(<span class="kw">net1</span>, layout=<span class="kw">cent.gtp.full</span>, full=<span class="fl">TRUE</span>, vertex.label=<span class="fl">NA</span>, vertex.size=<span class="fl">5</span>)
<span class="fu"><a href="../../../reference/plot.cna.html">plot.cna</a></span>(<span class="kw">net1</span>, layout=<span class="kw">cent.gtp</span>)
<span class="fu"><a href="../../../reference/plot.cna.html">plot.cna</a></span>(<span class="kw">net2</span>, layout=<span class="kw">cent.gdi.full</span>, full=<span class="fl">TRUE</span>, vertex.label=<span class="fl">NA</span>, vertex.size=<span class="fl">5</span>)
<span class="fu"><a href="../../../reference/plot.cna.html">plot.cna</a></span>(<span class="kw">net2</span>, layout=<span class="kw">cent.gdi</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_cna-transducin_files/figure-html/figure2-1.png" alt='Correlation networks for GTP "active" and GDI "inhibitory" conformational states of transducin. Networks are derived from ensemble NMA of available GTP and GDI crystallographic structures.' width="700"><p class="caption">
Correlation networks for GTP “active” and GDI “inhibitory” conformational states of transducin. Networks are derived from ensemble NMA of available GTP and GDI crystallographic structures.
</p>
</div>
</div>
<div id="document-details" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#document-details" class="anchor"></a>Document Details</h1>
<p>This document is shipped with the Bio3D package in both R and PDF formats. All code can be extracted and automatically executed to generate Figures and/or the PDF with the following commands:</p>
<div class="sourceCode" id="cb35"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://github.com/rstudio/rmarkdown">rmarkdown</a></span>)
<span class="fu"><a href="https://rdrr.io/pkg/rmarkdown/man/render.html">render</a></span>(<span class="st">"Bio3D_cna-transducin.Rmd"</span>, <span class="st">"all"</span>)
</pre></div>
</div>
<div id="information-about-the-current-bio3d-session" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#information-about-the-current-bio3d-session" class="anchor"></a>Information About the Current Bio3D Session</h1>
<div class="sourceCode" id="cb36"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>() 
</pre></div>
<pre><code>## R version 3.6.0 (2019-04-26)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS  10.15.5
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] bio3d_2.4-1.9000
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.5         digest_0.6.25      crayon_1.3.4       rprojroot_1.3-2   
##  [5] assertthat_0.2.1   grid_3.6.0         R6_2.4.1           backports_1.1.8   
##  [9] magrittr_1.5       evaluate_0.14      highr_0.8          stringi_1.4.6     
## [13] rlang_0.4.6.9000   fs_1.4.2           rmarkdown_2.3.2    pkgdown_1.5.1.9000
## [17] desc_1.2.0         tools_3.6.0        stringr_1.4.0      igraph_1.2.5      
## [21] parallel_3.6.0     yaml_2.2.1         xfun_0.15          compiler_3.6.0    
## [25] pkgconfig_2.0.3    memoise_1.1.0      htmltools_0.5.0    knitr_1.29</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references hanging-indent">
<div id="ref-grant06">
<p>Grant, B. J., A. P. D. C Rodrigues, K. M. Elsawy, A. J. Mccammon, and L. S. D. Caves. 2006. “Bio3d: An R Package for the Comparative Analysis of Protein Structures.” <em>Bioinformatics</em> 22: 2695–6. <a href="https://doi.org/10.1093/bioinformatics/btl461">https://doi.org/10.1093/bioinformatics/btl461</a>.</p>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>The latest version of the package, full documentation and further vignettes (including detailed installation instructions) can be obtained from the main Bio3D website: <a href="http://thegrantlab.org/bio3d/">http://thegrantlab.org/bio3d/</a>.<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>This vignette contains executable examples, see <code><a href="https://rdrr.io/r/utils/vignette.html">help(vignette)</a></code> for further details.<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="tidyverse">
  <p>Authored by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="author">
  <p>
    Built with ❤️ and <a href="https://pkgdown.r-lib.org">pkgdown</a> © 2020.
  </p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '1270c4079b26a138795263974bbf302d',
    indexName: 'tidyverse',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
