<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Protein Structure Networks with Bio3D • Bio3D</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../../../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../../../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../../../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../../../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../../../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../../../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="../../../tidyverse.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../../bootstrap-toc.css">
<script src="../../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../../pkgdown.css" rel="stylesheet">
<script src="../../../pkgdown.js"></script><link href="../../../tidyverse-2.css" rel="stylesheet">
<!--optional theme
<link href="../../../.css" rel="stylesheet">
--><!-- docsearch --><script src="../../../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../../../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Protein Structure Networks with Bio3D">
<meta property="og:description" content="bio3d">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../../../index.html">
        bio3d</a>
        <div class="info hidden-xs hidden-sm">
<!--          <span class="partof">part of the <a href="https://tidyverse.org">tidyverse</a></span>
-->
          <span class="version version-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4-1.9000</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">INTRODUCTORY TUTORIALS</li>
    <li>
      <a href="../../../articles/online/install_vignette/Bio3D_install.html">Installing Bio3D</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/user-guide">New user guide</a>
    </li>
    <li>
      <a href="../../../articles/online/intro_vignette/Bio3D_introduction.html">Getting started with Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/pdb_vignette/Bio3D_pdb.html">PDB structure manipulation and analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/pca_vignette/Bio3D_pca.html">Comparative sequence and structure analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/nma_vignette/Bio3D_nma.html">Enhanced Normal Modes Analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/traj_vignette/Bio3D_md.html">Beginning trajectory analysis with Bio3D </a>
    </li>
    <li class="divider">
    <li class="dropdown-header">ADVANCED TUTORIALS</li>
    <li>
      <a href="../../../articles/online/enma_vignettes/Bio3D_nma-dhfr-partI.html">Ensemble NMA of E.coli DHFR structures</a>
    </li>
    <li>
      <a href="../../../articles/online/enma_vignettes/Bio3D_nma-dhfr-partII.html">Ensemble NMA across multiple species of DHFR</a>
    </li>
    <li>
      <a href="../../../articles/online/cna_vignette/cna_vignette.spin.html">Protein structure network analysis with Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/eddm_vignette/Bio3D_eddm.spin.html">Ensemble Difference Distance Matrix (eDDM) analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">WEBAPP TUTORIALS</li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/download/webapp-tutorials?download=22:bio3d-webapp-pca">Online protein structure analysis with Bio3D-Web (PDF)</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/webapps">Online Normal Mode Analysis with Bio3D WebApps (To Update)</a>
    </li>
    <li class="divider">
    <li>
      <a href="../../../articles/index.html">All vignettes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Documentation
  </a>
</li>
        <li>
  <a href="../../../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../../../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    News
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket"></span>
     
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="cna_vignette.spin_files/header-attrs-2.3.2/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Protein Structure Networks with Bio3D</h1>
                        <h4 class="author">Xin-Qiu Yao, Guido Scarabelli, Lars Skjaerven &amp; Barry J. Grant</h4>
            
      
      
      <div class="hidden name"><code>cna_vignette.spin.Rmd</code></div>

    </div>

    
    
<div id="background" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#background" class="anchor"></a>Background</h1>
<p>Bio3D<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> is a family of R packages that provide interactive tools for structural bioinformatics. The primary focus of Bio3D is the analysis of bimolecular structure, sequence and simulation data.</p>
<p>The Bio3D package, <strong>bio3d.cna</strong>, contains functionality for the creation and analysis of protein structure networks. This vignette will introduce this functionality by building and analyzing networks obtained from atomic correlation data (named <em>correlation network analysis</em>) derived from normal mode analysis (NMA) and molecular dynamics (MD) simulation (<strong>Figure 1</strong>)<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>. Note that Bio3D contains extensive functionality for both NMA and MD analysis. An introduction to each of these areas is detailed in separate NMA and MD tutorals/vignettes available from <a href="http://thegrantlab.org/bio3d" class="uri">http://thegrantlab.org/bio3d</a>.</p>
<p><br></p>
<div class="figure">
<img src="figures/cna_fig1.png" alt=""><p class="caption"><strong>Overview of a typical Bio3D network analysis protocol with key functions detailed in blue</strong>. Normal mode and/or molecular dynamics input first undergoes <strong>correlation analysis</strong> (with the function <strong>dccm()</strong>). The output from these functions is typically a matrix of residue-by-residue cross-correlations. <strong>Correlation network analysis</strong> can then be performed (with the <strong>cna()</strong> function) to generate a correlation network with residues as nodes that are linked by weighted edges that are proportional to their degree of correlated motion. The <strong>cna()</strong> function also performs a <strong>community clustering analysis</strong> to identify <em>communities</em> of highly correlated residues. The full residue network and coarse-grained community network can be visualized and analyzed further. With the object returned by cna(), a <strong>suboptimal path analysis</strong> can be performed via the <strong>cnapath()</strong> function. This analysis delineates “communication” paths between distal sites and identifies potential key residues underlying them. Collectively, correlation network analysis can provide valuable insight not readily available from the measures of correlation alone.</p>
</div>
<div id="requirements" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#requirements" class="anchor"></a>Requirements:</h4>
<p>Detailed instructions for obtaining and installing the Bio3D packages on various platforms can be found in the <a href="http://thegrantlab.org/"><strong>Installing Bio3D Vignette</strong></a> available on-line. Note that this vignette also makes use of the <a href="http://igraph.org/">IGRAPH</a> R package, which can be installed with the command:</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"igraph"</span>)
</pre></div>
</div>
</div>
<div id="basic-network-generation-and-visualization" class="section level1">
<h1 class="hasAnchor">
<a href="#basic-network-generation-and-visualization" class="anchor"></a>Basic network generation and visualization</h1>
<p>In this section we will first focus on correlation network analysis derived from NMA, and in subsequent sections MD.</p>
<div id="network-generation-from-normal-mode-analysis-data" class="section level2">
<h2 class="hasAnchor">
<a href="#network-generation-from-normal-mode-analysis-data" class="anchor"></a>Network generation from normal mode analysis data</h2>
<p>The code snippet below first loads the Bio3D and related packages, then reads an example PDB structure from the RCSB online database. We then follow the flow in Figure 1 by performing NMA, then dynamic cross-correlation analysis, and finally network analysis with the <strong>cna()</strong> function.</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="http://thegrantlab.org/bio3d">bio3d</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">bio3d.cna</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">bio3d.nma</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="http://igraph.org">igraph</a></span>)
<span class="kw">pdbfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"examples/hivp.pdb"</span>, package = <span class="st">"bio3d"</span>)
<span class="kw">pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>( <span class="kw">pdbfile</span> )
<span class="kw">modes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb</span>)
<span class="kw">cij</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/dccm.html">dccm</a></span>(<span class="kw">modes</span>)
<span class="kw">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/cna.html">cna</a></span>(<span class="kw">cij</span>, cutoff.cij=<span class="fl">0.3</span>)
</pre></div>
<p>To enable quick examination of the network analysis results we provide <strong>cna</strong> specific variants of the <strong>print()</strong>, <strong>summary()</strong> and <strong>plot()</strong> functions, as demonstrated below:</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">net</span>)
</pre></div>
<pre><code>## 
## Call:
##   cna(cij = cij, cutoff.cij = 0.3)
## 
## Structure: 
##  - NETWORK NODES#:   198     EDGES#: 752 
##  - COMMUNITY NODES#: 8   EDGES#: 11 
## 
##  + attr: network, communities, community.network,
##         community.cij, cij, call</code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="co"># Summary information as a table </span>
<span class="kw">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span>(<span class="kw">net</span>)
</pre></div>
<pre><code>##  id size                      members
##   1   36      c(1:8, 89:107, 190:198)
##   2   20        c(9:11, 22:32, 83:88)
##   3   24              c(12:21, 60:73)
##   4   28       c(33:46, 55:59, 74:82)
##   5   16            c(47:54, 146:153)
##   6   17     c(108, 122:130, 183:189)
##   7   28          c(109:121, 159:173)
##   8   29 c(131:145, 154:158, 174:182)</code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="co"># Other useful things are returned from **summary.cna()**</span>
<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/graph_attr_names.html">attributes</a></span>(<span class="kw">x</span>)
</pre></div>
<pre><code>## $names
## [1] "id"      "size"    "members" "tbl"</code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="co"># List the residues belonging to the 5th community</span>
<span class="kw">x</span><span class="op">$</span><span class="kw">members</span>[<span class="fl">5</span>]
</pre></div>
<pre><code>## $`5`
##  47  48  49  50  51  52  53  54 146 147 148 149 150 151 152 153 
##  47  48  49  50  51  52  53  54 146 147 148 149 150 151 152 153</code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="co"># Plot both the ‘full’ all-residue network and simplified community network</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">net</span>, <span class="kw">pdb</span>, full = <span class="fl">TRUE</span>, vertex.label.cex=<span class="fl">0.7</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">net</span>, <span class="kw">pdb</span>)
</pre></div>
<div class="figure">
<img src="cna_vignette.spin_files/figure-html/figure2-1.png" alt="Full all-residue network and simplified community network" width="600"><p class="caption">
Full all-residue network and simplified community network
</p>
</div>
<div id="side-note" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#side-note" class="anchor"></a>Side note:</h4>
<p>The raw correlation matrix obtained from NMA (or indeed MD) typically has many small but non-zero value elements. By default the <strong>cna()</strong> function excludes some of the weakest values (close to zero) to avoid building an edge heavy network, which would considerably slow down further analysis. The <strong>cutoff.cij</strong> argument in the <strong>cna()</strong> function specifies the lower boundary for considering correlation values. All values whose absolute value is lower than <strong>cutoff.cij</strong> will be ignored when defining network edges.</p>
<p>Note that the above code plots both a <em>full</em> all-residue network as well as a more coarse grained community network - see <strong>Figure 2</strong>. In this example, and by default, the Girvan-Newman clustering method was used to map the full network into communities of highly intra-connected but loosely inter-connected nodes that were in turn used to derive the simplified network displayed in <strong>Figure 2 (right)</strong>. We will discuss community detection in a separate section below and for now simply highlight that a comparison of the full network and the distribution and connectivity of communities between similar biomolecular systems (e.g. in the presence and absence of a binding partner or mutation) has the potential to highlight differences in coupled motions that may be indicative of potential allosteric pathways. This approach has previously been applied to investigate allostery in tRNA–protein complexes, kinesin motor domains, G-proteins, thrombin, and other systems [1-5].</p>
<p>Another useful function for the inspection of community structure is the <strong>vmd.cna()</strong> function (invoked by calling <strong>vmd()</strong>), which permits interactive network visualization in the VMD molecular graphics program (see <strong>Figure 3</strong>).</p>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="co"># View the network in VMD (see Figure 3)</span>
<span class="fu"><a href="../../../reference/vmd.html">vmd</a></span>(<span class="kw">net</span>, <span class="kw">pdb</span>, launch = <span class="fl">TRUE</span>)
</pre></div>
<div class="figure">
<img src="figures/cna_fig3.png" alt=""><p class="caption">Example of <strong>vmd.cna()</strong> output. For this figure we have set the VMD display style to ‘Tube’ and color method to ‘Color by Chain’. This allows one to see corresponding regions of structure colored by community along with the community protein structure network</p>
</div>
<p>One of the main advantages of correlation network analysis is that it often facilitates the interpretation of correlated motions that can be challenging to rationalize from the output of a dynamic correlation map alone (e.g. the output of the <strong>dccm()</strong> function). For example, the identified communities of highly intra-connected residues can be tied into the visualization of the correlation data (<strong>Figure 4</strong>) and cross referenced to the network visualizations presented in <strong>Figures 2</strong> and <strong>3</strong>. In this example the colored marginal segments of <strong>Figure 4</strong> match the identified communities in <strong>Figures 2</strong> and <strong>3</strong>:</p>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="co"># Plot the correlation matrix with community annotation, see Figure 4. </span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">cij</span>, margin.segments = <span class="kw">net</span><span class="op">$</span><span class="kw">communities</span><span class="op">$</span><span class="kw">membership</span>, main=<span class="st">""</span>)
</pre></div>
<div class="figure">
<img src="cna_vignette.spin_files/figure-html/figNMAcij-1.png" alt="Example dynamical cross-correlation matrix (DCCM) plot with community annotation. Note that unique colors are assigned to communities (that are not necessary consecutive in sequence) and that these correspond to those annotated communities in all previous network figures and those used by VMD in Figure 3." width="600"><p class="caption">
Example dynamical cross-correlation matrix (DCCM) plot with community annotation. Note that unique colors are assigned to communities (that are not necessary consecutive in sequence) and that these correspond to those annotated communities in all previous network figures and those used by VMD in Figure 3.
</p>
</div>
</div>
</div>
<div id="the-structure-of-cna-network-objects" class="section level2">
<h2 class="hasAnchor">
<a href="#the-structure-of-cna-network-objects" class="anchor"></a>The structure of cna() network objects</h2>
<p>To more fully appreciate the results of network analysis one should become somewhat familiar with the structure of the object returned from the <strong>cna()</strong> function. This will allow you to generate customized visualizations beyond those in <strong>Figures 2</strong> to <strong>4</strong> and also enable you to more fully examine the effect of different parameters on network structure and topology.</p>
<p>The output of the <strong>cna()</strong> function is a list object of class <em>cna</em> with both network and community structure attributes. These are summarized at the bottom of the <strong>print(net)</strong> results above (see the line beginning with ‘+ attr’) and can be explicitly listed with the <strong>attributes(cna)</strong> command:</p>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/graph_attr_names.html">attributes</a></span>(<span class="kw">net</span>)
</pre></div>
<pre><code>## $names
## [1] "network"           "communities"       "community.network"
## [4] "community.cij"     "cij"               "call"             
## 
## $class
## [1] "cna"</code></pre>
<p>Here we introduce each of these components in turn:</p>
<ul>
<li><p><strong><code>$network</code></strong>: The <em>full</em> protein structure network typically with a node per residue and connecting edges weighted by their corresponding correlation values (i.e., input cij values optionally filtered by the user defined ‘cutoff.cij’). This filtered cij matrix is also returned in the output as <strong><code>$cij</code></strong>, as listed below.</p></li>
<li><p><strong><code>$communities</code></strong>: A community clustering object detailing the results of clustering the full <strong><code>$network</code></strong>. This is itself a list object with a number of attributes (see <em><code><a href="https://rdrr.io/pkg/igraph/man/graph_attr_names.html">attributes(net$communities)</a></code></em> for details). Notable attributes include <strong><code>$communities$membership</code></strong>, <strong><code>$communities$modularity</code></strong> and <strong><code>$communities$algorithm</code></strong>. These components detail the nodes (typically residues) belonging to each community; the modularity (which represents the difference in probability of intra- and intercommunity connections for a given network division); and the clustering algorithm used respectively.</p></li>
<li><p><strong><code>$cij</code></strong>: The filtered correlation matrix used to build the full $network.</p></li>
<li><p><strong><code>$community.network</code></strong>: A coarse-grained community network object with the number of nodes equal to the number of communities present in <strong><code>$communities</code></strong> (i.e., the community clustering of the full network discussed below) and edges based on the inter-community coupling as determined by the input ‘collapse.method’ (by default this is the maximum coupling between the original nodes of the respective communities).</p></li>
<li><p><strong><code>$community.cij</code></strong>: A cij matrix obtained by applying a “collapse.method” on <code>$cij</code>. The rows and columns match the number of communities in $communities. The individual values are based on the cij couplings of inter-community residues. The currently available collapse methods for defining these values include: max, mean, trimmed and median.</p></li>
</ul>
<p>Both the <strong><code>$network</code></strong> and <strong><code>$community.network</code></strong> attributes are <a href="https://urldefense.com/v3/__http://igraph.sourceforge.net__;!!Mih3wA!Vq7PIukjkapVBl7l5rylIUTj54SLL0NA4UaAh9K-Nw7WDKBxkEq8Fzzr79BqmbY%24">igraph compatible network objects</a> [6] and contain additional node (or vertex) and edge annotations that can be accessed with the <strong>V()</strong> and <strong>E()</strong> functions in the following way:</p>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="co"># Basic data access follows the igraph package scheme</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span>(<span class="kw">net</span><span class="op">$</span><span class="kw">network</span>)<span class="op">$</span><span class="kw">color</span>)
</pre></div>
<pre><code>## [1] "#0000FF" "#0000FF" "#0000FF" "#0000FF" "#0000FF" "#0000FF"</code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span>(<span class="kw">net</span><span class="op">$</span><span class="kw">community.network</span>)<span class="op">$</span><span class="kw">size</span>
</pre></div>
<pre><code>## [1] 36 20 24 28 16 17 28 29</code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/E.html">E</a></span>(<span class="kw">net</span><span class="op">$</span><span class="kw">community.network</span>)<span class="op">$</span><span class="kw">weight</span>
</pre></div>
<pre><code>##  [1] 0.4959895 0.4579519 0.5620385 0.5190477 0.7274177 0.4988692 0.3649480
##  [8] 0.3648937 0.6414050 0.5867744 0.4989730</code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span>(<span class="kw">net</span><span class="op">$</span><span class="kw">community.network</span>)<span class="op">$</span><span class="kw">name</span>
</pre></div>
<pre><code>## [1] "1" "2" "3" "4" "5" "6" "7" "8"</code></pre>
<p>In addition to various Bio3D functions, the full set of extensive <a href="http://igraph.sourceforge.net">igraph package</a> features can thus be used to further analyze these network objects.</p>
</div>
<div id="the-community-clustering-procedure" class="section level2">
<h2 class="hasAnchor">
<a href="#the-community-clustering-procedure" class="anchor"></a>The community clustering procedure</h2>
<p>The <strong><code>$communities</code></strong> attribute returned by the <strong>cna()</strong> function provides details of the community clustering procedure. This procedure aims to split the full network into highly correlated local substructures (referred to as communities) within which the network connections are dense but between which they are sparser. A number of community detection algorithms can be used to solve this so called graph-partitioning problem. The default Girvan-Newman edge-betweenness approach is a divisive algorithm that is based on the use of the edge betweenness as a partitioning criterion. “Betweenness” is calculated by finding the shortest path(s) between a pair of vertexes and scoring each of the edges on this/these path(s) with the inverse value of the number of shortest paths. (So if there was only one path of the shortest length, each edge on it would score 1 and if there were 10 paths of that length, each edge would score 1/10.) This is done for every pair of vertexes. In this way each edge accumulates a “betweenness” score for the whole network. The network is separated into clusters by removing the edge with the highest “betweenness”, then recalculating betweenness and repeating. The method is fully described in (Girvan and Newman, 2002 PNAS) [7].</p>
<p>A useful quantity used to measure the quality of a given community partition is the <strong>modularity</strong> (detailed in the <strong><code>$communities$modularity</code></strong> component of <strong>cna()</strong> output). The modularity represents the difference in probability of intra- and inter-community connections for a given network division. Modularity values fall in the range from 0 to 1, with higher values indicating higher quality of the community structure. The optimum community structures obtained for typical MD and NMA derived correlation networks fall in the 0.4 to 0.7 range.</p>
<div class="sourceCode" id="cb24"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">net</span><span class="op">$</span><span class="kw">communities</span><span class="op">$</span><span class="kw">modularity</span>)
</pre></div>
<pre><code>## [1] -5.770111e-03 -4.430715e-03 -3.616681e-03 -8.542082e-04  2.043481e-05
## [6]  2.076868e-03</code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span>(<span class="kw">net</span><span class="op">$</span><span class="kw">communities</span><span class="op">$</span><span class="kw">modularity</span>)
</pre></div>
<pre><code>## [1] 0.7381234</code></pre>
<p>The <strong>community.tree()</strong> function can be used to reconstruct the community membership vector for each clustering step. Here we plot modularity vs number of communities.</p>
<div class="sourceCode" id="cb28"><pre class="downlit">
<span class="co"># See Figure 5.</span>
<span class="kw">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/community.tree.html">community.tree</a></span>(<span class="kw">net</span>, rescale=<span class="fl">TRUE</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>( <span class="kw">tree</span><span class="op">$</span><span class="kw">num.of.comms</span>, <span class="kw">tree</span><span class="op">$</span><span class="kw">modularity</span>, xlab=<span class="st">"Communities"</span>, ylab=<span class="st">"Modularity"</span>) 
</pre></div>
<div class="figure">
<img src="cna_vignette.spin_files/figure-html/modularityPlot-1.png" alt="Network modularity upon removing edges during the betweenness clustering procedure for community annotation. Note the peak in modularity at k=8." width="540"><p class="caption">
Network modularity upon removing edges during the betweenness clustering procedure for community annotation. Note the peak in modularity at k=8.
</p>
</div>
<p>Note that there are many points near the maximum modularity value in <strong>Figure 5</strong>. In such cases a number of alternative community partitions may be equally high scoring and therefore worth inspecting more closely. For example we will often favor a partition point with a smaller number of overall communities but slightly lower modularity score based on some prior knowledge of the system (for example results from a related system).</p>
<p>Below we inspect the maximum modularity value partitioning and the number of communities (k) at the point of maximum modularity. Based on a quick inspection of the clustering dendrogram we explore a new membership partitioning (at k=3). Again this example is purely for demonstrating some of the available functionality within the package, and we encourage motivated readers to explore the true structure within their data by examining multiple structures together with the effects of different input parameters and clustering procedures.</p>
<div class="sourceCode" id="cb29"><pre class="downlit">
<span class="kw">max.mod.ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span>(<span class="kw">tree</span><span class="op">$</span><span class="kw">modularity</span>)
<span class="kw">tree</span><span class="op">$</span><span class="kw">num.of.comms</span>[ <span class="kw">max.mod.ind</span> ]
</pre></div>
<pre><code>## [1] 8</code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit">
<span class="co"># Membership vector at this partition point</span>
<span class="kw">tree</span><span class="op">$</span><span class="kw">tree</span>[<span class="kw">max.mod.ind</span>,]
</pre></div>
<pre><code>##   [1] 1 1 1 1 1 1 1 1 2 2 2 3 3 3 3 3 3 3 3 3 3 2 2 2 2 2 2 2 2 2 2 2 4 4 4 4 4
##  [38] 4 4 4 4 4 4 4 4 4 5 5 5 5 5 5 5 5 4 4 4 4 4 3 3 3 3 3 3 3 3 3 3 3 3 3 3 4
##  [75] 4 4 4 4 4 4 4 4 2 2 2 2 2 2 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 6 7 7 7
## [112] 7 7 7 7 7 7 7 7 7 7 6 6 6 6 6 6 6 6 6 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 5 5 5
## [149] 5 5 5 5 5 8 8 8 8 8 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 8 8 8 8 8 8 8 8 8 6 6 6
## [186] 6 6 6 6 1 1 1 1 1 1 1 1 1</code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit">
<span class="co"># Should be the same as that contained in the original CNA network object</span>
<span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span>( <span class="kw">net</span><span class="op">$</span><span class="kw">communities</span><span class="op">$</span><span class="kw">membership</span> <span class="op">==</span> <span class="kw">tree</span><span class="op">$</span><span class="kw">tree</span>[<span class="kw">max.mod.ind</span>,] )
</pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit">
<span class="co"># Inspect the clustering dendrogram (Figure 6). </span>
<span class="co"># To match colors use 'colors=vmd_colors()[memb.k3]'</span>
<span class="kw">h</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/as.hclust.html">as.hclust</a></span>(<span class="kw">net</span><span class="op">$</span><span class="kw">communities</span>)
<span class="fu"><a href="../../../reference/hclustplot.html">hclustplot</a></span>(<span class="kw">h</span>, k=<span class="fl">3</span>)
</pre></div>
<div class="figure">
<img src="cna_vignette.spin_files/figure-html/hplot-1.png" alt="Example of community partitioning viewed as a dendrogram with colored communities" width="480"><p class="caption">
Example of community partitioning viewed as a dendrogram with colored communities
</p>
</div>
<div class="sourceCode" id="cb36"><pre class="downlit">
<span class="co"># Inspect a new membership partitioning (at k=3)</span>
<span class="kw">memb.k3</span> <span class="op">&lt;-</span> <span class="kw">tree</span><span class="op">$</span><span class="kw">tree</span>[ <span class="kw">tree</span><span class="op">$</span><span class="kw">num.of.comms</span> <span class="op">==</span> <span class="fl">3</span>, ]
</pre></div>
<div class="sourceCode" id="cb37"><pre class="downlit">
<span class="co"># Produce a new k=3 community network (Figure 7)</span>
<span class="kw">net.3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/network.amendment.html">network.amendment</a></span>(<span class="kw">net</span>, <span class="kw">memb.k3</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">net.3</span>, <span class="kw">pdb</span>)
</pre></div>
<div class="figure">
<img src="cna_vignette.spin_files/figure-html/net3-1.png" alt="Example of forcing a k=3 community network" width="420"><p class="caption">
Example of forcing a k=3 community network
</p>
</div>
<div class="sourceCode" id="cb38"><pre class="downlit">
<span class="co"># View the network in VMD (see Figure 8)</span>
<span class="fu"><a href="../../../reference/vmd.html">vmd</a></span>(<span class="kw">net.3</span>, <span class="fu"><a href="../../../reference/trim.html">trim.pdb</a></span>(<span class="kw">pdb</span>, <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>,<span class="st">"calpha"</span>)), launch=<span class="fl">TRUE</span> )
</pre></div>
<div class="figure">
<img src="figures/cna_fig7.png" alt=""><p class="caption">Example of <strong>vmd()</strong> output for modified k=3 community network. Within VMD we have set display style to ‘Tube’ and color to ‘Color by Chain’ as well as removing community spheres by deleting graphical objects.</p>
</div>
</div>
<div id="network-generation-from-molecular-dynamics-data" class="section level2">
<h2 class="hasAnchor">
<a href="#network-generation-from-molecular-dynamics-data" class="anchor"></a>Network generation from molecular dynamics data</h2>
<p>For this example section we apply correlation network analysis to a short molecular dynamics trajectory of Human Immunodeficiency Virus aspartic protease (HIVpr). This trajectory is included with the bio3d package and stored in CHARMM/NAMD DCD format. To reduce file size and speed up example execution, non C-alpha atoms have been previously removed (see the <a href="http://thegrantlab.org/">Trajectory Analysis vignette</a> for full details).</p>
<p>The code snippet below first sets the file paths for the example HIVpr starting structure (pdbfile) and trajectory data (dcdfile), then reads these files (producing the objects dcd and pdb).</p>
<div class="sourceCode" id="cb39"><pre class="downlit">
<span class="kw">dcdfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"examples/hivp.dcd"</span>, package = <span class="st">"bio3d"</span>)
<span class="kw">pdbfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"examples/hivp.pdb"</span>, package = <span class="st">"bio3d"</span>)
</pre></div>
<div class="sourceCode" id="cb40"><pre class="downlit">
<span class="co"># Read MD data</span>
<span class="kw">dcd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.dcd.html">read.dcd</a></span>(<span class="kw">dcdfile</span>)
<span class="kw">pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="kw">pdbfile</span>)
</pre></div>
<div class="sourceCode" id="cb41"><pre class="downlit">
<span class="kw">inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>, resno = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">24</span><span class="op">:</span><span class="fl">27</span>, <span class="fl">85</span><span class="op">:</span><span class="fl">90</span>), elety = <span class="st">"CA"</span>)
<span class="kw">trj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/fit.xyz.html">fit.xyz</a></span>(fixed = <span class="kw">pdb</span><span class="op">$</span><span class="kw">xyz</span>, mobile = <span class="kw">dcd</span>,
               fixed.inds = <span class="kw">inds</span><span class="op">$</span><span class="kw">xyz</span>, mobile.inds = <span class="kw">inds</span><span class="op">$</span><span class="kw">xyz</span>)
</pre></div>
<p>Once we have the superposed trajectory frames we can asses the extent to which the atomic fluctuations of individual residues (in this very short example simulation) are correlated with one another and build a network from this data:</p>
<div class="sourceCode" id="cb42"><pre class="downlit">
<span class="co"># See Figure 9.</span>
<span class="kw">cij</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/dccm.html">dccm</a></span>(<span class="kw">trj</span>)
<span class="kw">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/cna.html">cna</a></span>(<span class="kw">cij</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">net</span>, <span class="kw">pdb</span>)
</pre></div>
<div class="figure">
<img src="cna_vignette.spin_files/figure-html/example2MD-1.png" alt="Example network analysis of MD data." width="480"><p class="caption">
Example network analysis of MD data.
</p>
</div>
<div id="side-note-1" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#side-note-1" class="anchor"></a>Side note:</h4>
<p>Due to the often noisy nature of correlations calculated from short MD simulations, we typically suggest running longer simulations and multiple replica simulations from which a consensus correlation matrix and consensus network can be generated and compared to results obtained from individual simulations. For example, see the following works: Scarabelli and Grant, (2014). Kinesin-5 allosteric inhibitors uncouple the dynamics of nucleotide, microtubule and neck-linker binding sites, Biophys J. 107:2204 and Yao, et al., and Grant, (2016). Dynamic coupling and allosteric networks in the alpha subunit of heterotrimeric G proteins. J. Biol. Chem. 291:4742. A similar approach can be applied to multiple correlation matrices calculated from an ensemble normal mode analysis. An example is given in the Section <strong>Ensemble correlation network analysis</strong> below.</p>
<p>Note that the <strong>dccm()</strong> function can be slow to run on very large trajectories. In such cases you may want to use multiple CPU cores for the calculation by setting the ‘ncore’ option to an appropriate value for your computer - see <em>help(dccm)</em> for details.</p>
<div class="sourceCode" id="cb43"><pre class="downlit">
<span class="co"># View the correlations in pymol (see Figure 10).</span>
<span class="fu"><a href="../../../reference/pymol.html">pymol</a></span>(<span class="kw">cij</span>, <span class="kw">pdb</span>, type = <span class="st">"launch"</span>)
</pre></div>
<div class="figure">
<img src="figures/cna_fig10.png" alt=""><p class="caption">Example of <strong>pymol()</strong> output for an MD trajectory</p>
</div>
<div class="sourceCode" id="cb44"><pre class="downlit">
<span class="co"># View the structure mapped network (see Figure 11).</span>
<span class="fu"><a href="../../../reference/vmd.html">vmd</a></span>(<span class="kw">net</span>, <span class="kw">pdb</span>, launch = <span class="fl">TRUE</span>)
</pre></div>
<div class="figure">
<img src="figures/cna_fig11.png" alt=""><p class="caption">The <strong>vmd()</strong> output for the HIVpr example MD trajectory</p>
</div>
<p>As noted above, maximization of modularity sometimes creates unexpected community partitions splitting visually obvious domains into many small community ‘islands’. In such cases we look into partitions with modularity close to the maximal value but with an overall smaller number of communities (see above).</p>
<div class="sourceCode" id="cb45"><pre class="downlit">
<span class="co"># See Figure 12.</span>
<span class="kw">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/community.tree.html">community.tree</a></span>(<span class="kw">net</span>, rescale=<span class="fl">TRUE</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>( <span class="kw">tree</span><span class="op">$</span><span class="kw">num.of.comms</span>, <span class="kw">tree</span><span class="op">$</span><span class="kw">modularity</span>, xlab=<span class="st">"Communities"</span>, ylab=<span class="st">"Modularity"</span> ) 
</pre></div>
<div class="figure">
<img src="cna_vignette.spin_files/figure-html/modularityPlot2-1.png" alt="Network modularity for example HIVpr MD derived network. Note the two peaks in modularity for this short trajectory." width="600"><p class="caption">
Network modularity for example HIVpr MD derived network. Note the two peaks in modularity for this short trajectory.
</p>
</div>
<div class="sourceCode" id="cb46"><pre class="downlit">
<span class="co"># Example function using community.tree() and network.amendment()</span>
<span class="kw">mod.select</span> <span class="op">&lt;-</span> <span class="fu">function</span>(<span class="kw">x</span>, <span class="kw">thres</span>=<span class="fl">0.1</span>) {
   <span class="kw">remodel</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/community.tree.html">community.tree</a></span>(<span class="kw">x</span>, rescale = <span class="fl">TRUE</span>)
   <span class="kw">n.max</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="kw">x</span><span class="op">$</span><span class="kw">communities</span><span class="op">$</span><span class="kw">membership</span>))
   <span class="kw">ind.max</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">remodel</span><span class="op">$</span><span class="kw">num.of.comms</span> <span class="op">==</span> <span class="kw">n.max</span>)
   <span class="kw">v</span> <span class="op">=</span> <span class="kw">remodel</span><span class="op">$</span><span class="kw">modularity</span>[<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw">remodel</span><span class="op">$</span><span class="kw">modularity</span>)<span class="op">:</span><span class="kw">ind.max</span>]
   <span class="kw">v</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rev.html">rev</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/diff.html">diff</a></span>(<span class="kw">v</span>))
   <span class="kw">fa</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">v</span><span class="op">&gt;=</span><span class="kw">thres</span>)[<span class="fl">1</span>] <span class="op">-</span> <span class="fl">1</span>
   <span class="kw">ncomm</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="kw">fa</span>), <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span>(<span class="kw">remodel</span><span class="op">$</span><span class="kw">num.of.comms</span>), <span class="kw">n.max</span> <span class="op">-</span> <span class="kw">fa</span>)

   <span class="kw">ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">remodel</span><span class="op">$</span><span class="kw">num.of.comms</span> <span class="op">==</span> <span class="kw">ncomm</span>)
   <span class="fu"><a href="../../../reference/network.amendment.html">network.amendment</a></span>(<span class="kw">x</span>, <span class="kw">remodel</span><span class="op">$</span><span class="kw">tree</span>[<span class="kw">ind</span>, ])
}
</pre></div>
<div class="sourceCode" id="cb47"><pre class="downlit">
<span class="kw">nnet</span> <span class="op">=</span> <span class="fu">mod.select</span>(<span class="kw">net</span>)
<span class="kw">nnet</span>
<span class="co"># See Figure 13.</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">nnet</span>, <span class="kw">pdb</span>)
</pre></div>
<div class="figure">
<img src="cna_vignette.spin_files/figure-html/modularity_selection-1.png" alt="Alternate community partition." width="420"><p class="caption">
Alternate community partition.
</p>
</div>
<div class="sourceCode" id="cb48"><pre class="downlit">
<span class="co"># See Figure 14.</span>
<span class="fu"><a href="../../../reference/vmd.html">vmd</a></span>(<span class="kw">nnet</span>, <span class="kw">pdb</span>, launch=<span class="fl">TRUE</span>)
</pre></div>
<div class="figure">
<img src="figures/cna_fig13.png" alt=""><p class="caption">The <strong>vmd()</strong> output for the HIVpr example MD trajectory after modularity inspection</p>
</div>
</div>
</div>
<div id="using-a-contact-map-filter" class="section level2">
<h2 class="hasAnchor">
<a href="#using-a-contact-map-filter" class="anchor"></a>Using a contact map filter</h2>
<p>In the original Luthey-Schulten and co-workers approach to correlation network analysis[1], edges were only included between “in contact” nodes. Contacting nodes were defined as those having any heavy atom within 4.5 Å for greater than 75% of the simulation. The weight of these contact edges was then taken from the cross-correlation data. This approach effectively ignores long range correlations between residues that are not in physical contact for the majority of the trajectory. To replicate this approach one can simply define a contact map with the <strong>cmap()</strong> function and provide this to the <strong>cna()</strong> function along with the required cross-correlation matrix, e.g.:</p>
<div class="sourceCode" id="cb49"><pre class="downlit">
<span class="kw">cm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/cmap.html">cmap</a></span>(<span class="kw">trj</span>, dcut = <span class="fl">4.5</span>, scut = <span class="fl">0</span>, pcut = <span class="fl">0.75</span>, mask.lower = <span class="fl">FALSE</span>)
<span class="kw">net.cut</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/cna.html">cna</a></span>(<span class="kw">cij</span>, cm = <span class="kw">cm</span>)
</pre></div>
<p>One could also simply multiple the correlation matrix by the contact map to zero out these long-range correlations. However, we currently prefer not to use this filtering step as we have found that this approach can remove potentially interesting correlations that might prove to be important for the types of long-range coupling we are most interested in exploring. For example, compare the plot of correlation values below in <strong>Figure 15</strong> to those in <strong>Figure 4</strong>.</p>
<div class="sourceCode" id="cb50"><pre class="downlit">
<span class="co"># See Figure 15.</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>((<span class="kw">cij</span> <span class="op">*</span> <span class="kw">cm</span>), main=<span class="st">""</span>)
</pre></div>
<div class="figure">
<img src="cna_vignette.spin_files/figure-html/plotDCCM-1.png" alt="Contact map filtered DCCM plot (compare to Figure 4)." width="600"><p class="caption">
Contact map filtered DCCM plot (compare to Figure 4).
</p>
</div>
<div class="sourceCode" id="cb51"><pre class="downlit">
<span class="co">## Plot non-filtered DCCM</span>
<span class="co">##plot.dccm(cij, margin.segments = net$communities$membership)</span>
</pre></div>
<p>It is also possible to use a consensus of contact map and dynamical correlation matrices from replica simulations with the functions <strong>filter.cmap()</strong> and <strong>filter.dccm()</strong> - see their respective help pages for further details. (A more advanced use of <strong>filter.dccm()</strong> for an ensemble correlation network analysis is available in the <strong>Section 4</strong>.)</p>
</div>
</div>
<div id="customized-network-visualization" class="section level1">
<h1 class="hasAnchor">
<a href="#customized-network-visualization" class="anchor"></a>Customized network visualization</h1>
<p>We have already introduced the <code><a href="../../../reference/plot.cna.html">plot.cna()</a></code> and <code><a href="../../../reference/vmd.html">vmd.cna()</a></code> functions in earlier sections. Here we focus on methods for customizing network layouts, colors, labels, nodes and edges for visualization.</p>
<div id="finding-layouts-for-network-visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#finding-layouts-for-network-visualization" class="anchor"></a>Finding layouts for network visualization</h2>
<p>The <strong>layout.cna()</strong> function can be used to determine the <em>geometric center</em> of protein residues and communities and thus provide 3D and 2D network and community graph layouts that most closely match the protein systems 3D coords.</p>
<div class="sourceCode" id="cb52"><pre class="downlit">
<span class="kw">layout3D</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/layout.cna.html">layout.cna</a></span>(<span class="kw">net</span>, <span class="kw">pdb</span>, k=<span class="fl">3</span>) 
<span class="kw">layout2D</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/layout.cna.html">layout.cna</a></span>(<span class="kw">net</span>, <span class="kw">pdb</span>, k=<span class="fl">2</span>)
</pre></div>
<p>Other network layout options are available via the igraph package, see <em>?igraph::layout</em> in R for details. In addition, one can provide any 2D plotting coordinate values to <strong>plot.cna()</strong> to generate alternate orientations.</p>
</div>
<div id="identifying-nodescommunities-and-their-composite-residues" class="section level2">
<h2 class="hasAnchor">
<a href="#identifying-nodescommunities-and-their-composite-residues" class="anchor"></a>Identifying nodes/communities and their composite residues</h2>
<p>The <strong>identify.cna()</strong> function (invoked by calling <strong>identify()</strong>) allows one to click on the nodes of a <strong>plot.cna()</strong> network plot to identify the residues within a give set of nodes. This can also be a useful first step to determine the placement of text labels for a plot -see <em>help(identify.cna)</em> for details.</p>
<div class="sourceCode" id="cb53"><pre class="downlit">
<span class="co"># Click with the right mouse to label, left button to exit.</span>
<span class="kw">xy</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">net</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/identify.html">identify</a></span>(<span class="kw">xy</span>, cna = <span class="kw">net</span>)
</pre></div>
</div>
<div id="color-settings-and-nodeedge-labels" class="section level2">
<h2 class="hasAnchor">
<a href="#color-settings-and-nodeedge-labels" class="anchor"></a>Color settings and node/edge labels</h2>
<p>The <strong>vmd_colors()</strong> function is used by default to define node color. This can be changed to any input color vector as demonstrated below. Note that the <strong>vmd_colors()</strong> function outputs a vector of colors that is ordered as they are in the VMD molecular graphics program and changing the colors will likely break this correspondence.</p>
<div class="sourceCode" id="cb54"><pre class="downlit">
<span class="kw">grp.col</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/bwr.colors.html">bwr.colors</a></span>(<span class="fl">20</span>)[<span class="kw">net</span><span class="op">$</span><span class="kw">communities</span><span class="op">$</span><span class="kw">membership</span>]
</pre></div>
<p>Useful options in <strong>plot.cna()</strong> to be aware of include the optional flags ‘mark.groups’ and ‘mark.col’, which let one draw and color areas around groups of residues corresponding to the community clustering or any other residue partitioning of interest (see <strong>Figure 16</strong>):</p>
<div class="sourceCode" id="cb55"><pre class="downlit">
<span class="kw">grp.col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>()
<span class="co">for</span> (<span class="kw">i</span> <span class="co">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span>(<span class="kw">net</span><span class="op">$</span><span class="kw">communities</span><span class="op">$</span><span class="kw">membership</span>)) {
  <span class="kw">grp.tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">net</span><span class="op">$</span><span class="kw">communities</span><span class="op">$</span><span class="kw">membership</span> <span class="op">==</span> <span class="kw">i</span>)
  <span class="kw">grp.col</span>[[<span class="kw">i</span>]] <span class="op">&lt;-</span> <span class="kw">grp.tmp</span>
}
</pre></div>
<div class="sourceCode" id="cb56"><pre class="downlit">
<span class="kw">colbar.full</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/vmd_colors.html">vmd_colors</a></span>()[<span class="kw">net</span><span class="op">$</span><span class="kw">communities</span><span class="op">$</span><span class="kw">membership</span>]
<span class="kw">colbar.comms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/vmd_colors.html">vmd_colors</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span>(<span class="kw">net</span><span class="op">$</span><span class="kw">communities</span><span class="op">$</span><span class="kw">membership</span>), alpha = <span class="fl">0.5</span>)
<span class="co"># See Figure 16.</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">net</span>, <span class="kw">pdb</span>, full = <span class="fl">TRUE</span>, mark.groups = <span class="kw">grp.col</span>, mark.col = <span class="kw">colbar.comms</span>)
</pre></div>
<div class="figure">
<img src="cna_vignette.spin_files/figure-html/plotNET-1.png" alt="Example of *mark.groups* and *mark.col* options to the **plot.cna()** function. In this case adding convex hulls that are colored according to communities." width="420"><p class="caption">
Example of <em>mark.groups</em> and <em>mark.col</em> options to the <strong>plot.cna()</strong> function. In this case adding convex hulls that are colored according to communities.
</p>
</div>
</div>
<div id="network-manipulation" class="section level2">
<h2 class="hasAnchor">
<a href="#network-manipulation" class="anchor"></a>Network manipulation</h2>
<p>The <strong>prune.cna()</strong> function allows one to remove communities under a certain size or with less less than a certain number of edges to other communities prior to visualization or further analysis. For example, to remove communities composed of only 1 residue (see <strong>Figure 17</strong>):</p>
<div class="sourceCode" id="cb57"><pre class="downlit">
<span class="kw">net.pruned</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/prune.cna.html">prune.cna</a></span>(<span class="kw">net</span>, size.min = <span class="fl">2</span>)
</pre></div>
<pre><code>##  id size                 members
##   1   21 c(1:7, 95:102, 193:198)
##   2    2                     8:9
##   3   24         c(10:23, 63:72)
##   4    4                   24:27
##   5   47         c(28:62, 73:84)
##   6    1                      85
##   7    1                      86
##   8    1                      87
##   9    2                   88:89
##  10    5                   90:94
##  11    5                 103:107
##  12   26     c(108:122, 162:172)
##  13    6                 123:128
##  14   44     c(129:161, 173:183)
##  15    2                 184:185
##  16    1                     186
##  17    2                 187:188
##  18    4                 189:192
## Removing Nodes: 7, 8, 9, 16, 17, 6 
## id   size    edges   members
## 6    1   1   85
## 7    1   0   86
## 8    1   0   87
## 9    2   0   88:89
## 16   1   0   186
## 17   2   0   187:188</code></pre>
<div class="sourceCode" id="cb59"><pre class="downlit">
<span class="co"># See Figure 17.</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">net</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">net.pruned</span>)
</pre></div>
<div class="figure">
<img src="cna_vignette.spin_files/figure-html/pruneNET-1.png" alt="Original (left) and pruned (right) networks" width="720"><p class="caption">
Original (left) and pruned (right) networks
</p>
</div>
<p>Applying this command to our network example, you can see how the communities composed by less than 2 residues are deleted from the graph.</p>
</div>
</div>
<div id="network-node-centrality-and-suboptimal-path-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#network-node-centrality-and-suboptimal-path-analysis" class="anchor"></a>Network node centrality and suboptimal path analysis</h1>
<div id="calculating-node-centrality" class="section level2">
<h2 class="hasAnchor">
<a href="#calculating-node-centrality" class="anchor"></a>Calculating node centrality</h2>
<p>Node centrality describes the distribution of the edges in the network. In protein networks, identifying the nodes with many edges (hubs) as well as the segments with a high number of connections can yield insights into the internal dynamic coordination of protein regions. Centrality measures can also be employed to characterize protein sections showing differences in coupled motions between networks derived from different states (e.g. ligand bound and unbound etc.).</p>
<p>There are a number of different measures of node centrality, from the simplest one that counts the number of edges of each node (node degree) to the more complex (such as betweenness centrality). For example, the betweenness centrality of a node is defined as the number of unique shortest paths crossing that node. This measure has the advantage of considering the whole network topology and not only the closest neighbors in its calculation (as is the case with node degree).</p>
<p>The igraph package provides different functions to perform these calculations (such as <strong>degree</strong>, <strong>betweenness</strong> and <strong>closeness</strong>). Below we illustrate an example output of the betweenness centrality on our simple HIVP network:</p>
<div class="sourceCode" id="cb60"><pre class="downlit">
<span class="kw">node.betweenness</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/betweenness.html">betweenness</a></span>(<span class="kw">net</span><span class="op">$</span><span class="kw">network</span>)
</pre></div>
<div class="sourceCode" id="cb61"><pre class="downlit">
<span class="co"># See Figure 18. </span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">node.betweenness</span>, xlab=<span class="st">"Residue No"</span>, ylab=<span class="st">"Centrality"</span>, type=<span class="st">"h"</span>)
</pre></div>
<div class="figure">
<img src="cna_vignette.spin_files/figure-html/centrality-1.png" alt="Betweenness centrality." width="600"><p class="caption">
Betweenness centrality.
</p>
</div>
<p>Interestingly, in extended trajectories of this system the regions with the highest centrality values are the two flaps of the HIVpr structure, whose dynamics and interactions are likely important of substrate processing. To examine this for your own system you could map these values (or any other metric of interest) to the B-factor column of a PDB file for viewing in VMD or other molecular graphics software that can color by this column:</p>
<div class="sourceCode" id="cb62"><pre class="downlit">
<span class="co"># Output a PDB file with normalized betweenness mapped to B-factor </span>
<span class="fu"><a href="../../../reference/write.pdb.html">write.pdb</a></span>(<span class="kw">pdb</span>, b=<span class="fu"><a href="../../../reference/normalize.vector.html">normalize.vector</a></span>(<span class="kw">node.betweenness</span>), file=<span class="st">"tmp.pdb"</span>)
</pre></div>
</div>
<div id="suboptimal-path-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#suboptimal-path-analysis" class="anchor"></a>Suboptimal path analysis</h2>
<p>Identifying residues potentially involved in the dynamic coupling of distal protein regions can be facilitated by calculating possible linking paths through the correlation network objects. In this analysis, multiple distinct shortest or (sub)optimal paths between pre-specified sites (termed ‘source’ and ‘sink’, which are usually functionally relevant such as the active site and an allosteric site) are searched. To run such an analysis, we can use the Bio3D function <strong>cnapath()</strong> and associated tools for printing and visualization (e.g., the <strong>cnapath</strong> variants of <strong>summary()</strong>, <strong>print()</strong>, <strong>plot()</strong>, and <strong>vmd()</strong>).</p>
<div class="sourceCode" id="cb63"><pre class="downlit">
<span class="kw">pa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/cnapath.html">cnapath</a></span>(<span class="kw">net</span>, from=<span class="fl">32</span>, to=<span class="fl">131</span>, k=<span class="fl">50</span>)
</pre></div>
<p>The ‘from’ and ‘to’ arguments in above function call represent the node IDs of the two ends (i.e., ‘source’ and ‘sink’, respectively) in the path analysis. The number of paths calculated is specified by the argument ‘k’ (Here we explore 50 suboptimal paths).</p>
<p>A path is a set of network edges connecting the two sites, and path length is defined by the sum of edge weights. For example, network edges can be weighted by <span class="math inline">\(-ln(|cij|)\)</span>, where cij is the correlation between residues forming the edge. In this case, path length distribution can be used to measure the overall coupling strength of the network (longer paths mean weaker coupling). Such information can be obtained by using the function <strong>print()</strong> (or <strong>summary()</strong> for a similar effect). Also generated by the function is the importance of a residue measured by normalized node degeneracy (i.e., the fraction of suboptimal paths going through the node or residue).</p>
<div class="sourceCode" id="cb64"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">pa</span>)
</pre></div>
<pre><code>## Number of paths in network(s):
##      50 
## 
## Path length distribution: 
## 
## [2.287,2.297] (2.297,2.306] (2.306,2.315] (2.315,2.324] (2.324,2.333] 
##             4             6            13            12            15 
## 
## Node degeneracy table: 
## 
##  32   46   47   48   53   54   55   56   131  147  152  153  155 
##  1.00 0.12 0.22 0.66 0.56 0.26 0.40 0.98 1.00 1.00 0.64 1.00 0.42</code></pre>
<div class="sourceCode" id="cb66"><pre class="downlit">
<span class="co"># Or, simply type 'pa'</span>
</pre></div>
<div class="sourceCode" id="cb67"><pre class="downlit">
<span class="co"># Compare two or more sets of paths </span>
<span class="co"># (e.g., from networks of the same system but under distinct states)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">pa1</span>, <span class="kw">pa2</span>, <span class="kw">...</span>)
</pre></div>
<p>To quickly inspect on path length distribution and node degeneracy, use <strong>plot.cnapath()</strong>:</p>
<div class="sourceCode" id="cb68"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">pa</span>, label=<span class="st">"HIVP"</span>, pdb=<span class="kw">pdb</span>)
</pre></div>
<p>Or, for the same effect:</p>
<div class="sourceCode" id="cb69"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">pa</span>, label=<span class="st">"HIVP"</span>, pdb=<span class="kw">pdb</span>, plot=<span class="fl">TRUE</span>)
</pre></div>
<p>Returning a value for further analyses (of, e.g., node degeneracy):</p>
<div class="sourceCode" id="cb70"><pre class="downlit">
<span class="kw">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">pa</span>, pdb=<span class="kw">pdb</span>)
<span class="kw">out</span><span class="op">$</span><span class="kw">degeneracy</span>
</pre></div>
<p>To visulize paths use the function <strong>vmd.cnapath()</strong> (NOTE: you need VMD installed properly).</p>
<div class="sourceCode" id="cb71"><pre class="downlit">
<span class="fu"><a href="../../../reference/vmd.html">vmd</a></span>(<span class="kw">pa</span>, pdb=<span class="kw">pdb</span>, spline=<span class="fl">TRUE</span>, col=<span class="st">'cyan'</span>, launch=<span class="fl">TRUE</span>)
</pre></div>
<div id="side-note-2" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#side-note-2" class="anchor"></a>Side note:</h4>
<p>Set <strong>spline=FALSE</strong> to get faster loading of paths in VMD. Also, set <strong>launch=FALSE</strong> if you want to view paths later. To do that, load the VMD script file generated by the function from the command line (e.g., in Unix-like systems, type <code>vmd -e vmd.cnapath.vmd</code>. Read VMD manual for more detailed instruction). Customized output file names are supported through the <strong>out.prefix</strong> argument of the function (See <code><a href="../../../reference/vmd.html">?vmd.cnapath</a></code> for more detail).</p>
<p>To view example applications of the method and how key residues involved in allosteric signaling were identified, please read the following work: Scarabelli and Grant, (2014). Kinesin-5 allosteric inhibitors uncouple the dynamics of nucleotide, microtubule and neck-linker binding sites, <em>Biophys J</em>, <strong>107</strong>, 2204-2213 and Yao, et al., and Grant, (2016). Dynamic coupling and allosteric networks in the alpha subunit of heterotrimeric G proteins. <em>J Biol Chem</em>, <strong>291</strong>, 4742-4753.</p>
<p>Although we encourage users to do suboptimal path analysis with Bio3D, users may have their own preferred software packages. In this case, Bio3D can be used to do all calculations during the network construction. Then, the results are interfaced with other software for subsequent path analysis. For example, users may choose the WISP program [8] to search shortest paths between select network nodes. The program requires an input adjacency matrix of the network, which can be calculated with Bio3D and saved in a file using the following command:</p>
<div class="sourceCode" id="cb72"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span>(<span class="kw">net</span><span class="op">$</span><span class="kw">cij</span>, quote=<span class="fl">FALSE</span>, row.names=<span class="fl">FALSE</span>, col.names=<span class="fl">FALSE</span>, file=<span class="st">"adj.txt"</span>)
</pre></div>
</div>
</div>
</div>
<div id="ensemble-correlation-network-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#ensemble-correlation-network-analysis" class="anchor"></a>Ensemble correlation network analysis</h1>
<p>In this section, we will introduce the use of Bio3D to perform an ensemble correlation network analysis (eCNA). The method represents an enhanced version of the protein structure or correlation network analysis introduced in preceding sections that is based on a single structure or correlation matrix. In eCNA, multiple correlation matrices are taken as input, either from multiple MD simulations or an ensemble NMA. Matrices sampled for the same state (e.g., from multiple simulation replicas or NMA of the same structural category) are examined to identify significant correlations for the state, which are then used to define edges of the state-specific network. The objectives here are to screen out spurious long-distance strong correlations due to statistical variations (i.e., false positives in a correlation network analysis using a single correlation matrix), and simultaneously to capture robustly strong correlations that may be excluded in contact map based method just because the residue distance is beyond an empirical cutoff.</p>
<div id="calculating-multiple-correlation-matrices" class="section level2">
<h2 class="hasAnchor">
<a href="#calculating-multiple-correlation-matrices" class="anchor"></a>Calculating multiple correlation matrices</h2>
<p>In the following example we will use the dataset named ‘transducin’ available from the <strong>bio3d</strong> package, which contains 53 pre-assembled crystallographic structures of G<span class="math inline">\(\alpha\)</span>i categorized into <em>active GTP-analog-bound</em>, <em>inactive GDP-bound</em>, and <em>inhibitory GDI-bound</em> states. The dataset includes aligned PDB coordinates (<code>pdbs</code>), structural invariant core positions (<code>core</code>), and annotations for each PDB structure (<code>annotation</code>). Further details of the dataset can be obtained via <code><a href="../../../reference/example.data.html">help(transducin)</a></code> along with the <strong>Comparative Protein Structure Analysis with Bio3D</strong> vignette available <a href="http://thegrantlab.org/bio3d">online</a>.</p>
<div class="sourceCode" id="cb73"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/attach.html">attach</a></span>(<span class="kw">transducin</span>)
</pre></div>
<p>In Bio3D, residue cross-correlations are calculated with the <strong>dccm()</strong> function. The dynamical data can be from MD, NMA, multiple experimental structures, or a single multi-model PDB from, e.g., NMR. In this example, we employ NMA applied to the pre-assembled aligned structures. For clarity, we only consider structures in the <strong>GTP</strong> and <strong>GDI</strong> state.</p>
<div class="sourceCode" id="cb74"><pre class="downlit">
<span class="kw">keep.inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>( <span class="kw">annotation</span>[, <span class="st">"state3"</span>] <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"GTP"</span>, <span class="st">"GDI"</span>) )
<span class="kw">annotation</span> <span class="op">&lt;-</span> <span class="kw">annotation</span>[<span class="kw">keep.inds</span>, ]
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/trim.html">trim</a></span>(<span class="kw">pdbs</span>, row.inds=<span class="kw">keep.inds</span>)
<span class="kw">modes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdbs</span>)
<span class="kw">cijs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/dccm.html">dccm</a></span>(<span class="kw">modes</span>)
</pre></div>
</div>
<div id="filtering-correlations" class="section level2">
<h2 class="hasAnchor">
<a href="#filtering-correlations" class="anchor"></a>Filtering correlations</h2>
<p>Significant correlations are then identified with the function <strong>filter.dccm()</strong>, which inspects both robustness of correlations across structures and spatial distance between related residues. Here, the correlation inspection is done for each structural category or group. Briefly, a correlation value (denoted by ‘cij’) is retained if:</p>
<ol style="list-style-type: lower-alpha">
<li>
<span class="math inline">\(cij\ge c\)</span> across all structures in the group (or all simulation replicas if MD data are used), where <span class="math inline">\(c\)</span> is the correlation cutoff, or</li>
<li>
<span class="math inline">\(cij\ge c\)</span> in at least one structure and the corresponding residue pair are in contact.</li>
</ol>
<p>A user-defined cutoff value <span class="math inline">\(c\)</span> can be provided. Otherwise, a <strong>roughly</strong> guessed cutoff value will be used, which approximately keeps 5% of all residue-residue correlations. A contact map is also required, which can be either provided by the user (pre-calculated using, e.g., the bio3d function <strong>cmap()</strong>) or calculated internally. The internal algorithm in filter.dccm() defines a contact by that the C<span class="math inline">\(\alpha\)</span>-C<span class="math inline">\(\alpha\)</span> distance is no greater than 10 Å for 75% of all the structures in a group. See <code><a href="../../../reference/filter.dccm.html">?filter.dccm</a></code> for more detail.</p>
<div class="sourceCode" id="cb75"><pre class="downlit">
<span class="co"># The 'scut=0' indicates to include all residue pairs for the contact map</span>
<span class="co"># calculation, including neighboring residues in sequence.</span>
<span class="kw">cij</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/filter.dccm.html">filter.dccm</a></span>(<span class="kw">cijs</span>, cutoff.cij=<span class="fl">0.35</span>, xyz=<span class="kw">pdbs</span>, fac=<span class="kw">annotation</span>[, <span class="st">"state3"</span>],
                   scut=<span class="fl">0</span>)
</pre></div>
<div id="side-note-3" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#side-note-3" class="anchor"></a>Side-note:</h4>
<p>The returned <strong>cij</strong> is a ‘list’ object containing average correlation matrices, each for a structural group. It is a <em>good practice</em> to inspect the correlation using the <strong>pymol()</strong> or the <strong>plot.dccm()</strong> function, before moving toward the next step.</p>
<div class="sourceCode" id="cb76"><pre class="downlit">
<span class="co"># Generate a 'pdb' as the reference</span>
<span class="kw">pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbs2pdb.html">pdbs2pdb</a></span>(<span class="kw">pdbs</span>, inds=<span class="fl">1</span>, rm.gaps=<span class="fl">TRUE</span>)[[<span class="fl">1</span>]]
<span class="fu"><a href="../../../reference/pymol.html">pymol</a></span>(<span class="kw">cij</span><span class="op">$</span><span class="kw">GTP</span>, pdb=<span class="kw">pdb</span>)
</pre></div>
</div>
</div>
<div id="building-networks-and-comparing-residue-communities" class="section level2">
<h2 class="hasAnchor">
<a href="#building-networks-and-comparing-residue-communities" class="anchor"></a>Building networks and comparing residue communities</h2>
<p>Correlation networks are finally constructed with the function <strong>cna()</strong>. Note that since correlation matrices are already filtered in the last step, here the ‘cutoff.cij’ is set to zero in <code><a href="../../../reference/cna.html">cna()</a></code>. Networks are then visualized with the function <strong>plot.cna()</strong>.</p>
<div class="sourceCode" id="cb77"><pre class="downlit">
<span class="kw">nets</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/cna.html">cna</a></span>(<span class="kw">cij</span>, cutoff.cij = <span class="fl">0</span>)
<span class="co">#plot(nets, pdb=pdb)</span>
</pre></div>
<p>In a plot by simply calling <code><a href="../../../reference/plot.cna.html">plot.cna()</a></code>, colors of community vertexes are not consistent across networks, which impedes network comparisons. This is due to small variations of community definition between networks. A convenient tool to correct it is the <strong>community.aln()</strong> function provided in the bio3d.cna package. This function finds equivalent communities from two or more networks and re-assign colors to them in a consistent way across networks.</p>
<div class="sourceCode" id="cb78"><pre class="downlit">
<span class="kw">nets.aln</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/community.aln.html">community.aln</a></span>(<span class="kw">nets</span>)
<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/layout_.html">layout</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span><span class="op">:</span><span class="fl">4</span>), <span class="fl">2</span>, <span class="fl">2</span>))
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(mar=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.1</span>, <span class="fl">0.1</span>, <span class="fl">3</span>, <span class="fl">0.1</span>))
<span class="kw">cent.gtp</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/layout.cna.html">layout.cna</a></span>(<span class="kw">nets</span><span class="op">$</span><span class="kw">GTP</span>, pdb=<span class="kw">pdb</span>, k=<span class="fl">3</span>)[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>]
<span class="kw">cent.gdi</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/layout.cna.html">layout.cna</a></span>(<span class="kw">nets</span><span class="op">$</span><span class="kw">GDI</span>, pdb=<span class="kw">pdb</span>, k=<span class="fl">3</span>)[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>]
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">nets</span><span class="op">$</span><span class="kw">GTP</span>, layout=<span class="kw">cent.gtp</span>, vertex.label=<span class="fl">NA</span>, main=<span class="st">"GTP"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">nets.aln</span><span class="op">$</span><span class="kw">GTP</span>, layout=<span class="kw">cent.gtp</span>, vertex.label=<span class="fl">NA</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">nets</span><span class="op">$</span><span class="kw">GDI</span>, layout=<span class="kw">cent.gdi</span>, vertex.label=<span class="fl">NA</span>, main=<span class="st">"GDI"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">nets.aln</span><span class="op">$</span><span class="kw">GDI</span>, layout=<span class="kw">cent.gdi</span>, vertex.label=<span class="fl">NA</span>)
</pre></div>
<div class="figure">
<img src="cna_vignette.spin_files/figure-html/fig_ecna-1.png" alt="**Correlation networks for GTP (active) and GDI (inhibited) conformational states of G-alpha-i.** Networks are derived from ensemble NMA of available GTP and GDI crystallographic structures. Top, original networks; Bottom, after community color corrections." width="875"><p class="caption">
<strong>Correlation networks for GTP (active) and GDI (inhibited) conformational states of G-alpha-i.</strong> Networks are derived from ensemble NMA of available GTP and GDI crystallographic structures. Top, original networks; Bottom, after community color corrections.
</p>
</div>
</div>
<div id="comparing-suboptimal-paths" class="section level2">
<h2 class="hasAnchor">
<a href="#comparing-suboptimal-paths" class="anchor"></a>Comparing suboptimal paths</h2>
<p>The eCNA method can be coupled with suboptimal path analysis for more detailed network comparisons. Results are visualized with <strong>vmd.cnapath()</strong>. Here, the source and sink nodes are chosen to represent the receptor (which catalyzes the G protein activation) binding site and the active site. It clearly shows that distinct suboptimal paths are present in the two networks connecting the source and sink nodes, and the paths in the GTP (active) network are overall much shorter (indicating much stronger coupling between the distal sites) than those in the GDI (inhibited) network.</p>
<div class="sourceCode" id="cb79"><pre class="downlit">
<span class="kw">pa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/cnapath.html">cnapath</a></span>(<span class="kw">nets</span>, from=<span class="fl">307</span>, to=<span class="fl">168</span>, k=<span class="fl">100</span>)
</pre></div>
<div class="sourceCode" id="cb80"><pre class="downlit">
<span class="fu"><a href="../../../reference/vmd.html">vmd</a></span>(<span class="kw">pa</span>, pdb=<span class="kw">pdb</span>, spline=<span class="fl">TRUE</span>)
</pre></div>
<div class="figure">
<img src="figures/cna_fig_cnapath_galpha.png" alt=""><p class="caption"><strong>Ensemble correlation network analysis of G-alpha-i reveals distinct coupling paths between GTP and GDI structural ensembles.</strong> Identified suboptimal paths are viewed as lines mapped on a G-alpha-i structure, color coded by the structural ensemble (blue, GDI; red, GTP) used to build the network. Line radii are scaled by path lengths (shorter paths are thicker).</p>
</div>
<p>Quantitative comparisons of path length distributions can be done by calling the <strong>plot.cnapath()</strong> function. More detailed results are returned by the <strong>print.cnapath()</strong> function, which contains normalized node degeneracy for identifying specific crucial residues underlying each set of coupling paths.</p>
<div class="sourceCode" id="cb81"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">pa</span>, pdb=<span class="kw">pdb</span>)
</pre></div>
<div class="sourceCode" id="cb82"><pre class="downlit">
<span class="kw">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">pa</span>, pdb=<span class="kw">pdb</span>)
</pre></div>
<div class="sourceCode" id="cb83"><pre class="downlit">
<span class="kw">out</span><span class="op">$</span><span class="kw">degeneracy</span>
</pre></div>
<pre><code>##      K31  L32  L33 L34 M194 F195 D196 V197 G198 T215 C216 I217 F219 I220 S259
## GDI 0.00 0.00 0.00 1.0 0.00 0.00 1.00 0.29    1 0.00 0.00 0.00 0.99    1    0
## GTP 0.41 0.36 0.47 0.1 0.23 0.52 0.96 0.38    1 0.44 0.53 0.17 0.00    0    1
##     N265 C321 A322 N327 V328 F330 V331 F332 D333 A334 V335 T336 D337 I338 I339
## GDI 0.91 0.92  0.1 0.89  0.2  0.4 0.48  0.2 0.24  0.3 0.47 0.32  0.1 0.18    1
## GTP 0.00 0.00  0.0 0.00  0.0  0.0 0.00  0.0 0.00  0.0 0.00 0.00  0.0 0.00    1</code></pre>
<div class="sourceCode" id="cb85"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/detach.html">detach</a></span>(<span class="kw">transducin</span>)
</pre></div>
</div>
</div>
<div id="summary" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#summary" class="anchor"></a>Summary</h1>
<p>In this vignette we have confined ourselves to introducing relatively simple correlation network generation protocols. However, the individual Bio3D functions for network analysis permit extensive customization of the network generation procedure. For example, one can chose different node representations (including all heavy atoms, residue center of mass, Calpha atoms for each residue, or collections of user defined atoms), different correlation methods (including Pearson and mutual information), different community detection methods (including Girvan-Newman betweenness, Random walk, and Greedy modularity optimization approaches), as well as customized network layouts and result visualizations. We encourage users to explore these options and provide feedback as appropriate.</p>
<p>Although the protocols detailed herein can provide compelling results in a number of cases, defining an optimal procedure for network generation is an area of active research. For example rearrangement of some residues in the communities and changes in the total number of communities may be observed among different community structures based on different independent MD simulations of the same system. These variations of community structures are partially due to the small changes in the calculated cij values that reflect statistical fluctuations among independent simulations. However, changes in mutual information or correlation values may still be relatively small and differences among independent community structures may be largely due to limitations of the modularity metric, a methodology that was originally developed for binary adjacency matrices, where edges are counted without taking into account their lengths. We are currently developing a modularity methodology that is more closely aligned to protein dynamics data. We are also developing integrated 3D viewing functions (so VMD and PyMol will no longer be absolutely required during the exploratory interactive phase of analysis), as well as functionality to facilitate analysis of the interactions that generate changes in the community network.</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<p>[1] Sethi A, Eargle J, Black AA, Luthey-Schulten Z. Proc Natl Acad Sci U S A.2009, 106(16):6620-5.</p>
<p>[2] Scarabelli G and Grant BJ, PLoS Comput Biol 9, e1003329.</p>
<p>[3] Yao X, Grant BJ, Biophys J 105, L08–L10.</p>
<p>[4] Yao X, Malik RU, Griggs NW, Skjaerven L, Traynor JR, Sivaramakrishnan S, and Grant BJ, J Biol Chem 291, 4742-4753.</p>
<p>[5] Gasper PM, Fuglestad B, Komives EA, Markwick PR, McCammon JA. Proc Natl Acad Sci U S A. 2012 Dec 26;109(52):21216-22.</p>
<p>[6] Gábor Csárdi, Tamás Nepusz. InterJournal Complex Systems, 1695, 2006.</p>
<p>[7] Girvan M, Newman ME. Proc Natl Acad Sci U S A. 2002 Jun 11;99(12):7821-6.</p>
<p>[8] Van Wart AT, Durrant J, Votapka L, Amaro RE. J. Chem. Theory Comput., 2014, 10 (2), pp 511–517.</p>
</div>
<div id="about-this-document" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#about-this-document" class="anchor"></a>About this document</h1>
<p>This document is available from the <a href="http://thegrantlab.org/bio3d">Bio3D website</a> in R, HTML, and PDF formats. All code can be extracted and automatically executed to generate Figures and/or the PDF with the following commands:</p>
<div class="sourceCode" id="cb86"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://github.com/rstudio/rmarkdown">rmarkdown</a></span>)
<span class="fu"><a href="https://rdrr.io/pkg/rmarkdown/man/render.html">render</a></span>(<span class="st">"cna_vignette.R"</span>)
</pre></div>
</div>
<div id="information-about-the-current-bio3d-session" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#information-about-the-current-bio3d-session" class="anchor"></a>Information about the current Bio3D session</h1>
<div class="sourceCode" id="cb87"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>(), <span class="fl">FALSE</span>)
</pre></div>
<pre><code>## R version 3.6.0 (2019-04-26)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS  10.15.5
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] igraph_1.2.5         bio3d.nma_0.1.0.9000 bio3d.cna_0.1.0.9000
## [4] bio3d_2.4-1.9000    
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.5         knitr_1.29         magrittr_1.5       lattice_0.20-41   
##  [5] R6_2.4.1           rlang_0.4.6.9000   highr_0.8          stringr_1.4.0     
##  [9] tools_3.6.0        parallel_3.6.0     grid_3.6.0         xfun_0.15         
## [13] htmltools_0.5.0    yaml_2.2.1         assertthat_0.2.1   rprojroot_1.3-2   
## [17] digest_0.6.25      pkgdown_1.5.1.9000 crayon_1.3.4       fs_1.4.2          
## [21] memoise_1.1.0      evaluate_0.14      rmarkdown_2.3.2    stringi_1.4.6     
## [25] compiler_3.6.0     desc_1.2.0         backports_1.1.8    pkgconfig_2.0.3</code></pre>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>The latest version of the package, full documentation and further vignettes (including detailed installation instructions) can be obtained from the main Bio3D website: [<a href="http://thegrantlab.org/bio3d/(http:/thegrantlab.org/bio3d/" class="uri">http://thegrantlab.org/bio3d/(http:/thegrantlab.org/bio3d/</a>)<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>This document contains executable code that generates all figures contained within this document. See help(vignette) within R for full details.<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="tidyverse">
  <p>Authored by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="author">
  <p>
    Built with ❤️ and <a href="https://pkgdown.r-lib.org">pkgdown</a> © 2020.
  </p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '1270c4079b26a138795263974bbf302d',
    indexName: 'tidyverse',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
