<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Beginning Trajectory Analysis with Bio3D • Bio3D</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../../../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../../../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../../../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../../../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../../../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../../../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="../../../tidyverse.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../../bootstrap-toc.css">
<script src="../../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../../pkgdown.css" rel="stylesheet">
<script src="../../../pkgdown.js"></script><link href="../../../tidyverse-2.css" rel="stylesheet">
<!--optional theme
<link href="../../../.css" rel="stylesheet">
--><!-- docsearch --><script src="../../../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../../../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Beginning Trajectory Analysis with Bio3D">
<meta property="og:description" content="bio3d">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../../../index.html">
        bio3d</a>
        <div class="info hidden-xs hidden-sm">
<!--          <span class="partof">part of the <a href="https://tidyverse.org">tidyverse</a></span>
-->
          <span class="version version-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4-1.9000</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">INTRODUCTORY TUTORIALS</li>
    <li>
      <a href="../../../articles/online/install_vignette/Bio3D_install.html">Installing Bio3D</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/user-guide">New user guide</a>
    </li>
    <li>
      <a href="../../../articles/online/intro_vignette/Bio3D_introduction.html">Getting started with Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/pdb_vignette/Bio3D_pdb.html">PDB structure manipulation and analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/pca_vignette/Bio3D_pca.html">Comparative sequence and structure analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/nma_vignette/Bio3D_nma.html">Enhanced Normal Modes Analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/traj_vignette/Bio3D_md.html">Beginning trajectory analysis with Bio3D </a>
    </li>
    <li class="divider">
    <li class="dropdown-header">ADVANCED TUTORIALS</li>
    <li>
      <a href="../../../articles/online/enma_vignettes/Bio3D_nma-dhfr-partI.html">Ensemble NMA of E.coli DHFR structures</a>
    </li>
    <li>
      <a href="../../../articles/online/enma_vignettes/Bio3D_nma-dhfr-partII.html">Ensemble NMA across multiple species of DHFR</a>
    </li>
    <li>
      <a href="../../../articles/online/cna_vignette/cna_vignette.spin.html">Protein structure network analysis with Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/eddm_vignette/Bio3D_eddm.spin.html">Ensemble Difference Distance Matrix (eDDM) analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">WEBAPP TUTORIALS</li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/download/webapp-tutorials?download=22:bio3d-webapp-pca">Online protein structure analysis with Bio3D-Web (PDF)</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/webapps">Online Normal Mode Analysis with Bio3D WebApps (To Update)</a>
    </li>
    <li class="divider">
    <li>
      <a href="../../../articles/index.html">All vignettes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Documentation
  </a>
</li>
        <li>
  <a href="../../../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../../../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    News
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket"></span>
     
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Bio3D_md_files/header-attrs-2.3.2/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Beginning Trajectory Analysis with Bio3D</h1>
                        <h4 class="author">Xin-Qiu Yao, Lars Skjaerven, &amp; Barry J. Grant</h4>
            
      
      
      <div class="hidden name"><code>Bio3D_md.Rmd</code></div>

    </div>

    
    
<div id="background" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#background" class="anchor"></a>Background</h1>
<p>Bio3D<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> is a group of R packages that provide interactive tools for the analysis of bimolecular structure, sequence and simulation data. The aim of this document, termed a vignette<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a> in R parlance, is to provide a brief task-oriented introduction to basic molecular dynamics trajectory analysis with the Bio3D core package, <strong>bio3d</strong> <span class="citation">(Grant et al. 2006)</span>.</p>
<div id="requirements" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#requirements" class="anchor"></a>Requirements</h2>
<p>Detailed instructions for obtaining and installing the Bio3D packages on various platforms can be found in the <a href="http://thegrantlab.org/bio3d">Installing Bio3D</a> vignette available online. This particular vignette was generated using <strong>bio3d version 2.4.1.9000</strong>.</p>
</div>
</div>
<div id="getting-started" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting Started</h1>
<p>Start R, load the bio3d package and use the command <code><a href="https://rdrr.io/r/utils/demo.html">demo("md")</a></code> to get a quick feel for some of the tasks that we will be introducing in the following sections.</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="http://thegrantlab.org/bio3d">bio3d</a></span>)
<span class="fu"><a href="https://rdrr.io/r/utils/demo.html">demo</a></span>(<span class="st">"md"</span>)
</pre></div>
<div id="side-note" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#side-note" class="anchor"></a>Side-note:</h4>
<p>Note that you will be prompted to hit the <code>RETURN</code> key at each step of the demo as this will allow you to see the particular functions being called. Also note that detailed documentation and example code for each function can be accessed via the <code>help()</code> and <code><a href="https://rdrr.io/r/utils/example.html">example()</a></code> commands (e.g. <code><a href="../../../reference/read.pdb.html">help(read.pdb)</a></code>). You can also copy and paste any of the example code from the documentation of a particular function, or indeed this vignette, directly into your R session to see how things work. You can also find this <a href="http://thegrantlab.org/bio3d">documentation online</a>. <br><br></p>
</div>
<div id="reading-example-trajectory-data" class="section level2">
<h2 class="hasAnchor">
<a href="#reading-example-trajectory-data" class="anchor"></a>Reading Example Trajectory Data</h2>
<p>A number of example data sets are shipped with the bio3d package. The main purpose of including this data is to allow users to more quickly appreciate the capabilities of various Bio3D functions that would otherwise require potentially time consuming data generation. In the examples below we will input, process and analyze a molecular dynamics trajectory of Human Immunodeficiency Virus aspartic protease (HIVpr). This trajectory is stored in CHARMM/NAMD DCD format and has had all solvent and non C-alpha protein atoms excluded to reduce overall file size.</p>
<p>The code snippet below sets the file paths for the example HIVpr starting structure (pdbfile) and trajectory data (dcdfile).</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="kw">dcdfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"examples/hivp.dcd"</span>, package=<span class="st">"bio3d"</span>)
<span class="kw">pdbfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"examples/hivp.pdb"</span>, package=<span class="st">"bio3d"</span>)
</pre></div>
<div id="side-note-1" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#side-note-1" class="anchor"></a>Side-note:</h4>
<p>Note that in the above example the <code><a href="https://rdrr.io/r/base/system.file.html">system.file()</a></code> command returns a character string corresponding to the file name of a PDB structure included with the bio3d package. This is required as users may install the package in different locations. When using your own input files the <code><a href="https://rdrr.io/r/base/system.file.html">system.file()</a></code> command will not be required, for example</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="kw">mydcdfile</span> <span class="op">&lt;-</span> <span class="st">"/path/to/my/data/myfile.dcd"</span>
</pre></div>
<p><br></p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="kw">dcd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.dcd.html">read.dcd</a></span>(<span class="kw">dcdfile</span>)
<span class="kw">pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="kw">pdbfile</span>)
</pre></div>
<p>The <code><a href="../../../reference/read.dcd.html">read.dcd()</a></code> and <code><a href="../../../reference/read.pdb.html">read.pdb()</a></code> commands processes the input files and returns their output to the new objects <code>dcd</code> and <code>pdb</code>. We can check the basic structure of these objects with the following commands:</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">pdb</span>)
</pre></div>
<pre><code>## 
##  Call:  read.pdb(file = pdbfile)
## 
##    Total Models#: 1
##      Total Atoms#: 198,  XYZs#: 594  Chains#: 2  (values: A B)
## 
##      Protein Atoms#: 198  (residues/Calpha atoms#: 198)
##      Nucleic acid Atoms#: 0  (residues/phosphate atoms#: 0)
## 
##      Non-protein/nucleic Atoms#: 0  (residues: 0)
##      Non-protein/nucleic resid values: [ none ]
## 
##    Protein sequence:
##       PQITLWQRPLVTIKIGGQLKEALLDTGADDTVLEEMSLPGRWKPKMIGGIGGFIKVRQYD
##       QILIEICGHKAIGTVLVGPTPVNIIGRNLLTQIGCTLNFPQITLWQRPLVTIKIGGQLKE
##       ALLDTGADDTVLEEMSLPGRWKPKMIGGIGGFIKVRQYDQILIEICGHKAIGTVLVGPTP
##       VNIIGRNLLTQIGCTLNF
## 
## + attr: atom, xyz, helix, sheet, calpha,
##         call</code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">pdb</span><span class="op">$</span><span class="kw">xyz</span>)
</pre></div>
<pre><code>## 
##    Total Frames#: 1
##    Total XYZs#:   594,  (Atoms#:  198)
## 
##     [1]  51.842  59.784  -6.815  &lt;...&gt;  55.089  55.232  -6.899  [594] 
## 
## + attr: Matrix DIM = 1 x 594</code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">dcd</span>)
</pre></div>
<pre><code>## 
##    Total Frames#: 117
##    Total XYZs#:   594,  (Atoms#:  198)
## 
##     [1]  51.842  59.784  -6.815  &lt;...&gt;  27.995  70.42  -33.908  [69498] 
## 
## + attr: Matrix DIM = 117 x 594</code></pre>
<p>The output of the above function is telling us that we have 351 trajectory frames and 594 coordinates. The trajectory information is stored in a <code>xyz</code> matrix object with one row per frame and columns for the Cartesian coordinates (x, y and z columns).</p>
</div>
<div id="question" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question" class="anchor"></a>Question:</h4>
<p>How many atoms are in the trajectory and PDB files?</p>
</div>
<div id="question-1" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-1" class="anchor"></a>Question:</h4>
<p>How would you extract the amino acid sequence of the HIVpr system in 1-letter and 3-letter forms? HINT: try <code><a href="https://rdrr.io/r/utils/help.search.html">help.search("PDB sequence")</a></code> for a Bio3D function that might help you.</p>
</div>
<div id="side-note-2" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#side-note-2" class="anchor"></a>Side-note:</h4>
<p>Note that typically one works with trajectory files that contain all protein atoms, or at the very least all backbone atoms. Solvent however can often be excluded prior to Bio3D input - it just depends upon your particular analysis questions. For example, we are not able to analyze hydrogen bonding patterns or details of water occupancy with the currently inputed data. <br><br></p>
</div>
</div>
</div>
<div id="trajectory-frame-superposition" class="section level1">
<h1 class="hasAnchor">
<a href="#trajectory-frame-superposition" class="anchor"></a>Trajectory Frame Superposition</h1>
<p>In this simple example we select all C-alpha atoms for trajectory frame superposition.</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="kw">ca.inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>, elety=<span class="st">"CA"</span>)
</pre></div>
<p>The returned <code>ca.inds</code> object is a list containing atom and xyz numeric indices that we can now use to superpose all frames of the trajectory on the selected indices (in this case corresponding to all alpha Carbon atoms). For this we will with the <code><a href="../../../reference/fit.xyz.html">fit.xyz()</a></code> function.</p>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="kw">xyz</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/fit.xyz.html">fit.xyz</a></span>(fixed=<span class="kw">pdb</span><span class="op">$</span><span class="kw">xyz</span>, mobile=<span class="kw">dcd</span>,
               fixed.inds=<span class="kw">ca.inds</span><span class="op">$</span><span class="kw">xyz</span>,
               mobile.inds=<span class="kw">ca.inds</span><span class="op">$</span><span class="kw">xyz</span>)
</pre></div>
<p>The above command performs the actual superposition and stores the new coordinates in the matrix object <code>xyz</code>. Note that the dimensions (i.e. number of rows and columns, which correspond to frames and coordinates respectively) of <code>xyz</code> match those of the input trajectory:</p>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="kw">xyz</span>) <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="kw">dcd</span>)
</pre></div>
<pre><code>## [1] TRUE TRUE</code></pre>
<div id="question-2" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-2" class="anchor"></a>Question:</h4>
<p>How would you fit trajectory frames on the Calpha atoms of residues 24 to 27 and 85 to 90 in both chains? HINT: See the example section of <code><a href="../../../reference/atom.select.html">help(atom.select)</a></code>.</p>
</div>
<div id="question-3" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-3" class="anchor"></a>Question:</h4>
<p>Would you expect the alternate fitting suggested above to alter your later results? HINT: You can come back to this question later after going through the other sections.</p>
</div>
<div id="side-note-3" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#side-note-3" class="anchor"></a>Side-note:</h4>
<p>A simple way to obtain the average structure from your fitted trajectory is to use the following command <code><a href="https://rdrr.io/r/base/colSums.html">colMeans(xyz)</a></code>. <br><br></p>
</div>
</div>
<div id="root-mean-square-deviation-rmsd" class="section level1">
<h1 class="hasAnchor">
<a href="#root-mean-square-deviation-rmsd" class="anchor"></a>Root Mean Square Deviation (RMSD)</h1>
<p>RMSD is a standard measure of structural distance between coordinate sets and is implemented in the Bio3D function <code><a href="../../../reference/rmsd.html">rmsd()</a></code>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="kw">rd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/rmsd.html">rmsd</a></span>(<span class="kw">xyz</span>[<span class="fl">1</span>,<span class="kw">ca.inds</span><span class="op">$</span><span class="kw">xyz</span>], <span class="kw">xyz</span>[,<span class="kw">ca.inds</span><span class="op">$</span><span class="kw">xyz</span>])

<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">rd</span>, typ=<span class="st">"l"</span>, ylab=<span class="st">"RMSD"</span>, xlab=<span class="st">"Frame No."</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="fu"><a href="https://rdrr.io/r/stats/lowess.html">lowess</a></span>(<span class="kw">rd</span>), typ=<span class="st">"l"</span>, col=<span class="st">"red"</span>, lty=<span class="fl">2</span>, lwd=<span class="fl">2</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_md_files/figure-html/unnamed-chunk-9-1.png" alt="Simple time series of RMSD from the initial structure (note periodic jumps that we will later see correspond to transient openings of the flap regions of HIVpr)" width="700"><p class="caption">
Simple time series of RMSD from the initial structure (note periodic jumps that we will later see correspond to transient openings of the flap regions of HIVpr)
</p>
</div>
<p>A quick histogram can be useful for examining the distribution of RMSD values.</p>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span>(<span class="kw">rd</span>, breaks=<span class="fl">40</span>, freq=<span class="fl">FALSE</span>, main=<span class="st">"RMSD Histogram"</span>, xlab=<span class="st">"RMSD"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/lines.html">lines</a></span>(<span class="fu"><a href="https://rdrr.io/r/stats/density.html">density</a></span>(<span class="kw">rd</span>), col=<span class="st">"gray"</span>, lwd=<span class="fl">3</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_md_files/figure-html/unnamed-chunk-10-1.png" alt="Note the spread of RMSD values and that the majority of sampled conformations are around 2 Angstroms from the starting structure" width="700"><p class="caption">
Note the spread of RMSD values and that the majority of sampled conformations are around 2 Angstroms from the starting structure
</p>
</div>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span>(<span class="kw">rd</span>)
</pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   0.000   1.834   1.967   2.151   2.167   4.112</code></pre>
<div id="question-4" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-4" class="anchor"></a>Question:</h4>
<p>How would you calculate the pairwise RMSD between all frames?</p>
</div>
<div id="question-5" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-5" class="anchor"></a>Question:</h4>
<p>What would be a good way to visualize and further analyze such a pairwise RMSD matrix? HINT: Have a look at the NMA vignette for some, hopefully, inspiring plots? <br><br></p>
</div>
</div>
<div id="root-mean-squared-fluctuations-rmsf" class="section level1">
<h1 class="hasAnchor">
<a href="#root-mean-squared-fluctuations-rmsf" class="anchor"></a>Root Mean Squared Fluctuations (RMSF)</h1>
<p>RMSF is an often used measure of conformational variance and is implemented in the Bio3D function <code><a href="../../../reference/rmsf.html">rmsf()</a></code>. This analysis will highlight the portions of structure that are fluctuating from their mean structure the most (and least).</p>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="kw">rf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/rmsf.html">rmsf</a></span>(<span class="kw">xyz</span>[,<span class="kw">ca.inds</span><span class="op">$</span><span class="kw">xyz</span>])
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">rf</span>, ylab=<span class="st">"RMSF"</span>, xlab=<span class="st">"Residue Position"</span>, typ=<span class="st">"l"</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_md_files/figure-html/unnamed-chunk-12-1.png" alt="Residue-wise RMSF indicates regions of high mobility" width="700"><p class="caption">
Residue-wise RMSF indicates regions of high mobility
</p>
</div>
<div id="question-6" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-6" class="anchor"></a>Question:</h4>
<p>If you had multiple simulations (which by the way we greatly encourage), how would you plot these on the same graph? HINT: See <code><a href="https://rdrr.io/r/graphics/points.html">help(points)</a></code>.</p>
</div>
<div id="question-7" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-7" class="anchor"></a>Question:</h4>
<p>If you noticed differences at particular sites (e.g. in the presence or absence of a ligand) how would you go about addressing the significance of these differences? HINT: this is why we always encourage a multiple simulation approach and implement Bio3D in one of the most advanced statistical analysis packages available. <br><br></p>
</div>
</div>
<div id="principal-component-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#principal-component-analysis" class="anchor"></a>Principal Component Analysis</h1>
<p>PCA can be employed to examine the relationship between different conformations sampled during the trajectory and is implemented in the Bio3D functions <code><a href="../../../reference/pca.xyz.html">pca.xyz()</a></code> and <code><a href="../../../reference/pca.tor.html">pca.tor()</a></code>. The application of PCA to both distributions of experimental structures and molecular dynamics trajectories will be covered in detail in other vignettes. Briefly, we will note here that this method can provide considerable insight into the nature of conformational differences with the resulting principal components (orthogonal eigenvectors) describing the axes of maximal variance of the distribution of structures. Projection of the distribution onto the subspace defined by the largest principal components results in a lower dimensional representation of the structural dataset (see <code>Figure 4</code>). The percentage of the total mean square displacement (or variance) of atom positional fluctuations captured in each dimension is characterized by their corresponding eigenvalue (see <code>Figure 4D</code>). Experience suggests that 3–5 dimensions are often sufficient to capture over 70 percent of the total variance in a given family of experimental structures or indeed a standard molecular dynamics trajectory. Thus, a handful of principal components are sufficient to provide a useful description while still retaining most of the variance in the original distribution <span class="citation">(Grant et al. 2006)</span>.</p>
<p>A quick overview of the results of <code><a href="../../../reference/pca.xyz.html">pca.xyz()</a></code> can be obtained by calling <code><a href="../../../reference/plot.pca.html">plot.pca()</a></code></p>
<div class="sourceCode" id="cb20"><pre class="downlit">
<span class="kw">pc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pca.xyz.html">pca.xyz</a></span>(<span class="kw">xyz</span>[,<span class="kw">ca.inds</span><span class="op">$</span><span class="kw">xyz</span>])
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">pc</span>, col=<span class="fu"><a href="../../../reference/bwr.colors.html">bwr.colors</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="kw">xyz</span>)) )
</pre></div>
<div class="figure">
<img src="Bio3D_md_files/figure-html/unnamed-chunk-13-1.png" alt="PCA results for our HIVpr trajectory with instantaneous conformations (i.e. trajectory frames) colored from blue to red in order of time" width="700"><p class="caption">
PCA results for our HIVpr trajectory with instantaneous conformations (i.e. trajectory frames) colored from blue to red in order of time
</p>
</div>
<p>Note that there are distinct groupings of conformations along the PC1 plane (one centered around +30 and a second, larger grouping, at -5). The continuous color scale (from blue to whit to red) indicates that there are periodic jumps between these conformers throughout the trajectory. Below we perform a quick clustering in PC-space to further highlight these distinct conformers.</p>
<div class="sourceCode" id="cb21"><pre class="downlit">
<span class="kw">hc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html">hclust</a></span>(<span class="fu"><a href="https://rdrr.io/r/stats/dist.html">dist</a></span>(<span class="kw">pc</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>]))
<span class="kw">grps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cutree.html">cutree</a></span>(<span class="kw">hc</span>, k=<span class="fl">2</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">pc</span>, col=<span class="kw">grps</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_md_files/figure-html/unnamed-chunk-14-1.png" alt="Simple clustering in PC subspace" width="700"><p class="caption">
Simple clustering in PC subspace
</p>
</div>
<div id="question-8" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-8" class="anchor"></a>Question:</h4>
<p>How can we find out which frames correspond to the distinct groups along PC1? HINT: Which variable/object created above tells us about cluster membership? <br><br></p>
<p>Bellow we call <code><a href="../../../reference/plot.bio3d.html">plot.bio3d()</a></code> to examine the contribution of each residue to the first two principal components.</p>
<div class="sourceCode" id="cb22"><pre class="downlit">
<span class="fu"><a href="../../../reference/plot.bio3d.html">plot.bio3d</a></span>(<span class="kw">pc</span><span class="op">$</span><span class="kw">au</span>[,<span class="fl">1</span>], ylab=<span class="st">"PC1 (A)"</span>, xlab=<span class="st">"Residue Position"</span>, typ=<span class="st">"l"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="kw">pc</span><span class="op">$</span><span class="kw">au</span>[,<span class="fl">2</span>], typ=<span class="st">"l"</span>, col=<span class="st">"blue"</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_md_files/figure-html/unnamed-chunk-15-1.png" alt="Residue-wise loadings for PC1 (black) and PC2 (blue)" width="700"><p class="caption">
Residue-wise loadings for PC1 (black) and PC2 (blue)
</p>
</div>
</div>
<div id="question-9" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-9" class="anchor"></a>Question:</h4>
<p>Why do you think there might be two major peaks in our RMSF plot? <br><br></p>
<p>To further aid interpretation, a PDB format trajectory can be produced that interpolates between the most dissimilar structures in the distribution along a given principal component. This involves dividing the difference between the conformers into a number of evenly spaced steps along the principal components, forming the frames of the output multi-model PDB trajectory. Such trajectories can be directly visualized in a molecular graphics program, such as VMD <span class="citation">(Humphrey 1996)</span>. Furthermore, the interpolated structures can be analyzed for possible domain and shear movements with other Bio3D functions, or used as initial seed structures for reaction path refinement methods (note you will likely want to perform all heavy atom PCA for such applications).</p>
<div class="sourceCode" id="cb23"><pre class="downlit">
<span class="kw">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/mktrj.html">mktrj.pca</a></span>(<span class="kw">pc</span>, pc=<span class="fl">1</span>, b=<span class="kw">pc</span><span class="op">$</span><span class="kw">au</span>[,<span class="fl">1</span>], file=<span class="st">"pc1.pdb"</span>)
<span class="kw">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/mktrj.html">mktrj.pca</a></span>(<span class="kw">pc</span>, pc=<span class="fl">2</span>, b=<span class="kw">pc</span><span class="op">$</span><span class="kw">au</span>[,<span class="fl">2</span>], file=<span class="st">"pc2.pdb"</span>)
</pre></div>
<p>You can also write these trajectory’s as AMBER NetCDF format files with the <code>write.ncdf</code> function. To view the PDB trajectories in VMD just open the files in the normal way and display as tube representation for example (see figure below).</p>
<div class="sourceCode" id="cb24"><pre class="downlit">
<span class="fu"><a href="../../../reference/write.ncdf.html">write.ncdf</a></span>(<span class="kw">p1</span>, <span class="st">"trj_pc1.nc"</span>)
</pre></div>
<div class="figure">
<img src="figures/hiv_pc1.png" alt=""><p class="caption">Visualization of the first principal component of HIV-1 protease. Color scale from blue to red depict low to high atomic displacements.</p>
</div>
</div>
<div id="question-10" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-10" class="anchor"></a>Question:</h4>
<p>Which regions are the most dynamic and why? <br><br></p>
</div>
</div>
<div id="cross-correlation-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#cross-correlation-analysis" class="anchor"></a>Cross-Correlation Analysis</h1>
<p>The extent to which the atomic fluctuations/displacements of a system are correlated with one another can be assessed by examining the magnitude of all pairwise cross-correlation coefficients. The Bio3D <code><a href="../../../reference/dccm.html">dccm()</a></code> function returns a matrix of all atom-wise cross-correlations whose elements may be displayed in a graphical representation frequently termed a dynamical cross-correlation map, or DCCM.</p>
<div class="sourceCode" id="cb25"><pre class="downlit">
<span class="kw">cij</span><span class="op">&lt;-</span><span class="fu"><a href="../../../reference/dccm.html">dccm</a></span>(<span class="kw">xyz</span>[,<span class="kw">ca.inds</span><span class="op">$</span><span class="kw">xyz</span>])
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">cij</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_md_files/figure-html/unnamed-chunk-18-1.png" alt="Dynamical cross-correlation map (DCCM) calculated with function **dccm()** using the MD simulation of HIV-1 protease as input." width="700"><p class="caption">
Dynamical cross-correlation map (DCCM) calculated with function <strong>dccm()</strong> using the MD simulation of HIV-1 protease as input.
</p>
</div>
<div id="question-11" class="section level4 unnumbered">
<h4 class="hasAnchor">
<a href="#question-11" class="anchor"></a>Question:</h4>
<p>What do the off-diagonal regions of negative correlation correspond to in this plot and which regions are involved? HINT: Negative values typical indicate regions that move in opposite directions. <br><br></p>
<p>A 3D visualization of these correlations can be provided through the function <code><a href="../../../reference/pymol.html">pymol.dccm()</a></code></p>
<div class="sourceCode" id="cb26"><pre class="downlit">
<span class="co"># View the correlations in pymol</span>
<span class="fu"><a href="../../../reference/pymol.html">pymol.dccm</a></span>(<span class="kw">cij</span>, <span class="kw">pdb</span>, type=<span class="st">"launch"</span>)
</pre></div>
<div class="figure">
<img src="figures/dccm_hivpr.png" alt=""><p class="caption">Visualization of residue-residue cross correlations (using function <strong>pymol.dccm()</strong>). Red and blue lines indicate correlated and anti-correlation motions, respectively.</p>
</div>
<p>See also the <code>Enhanced Methods for Normal Mode Analysis</code> for additional visualization examples. Also you might want to checkout the <code>Comparative Analysis of Protein Structures</code> vignette for relating results like these to available experimental data. The logical expansion of this analysis is described in the <code>Correlation Network Analysis</code> vignette. <br><br></p>
</div>
</div>
<div id="where-to-next" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#where-to-next" class="anchor"></a>Where to Next</h1>
<p>If you have read this far, congratulations! We are ready to have some fun and move to <a href="http://thegrantlab.org/bio3d">other package vignettes</a> that describe more interesting analysis including Correlation Network Analysis (where we will build and dissect dynamic networks form different correlated motion data), enhanced methods for Normal Mode Analysis (where we will explore the dynamics of large protein families and superfamilies), and advanced Comparative Structure Analysis (where we will mine available experimental data and supplement it with simulation results to map the conformational dynamics and coupled motions of proteins). <br><br></p>
</div>
<div id="document-details" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#document-details" class="anchor"></a>Document Details</h1>
<p>This document is available from the <a href="http://thegrantlab.org/bio3d">Bio3D website</a> in R markdown, HTML, and PDF formats. All code can be extracted and automatically executed to generate Figures and/or the PDF with the following commands:</p>
<div class="sourceCode" id="cb27"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://github.com/rstudio/rmarkdown">rmarkdown</a></span>)
<span class="fu"><a href="https://rdrr.io/pkg/rmarkdown/man/render.html">render</a></span>(<span class="st">"Bio3D_md.Rmd"</span>, <span class="st">"all"</span>)
</pre></div>
</div>
<div id="information-about-the-current-bio3d-session" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#information-about-the-current-bio3d-session" class="anchor"></a>Information About the Current Bio3D Session</h1>
<div class="sourceCode" id="cb28"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()
</pre></div>
<pre><code>## R version 3.6.0 (2019-04-26)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS  10.15.5
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] bio3d_2.4-1.9000
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.5         lattice_0.20-41    digest_0.6.25      crayon_1.3.4      
##  [5] rprojroot_1.3-2    assertthat_0.2.1   grid_3.6.0         R6_2.4.1          
##  [9] backports_1.1.8    magrittr_1.5       evaluate_0.14      highr_0.8         
## [13] stringi_1.4.6      rlang_0.4.6.9000   fs_1.4.2           rmarkdown_2.3.2   
## [17] pkgdown_1.5.1.9000 desc_1.2.0         tools_3.6.0        stringr_1.4.0     
## [21] parallel_3.6.0     yaml_2.2.1         xfun_0.15          compiler_3.6.0    
## [25] memoise_1.1.0      htmltools_0.5.0    knitr_1.29</code></pre>
<p><br></p>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references hanging-indent">
<div id="ref-grant06">
<p>Grant, B. J., A. P. D. C Rodrigues, K. M. Elsawy, A. J. Mccammon, and L. S. D. Caves. 2006. “Bio3d: An R Package for the Comparative Analysis of Protein Structures.” <em>Bioinformatics</em> 22: 2695–6.</p>
</div>
<div id="ref-vmd">
<p>Humphrey, et al., W. 1996. “VMD: Visual Molecular Dynamics.” <em>J. Mol. Graph</em> 14: 33–38.</p>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>The latest version of the package, full documentation and further vignettes (including detailed installation instructions) can be obtained from the main Bio3D website: <a href="http://thegrantlab.org/bio3d/">http://thegrantlab.org/bio3d/</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>This vignette contains executable examples, see <code><a href="https://rdrr.io/r/utils/vignette.html">help(vignette)</a></code> for further details.<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="tidyverse">
  <p>Authored by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="author">
  <p>
    Built with ❤️ and <a href="https://pkgdown.r-lib.org">pkgdown</a> © 2020.
  </p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '1270c4079b26a138795263974bbf302d',
    indexName: 'tidyverse',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
