<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Atom Selection from PDB and PRMTOP Structure Objects — atom.select • Bio3D</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>

<!-- Bootstrap -->
<link href="../tidyverse.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>
<link href="../tidyverse-2.css" rel="stylesheet">

<!--optional theme
<link href="../.css" rel="stylesheet">
-->


<!-- docsearch -->
<script src="../docsearch.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous" />
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>


<meta property="og:title" content="Atom Selection from PDB and PRMTOP Structure Objects — atom.select" />
<meta property="og:description" content="Return the &amp;#8216;atom&amp;#8217; and &amp;#8216;xyz&amp;#8217; coordinate indices of
  &amp;#8216;pdb&amp;#8217; or &amp;#8216;prmtop&amp;#8217; structure objects corresponding to the
  intersection of a hierarchical selection." />
<meta property="og:image" content="/logo.png" />
<meta name="twitter:card" content="summary" />


<meta name="robots" content="noindex">

<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../index.html">
        bio3d</a>
        <div class="info hidden-xs hidden-sm">
<!--          <span class="partof">part of the <a href="https://tidyverse.org">tidyverse</a></span>
-->
          <span class="version version-danger" data-toggle="tooltip" data-placement="bottom" title="In-development version">2.4-1.9000</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">INTRODUCTORY TUTORIALS</li>
    <li>
      <a href="../articles/index.html">All Vignettes (that compile)</a>
    </li>
    <li>
      <a href="../articles/bio3d_vignettes.html">Vignettes (old listing)</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">ADVANCED TUTORIALS</li>
    <li>
      <a href="../articles/b11.html">Title B1</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">WEBAPP TUTORIALS</li>
    <li>
      <a href="../articles/c11.html">Title C1</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Documentation
  </a>
</li>
        <li>
  <a href="../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    News
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket"></span>
     
  </a>
</li>
      </ul>
      
      <form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
        </div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Atom Selection from PDB and PRMTOP Structure Objects</h1>
    
    <div class="hidden name"><code>atom.select.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Return the &#8216;atom&#8217; and &#8216;xyz&#8217; coordinate indices of
  &#8216;pdb&#8217; or &#8216;prmtop&#8217; structure objects corresponding to the
  intersection of a hierarchical selection.</p>
    </div>

    <pre class="usage"><span class='fu'>atom.select</span>(<span class='kw'>...</span>)

<span class='co'># S3 method for pdb</span>
<span class='fu'>atom.select</span>(<span class='kw'>pdb</span>, string = <span class='kw'>NULL</span>,
                          type  = <span class='kw'>NULL</span>, eleno = <span class='kw'>NULL</span>, elety = <span class='kw'>NULL</span>,
                          resid = <span class='kw'>NULL</span>, chain = <span class='kw'>NULL</span>, resno = <span class='kw'>NULL</span>,
                          insert = <span class='kw'>NULL</span>, segid = <span class='kw'>NULL</span>, 
                          operator = <span class='st'>"AND"</span>, inverse = <span class='fl'>FALSE</span>,
                          value = <span class='fl'>FALSE</span>, verbose=<span class='fl'>FALSE</span>, <span class='kw'>...</span>)
<span class='co'># S3 method for pdbs</span>
<span class='fu'>atom.select</span>(<span class='kw'>pdbs</span>, string = <span class='kw'>NULL</span>, 
                           resno = <span class='kw'>NULL</span>, chain = <span class='kw'>NULL</span>, resid = <span class='kw'>NULL</span>,
                           operator=<span class='st'>"AND"</span>, inverse = <span class='fl'>FALSE</span>,
                           value = <span class='fl'>FALSE</span>, verbose=<span class='fl'>FALSE</span>, <span class='kw'>...</span>)
<span class='co'># S3 method for mol2</span>
<span class='fu'>atom.select</span>(<span class='kw'>mol</span>, string=<span class='kw'>NULL</span>,
                           eleno = <span class='kw'>NULL</span>, elena = <span class='kw'>NULL</span>, elety = <span class='kw'>NULL</span>,
                           resid = <span class='kw'>NULL</span>, chain = <span class='kw'>NULL</span>, resno = <span class='kw'>NULL</span>,
                           statbit = <span class='kw'>NULL</span>,
			   operator = "AND", inverse = FALS<span class='st'>"AND"</span><span class='fl'>FALSE</span>
                           value = <span class='fl'>FALSE</span>, verbose=<span class='fl'>FALSE</span>,  <span class='kw'>...</span>)

<span class='co'># S3 method for prmtop</span>
<span class='fu'>atom.select</span>(<span class='kw'>prmtop</span>, <span class='kw'>...</span>)

<span class='co'># S3 method for select</span>
<span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span>(<span class='kw'>x</span>, <span class='kw'>...</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>...</th>
      <td><p>arguments passed to <code>atom.select.pdb</code>,
    <code>atom.select.prmtop</code>, or <code>print</code>.</p></td>
    </tr>
    <tr>
      <th>pdb</th>
      <td><p>a structure object of class <code>"pdb"</code>, obtained from
    <code>read.pdb</code>.</p></td>
    </tr>
    <tr>
      <th>pdbs</th>
      <td><p>a numeric matrix of aligned C-alpha xyz Cartesian
    coordinates as obtained with <code>read.fasta.pdb</code> or <code>pdbaln</code>.</p></td>
    </tr>
    <tr>
      <th>string</th>
      <td><p>a single selection keyword from <code>calpha</code>
    <code>cbeta</code> <code>backbone</code> <code>sidechain</code> <code>protein</code>
    <code>nucleic</code> <code>ligand</code> <code>water</code> <code>h</code> or <code>noh</code>.</p></td>
    </tr>
    <tr>
      <th>type</th>
      <td><p>a single element character vector for selecting &#8216;ATOM&#8217;
    or &#8216;HETATM&#8217; record types.</p></td>
    </tr>
    <tr>
      <th>eleno</th>
      <td><p>a numeric vector of element numbers.</p></td>
    </tr>
    <tr>
      <th>elena</th>
      <td><p>a character vector of atom names.</p></td>
    </tr>
    <tr>
      <th>elety</th>
      <td><p>a character vector of atom names.</p></td>
    </tr>
    <tr>
      <th>resid</th>
      <td><p>a character vector of residue name identifiers.</p></td>
    </tr>
    <tr>
      <th>chain</th>
      <td><p>a character vector of chain identifiers.</p></td>
    </tr>
    <tr>
      <th>resno</th>
      <td><p>a numeric vector of residue numbers.</p></td>
    </tr>
    <tr>
      <th>insert</th>
      <td><p>a character vector of insert identifiers. Non-insert 
    residues can be selected with <code>NA</code> or &#8216;&#8217; values. The default value 
    of <code>NULL</code> will select both insert and non-insert residues.</p></td>
    </tr>
    <tr>
      <th>segid</th>
      <td><p>a character vector of segment identifiers. Empty segid
    values can be selected with <code>NA</code> or &#8216;&#8217; values. The default value
    of <code>NULL</code> will select both empty and non-empty segment identifiers.</p></td>
    </tr>
    <tr>
      <th>operator</th>
      <td><p>a single element character specifying either the AND
    or OR operator by which individual selection components should be
    combined. Allowed values are &#8216;"AND"&#8217; and &#8216;"OR"&#8217;.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>logical, if TRUE details of the selection are printed.</p></td>
    </tr>
    <tr>
      <th>inverse</th>
      <td><p>logical, if TRUE the inversed selection is retured
    (i.e. all atoms NOT in the selection).</p></td>
    </tr>
    <tr>
      <th>value</th>
      <td><p>logical, if FALSE, vectors containing the (integer) indices of
    the matches determined by <code>atom.select</code> are returned, and if TRUE,
    a <code>pdb</code> object containing the matching atoms themselves is
    returned.</p></td>
    </tr>
    <tr>
      <th>mol</th>
      <td><p>a structure object of class <code>"mol2"</code>, obtained from
    <code>read.mol2</code>.</p></td>
    </tr>
    <tr>
      <th>statbit</th>
      <td><p>a character vector of statbit identifiers.</p></td>
    </tr>
    <tr>
      <th>prmtop</th>
      <td><p>a structure object of class <code>"prmtop"</code>, obtained from
    <code>read.prmtop</code>.</p></td>
    </tr>
    <tr>
      <th>x</th>
      <td><p>a atom.select object as obtained from
    <code>atom.select</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function allows for the selection of atom and coordinate data
  corresponding to the intersection of various input criteria.</p>
<p>Input selection criteria include selection <code>string</code> keywords (such as
  <code>"calpha"</code>, <code>"backbone"</code>, <code>"sidechain"</code>,  <code>"protein"</code>,
  <code>"nucleic"</code>, <code>"ligand"</code>, etc.) and individual named
  selection components (including &#8216;chain&#8217;,  &#8216;resno&#8217;,
  &#8216;resid&#8217;, &#8216;elety&#8217; etc.).</p>
<p>For example, <code>atom.select(pdb, "calpha")</code> will return indices for
  all C-alpha (CA) atoms found in protein residues in the <code>pdb</code>
  object, <code>atom.select(pdb, "backbone")</code> will return indices
  for all protein N,CA,C,O atoms, and <code>atom.select(pdb, "cbeta")</code>
  for all protein N,CA,C,O,CB atoms.</p>
<p>Note that keyword <code>string</code> shortcuts can be combined with individual
  selection components, e.g. <code>atom.select(pdb, "protein", chain="A")</code>
  will select all protein atoms found in chain A.</p>
<p>Selection criteria are combined according to the provided
  <code>operator</code> argument. The default operator <code>AND</code> (or <code>&amp;</code>) will
  combine by intersection while <code>OR</code> (or <code><a href='https://rdrr.io/r/base/Logic.html'>|</a></code>) will take the union.</p>
<p>For example, <code>atom.select(pdb, "protein", elety=c("N", "CA", "C"),
  resno=65:103)</code> will select the N, CA, C atoms in the protein residues
  65 through 103, while <code>atom.select(pdb, "protein", resid="ATP",
  operator="OR")</code> will select all protein atoms as well as any ATP
  residue(s).</p>
<p>Other <code>string</code> shortcuts include:
  <code>"calpha"</code>, <code>"back"</code>, <code>"backbone"</code>, <code>"cbeta"</code>,
  <code>"protein"</code>, <code>"notprotein"</code>, <code>"ligand"</code>,
  <code>"water"</code>, <code>"notwater"</code>, <code>"h"</code>, <code>"noh"</code>,
  <code>"nucleic"</code>, and <code>"notnucleic"</code>.</p>
<p>In addition, the <code>combine.select</code> function can further combine atom
  selections using &#8216;AND&#8217;, &#8216;OR&#8217;, or &#8216;NOT&#8217; logical
  operations.</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>Protein atoms are defined as any atom in a residue matching the
  residue name in the attached <code>aa.table</code> data frame. See
  <code>aa.table$aa3</code> for a complete list of residue names.</p>
<p>Nucleic atoms are defined as all atoms found in residues with names A,
  U, G, C, T, I, DA, DU, DG, DC, DT, or DI.</p>
<p>Water atoms/residues are defined as those with residue names H2O, OH2,
  HOH, HHO, OHH, SOL, WAT, TIP, TIP, TIP3, or TIP4.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns a list of class <code>"select"</code> with the following components:</p>
<dt>atom</dt><dd><p>a numeric matrix of atomic indices.</p></dd>
  <dt>xyz </dt><dd><p>a numeric matrix of xyz indices.</p></dd>
  <dt>call </dt><dd><p>the matched call.</p></dd>

    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695--2696.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Barry Grant, Lars Skjaerven</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code>read.pdb</code>, <code>as.select</code>,
  <code>combine.select</code>, <code>trim.pdb</code>,
  <code>write.pdb</code>, <code>read.prmtop</code>,
  <code>read.crd</code>, <code>read.dcd</code>,
  <code>read.ncdf</code>.</p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'>##- PDB example</span>
<span class='co'># Read a PDB file</span>
<span class='kw'>pdb</span> <span class='op'>&lt;-</span> <span class='fu'>read.pdb</span>( <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"examples/1hel.pdb"</span>, package=<span class='st'>"bio3d"</span>) )

<span class='co'># Select protein atoms of chain A</span>
<span class='fu'>atom.select</span>(<span class='kw'>pdb</span>, <span class='st'>"protein"</span>, chain=<span class='st'>"A"</span>)</div><div class='output co'>#&gt; 
#&gt;  Call:  atom.select.pdb(pdb = pdb, string = "protein", chain = "A")
#&gt; 
#&gt;    Atom Indices#: 1001  ($atom)
#&gt;    XYZ  Indices#: 3003  ($xyz)
#&gt; 
#&gt; + attr: atom, xyz, call</div><div class='input'>
<span class='co'># Select all atoms except from the protein</span>
<span class='fu'>atom.select</span>(<span class='kw'>pdb</span>, <span class='st'>"protein"</span>, inverse=<span class='fl'>TRUE</span>, verbose=<span class='fl'>TRUE</span>)</div><div class='output co'>#&gt; 
#&gt;  .. 00001001 atom(s) from 'string' selection 
#&gt;  .. 00001001 atom(s) in final combined selection 
#&gt;  .. 00000000 atom(s) in inversed selection 
#&gt; </div><div class='output co'>#&gt; 
#&gt;  Call:  atom.select.pdb(pdb = pdb, string = "protein", inverse = TRUE, 
#&gt;     verbose = TRUE)
#&gt; 
#&gt;    Atom Indices#: 0  ($atom)
#&gt;    XYZ  Indices#: 0  ($xyz)
#&gt; 
#&gt; + attr: atom, xyz, call</div><div class='input'>
<span class='co'># Select all C-alpha atoms with residues numbers between 43 and 54</span>
<span class='kw'>sele</span> <span class='op'>&lt;-</span> <span class='fu'>atom.select</span>(<span class='kw'>pdb</span>, <span class='st'>"calpha"</span>, resno=<span class='fl'>43</span><span class='op'>:</span><span class='fl'>54</span>, verbose=<span class='fl'>TRUE</span>)</div><div class='output co'>#&gt; 
#&gt;  .. 00000129 atom(s) from 'string' selection 
#&gt;  .. 00000090 atom(s) from 'resno' selection 
#&gt;  .. 00000012 atom(s) in final combined selection 
#&gt; </div><div class='input'>
<span class='co'># Access the PDB data with the selection indices</span>
<span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span>( <span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>atom</span>[ <span class='kw'>sele</span><span class='op'>$</span><span class='kw'>atom</span>, <span class='st'>"resid"</span> ] )</div><div class='output co'>#&gt;  [1] "THR" "ASN" "ARG" "ASN" "THR" "ASP" "GLY" "SER" "THR" "ASP" "TYR" "GLY"</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span>( <span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>xyz</span>[ <span class='kw'>sele</span><span class='op'>$</span><span class='kw'>xyz</span> ] )</div><div class='output co'>#&gt;  [1] 10.705 15.992 17.773 12.828 18.909 18.863 15.061 19.112 21.827 15.247
#&gt; [11] 22.135 24.062 18.108 23.768 25.931 16.626 22.832 29.261 16.950 19.205
#&gt; [21] 28.182 13.311 18.716 27.712 11.958 17.610 24.179  8.767 18.608 22.181
#&gt; [31]  6.899 15.745 20.548  4.390 15.347 17.710</div><div class='input'>
<span class='co'># Trim PDB to selection</span>
<span class='kw'>ca.pdb</span> <span class='op'>&lt;-</span> <span class='fu'>trim.pdb</span>(<span class='kw'>pdb</span>, <span class='kw'>sele</span>)

<span class='co'>if</span> (<span class='fl'>FALSE</span>) {

<span class='co'>##- PRMTOP example</span>
<span class='kw'>prmtop</span> <span class='op'>&lt;-</span> <span class='fu'>read.prmtop</span>(<span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"examples/crambin.prmtop"</span>, package=<span class='st'>"bio3d"</span>))

<span class='co'>## Atom selection</span>
<span class='kw'>ca.inds</span> <span class='op'>&lt;-</span> <span class='fu'>atom.select</span>(<span class='kw'>prmtop</span>, <span class='st'>"calpha"</span>)

}</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="tidyverse">
  <p>Authored by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="author">
  <p>
    Built with &#10084;&#65039; and <a href="https://pkgdown.r-lib.org">pkgdown</a> &copy; 2020.
  </p>
</div>

      </footer>
   </div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script>
<script>
  docsearch({
    
    
    apiKey: '1270c4079b26a138795263974bbf302d',
    indexName: 'tidyverse',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>



  </body>
</html>


