<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>NCBI BLAST Sequence Search and Summary Plot of Hit Statistics — blast.pdb • Bio3D</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>

<!-- Bootstrap -->
<link href="../tidyverse.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>
<link href="../tidyverse-2.css" rel="stylesheet">

<!--optional theme
<link href="../.css" rel="stylesheet">
-->


<!-- docsearch -->
<script src="../docsearch.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous" />
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>


<meta property="og:title" content="NCBI BLAST Sequence Search and Summary Plot of Hit Statistics — blast.pdb" />
<meta property="og:description" content="Run NCBI blastp, on a given sequence, against the PDB, NR and
  swissprot sequence databases. Produce plots that facilitate hit selection
  from the match statistics of a BLAST result." />
<meta property="og:image" content="/logo.png" />
<meta name="twitter:card" content="summary" />


<meta name="robots" content="noindex">

<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../index.html">
        bio3d</a>
        <div class="info hidden-xs hidden-sm">
<!--          <span class="partof">part of the <a href="https://tidyverse.org">tidyverse</a></span>
-->
          <span class="version version-danger" data-toggle="tooltip" data-placement="bottom" title="In-development version">2.4-1.9000</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">INTRODUCTORY TUTORIALS</li>
    <li>
      <a href="../articles/index.html">All Vignettes (that compile)</a>
    </li>
    <li>
      <a href="../articles/bio3d_vignettes.html">Vignettes (old listing)</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">ADVANCED TUTORIALS</li>
    <li>
      <a href="../articles/b11.html">Title B1</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">WEBAPP TUTORIALS</li>
    <li>
      <a href="../articles/c11.html">Title C1</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Documentation
  </a>
</li>
        <li>
  <a href="../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    News
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket"></span>
     
  </a>
</li>
      </ul>
      
      <form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
        </div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>NCBI BLAST Sequence Search and Summary Plot of Hit Statistics</h1>
    
    <div class="hidden name"><code>blast.pdb.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Run NCBI blastp, on a given sequence, against the PDB, NR and
  swissprot sequence databases. Produce plots that facilitate hit selection
  from the match statistics of a BLAST result.</p>
    </div>

    <pre class="usage"><span class='fu'>blast.pdb</span>(<span class='kw'>seq</span>, database = <span class='st'>"pdb"</span>, time.out = <span class='kw'>NULL</span>, chain.single=<span class='fl'>TRUE</span>)

<span class='fu'>get.blast</span>(<span class='kw'>urlget</span>, time.out = <span class='kw'>NULL</span>, chain.single=<span class='fl'>TRUE</span>)

<span class='co'># S3 method for blast</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/plot.html'>plot</a></span>(<span class='kw'>x</span>, cutoff = <span class='kw'>NULL</span>, cut.seed=<span class='kw'>NULL</span>, cluster=<span class='fl'>TRUE</span>, mar=<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>2</span>, <span class='fl'>5</span>, <span class='fl'>1</span>, <span class='fl'>1</span>), cex=<span class='fl'>1.5</span>, <span class='kw'>...</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>seq</th>
      <td><p>a single element or multi-element character vector
    containing the query sequence. Alternatively a &#8216;fasta&#8217;
    object from function <code>get.seq</code> or &#8216;pdb&#8217; object from 
    function <code>read.pdb</code> can be provided.</p></td>
    </tr>
    <tr>
      <th>database</th>
      <td><p>a single element character vector specifying the
    database against which to search. Current options are &#8216;pdb&#8217;,
    &#8216;nr&#8217; and &#8216;swissprot&#8217;.</p></td>
    </tr>
    <tr>
      <th>time.out</th>
      <td><p>integer specifying the number of seconds to wait for
    the blast reply before a time out occurs.</p></td>
    </tr>
    <tr>
      <th>urlget</th>
      <td><p>the URL to retrieve BLAST results; Usually it is returned
    by blast.pdb if time.out is set and met.</p></td>
    </tr>
    <tr>
      <th>chain.single</th>
      <td><p>logical, if TRUE double NCBI character PDB database  
    chain identifiers are simplified to lowercase '1WF4_GG' &gt; '1WF4_g'. If 
    FALSE no conversion to match RCSB PDB files is performed.</p></td>
    </tr>
    <tr>
      <th>x</th>
      <td><p>BLAST results as obtained from the function
    <code>blast.pdb</code>.</p></td>
    </tr>
    <tr>
      <th>cutoff</th>
      <td><p>A numeric cutoff value, in terms of minus the log of
    the evalue, for returned hits.  If null then the function will
    try to find a suitable cutoff near &#8216;cut.seed&#8217; which can be used
    as an initial guide (see below).</p></td>
    </tr>
    <tr>
      <th>cut.seed</th>
      <td><p>A numeric seed cutoff value, used for initial
    cutoff estimation. If null then a seed position is set to the point of 
    largest drop-off in normalized scores (i.e. the biggest jump in 
    E-values).</p></td>
    </tr>
    <tr>
      <th>cluster</th>
      <td><p>Logical, if TRUE (and &#8216;cutoff&#8217; is null) a clustering 
    of normalized scores is performed to partition hits in groups by 
    similarity to query. If FALSE the partition point is set to the point of largest drop-off in normalized scores.</p></td>
    </tr>
    <tr>
      <th>mar</th>
      <td><p>A numerical vector of the form c(bottom, left, top, right)
    which gives the number of lines of margin to be specified on the
    four sides of the plot.</p></td>
    </tr>
    <tr>
      <th>cex</th>
      <td><p>a numerical single element vector giving the amount by
    which plot labels should be magnified relative to the default.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>extra plotting arguments.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The <code>blast.pdb</code> function employs direct HTTP-encoded requests to the NCBI web
  server to run BLASTP, the protein search algorithm of the BLAST
  software package.</p>
<p>BLAST, currently the most popular pairwise sequence comparison algorithm 
  for database searching, performs gapped local alignments via a heuristic 
  strategy: it identifies short nearly exact matches or hits, bidirectionally 
  extends non-overlapping hits resulting in ungapped extended hits or 
  high-scoring segment pairs(HSPs), and finally extends the highest scoring 
  HSP in both directions via a gapped alignment (Altschul et al., 1997)</p>
<p>For each pairwise alignment BLAST reports the raw score, bitscore
  and an E-value that assess the statistical significance of the
  raw score. Note that unlike the raw score E-values are normalized with
  respect to both the substitution matrix and the query and database lengths.</p>
<p>Here we also return a corrected normalized score (mlog.evalue) that in
  our experience is easier to handle and store than conventional
  E-values. In practice, this score is equivalent to minus the natural
  log of the E-value. Note that, unlike the raw score, this score is
  independent of the substitution matrix and and the query and database
  lengths, and thus is comparable between BLASTP searches.</p>
<p>Examining plots of BLAST alignment lengths, scores, E-values and normalized
  scores (-log(E-Value) from the <code>blast.pdb</code> function can aid in the
  identification sensible hit similarity thresholds. This is facilitated by 
  the <code>plot.blast</code> function.</p>
<p>If a &#8216;cutoff&#8217; value is not supplied then a basic hierarchical
  clustering of normalized scores is performed with initial group partitioning
  implemented at a hopefully sensible point in the vicinity of
  &#8216;h=cut.seed&#8217;. Inspection of the resultant plot can then be use
  to refine the value of &#8216;cut.seed&#8217; or indeed
  &#8216;cutoff&#8217;. As the &#8216;cutoff&#8217; value can vary depending on
  the desired application and indeed the properties of the system under
  study it is envisaged that &#8216;plot.blast&#8217; will be called multiple
  times to aid selection of a suitable &#8216;cutoff&#8217; value. See the
  examples below for further details.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>The function <code>blast.pdb</code> returns a list with three components,
  <code>hit.tbl</code>, <code>raw</code>, and <code>url</code>. 
  The function <code>plot.blast</code> produces a plot on the 
  active graphics device and returns a list object with four components,
  <code>hits</code>, <code>pdb.id</code>, <code>acc</code>, and <code>inds</code>. See below:</p> 
<p></p>
<dt>hit.tbl </dt><dd><p>a data frame summarizing BLAST results for each
    reported hit. It contains following major columns:
    <ul>
<li><p>&#8216;bitscore&#8217;, a numeric vector containing the raw score for each
    alignment.</p></li>
<li><p>&#8216;evalue&#8217;, a numeric vector containing the E-value of the raw
    score for each alignment.</p></li>
<li><p>&#8216;mlog.evalue&#8217;, a numeric vector containing minus the natural log
    of the E-value.</p></li>
<li><p>&#8216;acc&#8217;, a character vector containing the accession database identifier of
    each hit.</p></li>
<li><p>&#8216;pdb.id&#8217;, a character vector containing the PDB database identifier of
    each hit.</p></li>
</ul></p></dd>
  <dt>raw </dt><dd><p>a data frame containing the raw BLAST output. Note multiple
    hits may appear in the same row.</p></dd>
  <dt>url </dt><dd><p>a single element character vector with the NCBI result URL and 
    RID code. This can be passed to the get.blast function.</p></dd>

  <dt>hits</dt><dd><p>an ordered matrix detailing the subset of hits with a normalized
    score above the chosen cutoff. Database identifiers are listed along
    with their cluster group number.</p></dd>
  <dt>pdb.id</dt><dd><p>a character vector containing the PDB database identifier of
    each hit above the chosen threshold.</p></dd>
  <dt>acc</dt><dd><p>a character vector containing the accession database identifier of
    each hit above the chosen threshold.</p></dd>
  <dt>inds</dt><dd><p>a numeric vector containing the indices of the hits
    relative to the input blast object.</p></dd>

    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695--2696.</p>
<p>&#8216;BLAST&#8217; is the work of Altschul et al.:
  Altschul, S.F. et al. (1990) <em>J. Mol. Biol.</em> <b>215</b>, 403--410.</p>  
<p>Full details of the &#8216;BLAST&#8217; algorithm, along with download and
  installation instructions can be obtained from:<br />
  <a href='http://www.ncbi.nlm.nih.gov/BLAST/'>http://www.ncbi.nlm.nih.gov/BLAST/</a>.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Barry Grant</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>Online access is required to query NCBI blast services.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code>plot.blast</code>, <code>hmmer</code>, <code>seqaln</code>, <code>get.pdb</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>if</span> (<span class='fl'>FALSE</span>) {
<span class='kw'>pdb</span> <span class='op'>&lt;-</span> <span class='fu'>read.pdb</span>(<span class='st'>"4q21"</span>)
<span class='kw'>blast</span> <span class='op'>&lt;-</span> <span class='fu'>blast.pdb</span>( <span class='fu'>pdbseq</span>(<span class='kw'>pdb</span>) )

<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span>(<span class='kw'>blast</span><span class='op'>$</span><span class='kw'>hit.tbl</span>)
<span class='kw'>top.hits</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/graphics/plot.html'>plot</a></span>(<span class='kw'>blast</span>)
<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span>(<span class='kw'>top.hits</span><span class='op'>$</span><span class='kw'>hits</span>)

<span class='co'>## Use 'get.blast()' to retrieve results at a later time.</span>
<span class='co'>#x &lt;- get.blast(blast$url)</span>
<span class='co'>#head(x$hit.tbl)</span>

<span class='co'># Examine and download 'best' hits</span>
<span class='kw'>top.hits</span> <span class='op'>&lt;-</span> <span class='fu'>plot.blast</span>(<span class='kw'>blast</span>, cutoff=<span class='fl'>188</span>)
<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span>(<span class='kw'>top.hits</span><span class='op'>$</span><span class='kw'>hits</span>)
<span class='co'>#get.pdb(top.hits)</span>
}</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="tidyverse">
  <p>Authored by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="author">
  <p>
    Built with &#10084;&#65039; and <a href="https://pkgdown.r-lib.org">pkgdown</a> &copy; 2020.
  </p>
</div>

      </footer>
   </div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script>
<script>
  docsearch({
    
    
    apiKey: '1270c4079b26a138795263974bbf302d',
    indexName: 'tidyverse',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>



  </body>
</html>


