<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Identification of Invariant Core Positions — core.find • Bio3D</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>

<!-- Bootstrap -->
<link href="../tidyverse.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>
<link href="../tidyverse-2.css" rel="stylesheet">

<!--optional theme-->
<link href="../.css" rel="stylesheet">



<!-- docsearch -->
<script src="../docsearch.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous" />
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>


<meta property="og:title" content="Identification of Invariant Core Positions — core.find" />
<meta property="og:description" content="Perform iterated rounds of structural superposition to identify the
  most invariant region in an aligned set of protein structures." />
<meta property="og:image" content="/logo.png" />
<meta name="twitter:card" content="summary" />


<meta name="robots" content="noindex">

<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->

<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script>

  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../index.html">
        bio3d</a>
        <div class="info hidden-xs hidden-sm">
<!--          <span class="partof">part of the <a href="https://tidyverse.org">tidyverse</a></span>
-->
          <span class="version version-danger" data-toggle="tooltip" data-placement="bottom" title="In-development version">2.4-1.9000</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">INTRODUCTORY TUTORIALS</li>
    <li>
      <a href="../articles/bio3d_vignettes.html">Vignettes (old listing)</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">ADVANCED TUTORIALS</li>
    <li>
      <a href="../articles/b11.html">Title B1</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">WEBAPP TUTORIALS</li>
    <li>
      <a href="../articles/c11.html">Title C1</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Documentation
  </a>
</li>
        <li>
  <a href="../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    News
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket"></span>
     
  </a>
</li>
      </ul>
      
      <form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
        </div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Identification of Invariant Core Positions</h1>
    
    <div class="hidden name"><code>core.find.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Perform iterated rounds of structural superposition to identify the
  most invariant region in an aligned set of protein structures.</p>
    </div>

    <pre class="usage"><span class='fu'>core.find</span>(<span class='kw'>...</span>)

<span class='co'># S3 method for pdbs</span>
<span class='fu'>core.find</span>(<span class='kw'>pdbs</span>, shortcut = <span class='fl'>FALSE</span>, rm.island = <span class='fl'>FALSE</span>,
          verbose = <span class='fl'>TRUE</span>, stop.at = <span class='fl'>15</span>, stop.vol = <span class='fl'>0.5</span>,
          write.pdbs = <span class='fl'>FALSE</span>, outpath=<span class='st'>"core_pruned"</span>,
          ncore = <span class='fl'>1</span>, nseg.scale = <span class='fl'>1</span>, progress = <span class='kw'>NULL</span>, <span class='kw'>...</span>)

<span class='co'># S3 method for default</span>
<span class='fu'>core.find</span>(<span class='kw'>xyz</span>, <span class='kw'>...</span>)

<span class='co'># S3 method for pdb</span>
<span class='fu'>core.find</span>(<span class='kw'>pdb</span>, verbose=<span class='fl'>TRUE</span>, <span class='kw'>...</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>pdbs</th>
      <td><p>a numeric matrix of aligned C-alpha xyz Cartesian
    coordinates. For example an alignment data structure obtained with
    <code>read.fasta.pdb</code> or <code>pdbaln</code>.</p></td>
    </tr>
    <tr>
      <th>shortcut</th>
      <td><p>if TRUE, remove more than one position at a time.</p></td>
    </tr>
    <tr>
      <th>rm.island</th>
      <td><p>remove isolated fragments of less than three
    residues.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>logical, if TRUE a &#8220;core\_pruned&#8221; directory
    containing &#8216;core structures&#8217; for each iteraction is written 
    to the current directory.</p></td>
    </tr>
    <tr>
      <th>stop.at</th>
      <td><p>minimal core size at which iterations should be
    stopped.</p></td>
    </tr>
    <tr>
      <th>stop.vol</th>
      <td><p>minimal core volume at which iterations should be
    stopped.</p></td>
    </tr>
    <tr>
      <th>write.pdbs</th>
      <td><p>logical, if TRUE core coordinate files, containing
    only core positions for each iteration, are written to a location
    specified by <code>outpath</code>.</p></td>
    </tr>
    <tr>
      <th>outpath</th>
      <td><p>character string specifying the output directory when
    <code>write.pdbs</code> is TRUE.</p></td>
    </tr>
    <tr>
      <th>ncore </th>
      <td><p>number of CPU cores used to do the calculation.
    <code>ncore&gt;1</code> requires package &#8216;parallel&#8217; installed.</p></td>
    </tr>
    <tr>
      <th>nseg.scale </th>
      <td><p>split input data into specified number of segments
    prior to running multiple core calculation. See <code>fit.xyz</code>.</p></td>
    </tr>
    <tr>
      <th>progress</th>
      <td><p>progress bar for use with shiny web app.</p></td>
    </tr>
    <tr>
      <th>xyz</th>
      <td><p>a numeric matrix of xyz Cartesian coordinates,
    e.g. obtained from <code>read.dcd</code> or <code>read.ncdf</code>.</p></td>
    </tr>
    <tr>
      <th>pdb</th>
      <td><p>an object of type <code>pdb</code> as obtained from function
    <code>read.pdb</code> with multiple frames (&gt;=4) stored in its
    <code>xyz</code> component. Note that the function will attempt to
    identify C-alpha and phosphate atoms (for protein and nucleic acids,
    respectively) in which the calculation should be based.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>arguments passed to and from functions.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function attempts to iteratively refine an initial structural
  superposition determined from a multiple alignment.
  This involves iterated rounds of superposition, where at each round the
  position(s) displaying the largest differences is(are) excluded from the
  dataset. 
  The spatial variation at each aligned position is determined from the
  eigenvalues of their Cartesian coordinates (i.e. the variance of the
  distribution along its three principal directions). Inspired by the
  work of Gerstein <em>et al.</em> (1991, 1995), an ellipsoid of
  variance is determined from the eigenvalues, and its volume is taken as
  a measure of structural variation at a given position.</p>
<p>Optional &#8220;core PDB files&#8221; containing core positions, upon which
  superposition is based, can be written to a location specified by
  <code>outpath</code> by setting <code>write.pdbs=TRUE</code>.  These files are
  useful for examining the core filtering process by visualising them in a
  graphics program.</p>  
<p></p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns a list of class <code>"core"</code> with the following components:</p>
<dt>volume </dt><dd><p>total core volume at each fitting iteration/round.</p></dd>
  <dt>length </dt><dd><p>core length at each round.</p></dd>
  <dt>resno </dt><dd><p>residue number of core residues at each round (taken
    from the first aligned structure) or, alternatively, the numeric
    index of core residues at each round.</p></dd>
  <dt>step.inds</dt><dd><p>atom indices of core atoms at each round.</p></dd>
  <dt>atom </dt><dd><p>atom indices of core positions in the last round.</p></dd>
  <dt>xyz </dt><dd><p>xyz indices of core positions in the last round.</p></dd>
  <dt>c1A.atom </dt><dd><p>atom indices of core positions with a total volume
    under 1 Angstrom\^3.</p></dd>
  <dt>c1A.xyz </dt><dd><p>xyz indices of core positions with a total volume
    under 1 Angstrom\^3.</p></dd>
  <dt>c1A.resno </dt><dd><p>residue numbers of core positions with a total volume
    under 1 Angstrom\^3.</p></dd>
  <dt>c0.5A.atom </dt><dd><p>atom indices of core positions with a total volume
    under 0.5 Angstrom\^3.</p></dd>
  <dt>c0.5A.xyz </dt><dd><p>xyz indices of core positions with a total volume
    under 0.5 Angstrom\^3.</p></dd>
  <dt>c0.5A.resno </dt><dd><p>residue numbers of core positions with a total volume
    under 0.5 Angstrom\^3.</p></dd>

    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695--2696.</p>
<p>Gerstein and Altman (1995) <em>J. Mol. Biol.</em>  <b>251</b>, 161--175.</p>
<p>Gerstein and Chothia (1991) <em>J. Mol. Biol.</em> <b>220</b>, 133--149.</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>The relevance of the &#8216;core positions&#8217; identified by this
  procedure is dependent upon the number of input structures and their
  diversity.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Barry Grant</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code>read.fasta.pdb</code>, <code>plot.core</code>,
  <code>fit.xyz</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>if</span> (<span class='fl'>FALSE</span>) {
<span class='co'>##-- Generate a small kinesin alignment and read corresponding structures</span>
<span class='kw'>pdbfiles</span> <span class='op'>&lt;-</span> <span class='fu'>get.pdb</span>(<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"1bg2"</span>,<span class='st'>"2ncd"</span>,<span class='st'>"1i6i"</span>,<span class='st'>"1i5s"</span>), URLonly=<span class='fl'>TRUE</span>)
<span class='kw'>pdbs</span> <span class='op'>&lt;-</span> <span class='fu'>pdbaln</span>(<span class='kw'>pdbfiles</span>)

<span class='co'>##-- Find 'core' positions</span>
<span class='kw'>core</span> <span class='op'>&lt;-</span> <span class='fu'>core.find</span>(<span class='kw'>pdbs</span>)
<span class='fu'><a href='https://rdrr.io/r/graphics/plot.html'>plot</a></span>(<span class='kw'>core</span>)

<span class='co'>##-- Fit on these relatively invarient subset of positions </span>
<span class='co'>#core.inds &lt;- print(core, vol=1)</span>
<span class='kw'>core.inds</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span>(<span class='kw'>core</span>, vol=<span class='fl'>0.5</span>)
<span class='kw'>xyz</span> <span class='op'>&lt;-</span> <span class='fu'>pdbfit</span>(<span class='kw'>pdbs</span>, <span class='kw'>core.inds</span>, outpath=<span class='st'>"corefit_structures"</span>)

<span class='co'>##-- Compare to fitting on all equivalent positions</span>
<span class='kw'>xyz2</span> <span class='op'>&lt;-</span> <span class='fu'>pdbfit</span>(<span class='kw'>pdbs</span>)

<span class='co'>## Note that overall RMSD will be higher but RMSF will</span>
<span class='co'>##  be lower in core regions, which may equate to a</span>
<span class='co'>##  'better fit' for certain applications</span>
<span class='kw'>gaps</span> <span class='op'>&lt;-</span> <span class='fu'>gap.inspect</span>(<span class='kw'>pdbs</span><span class='op'>$</span><span class='kw'>xyz</span>)
<span class='fu'>rmsd</span>(<span class='kw'>xyz</span>[,<span class='kw'>gaps</span><span class='op'>$</span><span class='kw'>f.inds</span>])
<span class='fu'>rmsd</span>(<span class='kw'>xyz2</span>[,<span class='kw'>gaps</span><span class='op'>$</span><span class='kw'>f.inds</span>])

<span class='fu'><a href='https://rdrr.io/r/graphics/plot.html'>plot</a></span>(<span class='fu'>rmsf</span>(<span class='kw'>xyz</span>[,<span class='kw'>gaps</span><span class='op'>$</span><span class='kw'>f.inds</span>]), typ=<span class='st'>"l"</span>, col=<span class='st'>"blue"</span>, ylim=<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>0</span>,<span class='fl'>9</span>))
<span class='fu'><a href='https://rdrr.io/r/graphics/points.html'>points</a></span>(<span class='fu'>rmsf</span>(<span class='kw'>xyz2</span>[,<span class='kw'>gaps</span><span class='op'>$</span><span class='kw'>f.inds</span>]), typ=<span class='st'>"l"</span>, col=<span class='st'>"red"</span>)
}

<span class='co'>if</span> (<span class='fl'>FALSE</span>) {
<span class='co'>##-- Run core.find() on a multimodel PDB file</span>
<span class='kw'>pdb</span> <span class='op'>&lt;-</span> <span class='fu'>read.pdb</span>(<span class='st'>'1d1d'</span>, multi=<span class='fl'>TRUE</span>)
<span class='kw'>core</span> <span class='op'>&lt;-</span> <span class='fu'>core.find</span>(<span class='kw'>pdb</span>)

<span class='co'>##-- Run core.find() on a trajectory</span>
<span class='kw'>trtfile</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"examples/hivp.dcd"</span>, package=<span class='st'>"bio3d"</span>)
<span class='kw'>trj</span> <span class='op'>&lt;-</span> <span class='fu'>read.dcd</span>(<span class='kw'>trtfile</span>)

<span class='co'>## Read the starting PDB file to determine atom correspondence</span>
<span class='kw'>pdbfile</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"examples/hivp.pdb"</span>, package=<span class='st'>"bio3d"</span>)
<span class='kw'>pdb</span> <span class='op'>&lt;-</span> <span class='fu'>read.pdb</span>(<span class='kw'>pdbfile</span>)

<span class='co'>## select calpha coords from a manageable number of frames</span>
<span class='kw'>ca.ind</span> <span class='op'>&lt;-</span> <span class='fu'>atom.select</span>(<span class='kw'>pdb</span>, <span class='st'>"calpha"</span>)<span class='op'>$</span><span class='kw'>xyz</span>
<span class='kw'>frames</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>1</span>, <span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span>(<span class='kw'>trj</span>), by=<span class='fl'>10</span>)

<span class='kw'>core</span> <span class='op'>&lt;-</span> <span class='fu'>core.find</span>( <span class='kw'>trj</span>[<span class='kw'>frames</span>, <span class='kw'>ca.ind</span>], write.pdbs=<span class='fl'>TRUE</span> )

<span class='co'>## have a look at the various cores "vmd -m core_pruned/*.pdb"</span>

<span class='co'>## Lets use a 6A^3 core cutoff</span>
<span class='kw'>inds</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span>(<span class='kw'>core</span>, vol=<span class='fl'>6</span>)
<span class='fu'>write.pdb</span>(xyz=<span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>xyz</span>[<span class='kw'>inds</span><span class='op'>$</span><span class='kw'>xyz</span>],resno=<span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>atom</span>[<span class='kw'>inds</span><span class='op'>$</span><span class='kw'>atom</span>,<span class='st'>"resno"</span>], file=<span class='st'>"core.pdb"</span>)


<span class='co'>##- Fit trj onto starting structure based on core indices</span>
<span class='kw'>xyz</span> <span class='op'>&lt;-</span> <span class='fu'>fit.xyz</span>( fixed = <span class='kw'>pdb</span><span class='op'>$</span><span class='kw'>xyz</span>,
               mobile = <span class='kw'>trj</span>,
               fixed.inds  = <span class='kw'>inds</span><span class='op'>$</span><span class='kw'>xyz</span>,
               mobile.inds = <span class='kw'>inds</span><span class='op'>$</span><span class='kw'>xyz</span>)

<span class='co'>##write.pdb(pdb=pdb, xyz=xyz, file="new_trj.pdb")</span>
<span class='co'>##write.ncdf(xyz, "new_trj.nc")</span>

}
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="tidyverse">
  <p>Rough draft website demo based on [tidytemplate]()</p>
</div>

<div class="author">
  <p>
    Developed by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.
    Site built by <a href="https://pkgdown.r-lib.org">pkgdown</a>.
  </p>
</div>

      </footer>
   </div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script>
<script>
  docsearch({
    
    
    apiKey: '1270c4079b26a138795263974bbf302d',
    indexName: 'tidyverse',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>



  </body>
</html>


