<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ Bio3D</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>

<!-- Bootstrap -->
<link href="../tidyverse.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>
<link href="../tidyverse-2.css" rel="stylesheet">

<!--optional theme-->
<link href="../.css" rel="stylesheet">



<!-- docsearch -->
<script src="../docsearch.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous" />
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>


<meta property="og:title" content="Function reference" />
<meta property="og:image" content="/logo.png" />
<meta name="twitter:card" content="summary" />


<meta name="robots" content="noindex">

<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->

<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script>

  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../index.html">
        bio3d</a>
        <div class="info hidden-xs hidden-sm">
<!--          <span class="partof">part of the <a href="https://tidyverse.org">tidyverse</a></span>
-->
          <span class="version version-danger" data-toggle="tooltip" data-placement="bottom" title="In-development version">2.4-1.9000</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">INTRODUCTORY TUTORIALS</li>
    <li>
      <a href="../articles/bio3d_vignettes.html">Vignettes (old listing)</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">ADVANCED TUTORIALS</li>
    <li>
      <a href="../articles/b11.html">Title B1</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">WEBAPP TUTORIALS</li>
    <li>
      <a href="../articles/c11.html">Title C1</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Documentation
  </a>
</li>
        <li>
  <a href="../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    News
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket"></span>
     
  </a>
</li>
      </ul>
      
      <form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
        </div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-all-functions" class="hasAnchor"><a href="#section-all-functions" class="anchor"></a>All functions</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="aa.index.html">aa.index</a></code> </p>
        </td>
        <td><p>AAindex: Amino Acid Index Database</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="aa.table.html">aa.table</a></code> </p>
        </td>
        <td><p>Table of Relevant Amino Acids</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="aa123.html">aa123()</a></code> <code><a href="aa123.html">aa321()</a></code> </p>
        </td>
        <td><p>Convert Between 1-letter and 3-letter Aminoacid Codes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="aa2index.html">aa2index()</a></code> </p>
        </td>
        <td><p>Convert an Aminoacid Sequence to AAIndex Values</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="aa2mass.html">aa2mass()</a></code> </p>
        </td>
        <td><p>Amino Acid Residues to Mass Converter</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="aanma.pdb.html">aanma()</a></code> <code><a href="aanma.pdb.html">rtb()</a></code> </p>
        </td>
        <td><p>All Atom Normal Mode Analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="aanma.pdbs.html">aanma(<i>&lt;pdbs&gt;</i>)</a></code> </p>
        </td>
        <td><p>Ensemble Normal Mode Analysis with All-Atom ENM</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="aln2html.html">aln2html()</a></code> </p>
        </td>
        <td><p>Create a HTML Page For a Given Alignment</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="angle.xyz.html">angle.xyz()</a></code> </p>
        </td>
        <td><p>Calculate the Angle Between Three Atoms</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="as.fasta.html">as.fasta()</a></code> </p>
        </td>
        <td><p>Alignment to FASTA object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="as.pdb.html">as.pdb()</a></code> </p>
        </td>
        <td><p>Convert to PDB format</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="as.select.html">as.select()</a></code> </p>
        </td>
        <td><p>Convert Atomic Indices to a Select Object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="atom.index.html">atom.index</a></code> </p>
        </td>
        <td><p>Atom Names/Types</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="atom.select.html">atom.select()</a></code> <code><a href="atom.select.html">print(<i>&lt;select&gt;</i>)</a></code> </p>
        </td>
        <td><p>Atom Selection from PDB and PRMTOP Structure Objects</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="atom2ele.html">atom2ele()</a></code> </p>
        </td>
        <td><p>Atom Names/Types to Atomic Symbols Converter</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="atom2mass.html">atom2mass()</a></code> </p>
        </td>
        <td><p>Atom Names/Types to Mass Converter</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="atom2xyz.html">atom2xyz()</a></code> <code><a href="atom2xyz.html">xyz2atom()</a></code> </p>
        </td>
        <td><p>Convert Between Atom and xyz Indices</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="basename.pdb.html">basename.pdb()</a></code> </p>
        </td>
        <td><p>Manipulate PDB File Names</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="bhattacharyya.html">bhattacharyya()</a></code> </p>
        </td>
        <td><p>Bhattacharyya Coefficient</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="binding.site.html">binding.site()</a></code> </p>
        </td>
        <td><p>Binding Site Residues</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="bio3d.package.html">bio3d-package</a></code> </p>
        </td>
        <td><p>Biological Structure Analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="biounit.html">biounit()</a></code> </p>
        </td>
        <td><p>Biological Units Construction</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="blast.pdb.html">blast.pdb()</a></code> <code><a href="blast.pdb.html">get.blast()</a></code> <code><a href="blast.pdb.html">plot(<i>&lt;blast&gt;</i>)</a></code> </p>
        </td>
        <td><p>NCBI BLAST Sequence Search and Summary Plot of Hit Statistics</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="bounds.html">bounds()</a></code> </p>
        </td>
        <td><p>Bounds of a Numeric Vector</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="bounds.sse.html">bounds.sse()</a></code> </p>
        </td>
        <td><p>Obtain A SSE Object From An SSE Sequence Vector</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="bwr.colors.html">bwr.colors()</a></code> <code><a href="bwr.colors.html">mono.colors()</a></code> </p>
        </td>
        <td><p>Color Palettes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cat.pdb.html">cat.pdb()</a></code> </p>
        </td>
        <td><p>Concatenate Multiple PDB Objects</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="chain.pdb.html">chain.pdb()</a></code> </p>
        </td>
        <td><p>Find Possible PDB Chain Breaks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="check.utility.html">check.utility()</a></code> </p>
        </td>
        <td><p>Check on Missing Utility Programs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="clean.pdb.html">clean.pdb()</a></code> </p>
        </td>
        <td><p>Inspect And Clean Up A PDB Object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cmap.html">cmap()</a></code> </p>
        </td>
        <td><p>Contact Map</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cna.html">cna()</a></code> </p>
        </td>
        <td><p>Protein Dynamic Correlation Network Construction and Community Analysis.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cnapath.html">cnapath()</a></code> <code><a href="cnapath.html">summary(<i>&lt;cnapath&gt;</i>)</a></code> <code><a href="cnapath.html">print(<i>&lt;cnapath&gt;</i>)</a></code> <code><a href="cnapath.html">plot(<i>&lt;cnapath&gt;</i>)</a></code> <code><a href="cnapath.html">plot(<i>&lt;ecnapath&gt;</i>)</a></code> </p>
        </td>
        <td><p>Suboptimal Path Analysis for Correlation Networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="com.html">com()</a></code> </p>
        </td>
        <td><p>Center of Mass</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="combine.select.html">combine.select()</a></code> </p>
        </td>
        <td><p>Combine Atom Selections From PDB Structure</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="community.aln.html">community.aln()</a></code> </p>
        </td>
        <td><p>Align communities from two or more networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="community.tree.html">community.tree()</a></code> </p>
        </td>
        <td><p>Reconstruction of the Girvan-Newman Community Tree for a CNA Class Object.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="consensus.html">consensus()</a></code> </p>
        </td>
        <td><p>Sequence Consensus for an Alignment</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="conserv.html">conserv()</a></code> </p>
        </td>
        <td><p>Score Residue Conservation At Each Position in an Alignment</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="convert.pdb.html">convert.pdb()</a></code> </p>
        </td>
        <td><p>Renumber and Convert Between Various PDB formats</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="core.cmap.html">core.cmap()</a></code> </p>
        </td>
        <td><p>Identification of Contact Map Core Positions</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="core.find.html">core.find()</a></code> </p>
        </td>
        <td><p>Identification of Invariant Core Positions</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cov.nma.html">cov(<i>&lt;nma&gt;</i>)</a></code> <code><a href="cov.nma.html">cov(<i>&lt;enma&gt;</i>)</a></code> </p>
        </td>
        <td><p>Calculate Covariance Matrix from Normal Modes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="covsoverlap.html">covsoverlap()</a></code> </p>
        </td>
        <td><p>Covariance Overlap</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dccm.html">dccm()</a></code> </p>
        </td>
        <td><p>DCCM: Dynamical Cross-Correlation Matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dccm.enma.html">dccm(<i>&lt;enma&gt;</i>)</a></code> </p>
        </td>
        <td><p>Cross-Correlation for Ensemble NMA (eNMA)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dccm.gnm.html">dccm(<i>&lt;gnm&gt;</i>)</a></code> <code><a href="dccm.gnm.html">dccm(<i>&lt;egnm&gt;</i>)</a></code> </p>
        </td>
        <td><p>Dynamic Cross-Correlation from Gaussian Network Model</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dccm.nma.html">dccm(<i>&lt;nma&gt;</i>)</a></code> </p>
        </td>
        <td><p>Dynamic Cross-Correlation from Normal Modes Analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dccm.pca.html">dccm(<i>&lt;pca&gt;</i>)</a></code> </p>
        </td>
        <td><p>Dynamical Cross-Correlation Matrix from Principal Component Analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dccm.xyz.html">dccm(<i>&lt;xyz&gt;</i>)</a></code> </p>
        </td>
        <td><p>Dynamical Cross-Correlation Matrix from Cartesian Coordinates</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="deformation.nma.html">deformation.nma()</a></code> </p>
        </td>
        <td><p>Deformation Analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="diag.ind.html">diag.ind()</a></code> </p>
        </td>
        <td><p>Diagonal Indices of a Matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="difference.vector.html">difference.vector()</a></code> </p>
        </td>
        <td><p>Difference Vector</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dist.xyz.html">dist.xyz()</a></code> </p>
        </td>
        <td><p>Calculate the Distances Between the Rows of Two Matrices</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dm.html">dm()</a></code> </p>
        </td>
        <td><p>Distance Matrix Analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dssp.html">dssp()</a></code> <code><a href="dssp.html">stride()</a></code> <code><a href="dssp.html">print(<i>&lt;sse&gt;</i>)</a></code> </p>
        </td>
        <td><p>Secondary Structure Analysis with DSSP or STRIDE</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="elements.html">elements</a></code> </p>
        </td>
        <td><p>Periodic Table of the Elements</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="entropy.html">entropy()</a></code> </p>
        </td>
        <td><p>Shannon Entropy Score</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="example.data.html">kinesin</a></code> <code><a href="example.data.html">transducin</a></code> <code><a href="example.data.html">hivp</a></code> </p>
        </td>
        <td><p>Bio3d Example Data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filter.cmap.html">filter.cmap()</a></code> </p>
        </td>
        <td><p>Contact Map Consensus Filtering</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filter.dccm.html">filter.dccm()</a></code> </p>
        </td>
        <td><p>Filter for Cross-correlation Matrices (Cij)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filter.identity.html">filter.identity()</a></code> </p>
        </td>
        <td><p>Percent Identity Filter</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filter.rmsd.html">filter.rmsd()</a></code> </p>
        </td>
        <td><p>RMSD Filter</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fit.xyz.html">fit.xyz()</a></code> <code><a href="fit.xyz.html">rot.lsq()</a></code> </p>
        </td>
        <td><p>Coordinate Superposition</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fluct.nma.html">fluct.nma()</a></code> </p>
        </td>
        <td><p>NMA Fluctuations</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="formula2mass.html">formula2mass()</a></code> </p>
        </td>
        <td><p>Chemical Formula to Mass Converter</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="gap.inspect.html">gap.inspect()</a></code> </p>
        </td>
        <td><p>Alignment Gap Summary</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="geostas.html">geostas()</a></code> <code><a href="geostas.html">amsm.xyz()</a></code> <code><a href="geostas.html">print(<i>&lt;geostas&gt;</i>)</a></code> </p>
        </td>
        <td><p>GeoStaS Domain Finder</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get.pdb.html">get.pdb()</a></code> </p>
        </td>
        <td><p>Download PDB Coordinate Files</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get.seq.html">get.seq()</a></code> </p>
        </td>
        <td><p>Download FASTA Sequence Files</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="gnm.html">gnm()</a></code> </p>
        </td>
        <td><p>Gaussian Network Model</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="hclustplot.html">hclustplot()</a></code> </p>
        </td>
        <td><p>Dendrogram with Clustering Annotation</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="hmmer.html">hmmer()</a></code> </p>
        </td>
        <td><p>HMMER Sequence Search</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="identify.cna.html">identify(<i>&lt;cna&gt;</i>)</a></code> </p>
        </td>
        <td><p>Identify Points in a CNA Protein Structure Network Plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="inner.prod.html">inner.prod()</a></code> </p>
        </td>
        <td><p>Mass-weighted Inner Product</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="inspect.connectivity.html">inspect.connectivity()</a></code> </p>
        </td>
        <td><p>Check the Connectivity of Protein Structures</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="is.gap.html">is.gap()</a></code> </p>
        </td>
        <td><p>Gap Characters</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="is.mol2.html">is.mol2()</a></code> </p>
        </td>
        <td><p>Is an Object of Class &#8216;mol2&#8217;?</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="is.pdb.html">is.pdb()</a></code> <code><a href="is.pdb.html">is.pdbs()</a></code> </p>
        </td>
        <td><p>Is an Object of Class &#8216;pdb(s)&#8217;?</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="is.select.html">is.select()</a></code> </p>
        </td>
        <td><p>Is an Object of Class &#8216;select&#8217;?</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="is.xyz.html">is.xyz()</a></code> <code><a href="is.xyz.html">as.xyz()</a></code> </p>
        </td>
        <td><p>Is an Object of Class &#8216;xyz&#8217;?</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="layout.cna.html">layout.cna()</a></code> </p>
        </td>
        <td><p>Protein Structure Network Layout</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="lbio3d.html">lbio3d()</a></code> </p>
        </td>
        <td><p>List all Functions in the bio3d Package</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="load.enmff.html">load.enmff()</a></code> <code><a href="load.enmff.html">ff.calpha()</a></code> <code><a href="load.enmff.html">ff.anm()</a></code> <code><a href="load.enmff.html">ff.pfanm()</a></code> <code><a href="load.enmff.html">ff.sdenm()</a></code> <code><a href="load.enmff.html">ff.reach()</a></code> <code><a href="load.enmff.html">ff.aaenm()</a></code> <code><a href="load.enmff.html">ff.aaenm2()</a></code> </p>
        </td>
        <td><p>ENM Force Field Loader</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mask.dccm.html">mask()</a></code> </p>
        </td>
        <td><p>Mask a Subset of Atoms in a DCCM Object.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mktrj.html">mktrj()</a></code> </p>
        </td>
        <td><p>PCA / NMA Atomic Displacement Trajectory</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="motif.find.html">motif.find()</a></code> </p>
        </td>
        <td><p>Find Sequence Motifs.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mustang.html">mustang()</a></code> </p>
        </td>
        <td><p>Structure-based Sequence Alignment with MUSTANG</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="network.amendment.html">network.amendment()</a></code> </p>
        </td>
        <td><p>Amendment of a CNA Network According To A Input Community Membership Vector.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="nma.html">nma()</a></code> </p>
        </td>
        <td><p>Normal Mode Analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="nma.pdb.html">nma(<i>&lt;pdb&gt;</i>)</a></code> <code><a href="nma.pdb.html">build.hessian()</a></code> <code><a href="nma.pdb.html">print(<i>&lt;nma&gt;</i>)</a></code> </p>
        </td>
        <td><p>Normal Mode Analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="nma.pdbs.html">nma(<i>&lt;pdbs&gt;</i>)</a></code> <code><a href="nma.pdbs.html">print(<i>&lt;enma&gt;</i>)</a></code> </p>
        </td>
        <td><p>Ensemble Normal Mode Analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="normalize.vector.html">normalize.vector()</a></code> </p>
        </td>
        <td><p>Mass-Weighted Normalized Vector</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="orient.pdb.html">orient.pdb()</a></code> </p>
        </td>
        <td><p>Orient a PDB Structure</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="overlap.html">overlap()</a></code> </p>
        </td>
        <td><p>Overlap analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pairwise.html">pairwise()</a></code> </p>
        </td>
        <td><p>Pair Indices</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pca.html">pca()</a></code> </p>
        </td>
        <td><p>Principal Component Analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pca.array.html">pca(<i>&lt;array&gt;</i>)</a></code> </p>
        </td>
        <td><p>Principal Component Analysis of an array of matrices</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pca.pdbs.html">pca(<i>&lt;pdbs&gt;</i>)</a></code> </p>
        </td>
        <td><p>Principal Component Analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pca.tor.html">pca(<i>&lt;tor&gt;</i>)</a></code> </p>
        </td>
        <td><p>Principal Component Analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pca.xyz.html">pca(<i>&lt;xyz&gt;</i>)</a></code> <code><a href="pca.xyz.html">print(<i>&lt;pca&gt;</i>)</a></code> </p>
        </td>
        <td><p>Principal Component Analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pdb.annotate.html">pdb.annotate()</a></code> <code><a href="pdb.annotate.html">pdb.pfam()</a></code> </p>
        </td>
        <td><p>Get Customizable Annotations From PDB Or PFAM Databases</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pdb2aln.html">pdb2aln()</a></code> </p>
        </td>
        <td><p>Align a PDB structure to an existing alignment</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pdb2aln.ind.html">pdb2aln.ind()</a></code> </p>
        </td>
        <td><p>Mapping from alignment positions to PDB atomic indices</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pdb2sse.html">pdb2sse()</a></code> </p>
        </td>
        <td><p>Obtain An SSE Sequence Vector From A PDB Object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pdbaln.html">pdbaln()</a></code> </p>
        </td>
        <td><p>Sequence Alignment of PDB Files</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pdbfit.html">pdbfit()</a></code> </p>
        </td>
        <td><p>PDB File Coordinate Superposition</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pdbs2pdb.html">pdbs2pdb()</a></code> </p>
        </td>
        <td><p>PDBs to PDB Converter</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pdbs2sse.html">pdbs2sse()</a></code> </p>
        </td>
        <td><p>SSE annotation for a PDBs Object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pdbseq.html">pdbseq()</a></code> </p>
        </td>
        <td><p>Extract The Aminoacid Sequence From A PDB Object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pdbsplit.html">pdbsplit()</a></code> </p>
        </td>
        <td><p>Split a PDB File Into Separate Files, One For Each Chain.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pfam.html">pfam()</a></code> </p>
        </td>
        <td><p>Download Pfam FASTA Sequence Alignment</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot.bio3d.html">plotb3()</a></code> <code><a href="plot.bio3d.html">plot(<i>&lt;bio3d&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plots with marginal SSE annotation</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot.cmap.html">plot(<i>&lt;cmap&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot Contact Matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot.cna.html">plot(<i>&lt;cna&gt;</i>)</a></code> <code><a href="plot.cna.html">plot(<i>&lt;ecna&gt;</i>)</a></code> </p>
        </td>
        <td><p>Protein Structure Network Plots in 2D and 3D.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot.core.html">plot(<i>&lt;core&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot Core Fitting Progress</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot.dccm.html">plot(<i>&lt;dccm&gt;</i>)</a></code> </p>
        </td>
        <td><p>DCCM Plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot.dmat.html">plot(<i>&lt;dmat&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot Distance Matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot.enma.html">plot(<i>&lt;enma&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot eNMA Results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot.fasta.html">plot(<i>&lt;fasta&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot a Multiple Sequence Alignment</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot.fluct.html">plot(<i>&lt;fluct&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot Fluctuations</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot.geostas.html">plot(<i>&lt;geostas&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot Geostas Results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot.hmmer.html">plot(<i>&lt;hmmer&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot a Summary of HMMER Hit Statistics.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot.matrix.loadings.html">plot(<i>&lt;matrix.loadings&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot Residue-Residue Matrix Loadings</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot.nma.html">plot(<i>&lt;nma&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot NMA Results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot.pca.html">plot(<i>&lt;pca&gt;</i>)</a></code> <code><a href="plot.pca.html">plot(<i>&lt;pca.scree&gt;</i>)</a></code> <code><a href="plot.pca.html">plot(<i>&lt;pca.score&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot PCA Results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot.pca.loadings.html">plot(<i>&lt;pca.loadings&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot Residue Loadings along PC1 to PC3</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot.rmsip.html">plot(<i>&lt;rmsip&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot RMSIP Results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="print.cna.html">print(<i>&lt;cna&gt;</i>)</a></code> <code><a href="print.cna.html">summary(<i>&lt;cna&gt;</i>)</a></code> </p>
        </td>
        <td><p>Summarize and Print Features of a cna Network Graph</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="print.core.html">print(<i>&lt;core&gt;</i>)</a></code> </p>
        </td>
        <td><p>Printing Core Positions and Returning Indices</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="print.fasta.html">print(<i>&lt;fasta&gt;</i>)</a></code> <code><a href="print.fasta.html">.print.fasta.ali()</a></code> </p>
        </td>
        <td><p>Printing Sequence Alignments</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="print.xyz.html">print(<i>&lt;xyz&gt;</i>)</a></code> </p>
        </td>
        <td><p>Printing XYZ coordinates</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="project.pca.html">project.pca()</a></code> <code><a href="project.pca.html">z2xyz.pca()</a></code> <code><a href="project.pca.html">xyz2z.pca()</a></code> </p>
        </td>
        <td><p>Project Data onto Principal Components</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="prune.cna.html">prune.cna()</a></code> </p>
        </td>
        <td><p>Prune A cna Network Object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pymol.html">pymol()</a></code> </p>
        </td>
        <td><p>Biomolecular Visualization with PyMOL</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read.all.html">read.all()</a></code> </p>
        </td>
        <td><p>Read Aligned Structure Data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read.cif.html">read.cif()</a></code> </p>
        </td>
        <td><p>Read mmCIF File</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read.crd.html">read.crd()</a></code> </p>
        </td>
        <td><p>Read Coordinate Data from Amber or Charmm</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read.crd.amber.html">read.crd(<i>&lt;amber&gt;</i>)</a></code> </p>
        </td>
        <td><p>Read AMBER Coordinate files</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read.crd.charmm.html">read.crd(<i>&lt;charmm&gt;</i>)</a></code> </p>
        </td>
        <td><p>Read CRD File</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read.dcd.html">read.dcd()</a></code> </p>
        </td>
        <td><p>Read CHARMM/X-PLOR/NAMD Binary DCD files</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read.fasta.html">read.fasta()</a></code> </p>
        </td>
        <td><p>Read FASTA formated Sequences</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read.fasta.pdb.html">read.fasta.pdb()</a></code> </p>
        </td>
        <td><p>Read Aligned Structure Data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read.mol2.html">read.mol2()</a></code> <code><a href="read.mol2.html">print(<i>&lt;mol2&gt;</i>)</a></code> </p>
        </td>
        <td><p>Read MOL2 File</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read.ncdf.html">read.ncdf()</a></code> </p>
        </td>
        <td><p>Read AMBER Binary netCDF files</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read.pdb.html">read.pdb()</a></code> <code><a href="read.pdb.html">read.pdb2()</a></code> <code><a href="read.pdb.html">print(<i>&lt;pdb&gt;</i>)</a></code> <code><a href="read.pdb.html">summary(<i>&lt;pdb&gt;</i>)</a></code> </p>
        </td>
        <td><p>Read PDB File</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read.pdcBD.html">read.pdcBD()</a></code> </p>
        </td>
        <td><p>Read PQR output from pdcBD File</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read.pqr.html">read.pqr()</a></code> </p>
        </td>
        <td><p>Read PQR File</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read.prmtop.html">read.prmtop()</a></code> <code><a href="read.prmtop.html">print(<i>&lt;prmtop&gt;</i>)</a></code> </p>
        </td>
        <td><p>Read AMBER Parameter/Topology files</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="rgyr.html">rgyr()</a></code> </p>
        </td>
        <td><p>Radius of Gyration</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="rle2.html">rle2()</a></code> <code><a href="rle2.html">print(<i>&lt;rle2&gt;</i>)</a></code> </p>
        </td>
        <td><p>Run Length Encoding with Indices</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="rmsd.html">rmsd()</a></code> </p>
        </td>
        <td><p>Root Mean Square Deviation</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="rmsf.html">rmsf()</a></code> </p>
        </td>
        <td><p>Atomic RMS Fluctuations</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="rmsip.html">rmsip()</a></code> </p>
        </td>
        <td><p>Root Mean Square Inner Product</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sdENM.html">sdENM</a></code> </p>
        </td>
        <td><p>Index for the sdENM ff</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="seq2aln.html">seq2aln()</a></code> </p>
        </td>
        <td><p>Add a Sequence to an Existing Alignmnet</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="seqaln.html">seqaln()</a></code> </p>
        </td>
        <td><p>Sequence Alignment with MUSCLE</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="seqaln.pair.html">seqaln.pair()</a></code> </p>
        </td>
        <td><p>Sequence Alignment of Identical Protein Sequences</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="seqbind.html">seqbind()</a></code> </p>
        </td>
        <td><p>Combine Sequences by Rows Without Recycling</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="seqidentity.html">seqidentity()</a></code> </p>
        </td>
        <td><p>Percent Identity</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="setup.ncore.html">setup.ncore()</a></code> </p>
        </td>
        <td><p>Setup for Running Bio3D Functions using Multiple CPU Cores</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sip.html">sip()</a></code> </p>
        </td>
        <td><p>Square Inner Product</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sse.bridges.html">sse.bridges()</a></code> </p>
        </td>
        <td><p>SSE Backbone Hydrogen Bonding</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="store.atom.html">store.atom()</a></code> </p>
        </td>
        <td><p>Store all-atom data from a PDB object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="struct.aln.html">struct.aln()</a></code> </p>
        </td>
        <td><p>Structure Alignment Of Two PDB Files</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="torsion.pdb.html">torsion.pdb()</a></code> </p>
        </td>
        <td><p>Calculate Mainchain and Sidechain Torsion/Dihedral Angles</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="torsion.xyz.html">torsion.xyz()</a></code> </p>
        </td>
        <td><p>Calculate Torsion/Dihedral Angles</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="trim.mol2.html">trim(<i>&lt;mol2&gt;</i>)</a></code> </p>
        </td>
        <td><p>Trim a MOL2 Object To A Subset of Atoms.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="trim.pdb.html">trim()</a></code> </p>
        </td>
        <td><p>Trim a PDB Object To A Subset of Atoms.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="trim.pdbs.html">trim(<i>&lt;pdbs&gt;</i>)</a></code> </p>
        </td>
        <td><p>Filter or Trim a PDBs Object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="trim.xyz.html">trim(<i>&lt;xyz&gt;</i>)</a></code> </p>
        </td>
        <td><p>Trim a XYZ Object of Cartesian Coordinates.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="unbound.html">unbound()</a></code> </p>
        </td>
        <td><p>Sequence Generation from a Bounds Vector</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="uniprot.html">uniprot()</a></code> </p>
        </td>
        <td><p>Fetch UniProt Entry Data.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="var.xyz.html">var.xyz()</a></code> <code><a href="var.xyz.html">var.pdbs()</a></code> </p>
        </td>
        <td><p>Pairwise Distance Variance in Cartesian Coordinates</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="vec2resno.html">vec2resno()</a></code> </p>
        </td>
        <td><p>Replicate Per-residue Vector Values</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="vmd.cna.html">vmd()</a></code> </p>
        </td>
        <td><p>View CNA Protein Structure Network Community Output in VMD</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="vmd_colors.html">vmd_colors()</a></code> </p>
        </td>
        <td><p>VMD Color Palette</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="wrap.tor.html">wrap.tor()</a></code> </p>
        </td>
        <td><p>Wrap Torsion Angle Data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="write.crd.html">write.crd()</a></code> </p>
        </td>
        <td><p>Write CRD File</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="write.fasta.html">write.fasta()</a></code> </p>
        </td>
        <td><p>Write FASTA Formated Sequences</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="write.mol2.html">write.mol2()</a></code> </p>
        </td>
        <td><p>Write MOL2 Format Coordinate File</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="write.ncdf.html">write.ncdf()</a></code> </p>
        </td>
        <td><p>Write AMBER Binary netCDF files</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="write.pdb.html">write.pdb()</a></code> </p>
        </td>
        <td><p>Write PDB Format Coordinate File</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="write.pir.html">write.pir()</a></code> </p>
        </td>
        <td><p>Write PIR Formated Sequences</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="write.pqr.html">write.pqr()</a></code> </p>
        </td>
        <td><p>Write PQR Format Coordinate File</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="tidyverse">
  <p>Rough draft website demo based on [tidytemplate]()</p>
</div>

<div class="author">
  <p>
    Developed by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.
    Site built by <a href="https://pkgdown.r-lib.org">pkgdown</a>.
  </p>
</div>

      </footer>
   </div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script>
<script>
  docsearch({
    
    
    apiKey: '1270c4079b26a138795263974bbf302d',
    indexName: 'tidyverse',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>



  </body>
</html>


