<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Ensemble Normal Mode Analysis — nma.pdbs • Bio3D</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>

<!-- Bootstrap -->
<link href="../tidyverse.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>
<link href="../tidyverse-2.css" rel="stylesheet">

<!--optional theme
<link href="../.css" rel="stylesheet">
-->


<!-- docsearch -->
<script src="../docsearch.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous" />
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>


<meta property="og:title" content="Ensemble Normal Mode Analysis — nma.pdbs" />
<meta property="og:description" content="Perform normal mode analysis (NMA) on an ensemble of aligned protein
  structures." />
<meta property="og:image" content="/logo.png" />
<meta name="twitter:card" content="summary" />


<meta name="robots" content="noindex">

<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../index.html">
        bio3d</a>
        <div class="info hidden-xs hidden-sm">
<!--          <span class="partof">part of the <a href="https://tidyverse.org">tidyverse</a></span>
-->
          <span class="version version-danger" data-toggle="tooltip" data-placement="bottom" title="In-development version">2.4-1.9000</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">INTRODUCTORY TUTORIALS</li>
    <li>
      <a href="../articles/index.html">All Vignettes (that compile)</a>
    </li>
    <li>
      <a href="../articles/bio3d_vignettes.html">Vignettes (old listing)</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">ADVANCED TUTORIALS</li>
    <li>
      <a href="../articles/b11.html">Title B1</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">WEBAPP TUTORIALS</li>
    <li>
      <a href="../articles/c11.html">Title C1</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Documentation
  </a>
</li>
        <li>
  <a href="../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    News
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket"></span>
     
  </a>
</li>
      </ul>
      
      <form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
        </div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Ensemble Normal Mode Analysis</h1>
    
    <div class="hidden name"><code>nma.pdbs.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Perform normal mode analysis (NMA) on an ensemble of aligned protein
  structures.</p>
    </div>

    <pre class="usage"><span class='co'># S3 method for pdbs</span>
<span class='fu'>nma</span>(<span class='kw'>pdbs</span>, fit = <span class='fl'>TRUE</span>, full = <span class='fl'>FALSE</span>, subspace = <span class='kw'>NULL</span>,
         rm.gaps = <span class='fl'>TRUE</span>, varweight=<span class='fl'>FALSE</span>, 
         outpath = <span class='kw'>NULL</span>, ncore = <span class='fl'>1</span>, progress = <span class='kw'>NULL</span>, <span class='kw'>...</span>)

<span class='co'># S3 method for enma</span>
<span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span>(<span class='kw'>x</span>, <span class='kw'>...</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>pdbs</th>
      <td><p>a numeric matrix of aligned C-alpha xyz Cartesian
    coordinates. For example an alignment data structure obtained with
    <code>read.fasta.pdb</code> or <code>pdbaln</code>.</p></td>
    </tr>
    <tr>
      <th>fit</th>
      <td><p>logical, if TRUE coordinate superposition is performed
    prior to normal mode calculations.</p></td>
    </tr>
    <tr>
      <th>full</th>
      <td><p>logical, if TRUE return the complete, full structure, 
    &#8216;nma&#8217; objects.</p></td>
    </tr>
    <tr>
      <th>subspace</th>
      <td><p>number of eigenvectors to store for further
    analysis.</p></td>
    </tr>
    <tr>
      <th>rm.gaps</th>
      <td><p>logical, if TRUE obtain the hessian matrices for only
    atoms in the aligned positions (non-gap positions in all aligned
    structures). Thus, gap positions are removed from output.</p></td>
    </tr>
    <tr>
      <th>varweight</th>
      <td><p>logical, if TRUE perform weighing of the pair force
    constants. Alternatively, provide a NxN matrix containing the
    weights. See function <code>var.xyz</code>.</p></td>
    </tr>
    <tr>
      <th>outpath</th>
      <td><p>character string specifing the output directory to
    which the PDB structures should be written.</p></td>
    </tr>
    <tr>
      <th>ncore </th>
      <td><p>number of CPU cores used to do the calculation.
    <code>ncore&gt;1</code> requires package &#8216;parallel&#8217; installed.</p></td>
    </tr>
    <tr>
      <th>x</th>
      <td><p>an <code>enma</code> object obtained from <code>nma.pdbs</code>.</p></td>
    </tr>
    <tr>
      <th>progress</th>
      <td><p>progress bar for use with shiny web app.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>additional arguments to <code>nma</code>, <code>aa2mass</code>, and
    <code><a href='https://rdrr.io/r/base/print.html'>print</a></code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function performs normal mode analysis (NMA) on a set of aligned
  protein structures obtained with function <code>read.fasta.pdb</code> or
  <code>pdbaln</code>. The main purpose is to provide aligned atomic
  fluctuations and mode vectors in an automated fashion.</p>
<p>The normal modes are calculated on the full structures as provided 
  by object &#8216;pdbs&#8217;. With the input argument &#8216;full=TRUE&#8217;
  the full &#8216;nma&#8217; objects are returned together with output
  &#8216;U.subs&#8217; providing the aligned mode vectors. When
  &#8216;rm.gaps=TRUE&#8217; the unaligned atoms are ommited from
  output. With default arguments &#8216;rmsip&#8217; provides RMSIP
  values for all pairwise structures.</p>  
<p>See examples for more details.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns an &#8216;enma&#8217; object with the following components:</p>
<dt>fluctuations </dt><dd><p>a numeric matrix containing aligned atomic
    fluctuations with one row per input structure.</p></dd>
  <dt>rmsip</dt><dd><p>a numeric matrix of pair wise RMSIP values (only the ten
    lowest frequency modes are included in the calculation).</p></dd>
  <dt>U.subspace </dt><dd><p>a three-dimensional array with aligned
    eigenvectors  (corresponding to the subspace defined by the first N
    non-trivial eigenvectors (&#8216;U&#8217;) of the &#8216;nma&#8217; object).</p></dd>
  <dt>L</dt><dd><p>numeric matrix containing the raw eigenvalues with one row
    per input structure.</p></dd>
  <dt>xyz </dt><dd><p>an object of class &#8216;xyz&#8217; containing the Cartesian
    coordinates in which the calculation was performed. Coordinates are
    superimposed to the first structure of the <code>pdbs</code> object when
    &#8216;fit=TRUE&#8217;.</p></dd>   
  <dt>full.nma </dt><dd><p>a list with a <code>nma</code> object for each input
    structure.</p></dd>

    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Skjaerven, L. et al. (2014) <em>BMC Bioinformatics</em> <b>15</b>, 399.
  Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695--2696.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Lars Skjaerven</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>For normal mode analysis on single structure PDB:
  <code>nma.pdb</code></p>  
<p>For the analysis of the resulting &#8216;eNMA&#8217; object:
  <code>mktrj.enma</code>, <code>dccm.enma</code>,
  <code>plot.enma</code>, <code>cov.enma</code>.</p>    
<p>Similarity measures:
  <code>sip</code>, <code>covsoverlap</code>,
  <code>bhattacharyya</code>, <code>rmsip</code>.</p>  
<p>Related functionality:
  <code>pdbaln</code>, <code>read.fasta.pdb</code>.</p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># \donttest{</span>
<span class='co'># Needs MUSCLE installed - testing excluded</span>

<span class='co'>if</span>(<span class='fu'>check.utility</span>(<span class='st'>"muscle"</span>)) {

<span class='co'>## Fetch PDB files and split to chain A only PDB files</span>
<span class='kw'>ids</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"1a70_A"</span>, <span class='st'>"1czp_A"</span>, <span class='st'>"1frd_A"</span>, <span class='st'>"1fxi_A"</span>, <span class='st'>"1iue_A"</span>, <span class='st'>"1pfd_A"</span>)
<span class='kw'>files</span> <span class='op'>&lt;-</span> <span class='fu'>get.pdb</span>(<span class='kw'>ids</span>, split = <span class='fl'>TRUE</span>, path = <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span>())

<span class='co'>## Sequence Alignement</span>
<span class='kw'>pdbs</span> <span class='op'>&lt;-</span> <span class='fu'>pdbaln</span>(<span class='kw'>files</span>, outfile = <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempfile</a></span>())

<span class='co'>## Normal mode analysis on aligned data</span>
<span class='kw'>modes</span> <span class='op'>&lt;-</span> <span class='fu'>nma</span>(<span class='kw'>pdbs</span>, rm.gaps=<span class='fl'>FALSE</span>)

<span class='co'>## Plot fluctuation data</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/plot.html'>plot</a></span>(<span class='kw'>modes</span>, pdbs=<span class='kw'>pdbs</span>)

<span class='co'>## Cluster on Fluctuation similariy</span>
<span class='kw'>sip</span> <span class='op'>&lt;-</span> <span class='fu'>sip</span>(<span class='kw'>modes</span>)
<span class='kw'>hc</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/hclust.html'>hclust</a></span>(<span class='fu'><a href='https://rdrr.io/r/stats/dist.html'>dist</a></span>(<span class='kw'>sip</span>))
<span class='kw'>col</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/cutree.html'>cutree</a></span>(<span class='kw'>hc</span>, k=<span class='fl'>3</span>)

<span class='co'>## Plot fluctuation data</span>
<span class='fu'><a href='https://rdrr.io/r/graphics/plot.html'>plot</a></span>(<span class='kw'>modes</span>, pdbs=<span class='kw'>pdbs</span>, col=<span class='kw'>col</span>)

<span class='co'>## Remove gaps from output</span>
<span class='kw'>modes</span> <span class='op'>&lt;-</span> <span class='fu'>nma</span>(<span class='kw'>pdbs</span>, rm.gaps=<span class='fl'>TRUE</span>)

<span class='co'>## RMSIP is pre-calculated</span>
<span class='fu'><a href='https://rdrr.io/r/stats/heatmap.html'>heatmap</a></span>(<span class='fl'>1</span><span class='op'>-</span><span class='kw'>modes</span><span class='op'>$</span><span class='kw'>rmsip</span>)

<span class='co'>## Bhattacharyya coefficient</span>
<span class='kw'>bc</span> <span class='op'>&lt;-</span> <span class='fu'>bhattacharyya</span>(<span class='kw'>modes</span>)
<span class='fu'><a href='https://rdrr.io/r/stats/heatmap.html'>heatmap</a></span>(<span class='fl'>1</span><span class='op'>-</span><span class='kw'>bc</span>)

}</div><div class='output co'>#&gt; <span class='warning'>Warning: /var/folders/xf/qznxnpf91vb1wm4xwgnbt0xr0000gn/T//RtmpyTUj9o/1a70.pdb  exists. Skipping download</span></div><div class='output co'>#&gt; <span class='warning'>Warning: /var/folders/xf/qznxnpf91vb1wm4xwgnbt0xr0000gn/T//RtmpyTUj9o/1czp.pdb  exists. Skipping download</span></div><div class='output co'>#&gt; <span class='warning'>Warning: /var/folders/xf/qznxnpf91vb1wm4xwgnbt0xr0000gn/T//RtmpyTUj9o/1frd.pdb  exists. Skipping download</span></div><div class='output co'>#&gt; <span class='warning'>Warning: /var/folders/xf/qznxnpf91vb1wm4xwgnbt0xr0000gn/T//RtmpyTUj9o/1fxi.pdb  exists. Skipping download</span></div><div class='output co'>#&gt; <span class='warning'>Warning: /var/folders/xf/qznxnpf91vb1wm4xwgnbt0xr0000gn/T//RtmpyTUj9o/1iue.pdb  exists. Skipping download</span></div><div class='output co'>#&gt; <span class='warning'>Warning: /var/folders/xf/qznxnpf91vb1wm4xwgnbt0xr0000gn/T//RtmpyTUj9o/1pfd.pdb  exists. Skipping download</span></div><div class='output co'>#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |============                                                          |  17%  |                                                                              |=======================                                               |  33%  |                                                                              |===================================                                   |  50%  |                                                                              |===============================================                       |  67%  |                                                                              |==========================================================            |  83%  |                                                                              |======================================================================| 100%
#&gt; Reading PDB files:
#&gt; /var/folders/xf/qznxnpf91vb1wm4xwgnbt0xr0000gn/T//RtmpyTUj9o/split_chain/1a70_A.pdb
#&gt; /var/folders/xf/qznxnpf91vb1wm4xwgnbt0xr0000gn/T//RtmpyTUj9o/split_chain/1czp_A.pdb
#&gt; /var/folders/xf/qznxnpf91vb1wm4xwgnbt0xr0000gn/T//RtmpyTUj9o/split_chain/1frd_A.pdb
#&gt; /var/folders/xf/qznxnpf91vb1wm4xwgnbt0xr0000gn/T//RtmpyTUj9o/split_chain/1fxi_A.pdb
#&gt; /var/folders/xf/qznxnpf91vb1wm4xwgnbt0xr0000gn/T//RtmpyTUj9o/split_chain/1iue_A.pdb
#&gt; /var/folders/xf/qznxnpf91vb1wm4xwgnbt0xr0000gn/T//RtmpyTUj9o/split_chain/1pfd_A.pdb
#&gt; .   PDB has ALT records, taking A only, rm.alt=TRUE
#&gt; .....
#&gt; 
#&gt; Extracting sequences
#&gt; 
#&gt; pdb/seq: 1   name: /var/folders/xf/qznxnpf91vb1wm4xwgnbt0xr0000gn/T//RtmpyTUj9o/split_chain/1a70_A.pdb 
#&gt; pdb/seq: 2   name: /var/folders/xf/qznxnpf91vb1wm4xwgnbt0xr0000gn/T//RtmpyTUj9o/split_chain/1czp_A.pdb 
#&gt;    PDB has ALT records, taking A only, rm.alt=TRUE
#&gt; pdb/seq: 3   name: /var/folders/xf/qznxnpf91vb1wm4xwgnbt0xr0000gn/T//RtmpyTUj9o/split_chain/1frd_A.pdb 
#&gt; pdb/seq: 4   name: /var/folders/xf/qznxnpf91vb1wm4xwgnbt0xr0000gn/T//RtmpyTUj9o/split_chain/1fxi_A.pdb 
#&gt; pdb/seq: 5   name: /var/folders/xf/qznxnpf91vb1wm4xwgnbt0xr0000gn/T//RtmpyTUj9o/split_chain/1iue_A.pdb 
#&gt; pdb/seq: 6   name: /var/folders/xf/qznxnpf91vb1wm4xwgnbt0xr0000gn/T//RtmpyTUj9o/split_chain/1pfd_A.pdb 
#&gt; 
#&gt; Details of Scheduled Calculation:
#&gt;   ... 6 input structures 
#&gt;   ... storing 282 eigenvectors for each structure 
#&gt;   ... dimension of x$U.subspace: ( 300x282x6 )
#&gt;   ... coordinate superposition prior to NM calculation 
#&gt;   ... estimated memory usage of final 'eNMA' object: 3.9 Mb 
#&gt; 
#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |============                                                          |  17%  |                                                                              |=======================                                               |  33%  |                                                                              |===================================                                   |  50%  |                                                                              |===============================================                       |  67%  |                                                                              |==========================================================            |  83%  |                                                                              |======================================================================| 100%</div><div class='output co'>#&gt; <span class='message'>Extracting SSE from pdbs$sse attribute</span></div><div class='output co'>#&gt; <span class='message'>Extracting SSE from pdbs$sse attribute</span></div><div class='img'><img src='nma.pdbs-1.png' alt='' width='700' height='433' /></div><div class='img'><img src='nma.pdbs-2.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; 
#&gt; Details of Scheduled Calculation:
#&gt;   ... 6 input structures 
#&gt;   ... storing 282 eigenvectors for each structure 
#&gt;   ... dimension of x$U.subspace: ( 288x282x6 )
#&gt;   ... coordinate superposition prior to NM calculation 
#&gt;   ... aligned eigenvectors (gap containing positions removed)  
#&gt;   ... estimated memory usage of final 'eNMA' object: 3.7 Mb 
#&gt; 
#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |============                                                          |  17%  |                                                                              |=======================                                               |  33%  |                                                                              |===================================                                   |  50%  |                                                                              |===============================================                       |  67%  |                                                                              |==========================================================            |  83%  |                                                                              |======================================================================| 100%</div><div class='img'><img src='nma.pdbs-3.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; Calculating covariance matrices
#&gt; Calculating pairwise bhattacharyya coefs</div><div class='img'><img src='nma.pdbs-4.png' alt='' width='700' height='433' /></div><div class='input'><span class='co'># }</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="tidyverse">
  <p>Authored by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="author">
  <p>
    Built with &#10084;&#65039; and <a href="https://pkgdown.r-lib.org">pkgdown</a> &copy; 2020.
  </p>
</div>

      </footer>
   </div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script>
<script>
  docsearch({
    
    
    apiKey: '1270c4079b26a138795263974bbf302d',
    indexName: 'tidyverse',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>



  </body>
</html>


