<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Read PQR File — read.pqr • Bio3D</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>

<!-- Bootstrap -->
<link href="../tidyverse.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>
<link href="../tidyverse-2.css" rel="stylesheet">

<!--optional theme
<link href="../.css" rel="stylesheet">
-->


<!-- docsearch -->
<script src="../docsearch.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous" />
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>


<meta property="og:title" content="Read PQR File — read.pqr" />
<meta property="og:description" content="Read a PQR coordinate file." />
<meta property="og:image" content="/logo.png" />
<meta name="twitter:card" content="summary" />


<meta name="robots" content="noindex">

<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../index.html">
        bio3d</a>
        <div class="info hidden-xs hidden-sm">
<!--          <span class="partof">part of the <a href="https://tidyverse.org">tidyverse</a></span>
-->
          <span class="version version-danger" data-toggle="tooltip" data-placement="bottom" title="In-development version">2.4-1.9000</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">INTRODUCTORY TUTORIALS</li>
    <li>
      <a href="../articles/index.html">All Vignettes (that compile)</a>
    </li>
    <li>
      <a href="../articles/bio3d_vignettes.html">Vignettes (old listing)</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">ADVANCED TUTORIALS</li>
    <li>
      <a href="../articles/b11.html">Title B1</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">WEBAPP TUTORIALS</li>
    <li>
      <a href="../articles/c11.html">Title C1</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Documentation
  </a>
</li>
        <li>
  <a href="../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    News
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket"></span>
     
  </a>
</li>
      </ul>
      
      <form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
        </div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Read PQR File</h1>
    
    <div class="hidden name"><code>read.pqr.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Read a PQR coordinate file.</p>
    </div>

    <pre class="usage"><span class='fu'>read.pqr</span>(<span class='kw'>file</span>, maxlines = <span class='op'>-</span><span class='fl'>1</span>, multi = <span class='fl'>FALSE</span>, rm.insert = <span class='fl'>FALSE</span>,
         rm.alt = <span class='fl'>TRUE</span>, verbose = <span class='fl'>TRUE</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>file</th>
      <td><p>the name of the PQR file to be read.</p></td>
    </tr>
    <tr>
      <th>maxlines</th>
      <td><p>the maximum number of lines to read before giving up
    with large files. By default if will read up to the end of input on
    the connection.</p></td>
    </tr>
    <tr>
      <th>multi</th>
      <td><p>logical, if TRUE multiple ATOM records are read for all
    models in multi-model files.</p></td>
    </tr>
    <tr>
      <th>rm.insert</th>
      <td><p>logical, if TRUE PDB insert records are ignored.</p></td>
    </tr>
    <tr>
      <th>rm.alt</th>
      <td><p>logical, if TRUE PDB alternate records are ignored.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>print details of the reading process.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>PQR file format is basically the same as PDB format except for the fields of
  <code>o</code> and <code>b</code>. In PDB, these two fields are filled with &#8216;Occupancy&#8217;
  and &#8216;B-factor&#8217; values, respectively, with each field 6-column long.
  In PQR, they are atomic &#8216;partial charge&#8217; and &#8216;radii&#8217;
  values, respectively, with each field 8-column long.</p>
<p><code>maxlines</code> may require increasing for some large multi-model files.
  The preferred means of reading such data is via binary DCD format 
  trajectory files (see the <code>read.dcd</code> function).</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns a list of class <code>"pdb"</code> with the following components:</p>
<dt>atom</dt><dd><p>a data.frame containing all atomic coordinate ATOM and
    HETATM data, with a row per ATOM/HETATM and a column per record type.
    See below for details of the record type naming convention (useful for
    accessing columns).</p></dd>
  <dt>helix </dt><dd><p>&#8216;start&#8217;, &#8216;end&#8217; and &#8216;length&#8217; of H
    type sse, where start and end are residue numbers &#8220;resno&#8221;.</p></dd>
  <dt>sheet </dt><dd><p>&#8216;start&#8217;, &#8216;end&#8217; and &#8216;length&#8217; of E
    type sse, where start and end are residue numbers &#8220;resno&#8221;.</p></dd>
  <dt>seqres </dt><dd><p>sequence from SEQRES field.</p></dd>
  <dt>xyz </dt><dd><p>a numeric matrix of class <code>"xyz"</code> containing the
    ATOM and HETATM coordinate data.</p></dd>
  <dt>calpha </dt><dd><p>logical vector with length equal to <code><a href='https://rdrr.io/r/base/nrow.html'>nrow(atom)</a></code>
    with TRUE values indicating a C-alpha &#8220;elety&#8221;.</p></dd>
  <dt>call </dt><dd><p>the matched call.</p></dd>

    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695--2696.</p>  
<p>For a description of PDB format (version3.3) see:<br />
  <a href='http://www.wwpdb.org/documentation/format33/v3.3.html'>http://www.wwpdb.org/documentation/format33/v3.3.html</a>.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Barry Grant</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>For both <code>atom</code> and <code>het</code> list components the column names can be
  used as a convenient means of data access, namely:
  Atom serial number &#8220;eleno&#8221; ,
  Atom type &#8220;elety&#8221;,
  Alternate location indicator &#8220;alt&#8221;,
  Residue name &#8220;resid&#8221;,
  Chain identifier &#8220;chain&#8221;,
  Residue sequence number &#8220;resno&#8221;,
  Code for insertion of residues &#8220;insert&#8221;,
  Orthogonal coordinates &#8220;x&#8221;,
  Orthogonal coordinates &#8220;y&#8221;,
  Orthogonal coordinates &#8220;z&#8221;,
  Occupancy &#8220;o&#8221;, and
  Temperature factor &#8220;b&#8221;.
  See examples for further details.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code>atom.select</code>, <code>write.pqr</code>, 
  <code>read.pdb</code>, <code>write.pdb</code>,
  <code>read.dcd</code>, <code>read.fasta.pdb</code>,
  <code>read.fasta</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># \donttest{</span>
<span class='co'># PDB server connection required - testing excluded</span>

<span class='co'># Read a PDB file and write it as a PQR file</span>
<span class='kw'>pdb</span> <span class='op'>&lt;-</span> <span class='fu'>read.pdb</span>( <span class='st'>"4q21"</span> )</div><div class='output co'>#&gt;   Note: Accessing on-line PDB file</div><div class='output co'>#&gt; <span class='warning'>Warning: /var/folders/xf/qznxnpf91vb1wm4xwgnbt0xr0000gn/T//RtmpyTUj9o/4q21.pdb  exists. Skipping download</span></div><div class='input'><span class='kw'>outfile</span> <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span>(), <span class='st'>"eg.pqr"</span>)
<span class='fu'>write.pqr</span>(pdb=<span class='kw'>pdb</span>, file = <span class='kw'>outfile</span>)

<span class='co'># Read the PQR file</span>
<span class='kw'>pqr</span> <span class='op'>&lt;-</span> <span class='fu'>read.pqr</span>(<span class='kw'>outfile</span>)

<span class='co'>## Print a brief composition summary</span>
<span class='kw'>pqr</span></div><div class='output co'>#&gt; 
#&gt;  Call:  read.pqr(file = outfile)
#&gt; 
#&gt;    Total Models#: 1
#&gt;      Total Atoms#: 1447,  XYZs#: 4341  Chains#: 1  (values: A)
#&gt; 
#&gt;      Protein Atoms#: 1340  (residues/Calpha atoms#: 168)
#&gt;      Nucleic acid Atoms#: 0  (residues/phosphate atoms#: 0)
#&gt; 
#&gt;      Non-protein/nucleic Atoms#: 107  (residues: 80)
#&gt;      Non-protein/nucleic resid values: [ GDP (1), HOH (78), MG (1) ]
#&gt; 
#&gt;    Protein sequence:
#&gt;       MTEYKLVVVGAGGVGKSALTIQLIQNHFVDEYDPTIEDSYRKQVVIDGETCLLDILDTAG
#&gt;       QEEYSAMRDQYMRTGEGFLCVFAINNTKSFEDIHQYREQIKRVKDSDDVPMVLVGNKCDL
#&gt;       AARTVESRQAQDLARSYGIPYIETSAKTRQGVEDAFYTLVREIRQHKL
#&gt; 
#&gt; + attr: atom, helix, sheet, seqres, xyz,
#&gt;         calpha, call</div><div class='input'>
<span class='co'>## Examine the storage format (or internal *str*ucture)</span>
<span class='fu'><a href='https://rdrr.io/r/utils/str.html'>str</a></span>(<span class='kw'>pqr</span>)</div><div class='output co'>#&gt; List of 7
#&gt;  $ atom  :'data.frame':	1447 obs. of  16 variables:
#&gt;   ..$ type  : chr [1:1447] "ATOM" "ATOM" "ATOM" "ATOM" ...
#&gt;   ..$ eleno : num [1:1447] 1 2 3 4 5 6 7 8 9 10 ...
#&gt;   ..$ elety : chr [1:1447] "N" "CA" "C" "O" ...
#&gt;   ..$ alt   : chr [1:1447] NA NA NA NA ...
#&gt;   ..$ resid : chr [1:1447] "MET" "MET" "MET" "MET" ...
#&gt;   ..$ chain : chr [1:1447] "A" "A" "A" "A" ...
#&gt;   ..$ resno : num [1:1447] 1 1 1 1 1 1 1 1 2 2 ...
#&gt;   ..$ insert: chr [1:1447] NA NA NA NA ...
#&gt;   ..$ x     : num [1:1447] 64.1 64 63.7 64.4 65.4 ...
#&gt;   ..$ y     : num [1:1447] 50.5 51.6 52.8 53.1 51.8 ...
#&gt;   ..$ z     : num [1:1447] 32.5 33.4 32.7 31.7 34.2 ...
#&gt;   ..$ o     : num [1:1447] 1 1 1 1 1 1 1 1 1 1 ...
#&gt;   ..$ b     : num [1:1447] 28.7 29.2 30.3 34.9 28.5 ...
#&gt;   ..$ segid : chr [1:1447] NA NA NA NA ...
#&gt;   ..$ elesy : chr [1:1447] NA NA NA NA ...
#&gt;   ..$ charge: chr [1:1447] NA NA NA NA ...
#&gt;  $ helix :List of 4
#&gt;   ..$ start: num(0) 
#&gt;   ..$ end  : num(0) 
#&gt;   ..$ chain: chr(0) 
#&gt;   ..$ type : chr(0) 
#&gt;  $ sheet :List of 4
#&gt;   ..$ start: num(0) 
#&gt;   ..$ end  : num(0) 
#&gt;   ..$ chain: chr(0) 
#&gt;   ..$ sense: chr(0) 
#&gt;  $ seqres: NULL
#&gt;  $ xyz   : 'xyz' num [1, 1:4341] 64.1 50.5 32.5 64 51.6 ...
#&gt;  $ calpha: logi [1:1447] FALSE TRUE FALSE FALSE FALSE FALSE ...
#&gt;  $ call  : language read.pqr(file = outfile)
#&gt;  - attr(*, "class")= chr [1:2] "pdb" "sse"</div><div class='input'>
<span class='co'>## Print data for the first four atom</span>
<span class='kw'>pqr</span><span class='op'>$</span><span class='kw'>atom</span>[<span class='fl'>1</span><span class='op'>:</span><span class='fl'>4</span>,]</div><div class='output co'>#&gt;   type eleno elety  alt resid chain resno insert      x      y      z o     b
#&gt; 1 ATOM     1     N &lt;NA&gt;   MET     A     1   &lt;NA&gt; 64.080 50.529 32.509 1 28.66
#&gt; 2 ATOM     2    CA &lt;NA&gt;   MET     A     1   &lt;NA&gt; 64.044 51.615 33.423 1 29.19
#&gt; 3 ATOM     3     C &lt;NA&gt;   MET     A     1   &lt;NA&gt; 63.722 52.849 32.671 1 30.27
#&gt; 4 ATOM     4     O &lt;NA&gt;   MET     A     1   &lt;NA&gt; 64.359 53.119 31.662 1 34.93
#&gt;   segid elesy charge
#&gt; 1  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 2  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 3  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 4  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;</div><div class='input'>
<span class='co'>## Print some coordinate data</span>
<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span>(<span class='kw'>pqr</span><span class='op'>$</span><span class='kw'>atom</span>[, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"x"</span>,<span class='st'>"y"</span>,<span class='st'>"z"</span>)])</div><div class='output co'>#&gt;        x      y      z
#&gt; 1 64.080 50.529 32.509
#&gt; 2 64.044 51.615 33.423
#&gt; 3 63.722 52.849 32.671
#&gt; 4 64.359 53.119 31.662
#&gt; 5 65.373 51.805 34.158
#&gt; 6 65.122 52.780 35.269</div><div class='input'>
<span class='co'>## Print C-alpha coordinates (can also use 'atom.select' function)</span>
<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span>(<span class='kw'>pqr</span><span class='op'>$</span><span class='kw'>atom</span>[<span class='kw'>pqr</span><span class='op'>$</span><span class='kw'>calpha</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"resid"</span>,<span class='st'>"elety"</span>,<span class='st'>"x"</span>,<span class='st'>"y"</span>,<span class='st'>"z"</span>)])</div><div class='output co'>#&gt;    resid elety      x      y      z
#&gt; 2    MET    CA 64.044 51.615 33.423
#&gt; 10   THR    CA 62.439 54.794 32.359
#&gt; 17   GLU    CA 63.968 58.232 32.801
#&gt; 26   TYR    CA 61.817 61.333 33.161
#&gt; 38   LYS    CA 63.343 64.814 33.163
#&gt; 47   LEU    CA 61.321 67.068 35.557</div><div class='input'><span class='kw'>inds</span> <span class='op'>&lt;-</span> <span class='fu'>atom.select</span>(<span class='kw'>pqr</span>, elety=<span class='st'>"CA"</span>)
<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span>( <span class='kw'>pqr</span><span class='op'>$</span><span class='kw'>atom</span>[<span class='kw'>inds</span><span class='op'>$</span><span class='kw'>atom</span>, ] )</div><div class='output co'>#&gt;    type eleno elety  alt resid chain resno insert      x      y      z o     b
#&gt; 2  ATOM     2    CA &lt;NA&gt;   MET     A     1   &lt;NA&gt; 64.044 51.615 33.423 1 29.19
#&gt; 10 ATOM    10    CA &lt;NA&gt;   THR     A     2   &lt;NA&gt; 62.439 54.794 32.359 1 28.10
#&gt; 17 ATOM    17    CA &lt;NA&gt;   GLU     A     3   &lt;NA&gt; 63.968 58.232 32.801 1 30.95
#&gt; 26 ATOM    26    CA &lt;NA&gt;   TYR     A     4   &lt;NA&gt; 61.817 61.333 33.161 1 23.42
#&gt; 38 ATOM    38    CA &lt;NA&gt;   LYS     A     5   &lt;NA&gt; 63.343 64.814 33.163 1 21.34
#&gt; 47 ATOM    47    CA &lt;NA&gt;   LEU     A     6   &lt;NA&gt; 61.321 67.068 35.557 1 18.99
#&gt;    segid elesy charge
#&gt; 2   &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 10  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 17  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 26  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 38  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 47  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;</div><div class='input'>
<span class='co'>## The atom.select() function returns 'indices' (row numbers)</span>
<span class='co'>## that can be used for accessing subsets of PDB objects, e.g.</span>
<span class='kw'>inds</span> <span class='op'>&lt;-</span> <span class='fu'>atom.select</span>(<span class='kw'>pqr</span>,<span class='st'>"ligand"</span>)
<span class='kw'>pqr</span><span class='op'>$</span><span class='kw'>atom</span>[<span class='kw'>inds</span><span class='op'>$</span><span class='kw'>atom</span>,]</div><div class='output co'>#&gt;        type eleno elety  alt resid chain resno insert      x      y      z o
#&gt; 1341 HETATM  1342    MG &lt;NA&gt;    MG     A   273   &lt;NA&gt; 65.614 76.977 46.715 1
#&gt; 1342 HETATM  1343    PB &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 62.667 77.781 47.505 1
#&gt; 1343 HETATM  1344   O1B &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 61.587 77.413 46.626 1
#&gt; 1344 HETATM  1345   O2B &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 63.294 79.098 47.336 1
#&gt; 1345 HETATM  1346   O3B &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 63.804 76.731 47.410 1
#&gt; 1346 HETATM  1347   O3A &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 62.281 77.644 49.012 1
#&gt; 1347 HETATM  1348    PA &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 62.781 76.563 50.116 1
#&gt; 1348 HETATM  1349   O1A &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 64.200 76.858 50.463 1
#&gt; 1349 HETATM  1350   O2A &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 62.459 75.187 49.671 1
#&gt; 1350 HETATM  1351   O5' &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 61.927 76.929 51.222 1
#&gt; 1351 HETATM  1352   C5' &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 61.690 78.290 51.572 1
#&gt; 1352 HETATM  1353   C4' &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 61.260 78.393 53.002 1
#&gt; 1353 HETATM  1354   O4' &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 59.989 77.748 53.185 1
#&gt; 1354 HETATM  1355   C3' &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 62.181 77.747 54.015 1
#&gt; 1355 HETATM  1356   O3' &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 62.291 78.499 55.179 1
#&gt; 1356 HETATM  1357   C2' &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 61.548 76.420 54.295 1
#&gt; 1357 HETATM  1358   O2' &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 61.846 76.085 55.643 1
#&gt; 1358 HETATM  1359   C1' &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 60.078 76.792 54.224 1
#&gt; 1359 HETATM  1360    N9 &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 59.258 75.630 53.844 1
#&gt; 1360 HETATM  1361    C8 &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 59.255 75.041 52.612 1
#&gt; 1361 HETATM  1362    N7 &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 58.334 74.158 52.460 1
#&gt; 1362 HETATM  1363    C5 &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 57.550 74.278 53.590 1
#&gt; 1363 HETATM  1364    C6 &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 56.499 73.638 53.877 1
#&gt; 1364 HETATM  1365    O6 &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 56.005 72.734 53.233 1
#&gt; 1365 HETATM  1366    N1 &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 55.907 74.049 55.053 1
#&gt; 1366 HETATM  1367    C2 &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 56.436 74.939 55.901 1
#&gt; 1367 HETATM  1368    N2 &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 55.747 75.129 57.025 1
#&gt; 1368 HETATM  1369    N3 &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 57.574 75.569 55.662 1
#&gt; 1369 HETATM  1370    C4 &lt;NA&gt;   GDP     A   274   &lt;NA&gt; 58.110 75.138 54.493 1
#&gt;          b segid elesy charge
#&gt; 1341 20.79  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1342 31.08  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1343 30.69  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1344 24.13  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1345 29.87  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1346 29.39  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1347 32.94  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1348 38.15  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1349 39.73  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1350 37.11  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1351 37.93  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1352 36.58  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1353 40.35  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1354 36.25  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1355 38.09  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1356 39.40  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1357 43.75  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1358 40.06  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1359 39.43  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1360 38.59  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1361 38.35  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1362 35.85  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1363 37.59  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1364 39.03  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1365 38.56  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1366 36.65  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1367 34.76  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1368 37.24  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;
#&gt; 1369 37.60  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt;</div><div class='input'><span class='kw'>pqr</span><span class='op'>$</span><span class='kw'>xyz</span>[<span class='kw'>inds</span><span class='op'>$</span><span class='kw'>xyz</span>]</div><div class='output co'>#&gt;  [1] 65.614 76.977 46.715 62.667 77.781 47.505 61.587 77.413 46.626 63.294
#&gt; [11] 79.098 47.336 63.804 76.731 47.410 62.281 77.644 49.012 62.781 76.563
#&gt; [21] 50.116 64.200 76.858 50.463 62.459 75.187 49.671 61.927 76.929 51.222
#&gt; [31] 61.690 78.290 51.572 61.260 78.393 53.002 59.989 77.748 53.185 62.181
#&gt; [41] 77.747 54.015 62.291 78.499 55.179 61.548 76.420 54.295 61.846 76.085
#&gt; [51] 55.643 60.078 76.792 54.224 59.258 75.630 53.844 59.255 75.041 52.612
#&gt; [61] 58.334 74.158 52.460 57.550 74.278 53.590 56.499 73.638 53.877 56.005
#&gt; [71] 72.734 53.233 55.907 74.049 55.053 56.436 74.939 55.901 55.747 75.129
#&gt; [81] 57.025 57.574 75.569 55.662 58.110 75.138 54.493</div><div class='input'>
<span class='co'>## See the help page for atom.select() function for more details.</span>

<span class='co'># }</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="tidyverse">
  <p>Authored by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="author">
  <p>
    Built with &#10084;&#65039; and <a href="https://pkgdown.r-lib.org">pkgdown</a> &copy; 2020.
  </p>
</div>

      </footer>
   </div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script>
<script>
  docsearch({
    
    
    apiKey: '1270c4079b26a138795263974bbf302d',
    indexName: 'tidyverse',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>



  </body>
</html>


