<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Sequence Alignment with MUSCLE — seqaln • Bio3D</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>

<!-- Bootstrap -->
<link href="../tidyverse.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>
<link href="../tidyverse-2.css" rel="stylesheet">

<!--optional theme-->
<link href="../.css" rel="stylesheet">



<!-- docsearch -->
<script src="../docsearch.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous" />
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>


<meta property="og:title" content="Sequence Alignment with MUSCLE — seqaln" />
<meta property="og:description" content="Create multiple alignments of amino acid or nucleotide sequences
  according to the method of Edgar." />
<meta property="og:image" content="/logo.png" />
<meta name="twitter:card" content="summary" />


<meta name="robots" content="noindex">

<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->

<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script>

  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../index.html">
        bio3d</a>
        <div class="info hidden-xs hidden-sm">
<!--          <span class="partof">part of the <a href="https://tidyverse.org">tidyverse</a></span>
-->
          <span class="version version-danger" data-toggle="tooltip" data-placement="bottom" title="In-development version">2.4-1.9000</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">INTRODUCTORY TUTORIALS</li>
    <li>
      <a href="../articles/bio3d_vignettes.html">Vignettes (old listing)</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">ADVANCED TUTORIALS</li>
    <li>
      <a href="../articles/b11.html">Title B1</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">WEBAPP TUTORIALS</li>
    <li>
      <a href="../articles/c11.html">Title C1</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Documentation
  </a>
</li>
        <li>
  <a href="../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    News
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket"></span>
     
  </a>
</li>
      </ul>
      
      <form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
        </div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Sequence Alignment with MUSCLE</h1>
    
    <div class="hidden name"><code>seqaln.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Create multiple alignments of amino acid or nucleotide sequences
  according to the method of Edgar.</p>
    </div>

    <pre class="usage"><span class='fu'>seqaln</span>(<span class='kw'>aln</span>, id=<span class='kw'>NULL</span>, profile=<span class='kw'>NULL</span>, exefile=<span class='st'>"muscle"</span>, outfile=<span class='st'>"aln.fa"</span>, 
       protein=<span class='fl'>TRUE</span>, seqgroup=<span class='fl'>FALSE</span>, refine=<span class='fl'>FALSE</span>, extra.args=<span class='st'>""</span>,
       verbose=<span class='fl'>FALSE</span>, web.args = <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span>(), <span class='kw'>...</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>aln</th>
      <td><p>a sequence character matrix, as obtained from
    <code>seqbind</code>, or an alignment list object as obtained from
      <code>read.fasta</code>.</p></td>
    </tr>
    <tr>
      <th>id</th>
      <td><p>a vector of sequence names to serve as sequence
    identifers.</p></td>
    </tr>
    <tr>
      <th>profile</th>
      <td><p>a profile alignment of class &#8216;fasta&#8217;
    (e.g. obtained from <code>read.fasta</code>). The alignment
    <code>aln</code> will be added to the profile.</p></td>
    </tr>
    <tr>
      <th>exefile</th>
      <td><p>file path to the &#8216;MUSCLE&#8217; program on your system (i.e.
    how is &#8216;MUSCLE&#8217; invoked). Alternatively, &#8216;CLUSTALO&#8217; can
    be used. Also supported is using the &#8216;msa&#8217; package from Bioconductor
    (need to install packages using <code><a href='https://rdrr.io/pkg/BiocManager/man/install.html'>BiocManager::install()</a></code>). To do so, simply
    set <code>exefile="msa"</code>.</p></td>
    </tr>
    <tr>
      <th>outfile</th>
      <td><p>name of &#8216;FASTA&#8217; output file to which alignment
    should be written.</p></td>
    </tr>
    <tr>
      <th>protein</th>
      <td><p>logical, if TRUE the input sequences are assumed to be
    protein not DNA or RNA.</p></td>
    </tr>
    <tr>
      <th>seqgroup</th>
      <td><p>logical, if TRUE similar sequences are grouped
    together in the output.</p></td>
    </tr>
    <tr>
      <th>refine</th>
      <td><p>logical, if TRUE the input sequences are assumed to already
  be aligned, and only tree dependent refinement is performed.</p></td>
    </tr>
    <tr>
      <th>extra.args</th>
      <td><p>a single character string containing extra command
    line arguments for the alignment program.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>logical, if TRUE &#8216;MUSCLE&#8217; warning and error
    messages are printed.</p></td>
    </tr>
    <tr>
      <th>web.args</th>
      <td><p>a &#8216;list&#8217; object containing arguments to perform
    online sequence alignment using EMBL-EBI Web Services. 
    See below for details.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>additional arguments passed to the function
    <code><a href='https://rdrr.io/pkg/msa/man/msaMuscle.html'>msa::msaMuscle()</a></code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Sequence alignment attempts to arrange the sequences of protein,
  DNA or RNA, to highlight regions of shared similarity that may
  reflect functional, structural, and/or evolutionary relationships
  between the sequences.</p>
<p>Aligned sequences are represented as rows within a matrix. Gaps
  (&#8216;-&#8217;) are inserted between the aminoacids or nucleotides so that
  equivalent characters are positioned in the same column.</p>
<p>This function calls the &#8216;MUSCLE&#8217; program to perform a multiple 
  sequence alignment, which must be installed on your system and in the search 
  path for executables. If local &#8216;MUSCLE&#8217; can not be found, alignment
  can still be performed via online web services (see below) with limited
  features.</p>
<p>If you have a large number of input sequences (a few thousand), or they are
  very long, the default settings may be too slow for practical
  use. A good compromise between speed and accuracy is to run just the
  first two iterations of the &#8216;MUSCLE&#8217; algorithm by setting the
  <code>extra.args</code> argument to &#8220;-maxiters 2&#8221;.</p>
<p>You can set &#8216;MUSCLE&#8217; to improve an existing alignment by setting
  <code>refine</code> to TRUE.</p>
<p>To inspect the sequence clustering used by &#8216;MUSCLE&#8217; to produce
  alignments, include &#8220;-tree2 tree.out&#8221; in the <code>extra.args</code>
  argument. You can then load the &#8220;tree.out&#8221; file with the
  &#8216;read.tree&#8217; function from the &#8216;ape&#8217; package.</p>
<p>&#8216;CLUSTALO&#8217; can be used as an alternative to &#8216;MUSCLE&#8217; by
  specifiying <code>exefile='clustalo'</code>. This might be useful e.g. when
  adding several sequences to a profile alignment.</p>
<p>If local &#8216;MUSCLE&#8217; or &#8216;CLUSTALO&#8217; program is unavailable, the alignment
  can be performed via the &#8216;msa&#8217; package from the Bioconductor repository.
  To do so, set <code>exefile="msa"</code>. Note that both &#8216;msa&#8217; and 
  &#8216;Biostrings&#8217; packages need to be installed properly using <code><a href='https://rdrr.io/pkg/BiocManager/man/install.html'>BiocManager::install()</a></code>.</p>  
<p>If the access to any method metioned above fails, 
  the function will attempt to perform alignment via the EMBL-EBI Web Services 
  (See <a href='http://www.ebi.ac.uk'>http://www.ebi.ac.uk</a>). In this case, the argument <code>web.args</code> 
  cannot be empty and must contain at least user's E-Mail address. 
  Note that as stated by EBI, a fake email address may result 
  in your jobs being killed and your IP, organisation or entire domain being 
  black-listed (See <a href='http://www.ebi.ac.uk/Tools/webservices/help/faq'>http://www.ebi.ac.uk/Tools/webservices/help/faq</a>). 
  Possible parameters to be passed via <code>web.args</code> include:</p><dl>
    <dt>email</dt><dd><p>a string containing a valid E-Mail address. Required.</p></dd>
    <dt>title</dt><dd><p>a string for the title of the job to be submitted to
      the remote server. Optional.</p></dd>
    <dt>timeout</dt><dd><p>integer specifying the number of seconds to wait for
      the response of the server before a time out occurs. Default: 90.</p></dd>
  
</dl><p>An example of usage is <code>web.args=list(email='user_id@email.provider')</code>.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns a list of class <code>"fasta"</code> with the following components:</p>
<dt>ali</dt><dd><p>an alignment character matrix with a row per sequence and
    a column per equivalent aminoacid/nucleotide.</p></dd>
  <dt>id</dt><dd><p>sequence names as identifers.</p></dd>
  <dt>call</dt><dd><p>the matched call.</p></dd>

    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695--2696.</p>  
<p>&#8216;MUSCLE&#8217; is the work of Edgar:
  Edgar (2004) <em>Nuc. Acid. Res.</em> <b>32</b>, 1792--1797.</p>  
<p>Full details of the &#8216;MUSCLE&#8217; algorithm, along with download and
  installation instructions can be obtained from:<br />
  <a href='http://www.drive5.com/muscle'>http://www.drive5.com/muscle</a>.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Barry Grant</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>A system call is made to the &#8216;MUSCLE&#8217; program, which must be
  installed on your system and in the search path for executables.
  See <a href='http://thegrantlab.org/bio3d/tutorials/installing-bio3d'>http://thegrantlab.org/bio3d/tutorials/installing-bio3d</a> for 
  instructions of how to install this program.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code>read.fasta</code>, <code>read.fasta.pdb</code>,
  <code>get.seq</code>, <code>seqbind</code>, 
  <code>pdbaln</code>, <code>plot.fasta</code>,
  <code>blast.pdb</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>if</span> (<span class='fl'>FALSE</span>) {
<span class='co'>##-- Basic sequence alignemnt</span>
<span class='kw'>seqs</span> <span class='op'>&lt;-</span> <span class='fu'>get.seq</span>(<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"4q21_A"</span>, <span class='st'>"1ftn_A"</span>))
<span class='kw'>aln</span> <span class='op'>&lt;-</span> <span class='fu'>seqaln</span>(<span class='kw'>seqs</span>)

<span class='co'>##-- add a sequence to the (profile) alignment</span>
<span class='kw'>seq</span> <span class='op'>&lt;-</span> <span class='fu'>get.seq</span>(<span class='st'>"1tnd_A"</span>)
<span class='kw'>aln</span> <span class='op'>&lt;-</span> <span class='fu'>seqaln</span>(<span class='kw'>seq</span>, profile=<span class='kw'>aln</span>)


<span class='co'>##-- Read a folder/directory of PDB files</span>
<span class='co'>#pdb.path &lt;- "my_dir_of_pdbs"</span>
<span class='co'>#files  &lt;- list.files(path=pdb.path ,</span>
<span class='co'>#                     pattern=".pdb",</span>
<span class='co'>#                     full.names=TRUE)</span>

<span class='co'>##-- Use online files</span>
<span class='kw'>files</span> <span class='op'>&lt;-</span> <span class='fu'>get.pdb</span>(<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"4q21"</span>,<span class='st'>"1ftn"</span>), URLonly=<span class='fl'>TRUE</span>)

<span class='co'>##-- Extract and store sequences</span>
<span class='kw'>raw</span> <span class='op'>&lt;-</span> <span class='kw'>NULL</span>
<span class='co'>for</span>(<span class='kw'>i</span> <span class='co'>in</span> <span class='fl'>1</span><span class='op'>:</span><span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='kw'>files</span>)) {
  <span class='kw'>pdb</span> <span class='op'>&lt;-</span> <span class='fu'>read.pdb</span>(<span class='kw'>files</span>[<span class='kw'>i</span>])
  <span class='kw'>raw</span> <span class='op'>&lt;-</span> <span class='fu'>seqbind</span>(<span class='kw'>raw</span>, <span class='fu'>pdbseq</span>(<span class='kw'>pdb</span>) )
}

<span class='co'>##-- Align these sequences</span>
<span class='kw'>aln</span> <span class='op'>&lt;-</span> <span class='fu'>seqaln</span>(<span class='kw'>raw</span>, id=<span class='kw'>files</span>, outfile=<span class='st'>"seqaln.fa"</span>)

<span class='co'>##-- Read Aligned PDBs storing coordinate data</span>
<span class='kw'>pdbs</span> <span class='op'>&lt;-</span> <span class='fu'>read.fasta.pdb</span>(<span class='kw'>aln</span>) 

<span class='co'>## Sequence identity</span>
<span class='fu'>seqidentity</span>(<span class='kw'>aln</span>)

<span class='co'>## Note that all the above can be done with the pdbaln() function:</span>
<span class='co'>#pdbs &lt;- pdbaln(files)</span>


<span class='co'>##-- For identical sequences with masking use a custom matrix</span>
<span class='kw'>aa</span> <span class='op'>&lt;-</span> <span class='fu'>seqbind</span>(<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"X"</span>,<span class='st'>"C"</span>,<span class='st'>"X"</span>,<span class='st'>"X"</span>,<span class='st'>"A"</span>,<span class='st'>"G"</span>,<span class='st'>"K"</span>),
              <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"C"</span>,<span class='st'>"-"</span>,<span class='st'>"A"</span>,<span class='st'>"X"</span>,<span class='st'>"G"</span>,<span class='st'>"X"</span>,<span class='st'>"X"</span>,<span class='st'>"K"</span>))

<span class='kw'>aln</span> <span class='op'>&lt;-</span> <span class='fu'>seqaln</span>(aln=<span class='kw'>aln</span>, id=<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"a"</span>,<span class='st'>"b"</span>), outfile=<span class='st'>"temp.fas"</span>, protein=<span class='fl'>TRUE</span>,
               extra.args= <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste</a></span>(<span class='st'>"-matrix"</span>,
                <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"matrices/custom.mat"</span>, package=<span class='st'>"bio3d"</span>),
                <span class='st'>"-gapopen -3.0 "</span>,
                <span class='st'>"-gapextend -0.5"</span>,
                <span class='st'>"-center 0.0"</span>) )
}</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="tidyverse">
  <p>Rough draft website demo based on [tidytemplate]()</p>
</div>

<div class="author">
  <p>
    Developed by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.
    Site built by <a href="https://pkgdown.r-lib.org">pkgdown</a>.
  </p>
</div>

      </footer>
   </div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script>
<script>
  docsearch({
    
    
    apiKey: '1270c4079b26a138795263974bbf302d',
    indexName: 'tidyverse',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>



  </body>
</html>


