<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Ensemble NMA of *E.coli* DHFR structures • Bio3D</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../../../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../../../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../../../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../../../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../../../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../../../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="../../../tidyverse.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../../bootstrap-toc.css">
<script src="../../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../../pkgdown.css" rel="stylesheet">
<script src="../../../pkgdown.js"></script><link href="../../../tidyverse-2.css" rel="stylesheet">
<!--optional theme
<link href="../../../.css" rel="stylesheet">
--><!-- docsearch --><script src="../../../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../../../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Ensemble NMA of *E.coli* DHFR structures">
<meta property="og:description" content="bio3d">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../../../index.html">
        bio3d</a>
        <div class="info hidden-xs hidden-sm">
<!--          <span class="partof">part of the <a href="https://tidyverse.org">tidyverse</a></span>
-->
          <span class="version version-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4-1.9000</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">INTRODUCTORY TUTORIALS</li>
    <li>
      <a href="../../../articles/online/install_vignette/Bio3D_install.html">Installing Bio3D</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/user-guide">New user guide</a>
    </li>
    <li>
      <a href="../../../articles/online/intro_vignette/Bio3D_introduction.html">Getting started with Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/pdb_vignette/Bio3D_pdb.html">PDB structure manipulation and analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/pca_vignette/Bio3D_pca.html">Comparative sequence and structure analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/nma_vignette/Bio3D_nma.html">Enhanced Normal Modes Analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/traj_vignette/Bio3D_md.html">Beginning trajectory analysis with Bio3D </a>
    </li>
    <li class="divider">
    <li class="dropdown-header">ADVANCED TUTORIALS</li>
    <li>
      <a href="../../../articles/online/enma_vignettes/Bio3D_nma-dhfr-partI.html">Ensemble NMA of E.coli DHFR structures</a>
    </li>
    <li>
      <a href="../../../articles/online/enma_vignettes/Bio3D_nma-dhfr-partII.html">Ensemble NMA across multiple species of DHFR</a>
    </li>
    <li>
      <a href="../../../articles/online/cna_vignette/cna_vignette.spin.html">Protein structure network analysis with Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/eddm_vignette/Bio3D_eddm.spin.html">Ensemble Difference Distance Matrix (eDDM) analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">WEBAPP TUTORIALS</li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/download/webapp-tutorials?download=22:bio3d-webapp-pca">Online protein structure analysis with Bio3D-Web (PDF)</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/webapps">Online Normal Mode Analysis with Bio3D WebApps (To Update)</a>
    </li>
    <li class="divider">
    <li>
      <a href="../../../articles/index.html">All vignettes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Documentation
  </a>
</li>
        <li>
  <a href="../../../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../../../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    News
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket"></span>
     
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Bio3D_nma-dhfr-partI_files/header-attrs-2.3.2/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Ensemble NMA of <em>E.coli</em> DHFR structures</h1>
                        <h4 class="author">Lars Skjaerven, Xin-Qiu Yao, Guido Scarabelli &amp; Barry J. Grant</h4>
            
      
      
      <div class="hidden name"><code>Bio3D_nma-dhfr-partI.Rmd</code></div>

    </div>

    
    
<div id="background" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#background" class="anchor"></a>Background</h1>
<p>Bio3D<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> is an R package that provides interactive tools for structural bioinformatics. The primary focus of Bio3D is the analysis of bimolecular structure, sequence and simulation data <span class="citation">(Grant et al. 2006)</span>.</p>
<p>Normal mode analysis (NMA) is one of the major simulation techniques used to probe large-scale motions in biomolecules. Typical application is for the prediction of functional motions in proteins. Version 2.0 of the Bio3D package now includes extensive NMA facilities <span class="citation">(Skjaerven et al. 2015)</span>. These include a unique collection of multiple elastic network model force-fields, automated ensemble analysis methods, and variance weighted NMA (see also the <a href="http://thegrantlab.org/bio3d/tutorials"><strong>NMA Vignette</strong></a>). Here we provide an in-depth demonstration of ensemble NMA with working code that comprise complete executable examples<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>.</p>
<div id="requirements" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#requirements" class="anchor"></a>Requirements</h2>
<p>Detailed instructions for obtaining and installing the Bio3D package on various platforms can be found in the <a href="http://thegrantlab.org/bio3d/tutorials"><strong>Installing Bio3D Vignette</strong></a> available both on-line and from within the Bio3D package. In addition to Bio3D the <em>MUSCLE</em> multiple sequence alignment program (available from the <a href="http://www.drive5.com/muscle/">muscle home page</a> must be installed on your system and in the search path for executables. Please see the installation vignette for further details.</p>
</div>
<div id="about-this-document" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#about-this-document" class="anchor"></a>About this document</h2>
<p>This vignette was generated using <strong>Bio3D version 2.4.1.9000</strong>.</p>
</div>
</div>
<div id="part-i-ensemble-nma-of-e-coli-dhfr-structures" class="section level1">
<h1 class="hasAnchor">
<a href="#part-i-ensemble-nma-of-e-coli-dhfr-structures" class="anchor"></a>Part I: Ensemble NMA of <em>E.coli</em> DHFR structures</h1>
<p>In this vignette we perform <em>ensemble NMA</em> on the complete collection of <em>E.coli</em> Dihydrofolate reductase (DHFR) structures in the protein data-bank (PDB). Starting from only one PDB identifier (PDB ID 1rx2) we show how to search the PDB for related structures using BLAST, fetch and align the structures, and finally calculate the normal modes of each individual structure in order to probe for potential differences in structural flexibility.</p>
<div id="search-and-retrieve-dhfr-structures" class="section level2">
<h2 class="hasAnchor">
<a href="#search-and-retrieve-dhfr-structures" class="anchor"></a>Search and retrieve DHFR structures</h2>
<p>Below we perform a blast search of the PDB database to identify related structures to our query <em>E.coli</em> DHFR sequence. In this particular example we use function <strong>get.seq()</strong> to fetch the query sequence for chain A of the PDB ID 1RX2 and use this as input to <strong>blast.pdb()</strong>. Note that <strong>get.seq()</strong> would also allow the corresponding UniProt identifier.</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="http://thegrantlab.org/bio3d">bio3d</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">bio3d.nma</span>)
</pre></div>
<pre><code>## Registered S3 methods overwritten by 'bio3d.nma':
##   method     from 
##   dccm.egnm  bio3d
##   dccm.enma  bio3d
##   dccm.gnm   bio3d
##   dccm.nma   bio3d
##   mktrj.enma bio3d
##   mktrj.nma  bio3d
##   plot.enma  bio3d
##   plot.nma   bio3d
##   plot.rmsip bio3d
##   print.enma bio3d
##   print.nma  bio3d
##   pymol.nma  bio3d</code></pre>
<pre><code>## 
## Attaching package: 'bio3d.nma'</code></pre>
<pre><code>## The following objects are masked from 'package:bio3d':
## 
##     aanma, bhattacharyya, build.hessian, cov.enma, cov.nma,
##     covsoverlap, difference.vector, ff.aaenm, ff.aaenm2, ff.anm,
##     ff.calpha, ff.pfanm, ff.reach, ff.sdenm, fluct.nma, gnm,
##     load.enmff, nma, overlap, rmsip, rtb, sip, var.pdbs, var.xyz</code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="kw">aa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/get.seq.html">get.seq</a></span>(<span class="st">"1rx2_A"</span>)
</pre></div>
<pre><code>## Fetching... Please wait. Done.</code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="kw">blast</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/hmmer.html">hmmer</a></span>(<span class="kw">aa</span>, verbose=<span class="fl">FALSE</span>)
</pre></div>
<p>Function <strong>plot.blast()</strong> facilitates the visualization and filtering of the Blast results. It will attempt to set a seed position to the point of largest drop-off in normalized scores (i.e. the biggest jump in E-values). In this particular case we specify a cutoff (after initial plotting) of 225 to include only the relevant <em>E.coli</em> structures:</p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="kw">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">blast</span>, cutoff=<span class="fl">232</span>)
</pre></div>
<pre><code>##   * Possible cutoff values:    241 2 
##             Yielding Nhits:    144 766 
## 
##   * Chosen cutoff value of:    232 
##             Yielding Nhits:    144</code></pre>
<div class="figure">
<img src="Bio3D_nma-dhfr-partI_files/figure-html/fig1-1-1.png" alt="Blast results. Visualize and filter blast results through function **plot.blast()**. Here we proceed with only the top scoring hits (black)." width="480"><p class="caption">
Blast results. Visualize and filter blast results through function <strong>plot.blast()</strong>. Here we proceed with only the top scoring hits (black).
</p>
</div>
<p>The Blast search and subsequent filtering identified a total of 101 related PDB structures to our query sequence. The PDB identifiers of this collection are accessible through the <code>pdb.id</code> attribute to the <code>hits</code> object (<code>hits$pdb.id</code>). Note that adjusting the cutoff argument (to <strong>plot.blast()</strong>) will result in a decrease or increase of hits.</p>
<p>We can now use function <strong>get.pdb()</strong> and <strong>pdbslit()</strong> to fetch and parse the identified structures. Finally, we use <strong>pdbaln()</strong> to align the PDB structures.</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="co"># fetch PDBs</span>
<span class="kw">raw.files</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/get.pdb.html">get.pdb</a></span>(<span class="kw">hits</span><span class="op">$</span><span class="kw">pdb.id</span>, path = <span class="st">"raw_pdbs"</span>, gzip=<span class="fl">TRUE</span>)
</pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="co"># split by chain ID</span>
<span class="kw">files</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbsplit.html">pdbsplit</a></span>(<span class="kw">raw.files</span>, ids = <span class="kw">hits</span><span class="op">$</span><span class="kw">pdb.id</span>, path = <span class="st">"raw_pdbs/split_chain"</span>, ncore=<span class="fl">4</span>)
</pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="co"># align structures</span>
<span class="kw">pdbs.all</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbaln.html">pdbaln</a></span>(<span class="kw">files</span>, fit=<span class="fl">TRUE</span>)
</pre></div>
<p>The <em>pdbs.all</em> object now contains <em>aligned</em> C-alpha atom data, including Cartesian coordinates, residue numbers, residue types, and B-factors. The sequence alignment is also stored by default to the FASTA format file ‘aln.fa’ (to view this you can use an alignment viewer such as SEAVIEW, see <em>Requirements</em> section above). In cases where manual edits of the alignment are necessary you can re-read the sequence alignment and coordinate data with:</p>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="kw">aln</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.fasta.html">read.fasta</a></span>(<span class="st">'aln.fa'</span>)
<span class="kw">pdbs.all</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.fasta.pdb.html">read.fasta.pdb</a></span>(<span class="kw">aln</span>)
</pre></div>
<p>At this point the <em>pdbs.all</em> object contains all identified 101 structures. This might include structures with missing residues, and/or multiple structurally redundant conformers. For our subsequent NMA missing in-structure residues might bias the calculation, and redundant structures can be useful to omit to reduce the computational load. Below we inspect the connectivity of the PDB structures with a function call to <strong>inspect.connectivity()</strong>, and <strong>trim.pdbs()</strong> to filter out those structures from our <em>pdbs.all</em> object. Similarly, we omit structures that are conformationally redundant to reduce the computational load with function <strong>filter.rmsd()</strong>:</p>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="co"># remove structures with missing residues</span>
<span class="kw">conn</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/inspect.connectivity.html">inspect.connectivity</a></span>(<span class="kw">pdbs.all</span>, cut=<span class="fl">4.05</span>)
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/trim.html">trim</a></span>(<span class="kw">pdbs.all</span>, row.inds=<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">conn</span>))

<span class="co"># which structures are omitted</span>
<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="op">!</span><span class="kw">conn</span>)

<span class="co"># remove conformational redundant structures</span>
<span class="kw">rd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/filter.rmsd.html">filter.rmsd</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>, cutoff=<span class="fl">0.1</span>, fit=<span class="fl">TRUE</span>)
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/trim.html">trim</a></span>(<span class="kw">pdbs</span>, row.inds=<span class="kw">rd</span><span class="op">$</span><span class="kw">ind</span>)

<span class="co"># remove "humanized" e-coli dhfr</span>
<span class="kw">excl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"3QL0"</span>, <span class="st">"4GH8"</span>), <span class="kw">grep</span>, <span class="fu"><a href="https://rdrr.io/r/base/chartr.html">toupper</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span>)))
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/trim.html">trim</a></span>(<span class="kw">pdbs</span>, row.inds=<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="op">!</span>(<span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span>) <span class="op">%in%</span> <span class="kw">excl</span>)))

<span class="co"># a list of PDB codes of our final selection</span>
<span class="kw">ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span>), split=<span class="st">".pdb"</span>))
</pre></div>
<p>Use <strong>print()</strong> to see a short summary of the pdbs object:</p>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">pdbs</span>, alignment=<span class="fl">FALSE</span>)
</pre></div>
<pre><code>## 
## Call:
##   trim.pdbs(pdbs = pdbs, row.inds = which(!(1:length(pdbs$id) %in% 
##     excl)))
## 
## Class:
##   pdbs, fasta
## 
## Alignment dimensions:
##   121 sequence rows; 166 position columns (159 non-gap, 7 gap) 
## 
## + attr: id, xyz, resno, b, chain, ali, resid, call</code></pre>
</div>
<div id="annotate-collected-pdb-structures" class="section level2">
<h2 class="hasAnchor">
<a href="#annotate-collected-pdb-structures" class="anchor"></a>Annotate collected PDB structures</h2>
<p>Function <strong>pdb.annotate()</strong> provides a convenient way of annotating the PDB files we have collected. Below we use the function to annotate each structure to its source species. This will come in handy when annotating plots later on:</p>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="kw">anno</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdb.annotate.html">pdb.annotate</a></span>(<span class="kw">ids</span>)
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="kw">anno</span><span class="op">$</span><span class="kw">source</span>))
</pre></div>
<pre><code>## [1] "Escherichia coli"</code></pre>
</div>
<div id="principal-component-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#principal-component-analysis" class="anchor"></a>Principal component analysis</h2>
<p>A principal component analysis (PCA) can be performed on the structural ensemble (stored in the <em>pdbs</em> object) with function <strong>pca.xyz()</strong>. To obtain meaningful results we first superimpose all structures on the <em>invariant core</em> (function <strong>core.find()</strong>).</p>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="co"># find invariant core</span>
<span class="kw">core</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/core.find.html">core.find</a></span>(<span class="kw">pdbs</span>)

<span class="co"># superimpose all structures to core</span>
<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span> <span class="op">=</span> <span class="fu"><a href="../../../reference/pdbfit.html">pdbfit</a></span>(<span class="kw">pdbs</span>, <span class="kw">core</span><span class="op">$</span><span class="kw">c0.5A.xyz</span>)

<span class="co"># identify gaps, and perform PCA</span>
<span class="kw">gaps.pos</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/gap.inspect.html">gap.inspect</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>)
<span class="kw">gaps.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/gap.inspect.html">gap.inspect</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">ali</span>)
<span class="kw">pc.xray</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pca.xyz.html">pca.xyz</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>[,<span class="kw">gaps.pos</span><span class="op">$</span><span class="kw">f.inds</span>])
</pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit">
<span class="co"># plot PCA</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">pc.xray</span>)
</pre></div>
<p>Note that a call to the wrapper function <strong>pca()</strong> would provide identical results as the above code:</p>
<div class="sourceCode" id="cb21"><pre class="downlit">
<span class="kw">pc.xray</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pca.html">pca</a></span>(<span class="kw">pdbs</span>, core.find=<span class="fl">TRUE</span>)
</pre></div>
<p>Function <strong>rmsd()</strong> will calculate all pairwise RMSD values of the structural ensemble. This facilitates clustering analysis based on the pairwise structural deviation:</p>
<div class="sourceCode" id="cb22"><pre class="downlit">
<span class="kw">rd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/rmsd.html">rmsd</a></span>(<span class="kw">pdbs</span>)
<span class="kw">hc.rd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html">hclust</a></span>(<span class="fu"><a href="https://rdrr.io/r/stats/dist.html">dist</a></span>(<span class="kw">rd</span>))
<span class="kw">grps.rd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cutree.html">cutree</a></span>(<span class="kw">hc.rd</span>, k=<span class="fl">4</span>)
</pre></div>
<div class="sourceCode" id="cb23"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>], col=<span class="st">"grey50"</span>, pch=<span class="fl">16</span>, cex=<span class="fl">1.3</span>,
     ylab=<span class="st">"Prinipcal Component 2"</span>, xlab=<span class="st">"Principal Component 1"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>], col=<span class="kw">grps.rd</span>, pch=<span class="fl">16</span>, cex=<span class="fl">0.9</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_nma-dhfr-partI_files/figure-html/figi-1-1.png" alt="Projection of the X-ray conformers shows that the E.coli DHFR structures can be divided into three major groups along their two first eigenvectors: closed, open, and occluded conformations. Each dot is colored according to their cluster membership: occluded conformations (green), open conformations (black), and closed conformations (red)." width="480"><p class="caption">
Projection of the X-ray conformers shows that the E.coli DHFR structures can be divided into three major groups along their two first eigenvectors: closed, open, and occluded conformations. Each dot is colored according to their cluster membership: occluded conformations (green), open conformations (black), and closed conformations (red).
</p>
</div>
<p>To visualize the major structural variations in the ensemble function <strong>mktrj()</strong> can be used to generate a trajectory PDB file by interpolating along the eigenvector:</p>
<div class="sourceCode" id="cb24"><pre class="downlit">
<span class="fu"><a href="../../../reference/mktrj.html">mktrj</a></span>(<span class="kw">pc.xray</span>, pc=<span class="fl">1</span>,
      resno=<span class="kw">pdbs</span><span class="op">$</span><span class="kw">resno</span>[<span class="fl">1</span>, <span class="kw">gaps.res</span><span class="op">$</span><span class="kw">f.inds</span>],
      resid=<span class="kw">pdbs</span><span class="op">$</span><span class="kw">resid</span>[<span class="fl">1</span>, <span class="kw">gaps.res</span><span class="op">$</span><span class="kw">f.inds</span>])
</pre></div>
<p>Function <strong>pdbfit()</strong> can be used to write the PDB files to separate directories according to their cluster membership:</p>
<div class="sourceCode" id="cb25"><pre class="downlit">
<span class="fu"><a href="../../../reference/pdbfit.html">pdbfit</a></span>(<span class="fu"><a href="../../../reference/trim.html">trim</a></span>(<span class="kw">pdbs</span>, row.inds=<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">grps.rd</span><span class="op">==</span><span class="fl">1</span>)), outpath=<span class="st">"grps1"</span>) <span class="co">## closed</span>
<span class="fu"><a href="../../../reference/pdbfit.html">pdbfit</a></span>(<span class="fu"><a href="../../../reference/trim.html">trim</a></span>(<span class="kw">pdbs</span>, row.inds=<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">grps.rd</span><span class="op">==</span><span class="fl">2</span>)), outpath=<span class="st">"grps2"</span>) <span class="co">## open</span>
<span class="fu"><a href="../../../reference/pdbfit.html">pdbfit</a></span>(<span class="fu"><a href="../../../reference/trim.html">trim</a></span>(<span class="kw">pdbs</span>, row.inds=<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">grps.rd</span><span class="op">==</span><span class="fl">3</span>)), outpath=<span class="st">"grps3"</span>) <span class="co">## occluded</span>
</pre></div>
</div>
<div id="normal-modes-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#normal-modes-analysis" class="anchor"></a>Normal modes analysis</h2>
<p>Function <strong>nma.pdbs()</strong> will calculate the normal modes of each protein structure stored in the <em>pdbs</em> object. The normal modes are calculated on the full structures as provided by object <em>pdbs</em>. With the default argument <code>rm.gaps=TRUE</code> unaligned atoms are omitted from output:</p>
<div class="sourceCode" id="cb26"><pre class="downlit">
<span class="kw">modes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdbs</span>, rm.gaps=<span class="fl">TRUE</span>, ncore=<span class="fl">4</span>)
</pre></div>
<p>The <em>modes</em> object of class <em>enma</em> contains aligned normal mode data including fluctuations, RMSIP data, and aligned eigenvectors. A short summary of the <em>modes</em> object can be obtain by calling the function <strong>print()</strong>, and the aligned fluctuations can be plotted with function <strong>plot.enma()</strong>.</p>
<div class="sourceCode" id="cb27"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">modes</span>)
</pre></div>
<pre><code>## 
## Call:
##   nma.pdbs(pdbs = pdbs, rm.gaps = TRUE, ncore = 4)
## 
## Class:
##   enma
## 
## Number of structures:
##   121
## 
## Attributes stored:
##   - Root mean square inner product (RMSIP)
##   - Aligned atomic fluctuations
##   - Aligned eigenvectors (gaps removed)
##   - Dimensions of x$U.subspace: 477x471x121
## 
## Coordinates were aligned prior to NMA calculations
## 
## + attr: fluctuations, rmsip, U.subspace, L, full.nma, xyz,
##         call</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit">
<span class="co"># plot modes fluctuations</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">modes</span>, pdbs=<span class="kw">pdbs</span>, col=<span class="kw">grps.rd</span>, label=<span class="kw">NULL</span>)
</pre></div>
<pre><code>## Re-reading PDB (2inq_B) to extract SSE</code></pre>
<pre><code>## /usr/local/bin/mkdssp</code></pre>
<pre><code>## Warning in dssp.pdb(pdb.ref, ...): Non-protein residues detected in input PDB:
## MT1, HOH, DOD</code></pre>
<div class="figure">
<img src="Bio3D_nma-dhfr-partI_files/figure-html/fig1l-1-1.png" alt="Normal mode fluctuations of the E.coli DHFR ensemble. Each line represent the mode fluctuations for the individual structures. Color coding according to structural clustering: green (occluded), black (open), and red (closed). " width="576"><p class="caption">
Normal mode fluctuations of the E.coli DHFR ensemble. Each line represent the mode fluctuations for the individual structures. Color coding according to structural clustering: green (occluded), black (open), and red (closed).
</p>
</div>
<div class="sourceCode" id="cb33"><pre class="downlit">
<span class="kw">hc.nma</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html">hclust</a></span>(<span class="fu"><a href="https://rdrr.io/r/stats/dist.html">as.dist</a></span>(<span class="fl">1</span><span class="op">-</span><span class="kw">modes</span><span class="op">$</span><span class="kw">rmsip</span>))
<span class="kw">grps.nma</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cutree.html">cutree</a></span>(<span class="kw">hc.nma</span>, k=<span class="fl">4</span>)
</pre></div>
<div class="sourceCode" id="cb34"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/stats/heatmap.html">heatmap</a></span>(<span class="fl">1</span><span class="op">-</span><span class="kw">modes</span><span class="op">$</span><span class="kw">rmsip</span>, distfun = <span class="kw">as.dist</span>, labRow = <span class="kw">ids</span>, labCol = <span class="kw">ids</span>,
        ColSideColors=<span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="kw">grps.nma</span>), RowSideColors=<span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="kw">grps.rd</span>))
</pre></div>
<div class="figure">
<img src="Bio3D_nma-dhfr-partI_files/figure-html/fig1m-1-1.png" alt="Clustering of structures based on their normal modes similarity (in terms of pair-wise RMSIP values)." width="700"><p class="caption">
Clustering of structures based on their normal modes similarity (in terms of pair-wise RMSIP values).
</p>
</div>
</div>
<div id="fluctuation-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#fluctuation-analysis" class="anchor"></a>Fluctuation analysis</h2>
<p>Comparing the mode fluctuations of two groups of structures can reveal specific regions of distinct flexibility patterns. Below we focus on the differences between the open (black), closed (red) and occluded (green) conformations of the <em>E.coli</em> structures:</p>
<div class="sourceCode" id="cb35"><pre class="downlit">
<span class="kw">cols</span> <span class="op">&lt;-</span> <span class="kw">grps.rd</span>
<span class="kw">cols</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">cols</span>!=<span class="fl">1</span> <span class="op">&amp;</span> <span class="kw">cols</span>!=<span class="fl">2</span>)]<span class="op">=</span><span class="fl">NA</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">modes</span>, pdbs=<span class="kw">pdbs</span>, col=<span class="kw">cols</span>, signif=<span class="fl">TRUE</span>)
</pre></div>
<pre><code>## Re-reading PDB (2inq_B) to extract SSE</code></pre>
<pre><code>## /usr/local/bin/mkdssp</code></pre>
<div class="figure">
<img src="Bio3D_nma-dhfr-partI_files/figure-html/fig1n-1-1.png" alt="Comparison of mode fluctuations between open (black) and closed (red) conformers. Significant differences among the mode fluctuations between the two groups are marked with shaded blue regions." width="576"><p class="caption">
Comparison of mode fluctuations between open (black) and closed (red) conformers. Significant differences among the mode fluctuations between the two groups are marked with shaded blue regions.
</p>
</div>
<div class="sourceCode" id="cb38"><pre class="downlit">
<span class="kw">cols</span> <span class="op">&lt;-</span> <span class="kw">grps.rd</span>
<span class="kw">cols</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">cols</span>!=<span class="fl">1</span> <span class="op">&amp;</span> <span class="kw">cols</span>!=<span class="fl">3</span>)]<span class="op">=</span><span class="fl">NA</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">modes</span>, pdbs=<span class="kw">pdbs</span>, col=<span class="kw">cols</span>, signif=<span class="fl">TRUE</span>)
</pre></div>
<pre><code>## Re-reading PDB (2inq_B) to extract SSE</code></pre>
<pre><code>## /usr/local/bin/mkdssp</code></pre>
<div class="figure">
<img src="Bio3D_nma-dhfr-partI_files/figure-html/fig1n-2-1.png" alt="Comparison of mode fluctuations between open (black) and occluded (green) conformers. Significant differences among the mode fluctuations between the two groups are marked with shaded blue regions." width="576"><p class="caption">
Comparison of mode fluctuations between open (black) and occluded (green) conformers. Significant differences among the mode fluctuations between the two groups are marked with shaded blue regions.
</p>
</div>
<div class="sourceCode" id="cb41"><pre class="downlit">
<span class="kw">cols</span> <span class="op">&lt;-</span> <span class="kw">grps.rd</span>
<span class="kw">cols</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">grps.rd</span>!=<span class="fl">2</span> <span class="op">&amp;</span> <span class="kw">grps.rd</span>!=<span class="fl">3</span>)]<span class="op">=</span><span class="fl">NA</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">modes</span>, pdbs=<span class="kw">pdbs</span>, col=<span class="kw">cols</span>, signif=<span class="fl">TRUE</span>)
</pre></div>
<pre><code>## Re-reading PDB (3ql3_A) to extract SSE</code></pre>
<pre><code>##    PDB has ALT records, taking A only, rm.alt=TRUE</code></pre>
<pre><code>## /usr/local/bin/mkdssp</code></pre>
<div class="figure">
<img src="Bio3D_nma-dhfr-partI_files/figure-html/fig1n-3-1.png" alt="Comparison of mode fluctuations between closed (red) and occluded (green) conformers. Significant differences among the mode fluctuations between the two groups are marked with shaded blue regions." width="576"><p class="caption">
Comparison of mode fluctuations between closed (red) and occluded (green) conformers. Significant differences among the mode fluctuations between the two groups are marked with shaded blue regions.
</p>
</div>
</div>
<div id="compare-with-md-simulation" class="section level2">
<h2 class="hasAnchor">
<a href="#compare-with-md-simulation" class="anchor"></a>Compare with MD simulation</h2>
<p>The above analysis can also nicely be integrated with molecular dynamics (MD) simulations. Below we read in a 5 ns long MD trajectory (of PDB ID 1RX2). We visualize the conformational sampling by projecting the MD conformers onto the principal components (PCs) of the X-ray ensemble, and finally compare the PCs of the MD simulation to the normal modes:</p>
<div class="sourceCode" id="cb45"><pre class="downlit">
<span class="kw">pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"md-traj/1rx2-CA.pdb"</span>)
<span class="kw">trj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.ncdf.html">read.ncdf</a></span>(<span class="st">"md-traj/1rx2_5ns.nc"</span>)

<span class="kw">inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdb2aln.ind.html">pdb2aln.ind</a></span>(<span class="kw">pdbs</span>, <span class="kw">pdb</span>, <span class="kw">gaps.res</span><span class="op">$</span><span class="kw">f.inds</span>)
<span class="kw">inds.core</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdb2aln.ind.html">pdb2aln.ind</a></span>(<span class="kw">pdbs</span>, <span class="kw">pdb</span>, <span class="kw">core</span><span class="op">$</span><span class="kw">c0.5A.atom</span>)

<span class="co"># inds$a is for the pdbs object</span>
<span class="co"># inds$b is for the MD structure</span>

<span class="kw">trj.fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/fit.xyz.html">fit.xyz</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>[<span class="fl">1</span>,], <span class="kw">trj</span>, <span class="kw">core</span><span class="op">$</span><span class="kw">c0.5A.xyz</span>, <span class="kw">inds.core</span><span class="op">$</span><span class="kw">b</span><span class="op">$</span><span class="kw">xyz</span>)
<span class="kw">proj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/project.pca.html">project.pca</a></span>(<span class="kw">trj.fit</span>[, <span class="kw">inds</span><span class="op">$</span><span class="kw">b</span><span class="op">$</span><span class="kw">xyz</span>], <span class="kw">pc.xray</span>)
<span class="kw">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/densCols.html">densCols</a></span>(<span class="kw">proj</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>])
</pre></div>
<div class="sourceCode" id="cb46"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">proj</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>], col=<span class="kw">cols</span>, pch=<span class="fl">16</span>,
     ylab=<span class="st">"Prinipcal Component 2"</span>, xlab=<span class="st">"Principal Component 1"</span>,
     xlim=<span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">1</span>]), ylim=<span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">2</span>]))
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>], col=<span class="fl">1</span>, pch=<span class="fl">1</span>, cex=<span class="fl">1.1</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>], col=<span class="kw">grps.rd</span>, pch=<span class="fl">16</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_nma-dhfr-partI_files/figure-html/fig1o-1-1.png" alt="Projection of MD conformers onto the X-ray PC space provides a two dimensional representation of the conformational sampling along the MD simulation (blue dots)." width="432"><p class="caption">
Projection of MD conformers onto the X-ray PC space provides a two dimensional representation of the conformational sampling along the MD simulation (blue dots).
</p>
</div>
<div class="sourceCode" id="cb47"><pre class="downlit">
<span class="co"># PCA of the MD trajectory</span>
<span class="kw">pc.md</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pca.xyz.html">pca.xyz</a></span>(<span class="kw">trj.fit</span>[, <span class="kw">inds</span><span class="op">$</span><span class="kw">b</span><span class="op">$</span><span class="kw">xyz</span>])

<span class="co"># compare MD-PCA and NMA</span>
<span class="kw">r</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/rmsip.html">rmsip</a></span>(<span class="kw">pc.md</span><span class="op">$</span><span class="kw">U</span>, <span class="kw">modes</span><span class="op">$</span><span class="kw">U.subspace</span>[,,<span class="fl">1</span>])

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">r</span>)
</pre></div>
<pre><code>## $overlap
##        b1    b2    b3    b4    b5    b6    b7    b8    b9   b10
## a1  0.273 0.007 0.042 0.042 0.050 0.001 0.004 0.000 0.002 0.005
## a2  0.202 0.225 0.007 0.006 0.031 0.026 0.006 0.008 0.002 0.001
## a3  0.102 0.002 0.013 0.005 0.000 0.009 0.001 0.016 0.002 0.128
## a4  0.004 0.090 0.033 0.276 0.016 0.079 0.045 0.006 0.008 0.001
## a5  0.018 0.016 0.017 0.010 0.014 0.037 0.005 0.000 0.047 0.008
## a6  0.043 0.019 0.022 0.078 0.042 0.028 0.067 0.057 0.028 0.019
## a7  0.002 0.036 0.170 0.002 0.002 0.046 0.003 0.140 0.001 0.000
## a8  0.042 0.430 0.062 0.128 0.005 0.034 0.009 0.000 0.007 0.002
## a9  0.015 0.000 0.001 0.021 0.147 0.005 0.059 0.002 0.017 0.069
## a10 0.004 0.007 0.155 0.007 0.049 0.000 0.034 0.004 0.009 0.008
## 
## $rmsip
## [1] 0.6390165
## 
## attr(,"class")
## [1] "rmsip"</code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">r</span>, xlab=<span class="st">"MD PCA"</span>, ylab=<span class="st">"NMA"</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_nma-dhfr-partI_files/figure-html/fig1p-1-1.png" alt="Overlap map between normal modes and principal components of a 5 ns long MD simulation. The two subsets yields an RMSIP value of 0.64, where a value of 1 would idicate identical directionality." width="432"><p class="caption">
Overlap map between normal modes and principal components of a 5 ns long MD simulation. The two subsets yields an RMSIP value of 0.64, where a value of 1 would idicate identical directionality.
</p>
</div>
<div class="sourceCode" id="cb50"><pre class="downlit">
<span class="co"># compare MD-PCA and X-rayPCA</span>
<span class="kw">r</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/rmsip.html">rmsip</a></span>(<span class="kw">pc.md</span>, <span class="kw">pc.xray</span>)
</pre></div>
</div>
<div id="domain-analysis-with-geostas" class="section level2">
<h2 class="hasAnchor">
<a href="#domain-analysis-with-geostas" class="anchor"></a>Domain analysis with GeoStaS</h2>
<p>Identification of regions in the protein that move as rigid bodies is facilitated with the implementation of the GeoStaS algorithm <span class="citation">(Romanowska, Nowinski, and Trylska 2012)</span>. Below we demonstrate the use of function <strong>geostas()</strong> on data obtained from ensemble NMA, an ensemble of PDB structures, and a 5 ns long MD simulation. See <code><a href="../../../reference/geostas.html">help(geostas)</a></code> for more details and further examples.</p>
<div class="sourceCode" id="cb51"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">bio3d.geostas</span>)
</pre></div>
<p><strong>GeoStaS on a PDB ensemble</strong>: Below we input the <code>pdbs</code> object to function <strong>geostas()</strong> to identify dynamic domains. Here, we attempt to divide the structure into 2 sub-domains using argument <code>k=2</code>. Function <strong>geostas()</strong> will return a <code>grps</code> attribute which corresponds to the domain assignment for each C-alpha atom in the structure. Note that we use argument <code>fit=FALSE</code> to avoid re-fitting the coordinates since. Recall that the coordinates of the <code>pdbs</code> object has already been superimposed to the identified invariant core (see above). To visualize the domain assignment we write a PDB trajectory of the first principal component (of the Cartesian coordinates of the <code>pdbs</code> object), and add argument <code>chain=gs.xray$grps</code> to replace the chain identifiers with the domain assignment:</p>
<div class="sourceCode" id="cb52"><pre class="downlit">
<span class="co"># Identify dynamic domains</span>
<span class="kw">gs.xray</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/geostas.html">geostas</a></span>(<span class="kw">pdbs</span>, k=<span class="fl">2</span>, fit=<span class="fl">FALSE</span>)

<span class="co"># Visualize PCs with colored domains (chain ID)</span>
<span class="fu"><a href="../../../reference/mktrj.html">mktrj</a></span>(<span class="kw">pc.xray</span>, pc=<span class="fl">1</span>, chain=<span class="kw">gs.xray</span><span class="op">$</span><span class="kw">grps</span>)
</pre></div>
<p><strong>GeoStaS on ensemble NMA</strong>: We can also identify dynamic domains from the normal modes of the ensemble of 121 protein structures stored in the <code>modes</code> object. By using function <strong>mktrj.enma()</strong> we generate a trajectory from the first five modes of all structures. We then input this trajectory to function <strong>geostas()</strong>:</p>
<div class="sourceCode" id="cb53"><pre class="downlit">
<span class="co"># Build conformational ensemble</span>
<span class="kw">trj.nma</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/mktrj.html">mktrj</a></span>(<span class="kw">modes</span>, <span class="kw">pdbs</span>, m.inds=<span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, s.inds=<span class="kw">NULL</span>, mag=<span class="fl">10</span>, step=<span class="fl">2</span>, rock=<span class="fl">FALSE</span>)

<span class="co"># Run geostas to find domains</span>
<span class="kw">gs.nma</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/geostas.html">geostas</a></span>(<span class="kw">trj.nma</span>, k=<span class="fl">2</span>, fit=<span class="fl">FALSE</span>)
</pre></div>
<pre><code>##   .. 'xyz' coordinate data with 6655 frames 
##   .. coordinates are not superimposed prior to geostas calculation
##   .. calculating atomic movement similarity matrix ('amsm.xyz()') 
##   .. dimensions of AMSM are 159x159
##   .. clustering AMSM using 'kmeans'</code></pre>
<div class="sourceCode" id="cb55"><pre class="downlit">
<span class="co"># Write NMA generated trajectory with domain assignment</span>
<span class="fu"><a href="../../../reference/write.pdb.html">write.pdb</a></span>(xyz=<span class="kw">trj.nma</span>, chain=<span class="kw">gs.nma</span><span class="op">$</span><span class="kw">grps</span>)
</pre></div>
<p>Note that function <strong>geostas.enma()</strong> wraps these three main functions calls to faciliatet this calculation. See <code><a href="../../../reference/geostas.html">help(geostas)</a></code> for more information.</p>
<p><strong>GeoStaS on a MD trajectory</strong>: The domain analysis can also be performed on the trajectory data obtained from the MD simulation (see above). Recall that the MD trajectory has already been superimposed to the invariant core. We therefore use argument <code>fit=FALSE</code> below. We then perform a new PCA of the MD trajectory, and visualize the domain assingments with function <strong>mktrj()</strong>:</p>
<div class="sourceCode" id="cb56"><pre class="downlit">
<span class="kw">gs.md</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/geostas.html">geostas</a></span>(<span class="kw">trj</span>, k=<span class="fl">2</span>, fit=<span class="fl">FALSE</span>)
<span class="kw">pc.md</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pca.html">pca</a></span>(<span class="kw">trj</span>, fit=<span class="fl">FALSE</span>)
<span class="fu"><a href="../../../reference/mktrj.html">mktrj</a></span>(<span class="kw">pc.md</span>, pc=<span class="fl">1</span>, chain=<span class="kw">gs.md</span><span class="op">$</span><span class="kw">grps</span>)
</pre></div>
<div class="figure">
<img src="figures/geostas-domains.png" alt=""><p class="caption">Visualization of domain assignment using function <strong>geostas()</strong> on the first five normal modes of the entire ensemble of 82 DHFR structures.</p>
</div>
</div>
<div id="measures-for-modes-comparison" class="section level2">
<h2 class="hasAnchor">
<a href="#measures-for-modes-comparison" class="anchor"></a>Measures for modes comparison</h2>
<p>Bio3D now includes multiple measures for the assessment of similarity between two normal mode objects. This enables clustering of related proteins based on the predicted modes of motion. Below we demonstrate the use of root mean squared inner product (RMSIP), squared inner product (SIP), covariance overlap, bhattacharyya coefficient, and PCA of the corresponding covariance matrices.</p>
<div class="sourceCode" id="cb57"><pre class="downlit">
<span class="co"># Similarity of atomic fluctuations</span>
<span class="kw">sip</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/sip.html">sip</a></span>(<span class="kw">modes</span>)
<span class="kw">hc.sip</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html">hclust</a></span>(<span class="fu"><a href="https://rdrr.io/r/stats/dist.html">as.dist</a></span>(<span class="fl">1</span><span class="op">-</span><span class="kw">sip</span>), method=<span class="st">"ward.D2"</span>)
<span class="kw">grps.sip</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cutree.html">cutree</a></span>(<span class="kw">hc.sip</span>, k=<span class="fl">3</span>)
</pre></div>
<div class="sourceCode" id="cb58"><pre class="downlit">
<span class="fu"><a href="../../../reference/hclustplot.html">hclustplot</a></span>(<span class="kw">hc.sip</span>, k=<span class="fl">3</span>, colors=<span class="kw">grps.rd</span>, labels=<span class="kw">ids</span>, cex=<span class="fl">0.7</span>, main=<span class="st">"SIP"</span>, fillbox=<span class="fl">FALSE</span>)

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(fig=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">.65</span>, <span class="fl">1</span>, <span class="fl">.45</span>, <span class="fl">1</span>), new = <span class="fl">TRUE</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>], col=<span class="st">"grey50"</span>, pch=<span class="fl">16</span>, cex=<span class="fl">1.1</span>, 
     ylab=<span class="st">""</span>, xlab=<span class="st">""</span>, axes=<span class="fl">FALSE</span>, bg=<span class="st">"red"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>], col=<span class="kw">grps.sip</span>, pch=<span class="fl">16</span>, cex=<span class="fl">0.7</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/box.html">box</a></span>()
</pre></div>
<div class="figure">
<img src="Bio3D_nma-dhfr-partI_files/figure-html/fig1q-sip-1.png" alt="Dendrogram shows the results of hierarchical clustering of structures based on the similarity of atomic fluctuations calculated from NMA. Colors of the labels depict associated conformatial state: green (occluded), black (open), and red (closed). The inset shows the conformerplot (see Figure 2), with colors according to clustering based on pairwise SIP values." width="672"><p class="caption">
Dendrogram shows the results of hierarchical clustering of structures based on the similarity of atomic fluctuations calculated from NMA. Colors of the labels depict associated conformatial state: green (occluded), black (open), and red (closed). The inset shows the conformerplot (see Figure 2), with colors according to clustering based on pairwise SIP values.
</p>
</div>
<div class="sourceCode" id="cb59"><pre class="downlit">
<span class="co"># RMSIP</span>
<span class="kw">rmsip</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/rmsip.html">rmsip</a></span>(<span class="kw">modes</span>)
<span class="kw">hc.rmsip</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html">hclust</a></span>(<span class="fu"><a href="https://rdrr.io/r/stats/dist.html">dist</a></span>(<span class="fl">1</span><span class="op">-</span><span class="kw">rmsip</span>), method=<span class="st">"ward.D2"</span>)
<span class="kw">grps.rmsip</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cutree.html">cutree</a></span>(<span class="kw">hc.rmsip</span>, k=<span class="fl">3</span>)
</pre></div>
<div class="sourceCode" id="cb60"><pre class="downlit">
<span class="fu"><a href="../../../reference/hclustplot.html">hclustplot</a></span>(<span class="kw">hc.rmsip</span>, k=<span class="fl">3</span>, colors=<span class="kw">grps.rd</span>, labels=<span class="kw">ids</span>, cex=<span class="fl">0.7</span>, main=<span class="st">"RMSIP"</span>, fillbox=<span class="fl">FALSE</span>)

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(fig=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">.65</span>, <span class="fl">1</span>, <span class="fl">.45</span>, <span class="fl">1</span>), new = <span class="fl">TRUE</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>], col=<span class="st">"grey50"</span>, pch=<span class="fl">16</span>, cex=<span class="fl">1.1</span>, 
     ylab=<span class="st">""</span>, xlab=<span class="st">""</span>, axes=<span class="fl">FALSE</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>], col=<span class="kw">grps.rmsip</span>, pch=<span class="fl">16</span>, cex=<span class="fl">0.7</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/box.html">box</a></span>()
</pre></div>
<div class="figure">
<img src="Bio3D_nma-dhfr-partI_files/figure-html/fig1q-rmsip-1.png" alt="Dendrogram shows the results of hierarchical clustering of structures based on their pairwise RMSIP values (calculated from NMA). Colors of the labels depict associated conformatial state: green (occluded), black (open), and red (closed). The inset shows the conformerplot (see Figure 2), with colors according to clustering based on the pairwise RMSIP values." width="672"><p class="caption">
Dendrogram shows the results of hierarchical clustering of structures based on their pairwise RMSIP values (calculated from NMA). Colors of the labels depict associated conformatial state: green (occluded), black (open), and red (closed). The inset shows the conformerplot (see Figure 2), with colors according to clustering based on the pairwise RMSIP values.
</p>
</div>
<div class="sourceCode" id="cb61"><pre class="downlit">
<span class="co"># Covariance overlap</span>
<span class="kw">co</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/covsoverlap.html">covsoverlap</a></span>(<span class="kw">modes</span>, subset=<span class="fl">200</span>)
<span class="kw">hc.co</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html">hclust</a></span>(<span class="fu"><a href="https://rdrr.io/r/stats/dist.html">as.dist</a></span>(<span class="fl">1</span><span class="op">-</span><span class="kw">co</span>), method=<span class="st">"ward.D2"</span>)
<span class="kw">grps.co</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cutree.html">cutree</a></span>(<span class="kw">hc.co</span>, k=<span class="fl">3</span>)
</pre></div>
<div class="sourceCode" id="cb62"><pre class="downlit">
<span class="fu"><a href="../../../reference/hclustplot.html">hclustplot</a></span>(<span class="kw">hc.co</span>, k=<span class="fl">3</span>, colors=<span class="kw">grps.rd</span>, labels=<span class="kw">ids</span>, cex=<span class="fl">0.7</span>, main=<span class="st">"Covariance overlap"</span>, fillbox=<span class="fl">FALSE</span>)

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(fig=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">.65</span>, <span class="fl">1</span>, <span class="fl">.45</span>, <span class="fl">1</span>), new = <span class="fl">TRUE</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>], col=<span class="st">"grey50"</span>, pch=<span class="fl">16</span>, cex=<span class="fl">1.1</span>, 
     ylab=<span class="st">""</span>, xlab=<span class="st">""</span>, axes=<span class="fl">FALSE</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>], col=<span class="kw">grps.co</span>, pch=<span class="fl">16</span>, cex=<span class="fl">0.7</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/box.html">box</a></span>()
</pre></div>
<div class="figure">
<img src="Bio3D_nma-dhfr-partI_files/figure-html/fig1q-co-1.png" alt="Dendrogram shows the results of hierarchical clustering of structures based on their pairwise covariance overlap (calculated from NMA). Colors of the labels depict associated conformatial state: green (occluded), black (open), and red (closed). The inset shows the conformerplot (see Figure 2), with colors according to clustering of the Covariance overlap measure." width="672"><p class="caption">
Dendrogram shows the results of hierarchical clustering of structures based on their pairwise covariance overlap (calculated from NMA). Colors of the labels depict associated conformatial state: green (occluded), black (open), and red (closed). The inset shows the conformerplot (see Figure 2), with colors according to clustering of the Covariance overlap measure.
</p>
</div>
<div class="sourceCode" id="cb63"><pre class="downlit">
<span class="co"># Bhattacharyya coefficient</span>
<span class="kw">covs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/cov.nma.html">cov.enma</a></span>(<span class="kw">modes</span>)
<span class="kw">bc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/bhattacharyya.html">bhattacharyya</a></span>(<span class="kw">modes</span>, covs=<span class="kw">covs</span>)
<span class="kw">hc.bc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html">hclust</a></span>(<span class="fu"><a href="https://rdrr.io/r/stats/dist.html">dist</a></span>(<span class="fl">1</span><span class="op">-</span><span class="kw">bc</span>), method=<span class="st">"ward.D2"</span>)
<span class="kw">grps.bc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cutree.html">cutree</a></span>(<span class="kw">hc.bc</span>, k=<span class="fl">3</span>)
</pre></div>
<div class="sourceCode" id="cb64"><pre class="downlit">
<span class="fu"><a href="../../../reference/hclustplot.html">hclustplot</a></span>(<span class="kw">hc.bc</span>, k=<span class="fl">3</span>, colors=<span class="kw">grps.rd</span>, labels=<span class="kw">ids</span>, cex=<span class="fl">0.7</span>, main=<span class="st">"Bhattacharyya coefficient"</span>, fillbox=<span class="fl">FALSE</span>)

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(fig=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">.65</span>, <span class="fl">1</span>, <span class="fl">.45</span>, <span class="fl">1</span>), new = <span class="fl">TRUE</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>], col=<span class="st">"grey50"</span>, pch=<span class="fl">16</span>, cex=<span class="fl">1.1</span>, 
     ylab=<span class="st">""</span>, xlab=<span class="st">""</span>, axes=<span class="fl">FALSE</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>], col=<span class="kw">grps.bc</span>, pch=<span class="fl">16</span>, cex=<span class="fl">0.7</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/box.html">box</a></span>()
</pre></div>
<div class="figure">
<img src="Bio3D_nma-dhfr-partI_files/figure-html/fig1q-bc-1.png" alt="Dendrogram shows the results of hierarchical clustering of structures based on their pairwise Bhattacharyya coefficient (calculated from NMA). Colors of the labels depict associated conformatial state: green (occluded), black (open), and red (closed). The inset shows the conformerplot (see Figure 2), with colors according to clustering of the pairwise Bhattacharyya coefficients." width="672"><p class="caption">
Dendrogram shows the results of hierarchical clustering of structures based on their pairwise Bhattacharyya coefficient (calculated from NMA). Colors of the labels depict associated conformatial state: green (occluded), black (open), and red (closed). The inset shows the conformerplot (see Figure 2), with colors according to clustering of the pairwise Bhattacharyya coefficients.
</p>
</div>
<div class="sourceCode" id="cb65"><pre class="downlit">
<span class="co"># PCA of covariance matrices</span>
<span class="kw">pc.covs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pca.array.html">pca.array</a></span>(<span class="kw">covs</span>)
<span class="kw">hc.covs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html">hclust</a></span>(<span class="fu"><a href="https://rdrr.io/r/stats/dist.html">dist</a></span>(<span class="kw">pc.covs</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>]), method=<span class="st">"ward.D2"</span>)
<span class="kw">grps.covs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cutree.html">cutree</a></span>(<span class="kw">hc.covs</span>, k=<span class="fl">3</span>)
</pre></div>
<div class="sourceCode" id="cb66"><pre class="downlit">
<span class="fu"><a href="../../../reference/hclustplot.html">hclustplot</a></span>(<span class="kw">hc.covs</span>, k=<span class="fl">3</span>, colors=<span class="kw">grps.rd</span>, labels=<span class="kw">ids</span>, cex=<span class="fl">0.7</span>, main=<span class="st">"PCA of covariance matrices"</span>, fillbox=<span class="fl">FALSE</span>)

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(fig=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">.65</span>, <span class="fl">1</span>, <span class="fl">.45</span>, <span class="fl">1</span>), new = <span class="fl">TRUE</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>], col=<span class="st">"grey50"</span>, pch=<span class="fl">16</span>, cex=<span class="fl">1.1</span>, 
     ylab=<span class="st">""</span>, xlab=<span class="st">""</span>, axes=<span class="fl">FALSE</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="kw">pc.xray</span><span class="op">$</span><span class="kw">z</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span>], col=<span class="kw">grps.covs</span>, pch=<span class="fl">16</span>, cex=<span class="fl">0.7</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/box.html">box</a></span>()
</pre></div>
<div class="figure">
<img src="Bio3D_nma-dhfr-partI_files/figure-html/fig1q-pcaco-1.png" alt="Dendrogram shows the results of hierarchical clustering of structures based on the PCA of covariance matrices (calculated from NMA). Colors of the labels depict associated conformatial state: green (occluded), black (open), and red (closed). The inset shows the conformerplot (see Figure 2), with colors according to clustering based on PCA of covariance matrices." width="672"><p class="caption">
Dendrogram shows the results of hierarchical clustering of structures based on the PCA of covariance matrices (calculated from NMA). Colors of the labels depict associated conformatial state: green (occluded), black (open), and red (closed). The inset shows the conformerplot (see Figure 2), with colors according to clustering based on PCA of covariance matrices.
</p>
</div>
</div>
</div>
<div id="document-details" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#document-details" class="anchor"></a>Document Details</h1>
<p>This document is shipped with the Bio3D package in both R and PDF formats. All code can be extracted and automatically executed to generate Figures and/or the PDF with the following commands:</p>
<div class="sourceCode" id="cb67"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://github.com/rstudio/rmarkdown">rmarkdown</a></span>)
<span class="fu"><a href="https://rdrr.io/pkg/rmarkdown/man/render.html">render</a></span>(<span class="st">"Bio3D_nma-dhfr-partI.Rmd"</span>, <span class="st">"all"</span>)
</pre></div>
</div>
<div id="information-about-the-current-bio3d-session" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#information-about-the-current-bio3d-session" class="anchor"></a>Information About the Current Bio3D Session</h1>
<div class="sourceCode" id="cb68"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>(), <span class="fl">FALSE</span>)
</pre></div>
<pre><code>## R version 3.6.0 (2019-04-26)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS  10.15.5
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] bio3d.nma_0.1.0.9000 bio3d_2.4-1.9000    
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.5         knitr_1.29         magrittr_1.5       lattice_0.20-41   
##  [5] R6_2.4.1           rlang_0.4.6.9000   stringr_1.4.0      highr_0.8         
##  [9] tools_3.6.0        parallel_3.6.0     grid_3.6.0         xfun_0.15         
## [13] KernSmooth_2.23-17 htmltools_0.5.0    yaml_2.2.1         assertthat_0.2.1  
## [17] rprojroot_1.3-2    digest_0.6.25      pkgdown_1.5.1.9000 crayon_1.3.4      
## [21] fs_1.4.2           ncdf4_1.17         memoise_1.1.0      evaluate_0.14     
## [25] rmarkdown_2.3.2    stringi_1.4.6      compiler_3.6.0     desc_1.2.0        
## [29] backports_1.1.8</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references hanging-indent">
<div id="ref-grant06">
<p>Grant, B. J., A. P. D. C Rodrigues, K. M. Elsawy, A. J. Mccammon, and L. S. D. Caves. 2006. “Bio3d: An R Package for the Comparative Analysis of Protein Structures.” <em>Bioinformatics</em> 22: 2695–6. <a href="https://doi.org/10.1093/bioinformatics/btl461">https://doi.org/10.1093/bioinformatics/btl461</a>.</p>
</div>
<div id="ref-Romanowska2012a">
<p>Romanowska, Julia, Krzysztof S. Nowinski, and Joanna Trylska. 2012. “Determining geometrically stable domains in molecular conformation sets.” <em>Journal of Chemical Theory and Computation</em> 8 (8): 2588–99. <a href="https://doi.org/10.1021/ct300206j">https://doi.org/10.1021/ct300206j</a>.</p>
</div>
<div id="ref-skjaerven15">
<p>Skjaerven, L., X. Q. Yao, G. Scarabelli, and B. J. Grant. 2015. “Integrating Protein Structural Dynamics and Evolutionary Analysis with Bio3d.” <em>BMC Bioinformatics</em> 15: 399. <a href="https://doi.org/10.1186/s12859-014-0399-6">https://doi.org/10.1186/s12859-014-0399-6</a>.</p>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>The latest version of the package, full documentation and further vignettes (including detailed installation instructions) can be obtained from the main Bio3D website: <a href="http://thegrantlab.org/bio3d/">http://thegrantlab.org/bio3d/</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>This vignette contains executable examples, see <code><a href="https://rdrr.io/r/utils/vignette.html">help(vignette)</a></code> for further details.<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="tidyverse">
  <p>Authored by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="author">
  <p>
    Built with ❤️ and <a href="https://pkgdown.r-lib.org">pkgdown</a> © 2020.
  </p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '1270c4079b26a138795263974bbf302d',
    indexName: 'tidyverse',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
