<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Ensemble NMA across multiple species of DHFR • Bio3D</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../../../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../../../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../../../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../../../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../../../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../../../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="../../../tidyverse.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../../bootstrap-toc.css">
<script src="../../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../../pkgdown.css" rel="stylesheet">
<script src="../../../pkgdown.js"></script><link href="../../../tidyverse-2.css" rel="stylesheet">
<!--optional theme
<link href="../../../.css" rel="stylesheet">
--><!-- docsearch --><script src="../../../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../../../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Ensemble NMA across multiple species of DHFR">
<meta property="og:description" content="bio3d">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../../../index.html">
        bio3d</a>
        <div class="info hidden-xs hidden-sm">
<!--          <span class="partof">part of the <a href="https://tidyverse.org">tidyverse</a></span>
-->
          <span class="version version-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4-1.9000</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">INTRODUCTORY TUTORIALS</li>
    <li>
      <a href="../../../articles/online/install_vignette/Bio3D_install.html">Installing Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/intro_vignette/Bio3D_introduction.html">Getting started with Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/pdb_vignette/Bio3D_pdb.html">PDB structure manipulation and analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/pca_vignette/Bio3D_pca.html">Comparative sequence and structure analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/nma_vignette/Bio3D_nma.html">Enhanced Normal Modes Analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/traj_vignette/Bio3D_md.html">Beginning trajectory analysis with Bio3D </a>
    </li>
    <li class="divider">
    <li class="dropdown-header">ADVANCED TUTORIALS</li>
    <li>
      <a href="../../../articles/online/enma_vignettes/Bio3D_nma-dhfr-partI.html">Ensemble NMA of E.coli DHFR structures</a>
    </li>
    <li>
      <a href="../../../articles/online/enma_vignettes/Bio3D_nma-dhfr-partII.html">Ensemble NMA across multiple species of DHFR</a>
    </li>
    <li>
      <a href="../../../articles/online/cna_vignette/Bio3D_cna-transducin.html">Correlation network analysis with Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/cna_vignette/cna_vignette.spin.html">Protein structure network analysis with Bio3D ***</a>
    </li>
    <li>
      <a href="../../../articles/online/eddm_vignette/Bio3D_eddm.spin.html">Ensemble Difference Distance Matrix (eDDM) analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">WEBAPP TUTORIALS</li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/download/webapp-tutorials?download=22:bio3d-webapp-pca">Online protein structure analysis with Bio3D-Web (PDF)</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/webapps">Online Normal Mode Analysis with Bio3D WebApps (To Update)</a>
    </li>
    <li class="divider">
    <li>
      <a href="../../../articles/index.html">All vignettes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Documentation
  </a>
</li>
        <li>
  <a href="../../../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../../../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    News
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket"></span>
     
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Bio3D_nma-dhfr-partII_files/header-attrs-2.3.2/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Ensemble NMA across multiple species of DHFR</h1>
                        <h4 class="author">Lars Skjaerven, Xin-Qiu Yao, Guido Scarabelli &amp; Barry J. Grant</h4>
            
      
      
      <div class="hidden name"><code>Bio3D_nma-dhfr-partII.Rmd</code></div>

    </div>

    
    
<div id="background" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#background" class="anchor"></a>Background</h1>
<p>Bio3D<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> is an R package that provides interactive tools for structural bioinformatics. The primary focus of Bio3D is the analysis of bimolecular structure, sequence and simulation data <span class="citation">(Grant et al. 2006)</span>.</p>
<p>Normal mode analysis (NMA) is one of the major simulation techniques used to probe large-scale motions in biomolecules. Typical application is for the prediction of functional motions in proteins. Version 2.0 of the Bio3D package now includes extensive NMA facilities <span class="citation">(Skjaerven et al. 2015)</span>. These include a unique collection of multiple elastic network model force-fields, automated ensemble analysis methods, and variance weighted NMA (see also the <a href="http://thegrantlab.org/bio3d/tutorials"><strong>NMA Vignette</strong></a>). Here we provide an in-depth demonstration of ensemble NMA with working code that comprise complete executable examples<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>.</p>
<div id="requirements" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#requirements" class="anchor"></a>Requirements</h2>
<p>Detailed instructions for obtaining and installing the Bio3D package on various platforms can be found in the <a href="http://thegrantlab.org/bio3d/download/download.html"><strong>Installing Bio3D Vignette</strong></a> available both on-line and from within the Bio3D package. In addition to Bio3D the <em>MUSCLE</em> and <em>CLUSTALO</em> multiple sequence alignment programs (available from the <a href="http://www.drive5.com/muscle/">muscle home page</a> and <a href="http://www.clustal.org/omega">clustalo home page</a>) must be installed on your system and in the search path for executables. Please see the installation vignette for further details.</p>
</div>
<div id="about-this-document" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#about-this-document" class="anchor"></a>About this document</h2>
<p>This vignette was generated using <strong>Bio3D version 2.4.1.9000</strong>.</p>
</div>
</div>
<div id="part-ii-ensemble-nma-across-multiple-species-of-dhfr" class="section level1">
<h1 class="hasAnchor">
<a href="#part-ii-ensemble-nma-across-multiple-species-of-dhfr" class="anchor"></a>Part II: Ensemble NMA across multiple species of DHFR</h1>
<p>In this vignette we extend the analysis from Part I by including a more extensive search of distant homologues within the DHFR family. Based on a HMMER search we identify and collect protein species down to a pairwise sequence identity of 20%. Normal modes analysis (NMA) across these species reveals a remarkable similarity of the fluctuation profiles, but also features which are characteristic to specific species.</p>
<div id="hmmer-search-for-distantly-related-dhfr-species" class="section level2">
<h2 class="hasAnchor">
<a href="#hmmer-search-for-distantly-related-dhfr-species" class="anchor"></a>HMMER search for distantly related DHFR species</h2>
<p>Below we use the sequence of <em>E.coli</em> DHFR to perform an initial search against the Pfam HMM database with function <strong>hmmer()</strong>. The arguments <code>type</code> and <code>db</code> specifies the type of hmmer search and the database to search, respectively. In this particular example, our query sequence is searched against the Pfam profile HMM library (arguments <code>type="hmmscan"</code> and <code>db="pfam"</code>) to identify its respecitve protein family. The to <strong>hmmer()</strong> will return a data frame object containing the Pfam accession ID (<code>$acc</code>), description of the identified family (<code>$desc</code>), family name (<code>$name</code>), etc.</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="http://thegrantlab.org/bio3d">bio3d</a></span>)
</pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="co"># get sequence of Ecoli DHFR</span>
<span class="kw">seq</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/get.seq.html">get.seq</a></span>(<span class="st">"1rx2_A"</span>)
</pre></div>
<pre><code>## Warning in get.seq("1rx2_A"): Removing existing file: seqs.fasta</code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="co"># scan the Pfam database for our sequence</span>
<span class="kw">pfam</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/hmmer.html">hmmer</a></span>(<span class="kw">seq</span>, type=<span class="st">"hmmscan"</span>, db=<span class="st">"pfam"</span>)
<span class="kw">pfam</span><span class="op">$</span><span class="kw">hit.tbl</span>
</pre></div>
<p><strong>Sidenote:</strong> The <strong>hmmer()</strong> function facilitates four different types of searches at a multitude of databases. Use function <code><a href="../../../reference/hmmer.html">help(hmmer)</a></code> for a complete overview of the different options.</p>
<p>Having identified the Pfam entry of our query protein we can use function <strong>pfam()</strong> to fetch the curated sequence alignment of the DHFR family. Use function <strong>print.fasta()</strong> to print a short summary of the downloaded sequence alignment to the screen. Note that if argument <code>alignment=TRUE</code> the sequence alignmenment itself will be written to screen.</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="co"># download pfam alignment for the DHFR family</span>
<span class="kw">pfam.aln</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pfam.html">pfam</a></span>(<span class="kw">pfam</span><span class="op">$</span><span class="kw">hit.tbl</span><span class="op">$</span><span class="kw">acc</span>[<span class="fl">1</span>])
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">pfam.aln</span>, alignment=<span class="fl">FALSE</span>)
</pre></div>
<pre><code>## 
## Call:
##   pfam(id = pfam$hit.tbl$acc[1])
## 
## Class:
##   fasta
## 
## Alignment dimensions:
##   55 sequence rows; 206 position columns (114 non-gap, 92 gap) 
## 
## + attr: id, ali, call</code></pre>
<p>The next hmmer search builds a profile HMM from the Pfam multiple sequence alignment and uses this HMM to search against a target sequence database (use <code>type="hmmsearch"</code>). In this case our target sequence database is the PDB (<code>db="pdb"</code>), but there are also other options such as <code>"swissprot"</code> and <code>"uniprot"</code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="co"># use Pfam alignment in search </span>
<span class="kw">hmm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/hmmer.html">hmmer</a></span>(<span class="kw">pfam.aln</span>, type=<span class="st">"hmmsearch"</span>, db=<span class="st">"pdb"</span>)
</pre></div>
<p>Function <strong>plot.hmmer()</strong> (the equivalent to <strong>plot.blast()</strong>) provides a quick overview of the rearch results, and can aid in the identification of a sensible hit similarity threshold. The normalized scores (-log(E-Value)) are shown in the upper panel, and the lower panel provides an overview of the kingdom and specie each hit are associated with. Here we specify a cutoff of 56 yielding 655 hits:</p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="kw">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">hmm</span>, cutoff=<span class="fl">56</span>)
</pre></div>
<pre><code>##   * Possible cutoff values:    56 0 
##             Yielding Nhits:    696 770 
## 
##   * Chosen cutoff value of:    56 
##             Yielding Nhits:    696</code></pre>
<div class="figure">
<img src="Bio3D_nma-dhfr-partII_files/figure-html/fig3-2-1.png" alt="Overview  of hits obtained from the HMMER search. The two upper panels show the normalized scores and bitscore. The two lower panels show a bar graph of the distribution of significant hits according to the bitscores and taxonomic kingdom. The first bar plot corresponds to the hits above cutoff, while the second par plot corresponds to those hits below the cutoff. Observe e.g. that the majority of the top scoring hits is from bacteria (red) while there are several eukaryotes (green) with lower scores." width="576"><p class="caption">
Overview of hits obtained from the HMMER search. The two upper panels show the normalized scores and bitscore. The two lower panels show a bar graph of the distribution of significant hits according to the bitscores and taxonomic kingdom. The first bar plot corresponds to the hits above cutoff, while the second par plot corresponds to those hits below the cutoff. Observe e.g. that the majority of the top scoring hits is from bacteria (red) while there are several eukaryotes (green) with lower scores.
</p>
</div>
<p><strong>Side-note</strong>: To visualize the hmmer search results in a web-browser go to the URL in the <code>hmm$url</code> attribute:</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="co"># view hmmer results in web-browser</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">hmm</span><span class="op">$</span><span class="kw">url</span>)
</pre></div>
<pre><code>## [1] "http://www.ebi.ac.uk/Tools/hmmer/results/AEDEF77C-EBF3-11EA-BCA4-FABF53F04F9B"</code></pre>
<p>An summary over the hit species can be obtained by investigatin the <code>hmm$hit.tbl$species</code> attribute:</p>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="kw">ids</span> <span class="op">&lt;-</span> <span class="kw">hits</span><span class="op">$</span><span class="kw">acc</span>
<span class="kw">species</span> <span class="op">&lt;-</span> <span class="kw">hmm</span><span class="op">$</span><span class="kw">hit.tbl</span><span class="op">$</span><span class="kw">species</span>[<span class="kw">hits</span><span class="op">$</span><span class="kw">inds</span>]

<span class="co"># print collected species</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="kw">species</span>))
</pre></div>
<pre><code>##  [1] "Geobacillus stearothermophilus"                        
##  [2] "Bacillus anthracis"                                    
##  [3] "Coxiella burnetii"                                     
##  [4] "Bacillus anthracis str. Sterne (GCA_000832635)"        
##  [5] "Bacillus anthracis str. Sterne"                        
##  [6] "Moritella profunda"                                    
##  [7] "Yersinia pestis CO92"                                  
##  [8] "Klebsiella pneumoniae CG43"                            
##  [9] "Escherichia coli"                                      
## [10] "Escherichia coli str. K-12 substr. DH10B"              
## [11] "Escherichia coli str. K-12 substr. MC4100"             
## [12] "Escherichia coli (strain K12)"                         
## [13] "Escherichia coli CFT073"                               
## [14] "Enterococcus faecalis V583"                            
## [15] "Staphylococcus aureus"                                 
## [16] "Escherichia coli UTI89"                                
## [17] "Staphylococcus aureus MUF168"                          
## [18] "Staphylococcus aureus RF122"                           
## [19] "Staphylococcus aureus (strain bovine RF122 / ET3-1)"   
## [20] "Streptococcus pneumoniae"                              
## [21] "Lactobacillus casei"                                   
## [22] "Mycobacterium tuberculosis (strain ATCC 25177 / H37Ra)"
## [23] "Mycobacterium tuberculosis H37Rv (GCA_000667805)"      
## [24] "Mycobacterium tuberculosis"                            
## [25] "Mycobacterium tuberculosis (strain ATCC 25618 / H37Rv)"
## [26] "Mycolicibacterium smegmatis MC2 155"                   
## [27] "Mycobacterium avium"                                   
## [28] "synthetic construct"                                   
## [29] "Haloferax volcanii"                                    
## [30] "Klebsiella pneumoniae"                                 
## [31] "Schistosoma mansoni"                                   
## [32] "Pneumocystis carinii"                                  
## [33] "Mus musculus"                                          
## [34] "Gallus gallus"                                         
## [35] "Homo sapiens"                                          
## [36] "Cryptosporidium hominis"                               
## [37] "Candida albicans"                                      
## [38] "[Candida] glabrata CBS 138"                            
## [39] "[Candida] glabrata"                                    
## [40] "Toxoplasma gondii"                                     
## [41] "Babesia bovis T2Bo"                                    
## [42] "Babesia bovis"                                         
## [43] "Plasmodium vivax"                                      
## [44] "Plasmodium falciparum"</code></pre>
</div>
<div id="retrieve-and-process-structures-from-the-pdb" class="section level2">
<h2 class="hasAnchor">
<a href="#retrieve-and-process-structures-from-the-pdb" class="anchor"></a>Retrieve and process structures from the PDB</h2>
<p>Having identified relevant PDB structures through the hmmer search we proceed by fetching and pre-processing the PDB files with functions <strong>get.pdb()</strong> and <strong>pdbsplit()</strong>.</p>
<p>As in the previous vignette, we are interested in protein structures without missing in-structure residues, and we also want to limit the number of identifical conformers:</p>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="co"># fetch and split PDBs</span>
<span class="kw">raw.files</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/get.pdb.html">get.pdb</a></span>(<span class="kw">ids</span>, path = <span class="st">"raw_pdbs"</span>, gzip=<span class="fl">TRUE</span>)
<span class="kw">files</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbsplit.html">pdbsplit</a></span>(<span class="kw">raw.files</span>, ids = <span class="kw">ids</span>,
                  path = <span class="st">"raw_pdbs/split_chain"</span>, ncore=<span class="fl">4</span>)
<span class="kw">pdbs.all</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbaln.html">pdbaln</a></span>(<span class="kw">files</span>)

<span class="co"># exclude hits with fusion proteins</span>
<span class="kw">gaps</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/gap.inspect.html">gap.inspect</a></span>(<span class="kw">pdbs.all</span><span class="op">$</span><span class="kw">ali</span>)
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/trim.html">trim</a></span>(<span class="kw">pdbs.all</span>, row.inds=<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">gaps</span><span class="op">$</span><span class="kw">row</span> <span class="op">&gt;</span> <span class="fl">200</span>))

<span class="co"># exclude specific hits</span>
<span class="kw">excl.inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"5dxv"</span>), <span class="kw">grep</span>, <span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span>))
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/trim.html">trim</a></span>(<span class="kw">pdbs</span>, row.inds=<span class="op">-</span><span class="kw">excl.inds</span>)

<span class="co"># exclude structures with missing residues</span>
<span class="kw">conn</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/inspect.connectivity.html">inspect.connectivity</a></span>(<span class="kw">pdbs</span>, cut=<span class="fl">4.05</span>)
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/trim.html">trim</a></span>(<span class="kw">pdbs</span>, row.inds=<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">conn</span>))
          
<span class="co"># exclude conformational redundant structures</span>
<span class="kw">rd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/filter.rmsd.html">filter.rmsd</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>, cutoff=<span class="fl">0.25</span>, fit=<span class="fl">TRUE</span>, ncore=<span class="fl">4</span>)
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/trim.html">trim</a></span>(<span class="kw">pdbs</span>, row.inds=<span class="kw">rd</span><span class="op">$</span><span class="kw">ind</span>)
</pre></div>
<p>In this particular case a standard sequence alignment (e.g. through function <strong>pdbaln()</strong> or <strong>seqaln()</strong>) is not sufficient for a correct alignment. We will therefore make use of the Pfam profile alignment, and align our selected PDBs to this using argument <code>profile</code> to function <strong>seqaln()</strong>. Subsequently, we re-read the fasta file, and use function <strong>read.fasta.pdb()</strong> to obtain aligned C-alpha atom data (including coordinates etc.) for the PDB ensemble:</p>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="co"># align pdbs to Pfam-profile alignment</span>
<span class="kw">aln</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/seqaln.html">seqaln</a></span>(<span class="kw">pdbs</span>, profile=<span class="kw">pfam.aln</span>, exefile=<span class="st">"clustalo"</span>, extra.args=<span class="st">"--dealign"</span>)

<span class="co"># final alignment will also contain the profile</span>
<span class="co"># store only PDBs in alignment</span>
<span class="kw">aln</span><span class="op">$</span><span class="kw">ali</span> <span class="op">&lt;-</span> <span class="kw">aln</span><span class="op">$</span><span class="kw">ali</span>[<span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span>),]
<span class="kw">aln</span><span class="op">$</span><span class="kw">id</span>  <span class="op">&lt;-</span> <span class="kw">aln</span><span class="op">$</span><span class="kw">id</span>[<span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span>)]

<span class="co"># re-read PDBs to match the new alignment</span>
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.fasta.pdb.html">read.fasta.pdb</a></span>(<span class="kw">aln</span>)

<span class="co"># exclude gap-only columns</span>
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/trim.html">trim</a></span>(<span class="kw">pdbs</span>)

<span class="co"># refit coordinates</span>
<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbfit.html">pdbfit</a></span>(<span class="kw">pdbs</span>)
</pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="co"># refit coordinates, and write PDBs to disk</span>
<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbfit.html">pdbfit</a></span>(<span class="kw">pdbs</span>, outpath=<span class="st">"flsq/"</span>)
</pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="co"># fetch IDs again</span>
<span class="kw">ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/basename.pdb.html">basename.pdb</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span>)
<span class="kw">species</span> <span class="op">&lt;-</span> <span class="kw">hmm</span><span class="op">$</span><span class="kw">hit.tbl</span><span class="op">$</span><span class="kw">species</span>[<span class="kw">hmm</span><span class="op">$</span><span class="kw">hit.tbl</span><span class="op">$</span><span class="kw">acc</span> <span class="op">%in%</span> <span class="kw">ids</span>]
</pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit">
<span class="co"># labels for annotating plots</span>
<span class="kw">labs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span>(<span class="kw">species</span>, <span class="fl">1</span>,<span class="fl">1</span>), <span class="st">". "</span>, 
              <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span>(<span class="kw">species</span>, <span class="st">" "</span>), <span class="fu">function</span>(<span class="kw">x</span>) <span class="kw">x</span>[<span class="fl">2</span>]), sep=<span class="st">""</span>)
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="kw">labs</span>))
</pre></div>
<pre><code>##  [1] "G. stearothermophilus" "B. anthracis"          "M. profunda"          
##  [4] "Y. pestis"             "K. pneumoniae"         "E. coli"              
##  [7] "E. faecalis"           "S. aureus"             "S. pneumoniae"        
## [10] "L. casei"              "M. tuberculosis"       "M. smegmatis"         
## [13] "M. avium"              "H. volcanii"           "S. mansoni"           
## [16] "P. carinii"            "M. musculus"           "G. gallus"            
## [19] "H. sapiens"            "C. albicans"           "[. glabrata"</code></pre>
<p>The <em>pdbs</em> object now contains <em>aligned</em> C-alpha atom data, including Cartesian coordinates, residue numbers, residue types, and B-factors. The sequence alignment is also stored by default to the FASTA format file ‘aln.fa’ (to view this you can use an alignment viewer such as SEAVIEW, see <em>Requirements</em> section above).</p>
</div>
<div id="sequence-conservation-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#sequence-conservation-analysis" class="anchor"></a>Sequence conservation analysis</h2>
<p>Function <strong>seqidentity()</strong> can be used to calculate the sequence identity for the PDBs ensemble. Below we also print a summary of the calculated sequence identities, and perform a clustering of the structures based on sequence identity:</p>
<div class="sourceCode" id="cb20"><pre class="downlit">
<span class="kw">seqide</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/seqidentity.html">seqidentity</a></span>(<span class="kw">pdbs</span>)
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">seqide</span>))
</pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##  0.2130  0.3080  0.3380  0.4485  0.3910  1.0000</code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit">
<span class="kw">hc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html">hclust</a></span>(<span class="fu"><a href="https://rdrr.io/r/stats/dist.html">as.dist</a></span>(<span class="fl">1</span><span class="op">-</span><span class="kw">seqide</span>))
<span class="kw">grps.seq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cutree.html">cutree</a></span>(<span class="kw">hc</span>, h=<span class="fl">0.6</span>)
</pre></div>
<div class="sourceCode" id="cb23"><pre class="downlit">
<span class="fu"><a href="../../../reference/hclustplot.html">hclustplot</a></span>(<span class="kw">hc</span>, k=<span class="fl">3</span>, labels=<span class="kw">labs</span>, cex=<span class="fl">0.25</span>, fillbox=<span class="fl">FALSE</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_nma-dhfr-partII_files/figure-html/fig3-3-1.png" alt="Clustering of collected structures based on sequence identity." width="672"><p class="caption">
Clustering of collected structures based on sequence identity.
</p>
</div>
</div>
<div id="normal-modes-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#normal-modes-analysis" class="anchor"></a>Normal modes analysis</h2>
<p>Function <strong>nma.pdbs()</strong> will calculate the normal modes of each protein structures stored in the <em>pdbs</em> object. The normal modes are calculated on the full structures as provided by object <em>pdbs</em>. Use argument <code>rm.gaps=FALSE</code> to visualize fluctuations also of un-aligned residues:</p>
<div class="sourceCode" id="cb24"><pre class="downlit">
<span class="kw">modes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdbs</span>, rm.gaps=<span class="fl">FALSE</span>, ncore=<span class="fl">4</span>)
</pre></div>
<p>The <em>modes</em> object of class <em>enma</em> contains aligned normal mode data including fluctuations, RMSIP data (only when<code>rm.gaps=FALSE</code>), and aligned eigenvectors. A short summary of the <em>modes</em> object can be obtain by calling the function <strong>print()</strong>, and the aligned fluctuations can be plotted with function <strong>plot.enma()</strong>:</p>
<div class="sourceCode" id="cb25"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">modes</span>)
</pre></div>
<pre><code>## 
## Call:
##   nma.pdbs(pdbs = pdbs, rm.gaps = FALSE, ncore = 4)
## 
## Class:
##   enma
## 
## Number of structures:
##   259
## 
## Attributes stored:
##   - Aligned atomic fluctuations
##   - Aligned eigenvectors (gaps not removed)
##   - Dimensions of x$U.subspace: 720x381x259
## 
## Coordinates were aligned prior to NMA calculations
## 
## + attr: fluctuations, rmsip, U.subspace, L, full.nma, xyz,
##         call</code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit">
<span class="kw">cons</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/conserv.html">conserv</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">ali</span>)
<span class="kw">sse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbs2sse.html">pdbs2sse</a></span>(<span class="kw">pdbs</span>, rm.gaps=<span class="fl">FALSE</span>, ind=<span class="fl">1</span>)
</pre></div>
<pre><code>## Warning in dssp.pdb(pdb.ref, ...): Non-protein residues detected in input PDB:
## SO4, HOH</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit">
<span class="kw">resno</span> <span class="op">&lt;-</span> <span class="kw">pdbs</span><span class="op">$</span><span class="kw">resno</span>[<span class="fl">1</span>, ]
<span class="kw">resno</span>[<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="kw">resno</span>)] <span class="op">&lt;-</span> <span class="st">""</span>
<span class="kw">xlab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"Residue number (reference PDB: "</span>, <span class="fu"><a href="../../../reference/basename.pdb.html">basename.pdb</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span>[<span class="fl">1</span>]), <span class="st">")"</span>)

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(mfrow=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">2</span>,<span class="fl">1</span>), mar=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">4</span>, <span class="fl">4</span>, <span class="fl">2</span>, <span class="fl">2</span>))
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">modes</span>, <span class="kw">pdbs</span>, ylim=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>,<span class="fl">2</span>), col=<span class="kw">grps.seq</span>, label=<span class="kw">NULL</span>, xlab=<span class="kw">xlab</span>)
</pre></div>
<pre><code>## Warning in dssp.pdb(pdb.ref, ...): Non-protein residues detected in input PDB:
## SO4, HOH</code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit">
<span class="fu"><a href="../../../reference/plot.bio3d.html">plot.bio3d</a></span>(<span class="kw">cons</span>, resno=<span class="kw">resno</span>, sse=<span class="kw">sse</span>, ylab=<span class="st">"Conservation"</span>, xlab=<span class="kw">xlab</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_nma-dhfr-partII_files/figure-html/fig3-4-1.png" alt="Flexibility profiles and sequence conservation. The figure shows the modes fluctuations colored according their sequence identity. The lower panel shows the sequence conservation for the PDBs. The plot is generated with function **plot.enma()** along with function call to **conerv()**." width="576"><p class="caption">
Flexibility profiles and sequence conservation. The figure shows the modes fluctuations colored according their sequence identity. The lower panel shows the sequence conservation for the PDBs. The plot is generated with function <strong>plot.enma()</strong> along with function call to <strong>conerv()</strong>.
</p>
</div>
<p>In some cases it can be difficult to interpret the fluctuation plot when all lines are plotted on top of each other. Argument <code>spread=TRUE</code> adds a small gap between grouped fluctuation profiles. Use this argument in combination with a new groups (<code>grps</code>) variable to function <strong>plot.enma()</strong>:</p>
<div class="sourceCode" id="cb32"><pre class="downlit">
<span class="kw">grps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fl">NA</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw">grps.seq</span>))
<span class="kw">grps</span>[<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"coli"</span>, <span class="kw">labs</span>)]<span class="op">=</span><span class="fl">1</span>
<span class="kw">grps</span>[<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"aureus"</span>, <span class="kw">labs</span>)]<span class="op">=</span><span class="fl">2</span>
<span class="kw">grps</span>[<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"anthracis"</span>, <span class="kw">labs</span>)]<span class="op">=</span><span class="fl">3</span>
<span class="kw">grps</span>[<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"tubercu"</span>, <span class="kw">labs</span>)]<span class="op">=</span><span class="fl">4</span>
<span class="kw">grps</span>[<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"casei"</span>, <span class="kw">labs</span>)]<span class="op">=</span><span class="fl">5</span>
<span class="kw">grps</span>[<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"sapiens"</span>, <span class="kw">labs</span>)]<span class="op">=</span><span class="fl">6</span>
<span class="kw">grps</span>[<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"albicans"</span>, <span class="kw">labs</span>)]<span class="op">=</span><span class="fl">7</span>
<span class="kw">grps</span>[<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"glabrata"</span>, <span class="kw">labs</span>)]<span class="op">=</span><span class="fl">8</span>
<span class="kw">grps</span>[<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"carinii"</span>, <span class="kw">labs</span>)]<span class="op">=</span><span class="fl">9</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">modes</span>, pdbs=<span class="kw">pdbs</span>, col=<span class="kw">grps</span>, spread=<span class="fl">TRUE</span>, ylim=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>,<span class="fl">1.5</span>), label=<span class="kw">NULL</span>)
</pre></div>
<pre><code>## Warning in dssp.pdb(pdb.ref, ...): Non-protein residues detected in input PDB:
## NDP, TOP, HOH</code></pre>
<div class="figure">
<img src="Bio3D_nma-dhfr-partII_files/figure-html/fig3-5-1.png" alt="Flexibility profiles for selected species. The plot is generated with function **plot.enma()** with argument `spread=TRUE`." width="672"><p class="caption">
Flexibility profiles for selected species. The plot is generated with function <strong>plot.enma()</strong> with argument <code>spread=TRUE</code>.
</p>
</div>
</div>
<div id="visualize-modes" class="section level2">
<h2 class="hasAnchor">
<a href="#visualize-modes" class="anchor"></a>Visualize modes</h2>
<p>A function call to <strong>mktrj.enma()</strong> will generate a trajectory PDB file for the visualization of a specific normal mode for one of the structures in the <em>pdbs</em> object. This allows for a visual comparison of the calculated normal modes. Below we make a PDB trajectory of the first mode (argument <code>m.inds=1</code>) of 3 relevant species (e.g. argument <code>s.inds=1</code>). Note that we use <strong>grep()</strong> to fetch the indices (in the <em>modes</em> and <em>pdbs</em> objects) of the relevant species:</p>
<div class="sourceCode" id="cb34"><pre class="downlit">
<span class="kw">inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"coli"</span>, <span class="kw">species</span>)[<span class="fl">1</span>], 
          <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"sapiens"</span>, <span class="kw">species</span>)[<span class="fl">1</span>],
          <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"albicans"</span>, <span class="kw">species</span>)[<span class="fl">1</span>])

<span class="co"># E.coli</span>
<span class="fu"><a href="../../../reference/mktrj.html">mktrj</a></span>(<span class="kw">modes</span>, <span class="kw">pdbs</span>, m.inds=<span class="fl">1</span>, s.inds=<span class="kw">inds</span>[<span class="fl">1</span>], file=<span class="st">"ecoli-mode1.pdb"</span>)

<span class="co"># H. sapiens</span>
<span class="fu"><a href="../../../reference/mktrj.html">mktrj</a></span>(<span class="kw">modes</span>, <span class="kw">pdbs</span>, m.inds=<span class="fl">1</span>, s.inds=<span class="kw">inds</span>[<span class="fl">2</span>], file=<span class="st">"hsapiens-mode1.pdb"</span>)

<span class="co"># C. albicans</span>
<span class="fu"><a href="../../../reference/mktrj.html">mktrj</a></span>(<span class="kw">modes</span>, <span class="kw">pdbs</span>, m.inds=<span class="fl">1</span>, s.inds=<span class="kw">inds</span>[<span class="fl">3</span>], file=<span class="st">"calbicans-mode1.pdb"</span>)
</pre></div>
<div class="figure">
<img src="figures/visualize-4hfrs.png" alt=""><p class="caption">Mode comparison of <em>E.coli</em>, <em>H.sapiens</em>, and <em>C.albicans</em>. The trajectories are made with function <strong>mktrj.enma()</strong> and visualized in PyMol.</p>
</div>
</div>
</div>
<div id="document-details" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#document-details" class="anchor"></a>Document Details</h1>
<p>This document is shipped with the Bio3D package in both R and PDF formats. All code can be extracted and automatically executed to generate Figures and/or the PDF with the following commands:</p>
<div class="sourceCode" id="cb35"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://github.com/rstudio/rmarkdown">rmarkdown</a></span>)
<span class="fu"><a href="https://rdrr.io/pkg/rmarkdown/man/render.html">render</a></span>(<span class="st">"Bio3D_nma-dhfr-partII.Rmd"</span>, <span class="st">"all"</span>)
</pre></div>
<div style="page-break-after: always;"></div>
</div>
<div id="information-about-the-current-bio3d-session" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#information-about-the-current-bio3d-session" class="anchor"></a>Information About the Current Bio3D Session</h1>
<div class="sourceCode" id="cb36"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>(), <span class="fl">FALSE</span>)
</pre></div>
<pre><code>## R version 3.6.0 (2019-04-26)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS  10.15.5
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] bio3d_2.4-1.9000
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.5         digest_0.6.25      crayon_1.3.4       rprojroot_1.3-2   
##  [5] assertthat_0.2.1   grid_3.6.0         R6_2.4.1           backports_1.1.8   
##  [9] magrittr_1.5       evaluate_0.14      highr_0.8          stringi_1.4.6     
## [13] rlang_0.4.6.9000   fs_1.4.2           rmarkdown_2.3.2    pkgdown_1.5.1.9000
## [17] desc_1.2.0         tools_3.6.0        stringr_1.4.0      parallel_3.6.0    
## [21] yaml_2.2.1         xfun_0.15          compiler_3.6.0     memoise_1.1.0     
## [25] htmltools_0.5.0    knitr_1.29</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references hanging-indent">
<div id="ref-grant06">
<p>Grant, B. J., A. P. D. C Rodrigues, K. M. Elsawy, A. J. Mccammon, and L. S. D. Caves. 2006. “Bio3d: An R Package for the Comparative Analysis of Protein Structures.” <em>Bioinformatics</em> 22: 2695–6. <a href="https://doi.org/10.1093/bioinformatics/btl461">https://doi.org/10.1093/bioinformatics/btl461</a>.</p>
</div>
<div id="ref-skjaerven15">
<p>Skjaerven, L., X. Q. Yao, G. Scarabelli, and B. J. Grant. 2015. “Integrating Protein Structural Dynamics and Evolutionary Analysis with Bio3d.” <em>BMC Bioinformatics</em> 15: 399. <a href="https://doi.org/10.1186/s12859-014-0399-6">https://doi.org/10.1186/s12859-014-0399-6</a>.</p>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>The latest version of the package, full documentation and further vignettes (including detailed installation instructions) can be obtained from the main Bio3D website: <a href="http://thegrantlab.org/bio3d/">http://thegrantlab.org/bio3d/</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>This vignette contains executable examples, see <code><a href="https://rdrr.io/r/utils/vignette.html">help(vignette)</a></code> for further details.<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="tidyverse">
  <p>Authored by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="author">
  <p>
    Built with ❤️ and <a href="https://pkgdown.r-lib.org">pkgdown</a> © 2020.
  </p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '1270c4079b26a138795263974bbf302d',
    indexName: 'tidyverse',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
