<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Enhanced Normal Modes Analysis with Bio3D • Bio3D</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../../../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../../../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../../../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../../../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../../../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../../../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="../../../tidyverse.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../../bootstrap-toc.css">
<script src="../../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../../pkgdown.css" rel="stylesheet">
<script src="../../../pkgdown.js"></script><link href="../../../tidyverse-2.css" rel="stylesheet">
<!--optional theme
<link href="../../../.css" rel="stylesheet">
--><!-- docsearch --><script src="../../../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../../../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Enhanced Normal Modes Analysis with Bio3D">
<meta property="og:description" content="bio3d">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../../../index.html">
        bio3d</a>
        <div class="info hidden-xs hidden-sm">
<!--          <span class="partof">part of the <a href="https://tidyverse.org">tidyverse</a></span>
-->
          <span class="version version-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4-1.9000</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">INTRODUCTORY TUTORIALS</li>
    <li>
      <a href="../../../articles/online/install_vignette/Bio3D_install.html">Installing Bio3D</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/user-guide">New user guide</a>
    </li>
    <li>
      <a href="../../../articles/online/intro_vignette/Bio3D_introduction.html">Getting started with Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/pdb_vignette/Bio3D_pdb.html">PDB structure manipulation and analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/pca_vignette/Bio3D_pca.html">Comparative sequence and structure analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/nma_vignette/Bio3D_nma.html">Enhanced Normal Modes Analysis</a>
    </li>
    <li>
      <a href="../../../articles/online/traj_vignette/Bio3D_md.html">Beginning trajectory analysis with Bio3D </a>
    </li>
    <li class="divider">
    <li class="dropdown-header">ADVANCED TUTORIALS</li>
    <li>
      <a href="../../../articles/online/enma_vignettes/Bio3D_nma-dhfr-partI.html">Ensemble NMA of E.coli DHFR structures</a>
    </li>
    <li>
      <a href="../../../articles/online/enma_vignettes/Bio3D_nma-dhfr-partII.html">Ensemble NMA across multiple species of DHFR</a>
    </li>
    <li>
      <a href="../../../articles/online/cna_vignette/cna_vignette.spin.html">Protein structure network analysis with Bio3D</a>
    </li>
    <li>
      <a href="../../../articles/online/eddm_vignette/Bio3D_eddm.spin.html">Ensemble Difference Distance Matrix (eDDM) analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">WEBAPP TUTORIALS</li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/download/webapp-tutorials?download=22:bio3d-webapp-pca">Online protein structure analysis with Bio3D-Web (PDF)</a>
    </li>
    <li>
      <a href="http://thegrantlab.org/bio3d_v2/webapps">Online Normal Mode Analysis with Bio3D WebApps (To Update)</a>
    </li>
    <li class="divider">
    <li>
      <a href="../../../articles/index.html">All vignettes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Documentation
  </a>
</li>
        <li>
  <a href="../../../FAQ.html">
    <span class="fa fa-comments-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../../../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    News
  </a>
</li>
<li>
  <a href="http://thegrantlab.org/">
    <span class="fa fa-external-link"></span>
     
    Grant Lab
  </a>
</li>
<li>
  <a href="https://bitbucket.org/Grantlab/bio3d/src/master/">
    <span class="fa fa-bitbucket"></span>
     
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Bio3D_nma_files/header-attrs-2.3.2/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Enhanced Normal Modes Analysis with Bio3D</h1>
                        <h4 class="author">Lars Skjaerven, Xin-Qiu Yao, Guido Scarabelli &amp; Barry J. Grant</h4>
            
      
      
      <div class="hidden name"><code>Bio3D_nma.Rmd</code></div>

    </div>

    
    
<div style="page-break-after: always;"></div>
<div id="background" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#background" class="anchor"></a>Background</h1>
<p>Bio3D<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> is an R package that provides interactive tools for structural bioinformatics . The primary focus of Bio3D is the analysis of biomolecular structure, sequence and simulation data <span class="citation">(Grant et al. 2006)</span>.</p>
<p>Normal mode analysis (NMA) is one of the major simulation techniques used to probe large-scale motions in biomolecules. Typical application is for the prediction of functional motions in proteins. Version 2.0 of the Bio3D package now includes extensive NMA facilities <span class="citation">(Skjaerven et al. 2015)</span>. These include a unique collection of multiple elastic network model (ENM) force-fields (see <strong>Example 1</strong> below), automated ensemble analysis methods (<strong>Example 2</strong>), variance weighted NMA (<strong>Example 3</strong>), and NMA with user-defined force fields (<strong>Example 4</strong>). Here we demonstrate the use of these new features with working code that comprise complete executable examples<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>.</p>
<div id="requirements" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#requirements" class="anchor"></a>Requirements</h2>
<p>Detailed instructions for obtaining and installing the Bio3D package on various platforms can be found in the <a href="http://thegrantlab.org/bio3d/tutorials">Installing Bio3D Vignette</a> available both on-line and from within the Bio3D package. In addition to Bio3D the <em>MUSCLE</em> multiple sequence alignment program (available from the <a href="http://www.drive5.com/muscle/">muscle home page</a>) must be installed on your system and in the search path for executables. Please see the installation vignette for further details.</p>
</div>
<div id="about-this-document" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#about-this-document" class="anchor"></a>About this document</h2>
<p>This vignette was generated using <strong>Bio3D version 2.4.1.9000</strong>.</p>
</div>
</div>
<div id="example-1-basic-normal-mode-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#example-1-basic-normal-mode-analysis" class="anchor"></a>Example 1: Basic Normal Mode Analysis</h1>
<div id="example-1a-normal-mode-calculation" class="section level2">
<h2 class="hasAnchor">
<a href="#example-1a-normal-mode-calculation" class="anchor"></a>Example 1A: Normal mode calculation</h2>
<p>Normal mode analysis (NMA) of a single protein structure can be carried out by providing a PDB object to the function <strong>nma()</strong>. In the code below we first load the Bio3D package and then download an example structure of hen egg white lysozyme (PDB id <em>1hel</em>) with the function <strong>read.pdb()</strong>. Finally the function <strong>nma()</strong> is used perform the normal mode calculation:</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="http://thegrantlab.org/bio3d">bio3d</a></span>)
<span class="kw">pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"1hel"</span>)
</pre></div>
<pre><code>##   Note: Accessing on-line PDB file</code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="kw">modes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb</span>)
</pre></div>
<pre><code>##  Building Hessian...     Done in 0.015 seconds.
##  Diagonalizing Hessian...    Done in 0.076 seconds.</code></pre>
<p>A short summary of the returned <em>nma</em> object contained within the new variable <em>modes</em> can be obtained by simply calling the function <strong>print()</strong>:</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">modes</span>)
</pre></div>
<pre><code>## 
## Call:
##   nma.pdb(pdb = pdb)
## 
## Class:
##   VibrationalModes (nma)
## 
## Number of modes:
##   387 (6 trivial)
## 
## Frequencies:
##   Mode 7:    0.018
##   Mode 8:    0.019
##   Mode 9:    0.024
##   Mode 10:   0.025
##   Mode 11:   0.028
##   Mode 12:   0.029
## 
## + attr: modes, frequencies, force.constants, fluctuations,
##         U, L, xyz, mass, temp, triv.modes, natoms, call</code></pre>
<p>This reveals the function call resulting in the <em>nma</em> object along with the total number of stored normal modes. For PDB id <em>1hel</em> there are 129 amino acid residues, and thus <span class="math inline">\(387\)</span> modes (<span class="math inline">\(3*129=387\)</span>) in this object. The first six modes are so-called trivial modes with zero frequency and correspond to rigid-body rotation and translation. The frequency of the next six lowest-frequency modes is also printed.</p>
<p>Note that the returned <em>nma</em> object consists of a number of attributes listed on the <em>+attr:</em> line. These attributes contain the detailed results of the calculation and their complete description can be found on the <strong>nma()</strong> functions help page accessible with the command: <code><a href="../../../reference/nma.html">help(nma)</a></code>. To get a quick overview of the results one can simply call the <strong>plot()</strong> function on the returned <em>nma</em> object. This will produce a summary plot of (1) the eigenvalues, (2) the mode frequencies, and (3) the atomic fluctuations (See Figure 1).</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">modes</span>, sse=<span class="kw">pdb</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_nma_files/figure-html/ex1A-modesplot-1.png" alt="Summary plot of NMA results for hen egg white lysozyme (PDB id *1hel*). The optional `sse=pdb` argument provided to **plot.nma()** results in a secondary structure schematic being added to the top and bottom margins of the fluctuation plot (helices black and strands gray). Note the larger fluctuations predicted for loop regions." width="700"><p class="caption">
Summary plot of NMA results for hen egg white lysozyme (PDB id <em>1hel</em>). The optional <code>sse=pdb</code> argument provided to <strong>plot.nma()</strong> results in a secondary structure schematic being added to the top and bottom margins of the fluctuation plot (helices black and strands gray). Note the larger fluctuations predicted for loop regions.
</p>
</div>
</div>
<div id="example-1b-specifying-a-force-field" class="section level2">
<h2 class="hasAnchor">
<a href="#example-1b-specifying-a-force-field" class="anchor"></a>Example 1B: Specifying a force field</h2>
<p>The main Bio3D normal mode analysis function, <strong>nma()</strong>, requires a set of coordinates, as obtained from the <strong>read.pdb()</strong> function, and the specification of a force field describing the interactions between constituent atoms. By default the <em>calpha</em> force field originally developed by Konrad Hinsen is utilized <span class="citation">(Hinsen et al. 2000)</span>. This employs a spring force constant differentiating between nearest-neighbor pairs along the backbone and all other pairs. The force constant function was parameterized by fitting to a local minimum of a crambin model using the AMBER94 force field. However, a number of additional force fields are also available, as well as functionality for providing customized force constant functions. Full details of available force fields can be obtained with the command <code><a href="../../../reference/load.enmff.html">help(load.enmff)</a></code>. With the code below we briefly demonstrate their usage along with a simple comparison of the modes obtained from two of the most commonly used force fields:</p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/help.html">help</a></span>(<span class="kw">load.enmff</span>)
</pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="co"># Calculate modes with various force fields</span>
<span class="kw">modes.a</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb</span>, ff=<span class="st">"calpha"</span>)
<span class="kw">modes.b</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb</span>, ff=<span class="st">"anm"</span>)
<span class="kw">modes.c</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb</span>, ff=<span class="st">"pfanm"</span>)
<span class="kw">modes.d</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb</span>, ff=<span class="st">"reach"</span>)
<span class="kw">modes.e</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb</span>, ff=<span class="st">"sdenm"</span>)

<span class="co"># Root mean square inner product (RMSIP)</span>
<span class="kw">r</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/rmsip.html">rmsip</a></span>(<span class="kw">modes.a</span>, <span class="kw">modes.b</span>)
</pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">r</span>, xlab=<span class="st">"ANM"</span>, ylab=<span class="st">"C-alpha FF"</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_nma_files/figure-html/ex1B-modesplot-1.png" alt="Analysis of mode similarity between modes obtained from the *ANM* and *calpha* force fields by calculating mode overlap and root mean square inner product (RMSIP) with function **rmsip()**. An RMSIP value of *1* depicts identical directionality of the two mode subspaces. " width="700"><p class="caption">
Analysis of mode similarity between modes obtained from the <em>ANM</em> and <em>calpha</em> force fields by calculating mode overlap and root mean square inner product (RMSIP) with function <strong>rmsip()</strong>. An RMSIP value of <em>1</em> depicts identical directionality of the two mode subspaces.
</p>
</div>
</div>
<div id="example-1c-normal-mode-analysis-of-the-groel-subunit" class="section level2">
<h2 class="hasAnchor">
<a href="#example-1c-normal-mode-analysis-of-the-groel-subunit" class="anchor"></a>Example 1C: Normal mode analysis of the GroEL subunit</h2>
<p>Bio3D includes a number of functions for analyzing and visualizing the normal modes. In the example below we illustrate this functionality on the GroEL subunit. GroEL is a multimeric protein consisting of 14 identical subunits organized in three distinct domains inter-connected by two hinge regions facilitating large conformational changes.</p>
<p>We will investigate the normal modes through (<strong>1</strong>) mode visualization to illustrate the nature of the motions; (<strong>2</strong>) cross-correlation analysis to determine correlated regions; (<strong>3</strong>) deformation analysis to measure the local flexibility of the structure; (<strong>4</strong>) overlap analysis to determine which modes contribute to a given conformational change; and (<strong>5</strong>) domain analysis to identify regions of the protein moving as rigid parts.</p>
<div id="calculate-the-normal-modes" class="section level3">
<h3 class="hasAnchor">
<a href="#calculate-the-normal-modes" class="anchor"></a>Calculate the normal modes</h3>
<p>In the code below we download a structure of GroEL (PDB-id <em>1sx4</em>) and use <strong>atom.select()</strong> to select one of the 14 subunits prior to the call to <strong>nma()</strong>:</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="co"># Download PDB, calcualte normal modes of the open subunit</span>
<span class="kw">pdb.full</span>   <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"1sx4"</span>)
<span class="kw">pdb.open</span>   <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/trim.html">trim.pdb</a></span>(<span class="kw">pdb.full</span>, <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb.full</span>, chain=<span class="st">"A"</span>))
<span class="kw">modes</span>      <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb.open</span>)
</pre></div>
</div>
<div id="mode-visualization" class="section level3">
<h3 class="hasAnchor">
<a href="#mode-visualization" class="anchor"></a>Mode visualization</h3>
<p>With Bio3D you can visualize the normal modes either by generating a trajectory file which can be loaded into a molecular viewer program (e.g. VMD or PyMOL), or through a vector field representation in PyMOL. Both functions, <strong>mktrj.nma()</strong> and <strong>pymol.modes()</strong>, takes an <em>nma</em> object as input in addition to the mode index specifying which mode to visualize:</p>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="co"># Make a PDB trajectory</span>
<span class="fu"><a href="../../../reference/mktrj.html">mktrj</a></span>(<span class="kw">modes</span>, mode=<span class="fl">7</span>)

<span class="co"># Vector field representation (see Figure 3.)</span>
<span class="fu"><a href="../../../reference/pymol.html">pymol</a></span>(<span class="kw">modes</span>, mode=<span class="fl">7</span>)
</pre></div>
<div class="figure">
<img src="figures/groel-traj-vector.png" alt=""><p class="caption">Visualization of the first non-trivial mode of the GroEL subunit. Visualization is provided through a trajectory file (left), or vector field representation (right).</p>
</div>
</div>
<div id="cross-correlation-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#cross-correlation-analysis" class="anchor"></a>Cross-correlation analysis</h3>
<p>Function <strong>dccm.nma()</strong> calculates the cross-correlation matrix of the <em>nma</em> object. Function <strong>plot.dccm()</strong> will draw a correlation map, and 3D visualization of correlations is provided through function <strong>pymol.dccm()</strong>:</p>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="co"># Calculate the cross-correlation matrix</span>
<span class="kw">cm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/dccm.html">dccm</a></span>(<span class="kw">modes</span>)
</pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="co"># Plot a correlation map with plot.dccm(cm)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">cm</span>, sse=<span class="kw">pdb.open</span>, contour=<span class="kw">F</span>, col.regions=<span class="fu"><a href="../../../reference/bwr.colors.html">bwr.colors</a></span>(<span class="fl">20</span>), at=<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="op">-</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">0.1</span>) )
</pre></div>
<div class="figure">
<img src="Bio3D_nma_files/figure-html/ex1C-dccmplot-1.png" alt="Correlation map revealing correlated and anti-correlated regions in the protein structure." width="480"><p class="caption">
Correlation map revealing correlated and anti-correlated regions in the protein structure.
</p>
</div>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="co"># View the correlations in the structure (see Figure 5.)</span>
<span class="fu"><a href="../../../reference/pymol.html">pymol</a></span>(<span class="kw">cm</span>, <span class="kw">pdb.open</span>, type=<span class="st">"launch"</span>)
</pre></div>
<div class="figure">
<img src="figures/groel-correl.png" alt=""><p class="caption">Correlated (left) and anti-correlated (right) residues depicted with red and blue lines, respectively. The figures demonstrate the output of function <strong>pymol.dccm()</strong>.</p>
</div>
</div>
<div id="fluctuation-and-deformation-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#fluctuation-and-deformation-analysis" class="anchor"></a>Fluctuation and Deformation analysis</h3>
<p>Deformation analysis provides a measure for the amount of local flexibility in the protein structure - <em>i.e.</em> atomic motion relative to neighboring atoms. It differs from <em>fluctuations</em> (<em>e.g.</em> RMSF values) which provide amplitudes of the absolute atomic motion. Below we calculate deformation energies (with <strong>deformation.nma()</strong>) and atomic fluctuations (with <strong>fluct.nma()</strong>) of the first three modes and visualize the results in PyMOL:</p>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="co"># Deformation energies</span>
<span class="kw">defe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/deformation.nma.html">deformation.nma</a></span>(<span class="kw">modes</span>)
<span class="kw">defsums</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span>(<span class="kw">defe</span><span class="op">$</span><span class="kw">ei</span>[,<span class="fl">1</span><span class="op">:</span><span class="fl">3</span>])

<span class="co"># Fluctuations</span>
<span class="kw">flucts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/fluct.nma.html">fluct.nma</a></span>(<span class="kw">modes</span>, mode.inds=<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">7</span>,<span class="fl">9</span>))
</pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="co"># Write to PDB files (see Figure 6.)</span>
<span class="fu"><a href="../../../reference/write.pdb.html">write.pdb</a></span>(pdb=<span class="kw">NULL</span>, xyz=<span class="kw">modes</span><span class="op">$</span><span class="kw">xyz</span>, file=<span class="st">"R-defor.pdb"</span>, b=<span class="kw">defsums</span>)
<span class="fu"><a href="../../../reference/write.pdb.html">write.pdb</a></span>(pdb=<span class="kw">NULL</span>, xyz=<span class="kw">modes</span><span class="op">$</span><span class="kw">xyz</span>, file=<span class="st">"R-fluct.pdb"</span>, b=<span class="kw">flucts</span>)
</pre></div>
<div class="figure">
<img src="figures/groel_fluct-deformation.png" alt=""><p class="caption">Atomic fluctuations (left) and deformation energies (right) visualized in PyMOL.</p>
</div>
</div>
<div id="overlap-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#overlap-analysis" class="anchor"></a>Overlap analysis</h3>
<p>Finally, we illustrate overlap analysis to compare a conformational difference vector with the normal modes to identify which modes contribute to a given conformational change (i.e. the difference between the open and closed state of the GroEL subunit).</p>
<div class="sourceCode" id="cb18"><pre class="downlit">
<span class="co"># Closed state of the subunit</span>
<span class="kw">pdb.closed</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/trim.html">trim.pdb</a></span>(<span class="kw">pdb.full</span>, <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb.full</span>, chain=<span class="st">"H"</span>))

<span class="co"># Align closed and open PDBs</span>
<span class="kw">aln</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/struct.aln.html">struct.aln</a></span>(<span class="kw">pdb.open</span>, <span class="kw">pdb.closed</span>, max.cycles=<span class="fl">0</span>)
<span class="kw">pdb.closed</span><span class="op">$</span><span class="kw">xyz</span> <span class="op">&lt;-</span> <span class="kw">aln</span><span class="op">$</span><span class="kw">xyz</span>

<span class="co"># Caclulate a difference vector</span>
<span class="kw">xyz</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span>(<span class="kw">pdb.open</span><span class="op">$</span><span class="kw">xyz</span>[<span class="kw">aln</span><span class="op">$</span><span class="kw">a.inds</span><span class="op">$</span><span class="kw">xyz</span>], <span class="kw">pdb.closed</span><span class="op">$</span><span class="kw">xyz</span>[<span class="kw">aln</span><span class="op">$</span><span class="kw">a.inds</span><span class="op">$</span><span class="kw">xyz</span>])
<span class="kw">diff</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/difference.vector.html">difference.vector</a></span>(<span class="kw">xyz</span>)

<span class="co"># Calculate overlap</span>
<span class="kw">oa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/overlap.html">overlap</a></span>(<span class="kw">modes</span>, <span class="kw">diff</span>)
</pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">oa</span><span class="op">$</span><span class="kw">overlap</span>, type=<span class="st">'h'</span>, xlab=<span class="st">"Mode index"</span>, ylab=<span class="st">"Squared overlap"</span>, ylim=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>,<span class="fl">1</span>))
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="kw">oa</span><span class="op">$</span><span class="kw">overlap</span>, col=<span class="fl">1</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/lines.html">lines</a></span>(<span class="kw">oa</span><span class="op">$</span><span class="kw">overlap.cum</span>, type=<span class="st">'b'</span>, col=<span class="fl">2</span>, cex=<span class="fl">0.5</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">5</span>)<span class="op">+</span><span class="fl">.5</span>, <span class="kw">oa</span><span class="op">$</span><span class="kw">overlap</span>[<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">5</span>)], <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Mode 1"</span>, <span class="st">"Mode 5"</span>), adj=<span class="fl">0</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_nma_files/figure-html/ex1C-overlapplot-1.png" alt="Overlap analysis between the modes of the open subunit and the conformational difference vector between the closed-open state." width="700"><p class="caption">
Overlap analysis between the modes of the open subunit and the conformational difference vector between the closed-open state.
</p>
</div>
</div>
<div id="domain-analysis-with-geostas" class="section level3">
<h3 class="hasAnchor">
<a href="#domain-analysis-with-geostas" class="anchor"></a>Domain analysis with GeoStaS</h3>
<p>Identification of regions in the protein that move as rigid bodies is facilitated with the implementation of the GeoStaS algorithm <span class="citation">(Romanowska, Nowinski, and Trylska 2012)</span>. Below we demonstrate the use of function <strong>geostas()</strong> on an <em>nma</em> object, and an ensemble of X-ray structures. See <code><a href="../../../reference/geostas.html">help(geostas)</a></code> for more details and further examples.</p>
<p><strong>GeoStaS with NMA</strong>: Starting from the calculated normal modes, <strong>geostas</strong> generates a conformational ensemble by interpolating along the eigenvectors of the first 5 normal modes of the GroEL subunit. This ensemble is then used to search for rigid domains:</p>
<div class="sourceCode" id="cb20"><pre class="downlit">
<span class="co"># Run geostas to find domains</span>
<span class="kw">gs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/geostas.html">geostas</a></span>(<span class="kw">modes</span>, k=<span class="fl">4</span>)
</pre></div>
<div class="sourceCode" id="cb21"><pre class="downlit">
<span class="co"># Write NMA trajectory with domain assignment</span>
<span class="fu"><a href="../../../reference/mktrj.html">mktrj</a></span>(<span class="kw">modes</span>, mode=<span class="fl">7</span>, chain=<span class="kw">gs</span><span class="op">$</span><span class="kw">grps</span>)
</pre></div>
<p><strong>GeoStaS with X-ray structure ensemble</strong>: Alternatively the same analysis can be performed on an ensemble of X-ray structures obtained from the PDB:</p>
<div class="sourceCode" id="cb22"><pre class="downlit">
<span class="co"># Define the ensemble PDB-ids</span>
<span class="kw">ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"1sx4_[A,B,H,I]"</span>, <span class="st">"1xck_[A-B]"</span>, <span class="st">"1sx3_[A-B]"</span>, <span class="st">"4ab3_[A-B]"</span>)

<span class="co"># Download and split PDBs by chain ID</span>
<span class="kw">raw.files</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/get.pdb.html">get.pdb</a></span>(<span class="kw">ids</span>, <span class="st">"groel_pdbs"</span>, gzip=<span class="fl">TRUE</span>)
<span class="kw">files</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbsplit.html">pdbsplit</a></span>(<span class="kw">raw.files</span>, <span class="kw">ids</span>, path = <span class="st">"groel_pdbs"</span>)

<span class="co"># Align and superimpose coordinates</span>
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbaln.html">pdbaln</a></span>(<span class="kw">files</span>, fit=<span class="fl">TRUE</span>)

<span class="co"># Run geostast to find domains</span>
<span class="kw">gs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/geostas.html">geostas</a></span>(<span class="kw">pdbs</span>, k=<span class="fl">4</span>)
</pre></div>
<div class="sourceCode" id="cb23"><pre class="downlit">
<span class="co"># Plot a atomic movement similarity matrix</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">gs</span>, margin.segments=<span class="kw">gs</span><span class="op">$</span><span class="kw">grps</span>, contour=<span class="fl">FALSE</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_nma_files/figure-html/plot_geostas-amsm-1.png" alt="Atomic movement similarity matrix with domain annotation." width="480"><p class="caption">
Atomic movement similarity matrix with domain annotation.
</p>
</div>
<div class="sourceCode" id="cb24"><pre class="downlit">
<span class="co"># Principal component analysis </span>
<span class="kw">gaps.pos</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/gap.inspect.html">gap.inspect</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>)
<span class="kw">xyz</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/fit.xyz.html">fit.xyz</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>[<span class="fl">1</span>, <span class="kw">gaps.pos</span><span class="op">$</span><span class="kw">f.inds</span>],
               <span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>[, <span class="kw">gaps.pos</span><span class="op">$</span><span class="kw">f.inds</span>],
               fixed.inds=<span class="kw">gs</span><span class="op">$</span><span class="kw">fit.inds</span>,
               mobile.inds=<span class="kw">gs</span><span class="op">$</span><span class="kw">fit.inds</span>)

<span class="kw">pc.xray</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pca.xyz.html">pca.xyz</a></span>(<span class="kw">xyz</span>)
</pre></div>
<div class="sourceCode" id="cb25"><pre class="downlit">
<span class="co"># Visualize PCs with colored domains (chain ID)</span>
<span class="fu"><a href="../../../reference/mktrj.html">mktrj</a></span>(<span class="kw">pc.xray</span>, pc=<span class="fl">1</span>, chain=<span class="kw">gs</span><span class="op">$</span><span class="kw">grps</span>)
</pre></div>
<div class="figure">
<img src="figures/geostas.png" alt=""><p class="caption">Visualization of domain assignment obtained from function <strong>geostas()</strong> using (left) an ensemble of X-ray structures and (right) NMA.</p>
</div>
</div>
</div>
</div>
<div id="example-2-all-atom-normal-mode-analysis-enm" class="section level1">
<h1 class="hasAnchor">
<a href="#example-2-all-atom-normal-mode-analysis-enm" class="anchor"></a>Example 2: All-atom normal mode analysis (ENM)</h1>
<p>The coarse-grained NMA approach used above omits potentially important inter-atomic interactions in the normal modes calculations. To provide a more detailed model Bio3D now offers an ENM using all atoms. We have recently shown that this all-atom approach provides normal mode vectors with an improved agreement to normal modes calculated with the full empiric force field (i.e. the Amber99SB force field) <span class="citation">(Yao, Skjaerven, and Grant 2016)</span>. The followning examples demonstrates some of the capabilities with all-atom ENM in Bio3D.</p>
<div id="example-2a-dihydrofolate-reductase" class="section level2">
<h2 class="hasAnchor">
<a href="#example-2a-dihydrofolate-reductase" class="anchor"></a>Example 2A: Dihydrofolate reductase</h2>
<p>In the example below we load a structure of <em>E.coli</em> Dihydrofolate reductase (DHFR) and call function <strong>aanma()</strong> on the <em>pdb</em> object to build the all-atom elastic network and calculate the normal modes. In this particular case we use argument <code>outmodes="calpha"</code> to specify the diagonalization of the effective Hessian for the calpha atoms only <span class="citation">(Hinsen et al. 2000)</span>. In this way we can obtain normal mode vectors only for the calpha atoms which facilitates e.g. comparison with coarse grained NMA.</p>
<div class="sourceCode" id="cb26"><pre class="downlit">
<span class="co"># read e.coli DHFR</span>
<span class="kw">pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"1rg7"</span>)
</pre></div>
<pre><code>##   Note: Accessing on-line PDB file</code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit">
<span class="co"># keep only protein + methotrexate</span>
<span class="kw">pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/trim.html">trim</a></span>(<span class="kw">pdb</span>, <span class="st">"notwater"</span>)

<span class="co"># calculate all-atom NMA with ENM, output calpha only</span>
<span class="kw">m.aa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/aanma.html">aanma</a></span>(<span class="kw">pdb</span>, outmodes=<span class="st">"calpha"</span>)
</pre></div>
<pre><code>##  Building Hessian...     Done in 1.141 seconds.
##  Extracting effective Hessian..  Done in 23.753 seconds.
##  Diagonalizing Hessian...    Done in 0.141 seconds.</code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit">
<span class="co"># write summary</span>
<span class="kw">m.aa</span>
</pre></div>
<pre><code>## 
## Call:
##   aanma.pdb(pdb = pdb, outmodes = "calpha")
## 
## Class:
##   VibrationalModes (nma)
## 
## Number of modes:
##   477 (6 trivial)
## 
## Frequencies:
##   Mode 7:    0.02
##   Mode 8:    0.022
##   Mode 9:    0.023
##   Mode 10:   0.026
##   Mode 11:   0.027
##   Mode 12:   0.03
## 
## + attr: modes, frequencies, force.constants, fluctuations,
##         U, L, xyz, mass, temp, triv.modes, natoms, call</code></pre>
<p>We can compare the directionality of mode vectors with results obtained from coarse grained NMA using function <strong>rmsip()</strong>:</p>
<div class="sourceCode" id="cb32"><pre class="downlit">
<span class="co"># compare with c-alpha modes</span>
<span class="kw">m.ca</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb</span>)
</pre></div>
<pre><code>##  Building Hessian...     Done in 0.016 seconds.
##  Diagonalizing Hessian...    Done in 0.139 seconds.</code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit">
<span class="fu"><a href="../../../reference/rmsip.html">rmsip</a></span>(<span class="kw">m.aa</span>, <span class="kw">m.ca</span>)
</pre></div>
<pre><code>## $overlap
##        b1    b2    b3    b4    b5    b6    b7    b8    b9   b10
## a1  0.773 0.002 0.062 0.001 0.070 0.001 0.021 0.003 0.001 0.003
## a2  0.040 0.662 0.001 0.008 0.108 0.011 0.000 0.005 0.004 0.017
## a3  0.041 0.061 0.117 0.407 0.194 0.002 0.000 0.002 0.002 0.040
## a4  0.055 0.092 0.103 0.253 0.148 0.133 0.041 0.019 0.013 0.037
## a5  0.011 0.012 0.402 0.017 0.115 0.147 0.008 0.061 0.047 0.004
## a6  0.001 0.005 0.039 0.022 0.140 0.092 0.187 0.081 0.001 0.088
## a7  0.000 0.000 0.080 0.003 0.024 0.029 0.049 0.045 0.197 0.011
## a8  0.001 0.006 0.002 0.052 0.029 0.293 0.001 0.081 0.082 0.079
## a9  0.003 0.049 0.022 0.027 0.017 0.002 0.011 0.002 0.018 0.062
## a10 0.031 0.007 0.000 0.004 0.001 0.030 0.074 0.002 0.116 0.035
## 
## $rmsip
## [1] 0.8128787
## 
## attr(,"class")
## [1] "rmsip"</code></pre>
<p>Supplying argument <code>outmodes="noh"</code> will result in that the entire all-atom (heavy atoms) Hessian matrix is diagonalized. This will output a modes object containing all 3840 mode vectors facilitating e.g. trajetory output of all-atom NMA:</p>
<div class="sourceCode" id="cb36"><pre class="downlit">
<span class="co"># all-atom NMA, but output all atoms</span>
<span class="kw">m.noh</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/aanma.html">aanma</a></span>(<span class="kw">pdb</span>, outmodes=<span class="st">"noh"</span>)
</pre></div>
<pre><code>##  Building Hessian...     Done in 1.226 seconds.
##  Diagonalizing Hessian...    Done in 71.076 seconds.</code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit">
<span class="co"># output trajectory of first modes</span>
<span class="fu"><a href="../../../reference/mktrj.html">mktrj</a></span>(<span class="kw">m.noh</span>, mode = <span class="fl">7</span>, pdb=<span class="kw">pdb</span>)
</pre></div>
<div class="figure">
<img src="figures/dhfr_allatomENM.png" alt=""><p class="caption">Visualization of the first non-trivial mode of DHFR using an all-atom elastic network model for the normal mode calculation (function <strong>aanma()</strong>). Ligand MTX is shown in sapce filling representation to illustrate it’s present in the model and the normal modes calculation.</p>
</div>
</div>
<div id="example-2b-reducing-the-computational-load" class="section level2">
<h2 class="hasAnchor">
<a href="#example-2b-reducing-the-computational-load" class="anchor"></a>Example 2B: Reducing the computational load</h2>
<p>Although all-atom ENM provides normal modes with better agreement to Amber modes it is also a far more computational expensive approach (compare 20.7 and 0.2 seconds for the example above). To speed up the calculation of all-atom ENM Bio3D provides functionality for the rotation-translation block based approximation <span class="citation">(Durand, Trinquier, and Sanejouand 1994)</span> <span class="citation">(Tama et al. 2000)</span>. In this method, each residue is assumed to be a rigid body (or <em>block</em>) that has only rotational and translational degrees of freedom. Intra-residue deformation is thus ignored. This approach significantly speeds up the calculation of the normal modes while the results are largely identical:</p>
<div class="sourceCode" id="cb39"><pre class="downlit">
<span class="co"># use rotation and translation of blocks</span>
<span class="kw">m.rtb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/aanma.html">aanma</a></span>(<span class="kw">pdb</span>, rtb=<span class="fl">TRUE</span>)
</pre></div>
<pre><code>##  Building Hessian...     Done in 1.132 seconds.
##  Extracting effective Hessian with RTB.. Done in 3.682 seconds.
##  Diagonalizing Hessian with RTB...       Done in 0.15 seconds.</code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit">
<span class="fu"><a href="../../../reference/rmsip.html">rmsip</a></span>(<span class="kw">m.aa</span>, <span class="kw">m.rtb</span>)
</pre></div>
<pre><code>## $overlap
##     b1    b2    b3    b4 b5 b6 b7    b8    b9   b10
## a1   1 0.000 0.000 0.000  0  0  0 0.000 0.000 0.000
## a2   0 0.999 0.000 0.000  0  0  0 0.000 0.000 0.000
## a3   0 0.000 0.999 0.000  0  0  0 0.000 0.000 0.000
## a4   0 0.000 0.000 0.999  0  0  0 0.000 0.000 0.000
## a5   0 0.000 0.000 0.000  1  0  0 0.000 0.000 0.000
## a6   0 0.000 0.000 0.000  0  1  0 0.000 0.000 0.000
## a7   0 0.000 0.000 0.000  0  0  1 0.000 0.000 0.000
## a8   0 0.000 0.000 0.000  0  0  0 0.999 0.000 0.000
## a9   0 0.000 0.000 0.000  0  0  0 0.000 0.999 0.000
## a10  0 0.000 0.000 0.000  0  0  0 0.000 0.000 0.992
## 
## $rmsip
## [1] 0.9995298
## 
## attr(,"class")
## [1] "rmsip"</code></pre>
<p>Bio3D also provides a <em>reduced</em> atom model in which only a selection of all heavy atoms is used to build the ENM. More specifically, three to five atoms per residue constitute the model. Here the N, CA, C atoms represent the protein backbone, and zero (for Gly), one (for Lys, Ser, Cys, Pro, Ala, Met, Phe, and Tyr) or two (for the remaining amino acids) selected side chain atoms represent the side chain (selected based on side chain size and the distance to CA). This reduced-atom ENM has significantly improved computational efficiency and similar prediction accuracy with respect to the all-atom ENM.</p>
<div class="sourceCode" id="cb43"><pre class="downlit">
<span class="co"># use reduced-atom ENM</span>
<span class="kw">m.red</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/aanma.html">aanma</a></span>(<span class="kw">pdb</span>, reduced=<span class="fl">TRUE</span>)
</pre></div>
<pre><code>##  Building Hessian...     Done in 0.362 seconds.
##  Extracting effective Hessian..  Done in 3.867 seconds.
##  Diagonalizing Hessian...    Done in 0.141 seconds.</code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit">
<span class="fu"><a href="../../../reference/rmsip.html">rmsip</a></span>(<span class="kw">m.aa</span>, <span class="kw">m.red</span>)
</pre></div>
<pre><code>## $overlap
##        b1    b2    b3    b4    b5    b6    b7    b8    b9   b10
## a1  0.878 0.063 0.017 0.003 0.001 0.000 0.000 0.009 0.012 0.001
## a2  0.078 0.594 0.280 0.006 0.001 0.004 0.000 0.001 0.006 0.004
## a3  0.002 0.306 0.649 0.004 0.001 0.004 0.001 0.004 0.002 0.004
## a4  0.006 0.000 0.006 0.930 0.000 0.005 0.004 0.004 0.005 0.000
## a5  0.002 0.001 0.001 0.001 0.900 0.002 0.062 0.002 0.001 0.001
## a6  0.000 0.008 0.001 0.005 0.013 0.853 0.070 0.001 0.000 0.001
## a7  0.000 0.001 0.001 0.000 0.052 0.089 0.786 0.000 0.012 0.006
## a8  0.003 0.001 0.000 0.000 0.000 0.000 0.006 0.631 0.003 0.233
## a9  0.001 0.003 0.021 0.003 0.001 0.002 0.010 0.034 0.715 0.108
## a10 0.009 0.001 0.000 0.000 0.000 0.005 0.000 0.188 0.029 0.191
## 
## $rmsip
## [1] 0.9468653
## 
## attr(,"class")
## [1] "rmsip"</code></pre>
<p>The two approximations can also be combined to further decrease the computational expense:</p>
<div class="sourceCode" id="cb47"><pre class="downlit">
<span class="co"># use both 4-bead and RTB approach</span>
<span class="kw">m.rr</span>  <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/aanma.html">aanma</a></span>(<span class="kw">pdb</span>, reduced=<span class="fl">TRUE</span>, rtb=<span class="fl">TRUE</span>)
</pre></div>
<pre><code>##  Building Hessian...     Done in 0.284 seconds.
##  Extracting effective Hessian with RTB.. Done in 2.421 seconds.
##  Diagonalizing Hessian with RTB...       Done in 0.149 seconds.</code></pre>
</div>
</div>
<div id="example-3-ensemble-normal-mode-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#example-3-ensemble-normal-mode-analysis" class="anchor"></a>Example 3: Ensemble normal mode analysis</h1>
<p>The analysis of multiple protein structures (e.g. a protein family) can be accomplished with the <strong>nma.pdbs()</strong> function.<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a> This will take aligned input structures, as generated by the <strong>pdbaln()</strong> function for example, and perform NMA on each structure collecting the results in manner that facilitates the interpretation of similarity and dissimilarity trends in the structure set. Here we will analyze a collection of Dihydrofolate reductase (DHFR) structures with low sequence identity (<strong>Example 3A</strong>) and large set of closely related transducin heterotrimeric G protein family members (<strong>Example 3B</strong>).</p>
<div id="example-3a-dihydrofolate-reductase" class="section level2">
<h2 class="hasAnchor">
<a href="#example-3a-dihydrofolate-reductase" class="anchor"></a>Example 3A: Dihydrofolate reductase</h2>
<p>In the following code we collect 9 bacterial DHFR structures of 4 different species from the protein databank (using <strong>get.pdb()</strong>) with sequence identity down to 27% (see the call to function <strong>seqidentity()</strong> below), and align these with <strong>pdbaln()</strong>:</p>
<div class="sourceCode" id="cb49"><pre class="downlit">
<span class="co"># Select bacterial DHFR PDB IDs</span>
<span class="kw">ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"1rx2_A"</span>, <span class="st">"1rx4_A"</span>, <span class="st">"1rg7_A"</span>, 
         <span class="st">"3jw3_A"</span>, <span class="st">"3sai_A"</span>,
         <span class="st">"1df7_A"</span>, <span class="st">"4kne_A"</span>,
         <span class="st">"3fyv_X"</span>, <span class="st">"3sgy_B"</span>)

<span class="co"># Download and split by chain ID</span>
<span class="kw">raw.files</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/get.pdb.html">get.pdb</a></span>(<span class="kw">ids</span>, path=<span class="st">"raw_pdbs"</span>)
<span class="kw">files</span>     <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbsplit.html">pdbsplit</a></span>( <span class="kw">raw.files</span>, <span class="kw">ids</span> )

<span class="co"># Alignment of structures</span>
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbaln.html">pdbaln</a></span>(<span class="kw">files</span>)
</pre></div>
<div class="sourceCode" id="cb50"><pre class="downlit">
<span class="co"># Sequence identity</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span>( <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fu"><a href="../../../reference/seqidentity.html">seqidentity</a></span>(<span class="kw">pdbs</span>)) )
</pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##  0.2680  0.3460  0.3960  0.5265  0.9940  1.0000</code></pre>
<p>The <em>pdbs</em> object now contains <em>aligned</em> C-alpha atom data, including Cartesian coordinates, residue numbers, residue types, and B-factors. The sequence alignment is also stored by default to the FASTA format file ‘aln.fa’ (to view this you can use an alignment viewer such as SEAVIEW, see <em>Requirements</em> section above). Function <strong>nma.pdbs()</strong> will calculate the normal modes of each protein structures stored in the <em>pdbs</em> object. The normal modes are calculated on the full structures as provided by object <em>pdbs</em>. With the default argument <code>rm.gaps=TRUE</code> unaligned atoms are omitted from output in accordance with common practice <span class="citation">(Fuglebakk, Echave, and Reuter 2012)</span>.</p>
<div class="sourceCode" id="cb52"><pre class="downlit">
<span class="co"># NMA on all structures</span>
<span class="kw">modes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdbs</span>)
</pre></div>
<p>The <em>modes</em> object of class <em>enma</em> contains aligned normal mode data including fluctuations, RMSIP data, and aligned eigenvectors. A short summary of the <em>modes</em> object can be obtain by calling the function <strong>print()</strong>, and the aligned fluctuations can be plotted with function <strong>plot()</strong>:</p>
<div class="sourceCode" id="cb53"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">modes</span>)
</pre></div>
<pre><code>## 
## Call:
##   nma.pdbs(pdbs = pdbs)
## 
## Class:
##   enma
## 
## Number of structures:
##   9
## 
## Attributes stored:
##   - Root mean square inner product (RMSIP)
##   - Aligned atomic fluctuations
##   - Aligned eigenvectors (gaps removed)
##   - Dimensions of x$U.subspace: 456x450x9
## 
## Coordinates were aligned prior to NMA calculations
## 
## + attr: fluctuations, rmsip, U.subspace, L, full.nma, xyz,
##         call</code></pre>
<div class="sourceCode" id="cb55"><pre class="downlit">
<span class="co"># Plot fluctuation data</span>
<span class="kw">col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>, <span class="fl">2</span>,<span class="fl">2</span>, <span class="fl">3</span>,<span class="fl">3</span>, <span class="fl">4</span>,<span class="fl">4</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">modes</span>, pdbs=<span class="kw">pdbs</span>, col=<span class="kw">col</span>)
</pre></div>
<pre><code>## Extracting SSE from pdbs$sse attribute</code></pre>
<div class="sourceCode" id="cb57"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span>(<span class="st">"topleft"</span>, col=<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="kw">col</span>), lty=<span class="fl">1</span>,
       legend=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"E.Coli"</span>, <span class="st">"B.Anthracis"</span>, <span class="st">"M.Tubercolosis"</span>, <span class="st">"S.Aureus"</span>))
</pre></div>
<div class="figure">
<img src="Bio3D_nma_files/figure-html/ex2A-plot_enma1-1.png" alt="Results of ensemble NMA on four distinct bacterial species of the DHFR enzyme." width="960"><p class="caption">
Results of ensemble NMA on four distinct bacterial species of the DHFR enzyme.
</p>
</div>
<div class="sourceCode" id="cb58"><pre class="downlit">
<span class="co"># Alternatively, one can use 'rm.gaps=FALSE' to keep the gap containing columns</span>
<span class="kw">modes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.pdbs.html">nma.pdbs</a></span>(<span class="kw">pdbs</span>, rm.gaps=<span class="fl">FALSE</span>)
</pre></div>
<p>Cross-correlation analysis can be easily performed and the results contrasted for each member of the input ensemble. Below we calculate and plot the correlation matrices for each structure and then output correlations present only in all input structures.</p>
<div class="sourceCode" id="cb59"><pre class="downlit">
<span class="co"># Calculate correlation matrices for each structure</span>
<span class="kw">cij</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/dccm.html">dccm</a></span>(<span class="kw">modes</span>)
</pre></div>
<div class="sourceCode" id="cb60"><pre class="downlit">
<span class="co"># Determine correlations present only in all 9 input structures</span>
<span class="kw">cij.all</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/filter.dccm.html">filter.dccm</a></span>(<span class="kw">cij</span><span class="op">$</span><span class="kw">all.dccm</span>, cutoff.sims=<span class="fl">9</span>, cutoff.cij = <span class="fl">0</span>)
<span class="fu"><a href="../../../reference/plot.dccm.html">plot.dccm</a></span>(<span class="kw">cij.all</span>, main=<span class="st">"Consensus Residue Cross Correlation"</span>)
</pre></div>
</div>
<div id="example-3b-transducin" class="section level2">
<h2 class="hasAnchor">
<a href="#example-3b-transducin" class="anchor"></a>Example 3B: Transducin</h2>
<p>In this section we will demonstrate the use of <strong>nma.pdbs()</strong> on the example transducin family data that ships with the Bio3D package. This can be loaded with the command <em>data(transducin)</em> and contains an object <em>pdbs</em> consisting of aligned C-alpha coordinates for 53 transducin structures from the PDB as well their annotation (in the object <em>annotation</em>) as obtained from the <strong>pdb.annotate()</strong> function. Note that this data can be generated from scratch by following the <em>Comparative Structure Analysis with Bio3D Vignette</em> available both on-line and from within the Bio3D package.</p>
<div class="sourceCode" id="cb61"><pre class="downlit">
<span class="co"># Load data</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="kw">transducin</span>)
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="kw">transducin</span><span class="op">$</span><span class="kw">pdbs</span>
<span class="kw">annotation</span> <span class="op">&lt;-</span> <span class="kw">transducin</span><span class="op">$</span><span class="kw">annotation</span>

<span class="co"># Find gap positions</span>
<span class="kw">gaps.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/gap.inspect.html">gap.inspect</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">ali</span>)
<span class="kw">gaps.pos</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/gap.inspect.html">gap.inspect</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>)
</pre></div>
<div class="sourceCode" id="cb62"><pre class="downlit">
<span class="co"># Calculate normal modes of the 53 structures</span>
<span class="kw">modes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.pdbs.html">nma.pdbs</a></span>(<span class="kw">pdbs</span>, ncore=<span class="fl">4</span>)
</pre></div>
<div class="sourceCode" id="cb63"><pre class="downlit">
<span class="co"># Make fluctuation plot</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">modes</span>, col=<span class="kw">annotation</span>[, <span class="st">"color"</span>], pdbs=<span class="kw">pdbs</span>)
</pre></div>
<pre><code>## Extracting SSE from pdbs$sse attribute</code></pre>
<div class="sourceCode" id="cb65"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span>(<span class="st">"left"</span>, lty=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="fl">1</span>), lwd=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">2</span>, <span class="fl">2</span>),
    col=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"red"</span>, <span class="st">"green"</span>), legend=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"GTP"</span>, <span class="st">"GDP"</span>))
</pre></div>
<div class="figure">
<img src="Bio3D_nma_files/figure-html/ex2B-plot-1.png" alt="Structural dynamics of transducin. The calculation is based on NMA of 53 structures: 28 GTP-bound (red), and 25 GDP-bound (green)." width="864"><p class="caption">
Structural dynamics of transducin. The calculation is based on NMA of 53 structures: 28 GTP-bound (red), and 25 GDP-bound (green).
</p>
</div>
<p>The similarity of structural dynamics is calculated by RMSIP based on the 10 lowest frequency normal modes. The <em>rmsip</em> values are pre-calculated in the <em>modes</em> object and can be accessed through the attribute <code>modes$rmsip</code>. As a comparison, we also calculate the root mean square deviation (RMSD) of all pair-wise structures:</p>
<div class="sourceCode" id="cb66"><pre class="downlit">
<span class="co"># Plot a heat map with clustering dendogram</span>
<span class="kw">ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">id</span>), <span class="fl">1</span>, <span class="fl">6</span>)
<span class="fu"><a href="https://rdrr.io/r/stats/heatmap.html">heatmap</a></span>((<span class="fl">1</span><span class="op">-</span><span class="kw">modes</span><span class="op">$</span><span class="kw">rmsip</span>), labRow=<span class="kw">annotation</span>[, <span class="st">"state"</span>], labCol=<span class="kw">ids</span>, symm=<span class="fl">TRUE</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_nma_files/figure-html/ex2B-rmsip-1.png" alt="RMSIP matrix of the transducin family." width="700"><p class="caption">
RMSIP matrix of the transducin family.
</p>
</div>
<div class="sourceCode" id="cb67"><pre class="downlit">
<span class="co"># Calculate pair-wise RMSD values</span>
<span class="kw">rmsd.map</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/rmsd.html">rmsd</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>, a.inds=<span class="kw">gaps.pos</span><span class="op">$</span><span class="kw">f.inds</span>, fit=<span class="fl">TRUE</span>)
<span class="fu"><a href="https://rdrr.io/r/stats/heatmap.html">heatmap</a></span>(<span class="kw">rmsd.map</span>, labRow=<span class="kw">annotation</span>[, <span class="st">"state"</span>], labCol=<span class="kw">ids</span>, symm=<span class="fl">TRUE</span>)
</pre></div>
<div class="figure">
<img src="../../../docs/articles/online/nma_vignette/Bio3D_nma_files/figure-html/ex2B-rmsd-1.png" alt="RMSD matrix of the transducin family." width="700"><p class="caption">
RMSD matrix of the transducin family.
</p>
</div>
</div>
<div id="example-3c-all-atom-ensemble-nma" class="section level2">
<h2 class="hasAnchor">
<a href="#example-3c-all-atom-ensemble-nma" class="anchor"></a>Example 3C: All-atom Ensemble NMA</h2>
<p>We can also apply all-atom ENMs to the ensemble NMA approach using function <strong>aanma.pdbs()</strong>. Note the need for function <strong>read.all()</strong> to obtain an all-atom version of the <em>pdbs</em> object.</p>
<div class="sourceCode" id="cb68"><pre class="downlit">
<span class="co"># Select bacterial DHFR PDB IDs</span>
<span class="kw">ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"1rx2_A"</span>, <span class="st">"1df7_A"</span>, <span class="st">"4kne_A"</span>, <span class="st">"3sgy_B"</span>)

<span class="co"># Download and split by chain ID</span>
<span class="kw">raw.files</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/get.pdb.html">get.pdb</a></span>(<span class="kw">ids</span>, path=<span class="st">"raw_pdbs"</span>)
<span class="kw">files</span>     <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbsplit.html">pdbsplit</a></span>( <span class="kw">raw.files</span>, <span class="kw">ids</span> )

<span class="co"># Read and align structures</span>
<span class="kw">aln</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbaln.html">pdbaln</a></span>(<span class="kw">files</span>)
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.all.html">read.all</a></span>(<span class="kw">aln</span>)
</pre></div>
<div class="sourceCode" id="cb69"><pre class="downlit">
<span class="co"># calculate modes</span>
<span class="kw">modes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/aanma.html">aanma</a></span>(<span class="kw">pdbs</span>, rtb=<span class="fl">TRUE</span>)
</pre></div>
<pre><code>## Fitting pdb structuresdone
## 
## Details of Scheduled Calculation:
##   ... 4 input structures 
##   ... storing 450 eigenvectors for each structure 
##   ... dimension of x$U.subspace: ( 456x450x4 )
##   ... coordinate superposition prior to NM calculation 
##   ... aligned eigenvectors (gap containing positions removed)  
##   ... rotation-translation block (RTB) approximation will be applied 
##   ... estimated memory usage of final 'eNMA' object: 6.3 Mb</code></pre>
<div class="sourceCode" id="cb71"><pre class="downlit">
<span class="co"># Make fluctuation plot</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">modes</span>, pdbs=<span class="kw">pdbs</span>)
</pre></div>
<pre><code>## Extracting SSE from pdbs$sse attribute</code></pre>
<div class="figure">
<img src="Bio3D_nma_files/figure-html/ex2C-aa-enmaplot-1.png" alt="Fluctuation plot of 4 *E.coli* DHFR structures calculated using all-atom ENM." width="864"><p class="caption">
Fluctuation plot of 4 <em>E.coli</em> DHFR structures calculated using all-atom ENM.
</p>
</div>
</div>
</div>
<div id="example-4-variance-weighted-normal-mode-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#example-4-variance-weighted-normal-mode-analysis" class="anchor"></a>Example 4: Variance weighted normal mode analysis</h1>
<p>In this example we illustrate an approach of weighting the pair force constants based on the variance of the inter atomic distances obtained from an ensemble of structures (e.g. available X-ray structures). The motivation for such variance-weighting is to reduce the well known dependence of the force constants on the one structure upon which they are derived <span class="citation">(Tama and Sanejouand 2001)</span>.</p>
<div id="example-4a-groel" class="section level2">
<h2 class="hasAnchor">
<a href="#example-4a-groel" class="anchor"></a>Example 4A: GroEL</h2>
<p>We first calculate the normal modes of both the closed and open state of the GroEL subunit, and we illustrate the difference in the agreement towards the observed conformational changes (characterized by X-ray and EM studies). We will then use an ensemble of X-ray/EM structures as weights to the pair-force constants.</p>
<div class="sourceCode" id="cb73"><pre class="downlit">
<span class="co"># Define the ensemble PDB-ids</span>
<span class="kw">ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"1sx4_[A,B,H,I]"</span>, <span class="st">"1xck_[A-B]"</span>, <span class="st">"1sx3_[A-B]"</span>, <span class="st">"4ab3_[A-B]"</span>)

<span class="co"># Download and split PDBs by chain ID</span>
<span class="kw">raw.files</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/get.pdb.html">get.pdb</a></span>(<span class="kw">ids</span>, <span class="st">"groel_pdbs"</span>, gzip=<span class="fl">TRUE</span>)
<span class="kw">files</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbsplit.html">pdbsplit</a></span>(<span class="kw">raw.files</span>, <span class="kw">ids</span>, path = <span class="st">"groel_pdbs"</span>)

<span class="co"># Align and superimpose coordinates</span>
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbaln.html">pdbaln</a></span>(<span class="kw">files</span>, fit=<span class="fl">TRUE</span>)
</pre></div>
<div id="calculate-normal-modes" class="section level3">
<h3 class="hasAnchor">
<a href="#calculate-normal-modes" class="anchor"></a>Calculate normal modes</h3>
<p>Next we will calculate the normal modes of the open and closed conformational state. They are stored at indices 1 and 5, respectively, in our <em>pdbs</em> object. Use the <strong>pdbs2pdb()</strong> to fetch the pdb objects which is needed for the input to <strong>nma()</strong>.</p>
<div class="sourceCode" id="cb74"><pre class="downlit">
<span class="co"># Inspect gaps</span>
<span class="kw">gaps.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/gap.inspect.html">gap.inspect</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">ali</span>)
<span class="kw">gaps.pos</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/gap.inspect.html">gap.inspect</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>)

<span class="co"># Access PDB objects</span>
<span class="kw">pdb.list</span>   <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbs2pdb.html">pdbs2pdb</a></span>(<span class="kw">pdbs</span>, inds=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">5</span>,<span class="fl">9</span>), rm.gaps=<span class="fl">TRUE</span>)
</pre></div>
<p>Note that we are here using the argument <code>rm.gaps=TRUE</code> to omit residues in gap containing columns of the alignment. Consequently, the resulting three pdb objects we obtain will have the same lengths (523 residues), which is convenient for subsequent analysis.</p>
<div class="sourceCode" id="cb75"><pre class="downlit">
<span class="kw">pdb.open</span>   <span class="op">&lt;-</span> <span class="kw">pdb.list</span>[[<span class="st">"1sx4_A"</span>]]
<span class="kw">pdb.closed</span> <span class="op">&lt;-</span> <span class="kw">pdb.list</span>[[<span class="st">"1xck_A"</span>]]
<span class="kw">pdb.rstate</span> <span class="op">&lt;-</span> <span class="kw">pdb.list</span>[[<span class="st">"4ab3_A"</span>]]

<span class="co"># Calaculate normal modes</span>
<span class="kw">modes.open</span>   <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb.open</span>)
<span class="kw">modes.closed</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb.closed</span>)
<span class="kw">modes.rstate</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb.rstate</span>)
</pre></div>
</div>
<div id="overlap-analysis-1" class="section level3">
<h3 class="hasAnchor">
<a href="#overlap-analysis-1" class="anchor"></a>Overlap analysis</h3>
<p>Use overlap analysis to determine the agreement between the normal mode vectors and the conformational difference vector:</p>
<div class="sourceCode" id="cb76"><pre class="downlit">
<span class="co"># Difference vector 1: closed - open</span>
<span class="kw">diff.vec.1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/difference.vector.html">difference.vector</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>[<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">5</span>), <span class="kw">gaps.pos</span><span class="op">$</span><span class="kw">f.inds</span>])
<span class="co"># Difference vector 2: closed - rstate</span>
<span class="kw">diff.vec.2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/difference.vector.html">difference.vector</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>[<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">5</span>,<span class="fl">9</span>), <span class="kw">gaps.pos</span><span class="op">$</span><span class="kw">f.inds</span>])

<span class="co"># Calculate overlap</span>
<span class="kw">oa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/overlap.html">overlap</a></span>(<span class="kw">modes.open</span>,   <span class="kw">diff.vec.1</span>)
<span class="kw">ob</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/overlap.html">overlap</a></span>(<span class="kw">modes.closed</span>, <span class="kw">diff.vec.1</span>)
<span class="kw">oc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/overlap.html">overlap</a></span>(<span class="kw">modes.closed</span>, <span class="kw">diff.vec.2</span>)
</pre></div>
<div class="sourceCode" id="cb77"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">oa</span><span class="op">$</span><span class="kw">overlap.cum</span>[<span class="fl">1</span><span class="op">:</span><span class="fl">10</span>], type=<span class="st">'b'</span>, ylim=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>,<span class="fl">1</span>),
     ylab=<span class="st">"Squared overlap"</span>, xlab=<span class="st">"Mode index"</span>, lwd=<span class="fl">2</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/lines.html">lines</a></span>(<span class="kw">ob</span><span class="op">$</span><span class="kw">overlap.cum</span>[<span class="fl">1</span><span class="op">:</span><span class="fl">10</span>], type=<span class="st">'b'</span>, lty=<span class="fl">2</span>, col=<span class="fl">2</span>, lwd=<span class="fl">2</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/lines.html">lines</a></span>(<span class="kw">oc</span><span class="op">$</span><span class="kw">overlap.cum</span>[<span class="fl">1</span><span class="op">:</span><span class="fl">10</span>], type=<span class="st">'b'</span>, lty=<span class="fl">3</span>, col=<span class="fl">4</span>, lwd=<span class="fl">1</span>)

<span class="fu"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span>(<span class="st">"bottomright"</span>,
       <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Open to closed"</span>, <span class="st">"Closed to open"</span>, <span class="st">"Closed to r-state"</span>),
       col=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">4</span>), lty=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span>))
</pre></div>
<div class="figure">
<img src="Bio3D_nma_files/figure-html/ex3A_plotoverlap1-1.png" alt="Overlap anlaysis with function **overlap()**. The modes calculated on the open state of the GroEL subunit shows a high similarity to the conformational difference vector (black), while the agreement is lower when the normal modes are calculated on the closed state (red). Blue line correspond to the overlap between the closed state and the r-state (a semi-open state characterized by a rotation of the apical domain in the opposite direction as compared to the open state." width="672"><p class="caption">
Overlap anlaysis with function <strong>overlap()</strong>. The modes calculated on the open state of the GroEL subunit shows a high similarity to the conformational difference vector (black), while the agreement is lower when the normal modes are calculated on the closed state (red). Blue line correspond to the overlap between the closed state and the r-state (a semi-open state characterized by a rotation of the apical domain in the opposite direction as compared to the open state.
</p>
</div>
</div>
<div id="variance-weighting" class="section level3">
<h3 class="hasAnchor">
<a href="#variance-weighting" class="anchor"></a>Variance weighting</h3>
<p>From the overlap analysis above we see the good agreement (high overlap value) between the conformational difference vector and the normal modes calculated on the open structures. Contrary, the lowest frequency modes of the closed structures does not show the same behavior. We will thus proceed with the weighting of the force constants. First we will calculate the variance of all pairwise distances in the ensemble and use this as weights for the NMA:</p>
<div class="sourceCode" id="cb78"><pre class="downlit">
<span class="co"># Calcualte variance</span>
<span class="kw">wts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/var.xyz.html">var.xyz</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>[, <span class="kw">gaps.pos</span><span class="op">$</span><span class="kw">f.inds</span>])
</pre></div>
<p>Weights to the force constants can be included by the argument ‘fc.weights’ to function <strong>nma()</strong>. This needs be a matrix with dimensions NxN (where N is the number of C-alpha atoms). Here we will run a small for-loop with increasing the strength of the weighting at each step and store the new overlap values in the variable ‘ob.wtd’:</p>
<div class="sourceCode" id="cb79"><pre class="downlit">
<span class="kw">ob.wtd</span> <span class="op">&lt;-</span> <span class="kw">NULL</span>
<span class="co">for</span> ( <span class="kw">i</span> <span class="co">in</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span> ) {
  <span class="kw">modes.wtd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb.closed</span>, fc.weights=<span class="kw">wts</span><span class="op">**</span><span class="kw">i</span>)
  <span class="kw">ob.tmp</span>    <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/overlap.html">overlap</a></span>(<span class="kw">modes.wtd</span>,    <span class="kw">diff.vec.1</span>)
  <span class="kw">ob.wtd</span>    <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span>(<span class="kw">ob.wtd</span>, <span class="kw">ob.tmp</span><span class="op">$</span><span class="kw">overlap.cum</span>)
}
</pre></div>
<div class="sourceCode" id="cb80"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">oa</span><span class="op">$</span><span class="kw">overlap.cum</span>[<span class="fl">1</span><span class="op">:</span><span class="fl">10</span>], type=<span class="st">'b'</span>, ylim=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>,<span class="fl">1</span>),
     ylab=<span class="st">"Squared overlap"</span>, xlab=<span class="st">"Mode index"</span>, axes=<span class="kw">T</span>, lwd=<span class="fl">2</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/lines.html">lines</a></span>(<span class="kw">ob</span><span class="op">$</span><span class="kw">overlap.cum</span>[<span class="fl">1</span><span class="op">:</span><span class="fl">10</span>], type=<span class="st">'b'</span>, lty=<span class="fl">2</span>, col=<span class="fl">1</span>, lwd=<span class="fl">2</span>)

<span class="kw">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/palettes.html">rainbow</a></span>(<span class="fl">10</span>)
<span class="co">for</span> ( <span class="kw">i</span> <span class="co">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="kw">ob.wtd</span>) ) {
  <span class="fu"><a href="https://rdrr.io/r/graphics/lines.html">lines</a></span>(<span class="kw">ob.wtd</span>[<span class="kw">i</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">10</span>], type=<span class="st">'b'</span>, lty=<span class="fl">1</span>, col=<span class="kw">cols</span>[<span class="kw">i</span>])
}

<span class="fu"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span>(<span class="st">"bottomright"</span>,
       <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Open state"</span>, <span class="st">"Closed state"</span>, <span class="st">"Closed state (weighted)"</span>),
       col=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"black"</span>, <span class="st">"black"</span>, <span class="st">"green"</span>), lty=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">1</span>))
</pre></div>
<div class="figure">
<img src="Bio3D_nma_files/figure-html/ex3A_plotoverlap2-1.png" alt="Overlap plot with increasing strength on the weighting. The final weighted normal modes of the closed subunit shows as high overlap values as the modes for the open state." width="672"><p class="caption">
Overlap plot with increasing strength on the weighting. The final weighted normal modes of the closed subunit shows as high overlap values as the modes for the open state.
</p>
</div>
</div>
<div id="rmsip-calculation" class="section level3">
<h3 class="hasAnchor">
<a href="#rmsip-calculation" class="anchor"></a>RMSIP calculation</h3>
<p>RMSIP can be used to compare the mode subspaces:</p>
<div class="sourceCode" id="cb81"><pre class="downlit">
<span class="kw">ra</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/rmsip.html">rmsip</a></span>(<span class="kw">modes.open</span>, <span class="kw">modes.wtd</span>)
<span class="kw">rb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/rmsip.html">rmsip</a></span>(<span class="kw">modes.open</span>, <span class="kw">modes.closed</span>)
</pre></div>
<div class="sourceCode" id="cb82"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(mfrow=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">2</span>))
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">ra</span>, ylab=<span class="st">"NMA(open)"</span>, xlab=<span class="st">"NMA(weighted)"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">rb</span>, ylab=<span class="st">"NMA(open)"</span>, xlab=<span class="st">"NMA(closed)"</span>)
</pre></div>
<div class="figure">
<img src="Bio3D_nma_files/figure-html/ex3A_plot-rmsip-1.png" alt="RMSIP maps between (un)weighted normal modes obtained from the open and closed subunits." width="960"><p class="caption">
RMSIP maps between (un)weighted normal modes obtained from the open and closed subunits.
</p>
</div>
</div>
<div id="match-with-pca" class="section level3">
<h3 class="hasAnchor">
<a href="#match-with-pca" class="anchor"></a>Match with PCA</h3>
<p>Finally, we compare the calculated normal modes with principal components obtained from the ensemble of X-ray structures using function <strong>pca.xyz()</strong>:</p>
<div class="sourceCode" id="cb83"><pre class="downlit">
<span class="co"># Calculate the PCs</span>
<span class="kw">pc.xray</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pca.xyz.html">pca.xyz</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>[,<span class="kw">gaps.pos</span><span class="op">$</span><span class="kw">f.inds</span>])
</pre></div>
<div class="sourceCode" id="cb84"><pre class="downlit">
<span class="co"># or alternatively... </span>
<span class="kw">pc.xray</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pca.html">pca</a></span>(<span class="kw">pdbs</span>)

<span class="co"># Calculate RMSIP values</span>
<span class="fu"><a href="../../../reference/rmsip.html">rmsip</a></span>(<span class="kw">pc.xray</span>, <span class="kw">modes.open</span>)<span class="op">$</span><span class="kw">rmsip</span>
</pre></div>
<pre><code>## [1] 0.6208435</code></pre>
<div class="sourceCode" id="cb86"><pre class="downlit">
<span class="fu"><a href="../../../reference/rmsip.html">rmsip</a></span>(<span class="kw">pc.xray</span>, <span class="kw">modes.closed</span>)<span class="op">$</span><span class="kw">rmsip</span>
</pre></div>
<pre><code>## [1] 0.630455</code></pre>
<div class="sourceCode" id="cb88"><pre class="downlit">
<span class="fu"><a href="../../../reference/rmsip.html">rmsip</a></span>(<span class="kw">pc.xray</span>, <span class="kw">modes.rstate</span>)<span class="op">$</span><span class="kw">rmsip</span>
</pre></div>
<pre><code>## [1] 0.586484</code></pre>
<div class="sourceCode" id="cb90"><pre class="downlit">
<span class="fu"><a href="../../../reference/rmsip.html">rmsip</a></span>(<span class="kw">pc.xray</span>, <span class="kw">modes.wtd</span>)<span class="op">$</span><span class="kw">rmsip</span>
</pre></div>
<pre><code>## [1] 0.6606216</code></pre>
</div>
</div>
<div id="example-4b-transducin" class="section level2">
<h2 class="hasAnchor">
<a href="#example-4b-transducin" class="anchor"></a>Example 4B: Transducin</h2>
<p>This example will run <strong>nma()</strong> on transducin with variance weighted force constants. The modes predicted by NMA will be compared with principal component analysis (PCA) results over the transducin family. We load the transducin data via the command <em>data(transducin)</em> and calculate the normal modes for two structures corresponding to two nucleotide states, respectively: GDP (PDB id 1TAG) and GTP (PDB id 1TND). Again we use function <strong>pdbs2pdb()</strong> to build the <em>pdb</em> objects from the <em>pdbs</em> object (containing aligned structure/sequence information). The coordinates of the data set were fitted to all non-gap containing C-alpha positions.</p>
<div class="sourceCode" id="cb92"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="kw">transducin</span>)
<span class="kw">pdbs</span> <span class="op">&lt;-</span> <span class="kw">transducin</span><span class="op">$</span><span class="kw">pdbs</span>

<span class="kw">gaps.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/gap.inspect.html">gap.inspect</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">ali</span>)
<span class="kw">gaps.pos</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/gap.inspect.html">gap.inspect</a></span>(<span class="kw">pdbs</span><span class="op">$</span><span class="kw">xyz</span>)
</pre></div>
<div class="sourceCode" id="cb93"><pre class="downlit">
<span class="co"># Fit coordinates based on all non-gap positions</span>
<span class="co"># and do PCA</span>
<span class="kw">xyz</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbfit.html">pdbfit</a></span>(<span class="kw">pdbs</span>)
<span class="kw">pc.xray</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pca.xyz.html">pca.xyz</a></span>(<span class="kw">xyz</span>[, <span class="kw">gaps.pos</span><span class="op">$</span><span class="kw">f.inds</span>])
</pre></div>
<div class="sourceCode" id="cb94"><pre class="downlit">
<span class="co"># Fetch PDB objects</span>
<span class="kw">npdbs</span> <span class="op">&lt;-</span> <span class="kw">pdbs</span>
<span class="kw">npdbs</span><span class="op">$</span><span class="kw">xyz</span> <span class="op">&lt;-</span> <span class="kw">xyz</span>
<span class="kw">pdb.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/pdbs2pdb.html">pdbs2pdb</a></span>(<span class="kw">npdbs</span>, inds=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">2</span>, <span class="fl">7</span>), rm.gaps=<span class="fl">TRUE</span>)
<span class="kw">pdb.gdp</span> <span class="op">&lt;-</span> <span class="kw">pdb.list</span>[[ <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"1TAG_A"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">pdb.list</span>)) ]]
<span class="kw">pdb.gtp</span> <span class="op">&lt;-</span> <span class="kw">pdb.list</span>[[ <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"1TND_B"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">pdb.list</span>)) ]]

<span class="co"># Calculate normal modes</span>
<span class="kw">modes.gdp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb.gdp</span>)
<span class="kw">modes.gtp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb.gtp</span>)
</pre></div>
<p>Now, we calculate the pairwise distance variance based on the structural ensemble with the function <strong>var.xyz()</strong> as described above. This will be used to weight the force constants in the elastic network model.</p>
<div class="sourceCode" id="cb95"><pre class="downlit">
<span class="co"># Calculate weights </span>
<span class="kw">weights</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/var.xyz.html">var.xyz</a></span>(<span class="kw">xyz</span>[, <span class="kw">gaps.pos</span><span class="op">$</span><span class="kw">f.inds</span>])

<span class="co"># Calculate normal modes with weighted pair force constants</span>
<span class="kw">modes.gdp.b</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb.gdp</span>, fc.weights=<span class="kw">weights</span><span class="op">**</span><span class="fl">100</span>)
<span class="kw">modes.gtp.b</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb.gtp</span>, fc.weights=<span class="kw">weights</span><span class="op">**</span><span class="fl">100</span>)
</pre></div>
<p>To evaluate the results, we calculate the overlap (square dot product) between modes predicted by variance weighted or non-weighted NMA and the first principal component from PCA.</p>
<div class="sourceCode" id="cb96"><pre class="downlit">
<span class="kw">oa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/overlap.html">overlap</a></span>(<span class="kw">modes.gdp</span>, <span class="kw">pc.xray</span><span class="op">$</span><span class="kw">U</span>[,<span class="fl">1</span>])
<span class="kw">ob</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/overlap.html">overlap</a></span>(<span class="kw">modes.gtp</span>, <span class="kw">pc.xray</span><span class="op">$</span><span class="kw">U</span>[,<span class="fl">1</span>])
<span class="kw">oc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/overlap.html">overlap</a></span>(<span class="kw">modes.gdp.b</span>, <span class="kw">pc.xray</span><span class="op">$</span><span class="kw">U</span>[,<span class="fl">1</span>])
<span class="kw">od</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/overlap.html">overlap</a></span>(<span class="kw">modes.gtp.b</span>, <span class="kw">pc.xray</span><span class="op">$</span><span class="kw">U</span>[,<span class="fl">1</span>])
</pre></div>
<div class="sourceCode" id="cb97"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">oa</span><span class="op">$</span><span class="kw">overlap.cum</span>, type=<span class="st">'o'</span>, ylim=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>,<span class="fl">1</span>), col=<span class="st">"darkgreen"</span>, lwd=<span class="fl">2</span>, xlab=<span class="st">"Mode"</span>, 
     ylab=<span class="st">"Cummulative overlap"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/lines.html">lines</a></span>(<span class="kw">ob</span><span class="op">$</span><span class="kw">overlap.cum</span>, type=<span class="st">'o'</span>, ylim=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>,<span class="fl">1</span>), col=<span class="st">"red"</span>, lwd=<span class="fl">2</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/lines.html">lines</a></span>(<span class="kw">oc</span><span class="op">$</span><span class="kw">overlap.cum</span>, type=<span class="st">'b'</span>, ylim=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>,<span class="fl">1</span>), col=<span class="st">"darkgreen"</span>, lwd=<span class="fl">2</span>, lty=<span class="fl">2</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/lines.html">lines</a></span>(<span class="kw">od</span><span class="op">$</span><span class="kw">overlap.cum</span>, type=<span class="st">'b'</span>, ylim=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>,<span class="fl">1</span>), col=<span class="st">"red"</span>, lwd=<span class="fl">2</span>, lty=<span class="fl">2</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span>(<span class="fl">20</span>, <span class="kw">oa</span><span class="op">$</span><span class="kw">overlap.cum</span>[<span class="fl">20</span>], label=<span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="kw">oa</span><span class="op">$</span><span class="kw">overlap.cum</span>[<span class="fl">20</span>], <span class="fl">2</span>), pos=<span class="fl">3</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span>(<span class="fl">20</span>, <span class="kw">ob</span><span class="op">$</span><span class="kw">overlap.cum</span>[<span class="fl">20</span>], label=<span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="kw">ob</span><span class="op">$</span><span class="kw">overlap.cum</span>[<span class="fl">20</span>], <span class="fl">2</span>), pos=<span class="fl">3</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span>(<span class="fl">20</span>, <span class="kw">oc</span><span class="op">$</span><span class="kw">overlap.cum</span>[<span class="fl">20</span>], label=<span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="kw">oc</span><span class="op">$</span><span class="kw">overlap.cum</span>[<span class="fl">20</span>], <span class="fl">2</span>), pos=<span class="fl">3</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span>(<span class="fl">20</span>, <span class="kw">od</span><span class="op">$</span><span class="kw">overlap.cum</span>[<span class="fl">20</span>], label=<span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="kw">od</span><span class="op">$</span><span class="kw">overlap.cum</span>[<span class="fl">20</span>], <span class="fl">2</span>), pos=<span class="fl">3</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span>(<span class="st">"topleft"</span>, pch=<span class="fl">1</span>, lty=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">2</span>), col=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"darkgreen"</span>, <span class="st">"red"</span>, 
       <span class="st">"darkgreen"</span>, <span class="st">"red"</span>), legend=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"GDP"</span>, <span class="st">"GTP"</span>, <span class="st">"Weighted GDP"</span>, <span class="st">"Weighted GTP"</span>))
</pre></div>
<div class="figure">
<img src="Bio3D_nma_files/figure-html/ex3B-plotoverlap-1.png" alt="Variance weighted force constants improve NMA prediction" width="672"><p class="caption">
Variance weighted force constants improve NMA prediction
</p>
</div>
</div>
</div>
<div id="example-5-user-defined-pair-force-constant-functions" class="section level1">
<h1 class="hasAnchor">
<a href="#example-5-user-defined-pair-force-constant-functions" class="anchor"></a>Example 5: User-defined pair force constant functions</h1>
<p>In this example we demonstrate the interface for defining custom functions for the pair spring force constants. A custom function can be obtained through simple scripting as shown below.</p>
<div id="example-5a-specifying-a-simple-function" class="section level2">
<h2 class="hasAnchor">
<a href="#example-5a-specifying-a-simple-function" class="anchor"></a>Example 5A: Specifying a simple function</h2>
<p>We first show how to define a simple force constant function by building a revised version of the parameter-free ANM force field. The function <strong>my.ff()</strong> below takes as input <em>r</em> which is a vector of inter-atomic (calpha) distances (i.e. distances from atom <em>i</em>, to all other atoms in the system; this function will thus be called N times, where N is the number of calpha atoms). It will in this case return 0 for the pairs with a distance larger than 10 , and <span class="math inline">\(r^{-2}\)</span> for all other pairs. Our simple function will thus look like:</p>
<div class="sourceCode" id="cb98"><pre class="downlit">
<span class="co"># Define function for spring force constants</span>
<span class="st">"my.ff"</span> <span class="op">&lt;-</span> <span class="fu">function</span>(<span class="kw">r</span>, <span class="kw">...</span>) {
  <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>( <span class="kw">r</span><span class="op">&gt;</span><span class="fl">10</span>, <span class="fl">0</span>, <span class="kw">r</span><span class="op">^</span>(<span class="op">-</span><span class="fl">2</span>) )
}
</pre></div>
<p>Once the function is in place we can feed it to function <strong>nma()</strong> to calculate the normal modes based on the particular force constants built with our new function. Below we apply it to the lysozyme structure (PDB id <em>1hel</em>) from Example 1:</p>
<div class="sourceCode" id="cb99"><pre class="downlit">
<span class="co"># Download PDB and calculate normal modes</span>
<span class="kw">pdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/read.pdb.html">read.pdb</a></span>(<span class="st">"1hel"</span>)
<span class="kw">modes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb</span>, pfc.fun=<span class="kw">my.ff</span>)
</pre></div>
<p>Alternatively we can take a more manual approach by calling <strong>build.hessian()</strong> if we want to investigate the Hessian matrix further (note that <strong>build.hessian</strong> is called from within function <strong>nma()</strong> which will diagonalize the hessian to obtain the normal modes and thus not return it to the user). In the code below we first build the hessian and illustrate how to obtain the normal modes through calls to either <strong>eigen()</strong> or <strong>nma()</strong> (which can also take a Hessian matrix as input):</p>
<div class="sourceCode" id="cb100"><pre class="downlit">
<span class="co"># Indices for CA atoms</span>
<span class="kw">ca.inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/atom.select.html">atom.select</a></span>(<span class="kw">pdb</span>, <span class="st">'calpha'</span>)

<span class="co"># Build hessian matrix</span>
<span class="kw">h</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.pdb.html">build.hessian</a></span>(<span class="kw">pdb</span><span class="op">$</span><span class="kw">xyz</span>[ <span class="kw">ca.inds</span><span class="op">$</span><span class="kw">xyz</span> ], pfc.fun=<span class="kw">my.ff</span>)

<span class="co"># Diagonalize and obtain eigenvectors and eigenvalues</span>
<span class="kw">modes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/eigen.html">eigen</a></span>(<span class="kw">h</span>, symmetric=<span class="fl">TRUE</span>)

<span class="co"># ... or feed the Hessian to function 'nma()'</span>
<span class="kw">modes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb</span>, hessian=<span class="kw">h</span>, mass=<span class="fl">FALSE</span>)
</pre></div>
<p>Note that function <strong>nma()</strong> assumes the Hessian to be mass-weighted and we therefore have to specify <em>mass=FALSE</em> in this particular case. To obtain a mass-weighted Hessian pass the amino acid masses through argument <em>aa.mass</em> to function <strong>build.hessian()</strong>.</p>
</div>
<div id="example-5b-specific-force-constants-for-disulfide-bridges" class="section level2">
<h2 class="hasAnchor">
<a href="#example-5b-specific-force-constants-for-disulfide-bridges" class="anchor"></a>Example 5B: Specific force constants for disulfide bridges</h2>
<p>In the following code we illustrate a more advanced force constant function making use of arguments <em>atom.id</em> and <em>ssdat</em> which is passed from function <strong>build.hessian()</strong> by default. This allows users to access the protein sequence (<code>ssdat$seq</code>), secondary structure data (<code>ssdat$sse</code>), beta bridges (<code>ssdat$beta.bridges</code>), helix 1-4 (<code>ssdat$helix14</code>), and disulfide bridges (ss bonds; <code>ssdat$ss.bonds</code>) when building the force constants.</p>
<p>First we define our new function (<strong>ff.custom()</strong>) and specify the force constants which should be applied to bonded and non-bonded interactions (<code>k.bonded</code> and <code>k.nonbonded</code>, respectively). Next we define the the force constant for the disulfide bridges (<code>k.ssbond</code>):</p>
<div class="sourceCode" id="cb101"><pre class="downlit">
<span class="st">"ff.custom"</span> <span class="op">&lt;-</span> <span class="fu">function</span>(<span class="kw">r</span>, <span class="kw">atom.id</span>, <span class="kw">pdb</span>=<span class="kw">NULL</span>, <span class="kw">ss.bonds</span>=<span class="kw">NULL</span>, <span class="kw">...</span>) {
  <span class="co"># Default force constants (Hinsen et al 2000)</span>
  <span class="kw">k.bonded</span>    <span class="op">&lt;-</span> (<span class="kw">r</span> <span class="op">*</span> <span class="fl">8.6</span> <span class="op">*</span> <span class="fl">10</span><span class="op">^</span><span class="fl">2</span>) <span class="op">-</span> (<span class="fl">2.39</span> <span class="op">*</span> <span class="fl">10</span><span class="op">^</span><span class="fl">3</span>)
  <span class="kw">k.nonbonded</span> <span class="op">&lt;-</span> (<span class="fl">128</span> <span class="op">*</span> <span class="fl">10</span><span class="op">^</span><span class="fl">4</span>) <span class="op">*</span> <span class="kw">r</span><span class="op">^</span>(<span class="op">-</span><span class="fl">6</span>)

  <span class="co"># Special force constant for SS-bonds</span>
  <span class="kw">k.ssbond</span>    <span class="op">&lt;-</span> <span class="fl">50</span>;

  <span class="co"># Calculate default values (equivalent to the calpha ff)</span>
  <span class="kw">ks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="kw">r</span> <span class="op">&lt;</span> <span class="fl">4.0</span>,
               <span class="kw">k.bonded</span>,
               <span class="kw">k.nonbonded</span>)

  <span class="co">if</span>(<span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html">is.null</a></span>(<span class="kw">ss.bonds</span>)) {
    <span class="co"># If atom.id is part off a ssbond..</span>
    <span class="kw">inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">ss.bonds</span> <span class="op">==</span> <span class="kw">atom.id</span>, arr.ind=<span class="fl">TRUE</span>)
    
    <span class="co">if</span>(<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw">inds</span>) <span class="op">&gt;</span> <span class="fl">0</span>) {
      <span class="co"># Find ss-bond pair</span>
      <span class="kw">inds.paired</span> <span class="op">&lt;-</span> <span class="kw">ss.bonds</span>[ <span class="kw">inds</span>[<span class="fl">1</span>, <span class="st">"row"</span>], ]
      <span class="kw">j</span> <span class="op">&lt;-</span> <span class="kw">inds.paired</span>[ <span class="kw">inds.paired</span> != <span class="kw">atom.id</span> ]

      <span class="co"># and change the spring force constant</span>
      <span class="kw">ks</span>[<span class="kw">j</span>] <span class="op">&lt;-</span> <span class="kw">k.ssbond</span>
    }
  }
  <span class="fu"><a href="https://rdrr.io/r/base/function.html">return</a></span>(<span class="kw">ks</span>)
}
</pre></div>
<p>The disulfide bridges can be supplied as input to <strong>nma()</strong> function via a simple two-column matrix:</p>
<div class="sourceCode" id="cb102"><pre class="downlit">
<span class="co"># Define SS-bonds in a two-column matrix</span>
<span class="kw">ss.bonds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">76</span>,<span class="fl">94</span>, <span class="fl">64</span>,<span class="fl">80</span>, <span class="fl">30</span>,<span class="fl">115</span>, <span class="fl">6</span>,<span class="fl">127</span>),
                   ncol=<span class="fl">2</span>, byrow=<span class="fl">TRUE</span>)

<span class="co"># Calculate modes with custom force field</span>
<span class="kw">modes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../../reference/nma.html">nma</a></span>(<span class="kw">pdb</span>, pfc.fun=<span class="kw">ff.custom</span>, ss.bonds=<span class="kw">ss.bonds</span>)
</pre></div>
</div>
</div>
<div id="document-details" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#document-details" class="anchor"></a>Document Details</h1>
<p>This document is shipped with the Bio3D package in both R and PDF formats. All code can be extracted and automatically executed to generate Figures and/or the PDF with the following commands:</p>
<div class="sourceCode" id="cb103"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://github.com/rstudio/rmarkdown">rmarkdown</a></span>)
<span class="fu"><a href="https://rdrr.io/pkg/rmarkdown/man/render.html">render</a></span>(<span class="st">"Bio3D_nma.Rmd"</span>, <span class="st">"all"</span>)
</pre></div>
</div>
<div id="information-about-the-current-bio3d-session" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#information-about-the-current-bio3d-session" class="anchor"></a>Information About the Current Bio3D Session</h1>
<div class="sourceCode" id="cb104"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>() 
</pre></div>
<pre><code>## R version 3.6.0 (2019-04-26)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS  10.15.5
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] bio3d_2.4-1.9000
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.5         lattice_0.20-41    digest_0.6.25      crayon_1.3.4      
##  [5] rprojroot_1.3-2    assertthat_0.2.1   grid_3.6.0         R6_2.4.1          
##  [9] backports_1.1.8    magrittr_1.5       evaluate_0.14      highr_0.8         
## [13] stringi_1.4.6      rlang_0.4.6.9000   fs_1.4.2           rmarkdown_2.3.2   
## [17] pkgdown_1.5.1.9000 desc_1.2.0         tools_3.6.0        stringr_1.4.0     
## [21] parallel_3.6.0     yaml_2.2.1         xfun_0.15          compiler_3.6.0    
## [25] memoise_1.1.0      htmltools_0.5.0    knitr_1.29</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references hanging-indent">
<div id="ref-Durand1994">
<p>Durand, P., G. Trinquier, and Y. H. Sanejouand. 1994. “A New Approach for Determining Low-Frequency Normal Modes in Macromolecules.” <em>Biopolymers</em> 34: 759–71. <a href="https://doi.org/10.1002/bip.360340608">https://doi.org/10.1002/bip.360340608</a>.</p>
</div>
<div id="ref-Fuglebakk2012">
<p>Fuglebakk, Edvin, Julián Echave, and Nathalie Reuter. 2012. “Measuring and comparing structural fluctuation patterns in large protein datasets.” <em>Bioinformatics (Oxford, England)</em> 28 (19): 2431–40. <a href="https://doi.org/10.1093/bioinformatics/bts445">https://doi.org/10.1093/bioinformatics/bts445</a>.</p>
</div>
<div id="ref-grant06">
<p>Grant, B. J., A. P. D. C Rodrigues, K. M. Elsawy, A. J. Mccammon, and L. S. D. Caves. 2006. “Bio3d: An R Package for the Comparative Analysis of Protein Structures.” <em>Bioinformatics</em> 22: 2695–6. <a href="https://doi.org/10.1093/bioinformatics/btl461">https://doi.org/10.1093/bioinformatics/btl461</a>.</p>
</div>
<div id="ref-Hinsen2000p">
<p>Hinsen, K, A J Petrescu, S Dellerue, M C Bellissent-Funel, and G R Kneller. 2000. “Harmonicity in slow protein dynamics.” <em>Chemical Physics</em> 261 (1-2): 25–37. <a href="https://doi.org/10.1016/S0301-0104(00)00222-6">https://doi.org/10.1016/S0301-0104(00)00222-6</a>.</p>
</div>
<div id="ref-Romanowska2012a">
<p>Romanowska, Julia, Krzysztof S. Nowinski, and Joanna Trylska. 2012. “Determining geometrically stable domains in molecular conformation sets.” <em>Journal of Chemical Theory and Computation</em> 8 (8): 2588–99. <a href="https://doi.org/10.1021/ct300206j">https://doi.org/10.1021/ct300206j</a>.</p>
</div>
<div id="ref-skjaerven15">
<p>Skjaerven, L., X. Q. Yao, G. Scarabelli, and B. J. Grant. 2015. “Integrating Protein Structural Dynamics and Evolutionary Analysis with Bio3d.” <em>BMC Bioinformatics</em> 15: 399. <a href="https://doi.org/10.1186/s12859-014-0399-6">https://doi.org/10.1186/s12859-014-0399-6</a>.</p>
</div>
<div id="ref-Tama2000">
<p>Tama, F., F. X. Gadea, O. Marques, and Y. H. Sanejouand. 2000. “Building-Block Approach for Determining Low-Frequency Normal Modes of Macromolecules.” <em>Proteins</em> 41: 1–7. <a href="https://doi.org/10.1002/1097-0134(20001001)41:1%3C1::AID-PROT10%3E3.0.CO;2-P">https://doi.org/10.1002/1097-0134(20001001)41:1&lt;1::AID-PROT10&gt;3.0.CO;2-P</a>.</p>
</div>
<div id="ref-Tama2001p">
<p>Tama, F, and Y H Sanejouand. 2001. “Conformational change of proteins arising from normal mode calculations.” <em>Protein Eng</em> 14 (1): 1–6. <a href="https://doi.org/10.1093/protein/14.1.1">https://doi.org/10.1093/protein/14.1.1</a>.</p>
</div>
<div id="ref-Yao2016a">
<p>Yao, X. Q., L. Skjaerven, and B. J. Grant. 2016. “Rapid Characterization of Allosteric Networks with Ensemble Normal Mode Analysis.” <em>The Journal of Physical Chemistry B</em> 120: 8276–88. <a href="https://doi.org/10.1021/acs.jpcb.6b01991">https://doi.org/10.1021/acs.jpcb.6b01991</a>.</p>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>The latest version of the package, full documentation and further vignettes (including detailed installation instructions) can be obtained from the main Bio3D website: <a href="http://thegrantlab.org/bio3d/">http://thegrantlab.org/bio3d/</a>.<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>This vignette contains executable examples, see <code><a href="https://rdrr.io/r/utils/vignette.html">help(vignette)</a></code> for further details.<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p>See also dedicated vignettes for <em>ensemble NMA</em> provided with the Bio3D package.<a href="#fnref3" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="tidyverse">
  <p>Authored by Barry Grant, Xin-Qiu Yao, Lars Skjaerven, Julien Ide.</p>
</div>

<div class="author">
  <p>
    Built with ❤️ and <a href="https://pkgdown.r-lib.org">pkgdown</a> © 2020.
  </p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '1270c4079b26a138795263974bbf302d',
    indexName: 'tidyverse',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
